// @opentiny/vue-time@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/time/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import TimeSpinner from '@opentiny/vue-time-spinner';
import Button from '@opentiny/vue-button';
import { resolveComponent, openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, createVNode, createTextVNode, toDisplayString, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["dodestroy", "pick", "select-range"],
  components: {
    TimeSpinner,
    TinyButton: Button
  },
  props: [].concat(props, ["show", "timeArrowControl", "emitter", "value", "step"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-time-panel__footer",
  class: "border-t border-t-color-bg-3 py-3 px-4 h-14 leading-7 box-border flex justify-between"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_time_spinner = resolveComponent("time-spinner");
  var _component_tiny_button = resolveComponent("tiny-button");
  return openBlock(), createBlock(Transition, {
    name: "tiny-zoom-in-top",
    onAfterLeave: _cache[1] || (_cache[1] = function($event) {
      return _ctx.$emit("dodestroy");
    }),
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          "data-tag": "tiny-time-panel tiny-popper us",
          class: normalizeClass(_ctx.m("my-1 bg-color-bg-1 shadow-md rounded absolute w-60 right-0 z-[1000] box-content select-none", _ctx.state.popperClass))
        },
        [createElementVNode(
          "div",
          {
            "data-tag": "tiny-time-panel__content",
            class: normalizeClass(["relative overflow-hidden", {
              "has-seconds": _ctx.state.showSeconds
            }])
          },
          [createVNode(_component_time_spinner, {
            ref: "spinner",
            step: _ctx.step,
            onChange: _ctx.handleChange,
            "arrow-control": _ctx.state.useArrow,
            "show-seconds": _ctx.state.showSeconds,
            "am-pm-mode": _ctx.state.amPmMode,
            onSelectRange: _ctx.setSelectionRange,
            date: _ctx.state.date
          }, null, 8, ["step", "onChange", "arrow-control", "show-seconds", "am-pm-mode", "onSelectRange", "date"])],
          2
          /* CLASS */
        ), createElementVNode("div", _hoisted_1, [createVNode(_component_tiny_button, {
          "data-tag": "tiny-time-panel__btn cancel",
          type: "text",
          "custom-class": "text-color-brand text-left p-0",
          onClick: _ctx.handleCancel
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.datepicker.cancel")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        }, 8, ["onClick"]), createVNode(_component_tiny_button, {
          "data-tag": "tiny-time-panel__btn",
          type: "primary",
          disabled: _ctx.state.disabled,
          onClick: _cache[0] || (_cache[0] = function($event) {
            return _ctx.handleConfirm();
          })
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.datepicker.confirm")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        }, 8, ["disabled"])])],
        2
        /* CLASS */
      ), [[vShow, _ctx.state.visible]])];
    }),
    _: 1
    /* STABLE */
  });
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
