// @opentiny/vue-badge@3.20.0/lib/mobile.js
import { renderless, api } from '@opentiny/vue-renderless/badge/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import '@opentiny/vue-theme-mobile/badge/index.css';
import { openBlock, createElementBlock, renderSlot, normalizeClass, createElementVNode, toDisplayString, createCommentVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["isDot", "isFixed", "isMini", "hidden", "max", "type", "value", "modelValue", "href", "target"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-mobile-badge"
};
var _hoisted_2 = {
  key: 0
};
var _hoisted_3 = ["href", "target"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [renderSlot(_ctx.$slots, "default"), !_ctx.hidden && (_ctx.value > 0 || _ctx.isDot) ? (openBlock(), createElementBlock(
    "div",
    {
      key: 0,
      class: normalizeClass(["tiny-mobile-badge__content", [{
        "is-dot": _ctx.isDot,
        "is-fixed": _ctx.isFixed,
        "is-mini": _ctx.isMini
      }, _ctx.value < 10 ? "is-circle" : "", _ctx.type ? "tiny-mobile-badge--" + _ctx.type : ""]])
    },
    [!_ctx.isDot ? (openBlock(), createElementBlock("span", _hoisted_2, [renderSlot(_ctx.$slots, "content", {}, function() {
      return [createElementVNode("a", {
        href: _ctx.state.href,
        target: _ctx.target,
        rel: "noopener noreferrer",
        class: "tiny-mobile-badge__link"
      }, toDisplayString(_ctx.state.content), 9, _hoisted_3)];
    })])) : createCommentVNode("v-if", true)],
    2
    /* CLASS */
  )) : createCommentVNode("v-if", true)]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobile as default };
