// @opentiny/vue-time-line-new@3.20.0/lib/mobile.js
import { renderless, api } from '@opentiny/vue-renderless/time-line-new/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import '@opentiny/vue-theme-mobile/steps/index.css';
import { openBlock, createElementBlock, Fragment, renderList, normalizeStyle, normalizeClass, createElementVNode, renderSlot, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["nameField", "timeField", "data", "space"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: /* @__PURE__ */ normalizeClass(["tiny-mobile-timeline"])
};
var _hoisted_2 = {
  class: "tiny-mobile-timeline__list"
};
var _hoisted_3 = {
  class: "tiny-mobile-timeline__date-time"
};
var _hoisted_4 = {
  class: "tiny-mobile-timeline__date"
};
var _hoisted_5 = {
  class: "tiny-mobile-timeline__time"
};
var _hoisted_6 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "tiny-mobile-timeline__icon"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_7 = [_hoisted_6];
var _hoisted_8 = {
  class: "tiny-mobile-timeline__content"
};
var _hoisted_9 = {
  class: "tiny-mobile-timeline__title"
};
var _hoisted_10 = {
  class: "tiny-mobile-timeline__cycle"
};
var _hoisted_11 = {
  class: "tiny-mobile-timeline__user"
};
var _hoisted_12 = {
  class: "tiny-mobile-timeline__tip"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [(openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.state.nodes, function(node, index) {
      return openBlock(), createElementBlock(
        "div",
        {
          key: index,
          style: normalizeStyle({
            height: index === _ctx.state.nodes.length - 1 ? "" : _ctx.space ? _ctx.space + "px" : "88px"
          }),
          class: normalizeClass(["tiny-mobile-timeline__item", node.status])
        },
        [createElementVNode("ul", _hoisted_2, [renderSlot(_ctx.$slots, "left", {
          slotScope: node
        }, function() {
          return [createElementVNode("li", _hoisted_3, [createElementVNode(
            "span",
            _hoisted_4,
            toDisplayString(_ctx.getDate(node[_ctx.timeField]).date),
            1
            /* TEXT */
          ), createElementVNode(
            "span",
            _hoisted_5,
            toDisplayString(_ctx.getDate(node[_ctx.timeField]).time),
            1
            /* TEXT */
          )])];
        }), createElementVNode(
          "li",
          {
            style: normalizeStyle({
              height: index === _ctx.state.nodes.length - 1 ? "" : _ctx.space ? _ctx.space + "px" : "88px"
            }),
            class: "tiny-mobile-timeline__line"
          },
          _hoisted_7,
          4
          /* STYLE */
        ), renderSlot(_ctx.$slots, "right", {
          slotScope: node
        }, function() {
          return [createElementVNode("li", _hoisted_8, [createElementVNode("div", null, [createElementVNode(
            "span",
            _hoisted_9,
            toDisplayString(node[_ctx.nameField]),
            1
            /* TEXT */
          ), createElementVNode(
            "span",
            _hoisted_10,
            toDisplayString(node.cycle),
            1
            /* TEXT */
          )]), createElementVNode("div", null, [createElementVNode(
            "span",
            _hoisted_11,
            toDisplayString(node.personInfo),
            1
            /* TEXT */
          ), createElementVNode(
            "span",
            _hoisted_12,
            toDisplayString(node.overdue),
            1
            /* TEXT */
          )])])];
        })])],
        6
        /* CLASS, STYLE */
      );
    }),
    128
    /* KEYED_FRAGMENT */
  ))]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobile as default };
