// @opentiny/vue-tabbar@3.20.0/lib/mobile-first.js
import { defineComponent, $prefix, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/tabbar/vue';
import TabbarItem from '@opentiny/vue-tabbar-item';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, renderSlot, Fragment, renderList, createBlock, mergeProps } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "Tabbar",
  componentName: "Tabbar",
  props: {
    activeColor: String,
    border: {
      type: Boolean,
      default: true
    },
    inactiveColor: String,
    modelValue: {
      type: [Number, String],
      default: 0
    },
    showNumber: {
      type: Number,
      default: 5
    },
    itemList: {
      type: Array,
      default: function _default() {
        return [];
      }
    }
  },
  components: {
    TabbarItem
  },
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tabbar_item = resolveComponent("tabbar-item");
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-tabbar",
      class: normalizeClass(["z-10 flex box-content w-full h-14 bg-color-bg-1 fixed bottom-0 left-0", {
        "border-t border-color-border border-solid": _ctx.border
      }, {
        unfit: !_ctx.state.fit
      }]),
      ref: "tabbar"
    },
    [renderSlot(_ctx.$slots, "default", {}, function() {
      return [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.itemList, function(item, index) {
          return openBlock(), createBlock(
            _component_tabbar_item,
            mergeProps({
              key: index
            }, item),
            null,
            16
            /* FULL_PROPS */
          );
        }),
        128
        /* KEYED_FRAGMENT */
      ))];
    })],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
