// @opentiny/vue-virtual-tree@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import { numericProp, makeNumericProp, unknownProp } from "@opentiny/vue-renderless/common/prop-util";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var VirtualTree = defineComponent({
  name: $prefix + "VirtualTree",
  props: _extends({}, $props, {
    width: numericProp,
    height: numericProp,
    rowHeight: makeNumericProp(36),
    scrollbarSize: makeNumericProp(6),
    treeOp: unknownProp
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
VirtualTree.install = function(Vue) {
  Vue.component(VirtualTree.name, VirtualTree);
};
VirtualTree.version = version;
export {
  VirtualTree as default
};
