// @opentiny/vue-input@3.20.0/lib/tall-storage.js
import { defineComponent, $prefix, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/tall-storage/vue';
import '@opentiny/vue-theme/tall-storage/index.css';
import { openBlock, createElementBlock, createElementVNode, withKeys, Fragment, renderList, normalizeClass, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "TallStorage",
  inheritAttrs: false,
  props: {
    isMemoryStorage: Boolean,
    localstorageData: Array
  },
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-storage-list-style"
};
var _hoisted_2 = ["onClick"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-tall-storage",
      onMousedown: _cache[1] || (_cache[1] = function() {
        return _ctx.mousedown && _ctx.mousedown.apply(_ctx, arguments);
      })
    },
    [createElementVNode("div", _hoisted_1, [createElementVNode(
      "ul",
      {
        onKeydown: _cache[0] || (_cache[0] = withKeys(function() {
          return _ctx.keydownEvent && _ctx.keydownEvent.apply(_ctx, arguments);
        }, ["enter"])),
        class: "tiny-storage-list"
      },
      [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.localstorageData, function(item) {
          return openBlock(), createElementBlock("li", {
            key: item,
            class: normalizeClass(["tiny-storage-item", [_ctx.state.hoverValue === item ? "item-hover" : ""]]),
            onClick: function onClick($event) {
              return _ctx.selectItem(item);
            }
          }, toDisplayString(item), 11, _hoisted_2);
        }),
        128
        /* KEYED_FRAGMENT */
      ))],
      32
      /* NEED_HYDRATION */
    )])],
    32
    /* NEED_HYDRATION */
  );
}
var tallStorage = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { tallStorage as default };
