// @opentiny/vue-time-range@3.20.0/lib/index.js
import { renderless, api } from '@opentiny/vue-renderless/time-range/vue';
import { defineComponent, $prefix, setup } from '@opentiny/vue-common';
import TimeSpinner from '@opentiny/vue-time-spinner';
import Button from '@opentiny/vue-button';
import { resolveComponent, openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, toDisplayString, createElementBlock, createCommentVNode, createVNode, createTextVNode, vShow } from 'vue';
import '@opentiny/vue-theme/time-range/index.css';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "TimeRange",
  emits: ["dodestroy", "pick", "select-range"],
  components: {
    TimeSpinner,
    TinyButton: Button
  },
  props: {
    emitter: Object
  },
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  class: "tiny-time-range-picker__content"
};
var _hoisted_2 = {
  class: "tiny-time-range-picker__cell"
};
var _hoisted_3 = {
  class: "tiny-time-range-picker__header"
};
var _hoisted_4 = {
  class: "tiny-time-range-picker__content-date"
};
var _hoisted_5 = {
  class: "tiny-time-range-picker__header-title"
};
var _hoisted_6 = {
  key: 0
};
var _hoisted_7 = {
  class: "tiny-time-range-picker__cell"
};
var _hoisted_8 = {
  class: "tiny-time-range-picker__header"
};
var _hoisted_9 = {
  class: "tiny-time-range-picker__header-title"
};
var _hoisted_10 = {
  key: 0
};
var _hoisted_11 = {
  class: "tiny-time-range-picker__footer"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_time_spinner = resolveComponent("time-spinner");
  var _component_tiny_button = resolveComponent("tiny-button");
  return openBlock(), createBlock(Transition, {
    name: "tiny-zoom-in-top",
    onAfterLeave: _cache[2] || (_cache[2] = function($event) {
      return _ctx.$emit("dodestroy");
    }),
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          class: normalizeClass(["tiny-time-range tiny-time-range-picker tiny-picker-panel tiny-popper", _ctx.state.popperClass])
        },
        [createElementVNode("div", _hoisted_1, [createElementVNode("div", _hoisted_2, [createElementVNode("div", _hoisted_3, [createElementVNode(
          "span",
          null,
          toDisplayString(_ctx.t("ui.datepicker.startTime")),
          1
          /* TEXT */
        )]), createElementVNode("div", _hoisted_4, [createElementVNode("div", _hoisted_5, [createElementVNode(
          "span",
          null,
          toDisplayString(_ctx.t("ui.datepicker.hour")),
          1
          /* TEXT */
        ), createElementVNode(
          "span",
          null,
          toDisplayString(_ctx.t("ui.datepicker.minute")),
          1
          /* TEXT */
        ), _ctx.state.showSeconds ? (openBlock(), createElementBlock(
          "span",
          _hoisted_6,
          toDisplayString(_ctx.t("ui.datepicker.second")),
          1
          /* TEXT */
        )) : createCommentVNode("v-if", true)]), createElementVNode(
          "div",
          {
            class: normalizeClass([{
              "has-seconds": _ctx.state.showSeconds,
              "is-arrow": _ctx.state.arrowControl
            }, "tiny-time-range-picker__body tiny-time-panel__content"])
          },
          [createVNode(_component_time_spinner, {
            ref: "minSpinner",
            "show-seconds": _ctx.state.showSeconds,
            "am-pm-mode": _ctx.state.amPmMode,
            onChange: _ctx.handleMinChange,
            "end-date": _ctx.state.maxDate,
            "arrow-control": _ctx.state.arrowControl,
            onSelectRange: _ctx.setMinSelectionRange,
            date: _ctx.state.minDate
          }, null, 8, ["show-seconds", "am-pm-mode", "onChange", "end-date", "arrow-control", "onSelectRange", "date"])],
          2
          /* CLASS */
        )])]), createElementVNode("div", _hoisted_7, [createElementVNode("div", _hoisted_8, [createElementVNode(
          "span",
          null,
          toDisplayString(_ctx.t("ui.datepicker.endTime")),
          1
          /* TEXT */
        )]), createElementVNode("div", _hoisted_9, [createElementVNode(
          "span",
          null,
          toDisplayString(_ctx.t("ui.datepicker.hour")),
          1
          /* TEXT */
        ), createElementVNode(
          "span",
          null,
          toDisplayString(_ctx.t("ui.datepicker.minute")),
          1
          /* TEXT */
        ), _ctx.state.showSeconds ? (openBlock(), createElementBlock(
          "span",
          _hoisted_10,
          toDisplayString(_ctx.t("ui.datepicker.second")),
          1
          /* TEXT */
        )) : createCommentVNode("v-if", true)]), createElementVNode(
          "div",
          {
            class: normalizeClass([{
              "has-seconds": _ctx.state.showSeconds,
              "is-arrow": _ctx.state.arrowControl
            }, "tiny-time-range-picker__body tiny-time-panel__content"])
          },
          [createVNode(_component_time_spinner, {
            ref: "maxSpinner",
            "show-seconds": _ctx.state.showSeconds,
            "am-pm-mode": _ctx.state.amPmMode,
            onChange: _ctx.handleMaxChange,
            "start-date": _ctx.state.minDate,
            "arrow-control": _ctx.state.arrowControl,
            onSelectRange: _ctx.setMaxSelectionRange,
            date: _ctx.state.maxDate
          }, null, 8, ["show-seconds", "am-pm-mode", "onChange", "start-date", "arrow-control", "onSelectRange", "date"])],
          2
          /* CLASS */
        )])]), createElementVNode("div", _hoisted_11, [!_ctx.state.showTimePickerRangeButton ? (openBlock(), createBlock(_component_tiny_button, {
          key: 0,
          class: "tiny-time-range-picker__btn cancel",
          onClick: _cache[0] || (_cache[0] = function($event) {
            return _ctx.handleCancel();
          })
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.datepicker.cancel")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        })) : createCommentVNode("v-if", true), createVNode(_component_tiny_button, {
          onClick: _cache[1] || (_cache[1] = function($event) {
            return _ctx.handleConfirm();
          }),
          disabled: _ctx.state.btnDisabled,
          size: "small",
          class: normalizeClass({
            "tiny-time-range-picker__btn": !_ctx.state.showTimePickerRangeButton,
            confirm: !_ctx.state.disabled && !_ctx.state.showTimePickerRangeButton
          })
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.datepicker.confirm")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        }, 8, ["disabled", "class"])])],
        2
        /* CLASS */
      ), [[vShow, _ctx.state.visible]])];
    }),
    _: 1
    /* STABLE */
  });
}
var TimeRange = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.20.0";
TimeRange.install = function(Vue) {
  Vue.component(TimeRange.name, TimeRange);
};
TimeRange.version = version;

export { TimeRange as default };
