// @opentiny/vue-pager-item@3.20.0/lib/mobile-first.js
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/pager-item/vue';
import { IconPopup, IconDoubleLeft, IconDoubleRight } from '@opentiny/vue-icon';
import { openBlock, createElementBlock, normalizeClass, createElementVNode, createCommentVNode, createBlock, resolveDynamicComponent, Fragment, renderList, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var classes = {
  "pager-group": "inline-block align-middle text-xs h-7",
  "li": "bg-left-top inline-block text-xs ml-2 cursor-pointer text-center h-7 leading-7 min-w-[1.75rem] rounded-sm text-color-text-primary px-1 box-border transition-all duration-300 ease-in",
  "li-hover": "hover:text-color-brand-hover",
  "is-active": "text-color-brand border border-solid border-color-brand",
  "quick-icon": "h-7 w-3 fill-color-icon-primary hover:fill-color-brand",
  "dot": "w-4 min-w-0 p-0 border-none"
};
var _sfc_main = defineComponent({
  emits: ["click", "change"],
  components: {
    IconPopup: IconPopup(),
    IconDoubleLeft: IconDoubleLeft(),
    IconDoubleRight: IconDoubleRight()
  },
  props: [].concat(props, ["disabled", "currentPage", "pageCount", "pagerCount", "popupIcon", "doubleLeftIcon", "doubleRightIcon", "isBeforePageChange"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = ["textContent"];
var _hoisted_2 = ["textContent"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-pager-item",
      class: normalizeClass(_ctx.gcls("pager-group"))
    },
    [createElementVNode("ul", {
      "data-tag": "pager-item-content",
      class: "inline-flex text-xs",
      onClick: _cache[4] || (_cache[4] = function() {
        return _ctx.onPagerClick && _ctx.onPagerClick.apply(_ctx, arguments);
      })
    }, [_ctx.pageCount > 0 ? (openBlock(), createElementBlock(
      "li",
      {
        key: 0,
        "data-tag": "pager-item-hover",
        class: normalizeClass(_ctx.m(_ctx.gcls("li"), _ctx.gcls("li-hover"), _ctx.gcls({
          "is-active": _ctx.currentPage === 1
        }))),
        textContent: "1"
      },
      null,
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true), _ctx.state.showPrevMore ? (openBlock(), createElementBlock(
      "li",
      {
        key: 1,
        ref: "prev",
        "data-tag": "pager-item-premore",
        class: normalizeClass(_ctx.m(_ctx.gcls("li"), _ctx.gcls("dot"))),
        onMouseenter: _cache[0] || (_cache[0] = function($event) {
          return _ctx.onMouseenter("left");
        }),
        onMouseleave: _cache[1] || (_cache[1] = function($event) {
          return _ctx.state.quickprevIconClass = _ctx.popupIcon;
        })
      },
      [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.quickprevIconClass), {
        class: normalizeClass(_ctx.gcls("quick-icon"))
      }, null, 8, ["class"]))],
      34
      /* CLASS, NEED_HYDRATION */
    )) : createCommentVNode("v-if", true), (openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.pagers, function(pager) {
        return openBlock(), createElementBlock("li", {
          "data-tag": "pager-item-pager",
          key: pager,
          class: normalizeClass(_ctx.m(_ctx.gcls("li"), _ctx.gcls("li-hover"), _ctx.gcls({
            "is-active": _ctx.currentPage === pager
          }))),
          textContent: toDisplayString("" + pager)
        }, null, 10, _hoisted_1);
      }),
      128
      /* KEYED_FRAGMENT */
    )), _ctx.state.showNextMore ? (openBlock(), createElementBlock(
      "li",
      {
        key: 2,
        ref: "next",
        "data-tag": "pager-item-nextmore",
        class: normalizeClass(_ctx.m(_ctx.gcls("li"), _ctx.gcls("dot"))),
        onMouseenter: _cache[2] || (_cache[2] = function($event) {
          return _ctx.onMouseenter("right");
        }),
        onMouseleave: _cache[3] || (_cache[3] = function($event) {
          return _ctx.state.quicknextIconClass = _ctx.popupIcon;
        })
      },
      [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.quicknextIconClass), {
        class: normalizeClass(_ctx.gcls("quick-icon"))
      }, null, 8, ["class"]))],
      34
      /* CLASS, NEED_HYDRATION */
    )) : createCommentVNode("v-if", true), _ctx.pageCount > 1 ? (openBlock(), createElementBlock("li", {
      key: 3,
      "data-tag": "pager-item-active",
      class: normalizeClass(["hover:text-color-brand-hover", _ctx.m(_ctx.gcls("li"), _ctx.gcls({
        "is-active": _ctx.pageCount === _ctx.currentPage
      }))]),
      textContent: toDisplayString("" + _ctx.pageCount)
    }, null, 10, _hoisted_2)) : createCommentVNode("v-if", true)])],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
