// @opentiny/vue-scroll-text@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/scroll-text/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  ANIMATION_DURATION: "animation-duration"
};
var ScrollText = defineComponent({
  name: $prefix + "ScrollText",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    text: {
      type: String,
      default: "ScrollText"
    },
    time: {
      type: Number,
      default: 5
    },
    styles: {
      type: String,
      default: ""
    },
    direction: {
      type: String,
      default: "left",
      validator: function validator(value) {
        return Boolean(~["up", "left", "down", "right"].indexOf(value));
      }
    },
    hoverStop: {
      type: Boolean,
      default: true
    },
    widthAdapt: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
ScrollText.install = function(Vue) {
  Vue.component(ScrollText.name, ScrollText);
};
ScrollText.version = version;
export {
  ScrollText as default
};
