// @opentiny/vue-scroll-text@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/scroll-text/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { openBlock, createElementBlock, createElementVNode, normalizeClass, normalizeStyle, renderSlot, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["text", "time", "styles", "direction", "hoverStop", "widthAdapt"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-scroll-text__wrapper"
};
var _hoisted_2 = {
  ref: "content",
  class: "tiny-scroll-text__content"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "div",
    {
      ref: "wrap",
      class: normalizeClass(["tiny-scroll-text", {
        "width-adapt": _ctx.widthAdapt
      }]),
      onMouseenter: _cache[0] || (_cache[0] = function() {
        return _ctx.stopAnimation && _ctx.stopAnimation.apply(_ctx, arguments);
      }),
      onMouseleave: _cache[1] || (_cache[1] = function() {
        return _ctx.startAnimation && _ctx.startAnimation.apply(_ctx, arguments);
      })
    },
    [createElementVNode(
      "div",
      _hoisted_2,
      [createElementVNode(
        "div",
        {
          class: normalizeClass([{
            stop: _ctx.state.isStop
          }, _ctx.direction]),
          style: normalizeStyle(_ctx.state.bindStyle)
        },
        [renderSlot(_ctx.$slots, "default", {}, function() {
          return [createElementVNode(
            "p",
            null,
            toDisplayString(_ctx.text),
            1
            /* TEXT */
          )];
        })],
        6
        /* CLASS, STYLE */
      )],
      512
      /* NEED_PATCH */
    )],
    34
    /* CLASS, NEED_HYDRATION */
  )]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
