// @opentiny/vue-layout@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/layout/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Layout = defineComponent({
  name: $prefix + "Layout",
  provide: function provide() {
    return {
      layout: this
    };
  },
  props: _extends({}, $props, {
    tag: {
      type: String,
      default: "div"
    },
    cols: {
      type: Number,
      default: 12,
      validator: function validator(value) {
        return Boolean(~[12, 24].indexOf(value));
      }
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Layout.install = function(Vue) {
  Vue.component(Layout.name, Layout);
};
Layout.version = version;
export {
  Layout as default
};
