// @opentiny/vue-toast@3.20.0/lib/mobile.js
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/toast/vue';
import '@opentiny/vue-theme-mobile/toast/index.css';
import { IconMobileErrorWhite, IconMobileSuccessWhite } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, createElementVNode, normalizeClass, createBlock, createCommentVNode, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    IconMobileErrorWhite: IconMobileErrorWhite(),
    IconMobileSuccessWhite: IconMobileSuccessWhite()
  },
  props: [].concat(props, ["type", "zIndex", "text", "time", "timeout"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-mobile-toast"
};
var _hoisted_2 = {
  key: 0,
  class: "tiny-mobile-toast-icon"
};
var _hoisted_3 = {
  class: "tiny-mobile-toast-text"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_mobile_error_white = resolveComponent("icon-mobile-error-white");
  var _component_icon_mobile_success_white = resolveComponent("icon-mobile-success-white");
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "div",
    {
      class: normalizeClass(["tiny-mobile-toast-content", "tiny-mobile-toast-content-" + _ctx.state.type])
    },
    [_ctx.state.type !== "text" ? (openBlock(), createElementBlock("div", _hoisted_2, [_ctx.state.type === "error" ? (openBlock(), createBlock(_component_icon_mobile_error_white, {
      key: 0
    })) : createCommentVNode("v-if", true), _ctx.state.type === "correct" ? (openBlock(), createBlock(_component_icon_mobile_success_white, {
      key: 1
    })) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true), createElementVNode(
      "div",
      _hoisted_3,
      toDisplayString(_ctx.state.text),
      1
      /* TEXT */
    )],
    2
    /* CLASS */
  )]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobile as default };
