// @opentiny/vue-index-bar@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileTemplate from "./mobile.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileTemplate;
};
var IndexBar = defineComponent({
  name: $prefix + "IndexBar",
  props: _extends({}, $props, {
    indexList: {
      type: Array,
      default: function _default() {
        return [];
      }
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
IndexBar.install = function(Vue) {
  Vue.component(IndexBar.name, IndexBar);
};
IndexBar.version = version;
export {
  IndexBar as default
};
