// @opentiny/vue-bulletin-board@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/bulletin-board/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import Tabs from '@opentiny/vue-tabs';
import TabItem from '@opentiny/vue-tab-item';
import { iconChevronRight } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, createElementVNode, toDisplayString, createVNode, withCtx, Fragment, renderList, createBlock, createTextVNode, resolveDynamicComponent, createCommentVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["title", "icon", "showMore", "moreLink", "tabTitle", "activeName", "data"]),
  components: {
    Tabs,
    TabItem,
    IconChevronRight: iconChevronRight()
  },
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-bulletin-board"
};
var _hoisted_2 = {
  class: "tiny-bulletin-board__title"
};
var _hoisted_3 = {
  class: "tiny-bulletin-board__contain"
};
var _hoisted_4 = ["href", "target", "onClick"];
var _hoisted_5 = {
  key: 0,
  class: "tiny-bulletin-board__new"
};
var _hoisted_6 = ["onClick"];
var _hoisted_7 = {
  key: 0,
  class: "tiny-bulletin-board__new"
};
var _hoisted_8 = {
  class: "tiny-bulletin-board__textDate"
};
var _hoisted_9 = {
  key: 0,
  class: "tiny-bulletin-board__more"
};
var _hoisted_10 = ["href", "target"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_chevron_right = resolveComponent("icon-chevron-right");
  var _component_tab_item = resolveComponent("tab-item");
  var _component_tabs = resolveComponent("tabs");
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "p",
    _hoisted_2,
    toDisplayString(_ctx.title),
    1
    /* TEXT */
  ), createVNode(_component_tabs, {
    modelValue: _ctx.state.actName,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.actName = $event;
    })
  }, {
    default: withCtx(function() {
      return [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.tabTitle, function(item, index) {
          return openBlock(), createBlock(_component_tab_item, {
            key: index,
            name: (index + 1).toString()
          }, {
            title: withCtx(function() {
              return [createElementVNode(
                "span",
                null,
                toDisplayString(item),
                1
                /* TEXT */
              )];
            }),
            default: withCtx(function() {
              return [createElementVNode("div", _hoisted_3, [(openBlock(true), createElementBlock(
                Fragment,
                null,
                renderList(_ctx.state.dataList, function(subItem, subIndex) {
                  return openBlock(), createElementBlock("div", {
                    key: subIndex,
                    class: "tiny-bulletin-board__item"
                  }, [subItem.url || subItem.route ? (openBlock(), createElementBlock("a", {
                    key: 0,
                    href: subItem.url || _ctx.getRoute(subItem.route),
                    target: subItem.target,
                    class: "tiny-bulletin-board__textTitle",
                    rel: "noopener noreferrer",
                    onClick: function onClick($event) {
                      return _ctx.handleBulletinBoardClick(subItem);
                    }
                  }, [createTextVNode(
                    toDisplayString(subItem.text) + " ",
                    1
                    /* TEXT */
                  ), subIndex === 0 ? (openBlock(), createElementBlock("span", _hoisted_5, [(openBlock(), createBlock(resolveDynamicComponent(_ctx.icon), {
                    class: "tiny-svg-size"
                  })), createTextVNode(
                    toDisplayString(_ctx.icon ? "" : "NEW"),
                    1
                    /* TEXT */
                  )])) : createCommentVNode("v-if", true)], 8, _hoisted_4)) : (openBlock(), createElementBlock("span", {
                    key: 1,
                    class: "tiny-bulletin-board__textTitle",
                    onClick: function onClick($event) {
                      return _ctx.handleBulletinBoardClick(subItem);
                    }
                  }, [createTextVNode(
                    toDisplayString(subItem.text) + " ",
                    1
                    /* TEXT */
                  ), subIndex === 0 ? (openBlock(), createElementBlock("span", _hoisted_7, [(openBlock(), createBlock(resolveDynamicComponent(_ctx.icon), {
                    class: "tiny-svg-size"
                  })), createTextVNode(
                    toDisplayString(_ctx.icon ? "" : "NEW"),
                    1
                    /* TEXT */
                  )])) : createCommentVNode("v-if", true)], 8, _hoisted_6)), createElementVNode(
                    "p",
                    _hoisted_8,
                    toDisplayString(subItem.date),
                    1
                    /* TEXT */
                  )]);
                }),
                128
                /* KEYED_FRAGMENT */
              )), _ctx.showMore ? (openBlock(), createElementBlock("div", _hoisted_9, [_ctx.showMore && _ctx.state.moreLink && (_ctx.state.moreLink.url || _ctx.state.moreLink.route) ? (openBlock(), createElementBlock("a", {
                key: 0,
                href: _ctx.state.moreLink.url || _ctx.getRoute(_ctx.state.moreLink.route),
                class: "tiny-bulletin-board__more-link",
                target: _ctx.state.moreLink.target,
                rel: "noopener noreferrer"
              }, [createElementVNode(
                "span",
                null,
                toDisplayString(_ctx.state.moreLink.text || "more"),
                1
                /* TEXT */
              ), createVNode(_component_icon_chevron_right, {
                class: "tiny-svg-size"
              })], 8, _hoisted_10)) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true)])];
            }),
            _: 2
            /* DYNAMIC */
          }, 1032, ["name"]);
        }),
        128
        /* KEYED_FRAGMENT */
      ))];
    }),
    _: 1
    /* STABLE */
  }, 8, ["modelValue"])]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
