// @opentiny/vue-carousel-item@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/carousel-item/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { withDirectives, openBlock, createElementBlock, normalizeClass, normalizeStyle, vShow, createCommentVNode, renderSlot, createElementVNode, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["name", "title", "label"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 1,
  "data-tag": "tiny-carousel-item-title",
  class: "absolute bottom-0 h-6 w-full text-left text-color-text-inverse leading-6 bg-color-text-disabled"
};
var _hoisted_2 = {
  class: "py-0 px-3 text-xs w-4/5 inline-block text-ellipsis overflow-hidden whitespace-nowrap"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return withDirectives((openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-carousel-item",
      role: "carousel-item",
      class: normalizeClass(_ctx.m(["absolute top-0 left-0 w-full inline-block overflow-hidden z-0", _ctx.state.carouselParent.type === "card" ? "w-1/2 transition-transform duration-300 ease-in-out" : "", _ctx.state.hover ? "opacity-10" : "", _ctx.state.animate && !_ctx.state.moving ? "transition-transform duration-300 ease-in-out" : "", _ctx.state.inStage && _ctx.state.carouselParent.type === "card" ? "cursor-pointer z-[1] group" : "", _ctx.state.active ? "z-[2]" : ""])),
      style: normalizeStyle(_ctx.state.getTransform),
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.handleItemClick && _ctx.handleItemClick.apply(_ctx, arguments);
      })
    },
    [_ctx.state.carouselParent.type === "card" ? withDirectives((openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        "data-tag": "tiny-carousel-item-card",
        class: normalizeClass(["absolute h-full top-0 left-0 w-full opacity-20 duration-200 group-hover:opacity-10", {
          "opacity-10": _ctx.state.inStage && _ctx.state.carouselParent.type === "card" && _ctx.state.hover
        }])
      },
      null,
      2
      /* CLASS */
    )), [[vShow, !_ctx.state.active]]) : createCommentVNode("v-if", true), renderSlot(_ctx.$slots, "default"), _ctx.state.hasTitle ? (openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
      "span",
      _hoisted_2,
      toDisplayString(_ctx.title),
      1
      /* TEXT */
    )])) : createCommentVNode("v-if", true)],
    6
    /* CLASS, STYLE */
  )), [[vShow, _ctx.state.ready]]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
