// @opentiny/vue-carousel-item@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/carousel-item/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { withDirectives, openBlock, createElementBlock, normalizeClass, normalizeStyle, vShow, createCommentVNode, renderSlot, createElementVNode, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["name", "title", "label"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 0,
  class: "tiny-carousel__mask"
};
var _hoisted_2 = {
  key: 1,
  class: "item-title"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _ref;
  return withDirectives((openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass([(_ref = {
        "is-active": _ctx.state.active,
        "tiny-carousel__item--card": _ctx.state.carouselParent.type === "card",
        "is-in-stage": _ctx.state.inStage,
        "is-hover": _ctx.state.hover,
        "is-animating": _ctx.state.animating,
        "tiny-oblique": _ctx.state.isOblique
      }, _ref["tiny-oblique"] = _ctx.state.carouselParent.type === "card" && _ctx.state.carouselParent.setUserCls, _ref), "tiny-carousel__item"]),
      style: normalizeStyle(_ctx.state.getTransform),
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.handleItemClick && _ctx.handleItemClick.apply(_ctx, arguments);
      })
    },
    [_ctx.state.carouselParent.type === "card" ? withDirectives((openBlock(), createElementBlock(
      "div",
      _hoisted_1,
      null,
      512
      /* NEED_PATCH */
    )), [[vShow, !_ctx.state.active]]) : createCommentVNode("v-if", true), renderSlot(_ctx.$slots, "default"), _ctx.state.hasTitle ? (openBlock(), createElementBlock("div", _hoisted_2, [createElementVNode(
      "span",
      null,
      toDisplayString(_ctx.title),
      1
      /* TEXT */
    )])) : createCommentVNode("v-if", true)],
    6
    /* CLASS, STYLE */
  )), [[vShow, _ctx.state.ready]]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
