// @opentiny/vue-area@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/area/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var areaProps = _extends({}, $props, {
  modelValue: {},
  placeholder: {
    type: String,
    default: ""
  },
  size: String,
  disabled: {
    type: Boolean,
    default: false
  },
  props: {
    type: Object,
    default: function _default() {
      return {
        label: "name_cn",
        value: "org_id"
      };
    }
  },
  fetchJcr: {
    type: Function
  },
  fetchRegion: {
    type: Function
  },
  fetchRep: {
    type: Function
  },
  fetchOffice: {
    type: Function
  },
  popperClass: String,
  popperAppendToBody: {
    type: Boolean,
    default: true
  }
});
var Area = defineComponent({
  name: $prefix + "Area",
  props: areaProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Area.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Area.install = function(Vue) {
  Vue.component(Area.name, Area);
};
Area.version = version;
export {
  Area as default
};
