// @opentiny/fluent-editor@3.24.0/es/table/table-config.es.js
const TABLE_CELL_LINE_CLASS = "qlrte__better-table-cell-line";
const ROW_TOOL_CLASS = "qlrte__row-tool";
const ROW_TOOL_CELL_CLASS = "qlrte__row-tool-cell";
const ROW_TOOL_CELL_HOLDER_CLASS = "qlrte__row-tool-cell-holder";
const COL_TOOL_CLASS = "qlrte__col-tool";
const COL_TOOL_CELL_CLASS = "qlrte__col-tool-cell";
const COL_TOOL_CELL_HOLDER_CLASS = "qlrte__col-tool-cell-holder";
const SELECT_LINE_CLASS = "qlrte__selection-line";
const OPERATE_MENU_CLASS = "qlrte__op-menu";
const OPERATE_MENU_ITEM_CLASS = "qlrte__op-menu-item";
const OPERATE_MENU_ICON_CLASS = "qlrte__op-menu-icon";
const OPERATE_MENU_ICON_PREFIX_CLASS = "qlrte__op-menu-icon-prefix";
const OPERATE_MENU_TEXT_CLASS = "qlrte__op-menu-text";
const OPERATE_MENU_DIVIDING_CLASS = "qlrte__op-menu-dividing";
const OPERATE_MENU_SUBTITLE_CLASS = "qlrte__op-menu-subtitle";
const OPERATE_MENU_COLORPICKER_CLASS = "qlrte__op-menu-color-picker";
const OPERATE_MENU_COLORPICKER_ITEM_CLASS = "qlrte__op-menu-color-picker-item";
const COL_ATTRIBUTES = ["width"];
const COL_DEFAULT = {
  width: 100
};
const CELL_IDENTITY_KEYS = ["row", "cell"];
const CELL_ATTRIBUTES = ["rowspan", "colspan", "cell-bg", "list"];
const CELL_DEFAULT = {
  rowspan: 1,
  colspan: 1
};
const ERROR_LIMIT = 5;
const SCROLL_BAR_PANEL_HEIGHT = 32;
const ROW_TOOL_WIDTH = 16;
const COL_TOOL_HEIGHT = 16;
const CELL_MIN_WIDTH = 50;
const CELL_MIN_HEIGHT = 29;
const PRIMARY_COLOR = ["#5170ff", "var(--brand-1, #5170ff)"];
const MENU_ITEM_HEIGHT = 40;
const MENU_MIN_HEIGHT = 150;
const MENU_WIDTH = 205;
const LINE_POSITIONS = ["left", "right", "top", "bottom"];
export {
  CELL_ATTRIBUTES,
  CELL_DEFAULT,
  CELL_IDENTITY_KEYS,
  CELL_MIN_HEIGHT,
  CELL_MIN_WIDTH,
  COL_ATTRIBUTES,
  COL_DEFAULT,
  COL_TOOL_CELL_CLASS,
  COL_TOOL_CELL_HOLDER_CLASS,
  COL_TOOL_CLASS,
  COL_TOOL_HEIGHT,
  ERROR_LIMIT,
  LINE_POSITIONS,
  MENU_ITEM_HEIGHT,
  MENU_MIN_HEIGHT,
  MENU_WIDTH,
  OPERATE_MENU_CLASS,
  OPERATE_MENU_COLORPICKER_CLASS,
  OPERATE_MENU_COLORPICKER_ITEM_CLASS,
  OPERATE_MENU_DIVIDING_CLASS,
  OPERATE_MENU_ICON_CLASS,
  OPERATE_MENU_ICON_PREFIX_CLASS,
  OPERATE_MENU_ITEM_CLASS,
  OPERATE_MENU_SUBTITLE_CLASS,
  OPERATE_MENU_TEXT_CLASS,
  PRIMARY_COLOR,
  ROW_TOOL_CELL_CLASS,
  ROW_TOOL_CELL_HOLDER_CLASS,
  ROW_TOOL_CLASS,
  ROW_TOOL_WIDTH,
  SCROLL_BAR_PANEL_HEIGHT,
  SELECT_LINE_CLASS,
  TABLE_CELL_LINE_CLASS
};
