// @opentiny/vue-dropdown-menu@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/dropdown-menu/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import Tooltip from '@opentiny/vue-tooltip';
import { resolveComponent, openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, normalizeStyle, createVNode, renderSlot, createCommentVNode, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["visibleArrow", "arrowOffset", "placement", "popperClass", "popperAppendToBody", "checkedStatus", "multiStage", "maxHeight"]),
  emits: ["menu-item-click", "created"],
  components: {
    TinyTooltip: Tooltip
  },
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_tooltip = resolveComponent("tiny-tooltip");
  return openBlock(), createBlock(Transition, {
    name: "tiny-zoom-in-top",
    duration: 150,
    onAfterLeave: _ctx.doDestroy,
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "ul",
        {
          "data-tag": "tiny-dropdown-menu",
          class: normalizeClass(_ctx.m("min-w-[theme(spacing.18)] max-w-[theme(spacing.52)] absoulte bg-color-bg-1", {
            "py-1.5": _ctx.state.size === "medium"
          }, {
            "py-1.5": _ctx.state.size === "small"
          }, {
            "py-0.75": _ctx.state.size === "mini"
          }, {
            "overflow-x-hidden scrollbar-size-0": _ctx.maxHeight
          }, _ctx.multiStage ? "!block  rounded-none" : "rounded py-1 shadow-sm sm:shadow-md px-1", _ctx.state.canvasHeight ? "sm:mb-6" : !_ctx.multiStage && "sm:my-1 my-2", _ctx.popperClass)),
          style: normalizeStyle({
            maxHeight: _ctx.maxHeight + "px"
          })
        },
        [createVNode(_component_tiny_tooltip, {
          ref: "tooltip",
          effect: "light",
          content: _ctx.state.label,
          placement: "top-start",
          manual: true,
          modelValue: _ctx.state.showContent,
          "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
            return _ctx.state.showContent = $event;
          })
        }, null, 8, ["content", "modelValue"]), _ctx.state.initShowPopper || _ctx.state.showPopper ? renderSlot(_ctx.$slots, "default", {
          key: 0,
          selectedIndex: _ctx.state.selectedIndex
        }) : createCommentVNode("v-if", true)],
        6
        /* CLASS, STYLE */
      ), [[vShow, _ctx.state.showPopper]])];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["onAfterLeave"]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
