// @opentiny/vue-drop-roles@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/drop-roles/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var DropRoles = defineComponent({
  name: $prefix + "DropRoles",
  props: _extends({}, $props, {
    modelValue: {},
    placeholder: {
      type: String,
      default: ""
    },
    size: String,
    fetchRole: Function,
    fetchCurrentRole: Function,
    fields: Object,
    popperClass: String,
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    displayOnly: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
DropRoles.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
DropRoles.install = function(Vue) {
  Vue.component(DropRoles.name, DropRoles);
};
DropRoles.version = version;
export {
  DropRoles as default
};
