// @opentiny/huicharts@1.0.1/components/JadeJueChart/index.js
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import init from '../../option/init/index.js';
import BaseOption from './BaseOption.js';
import cloneDeep from '../../util/cloneDeep.js';
import { setSeriesData } from './handleSeries.js';
import { handleLabelFormatter } from './labelFormatter.js';
import PolarCoordSys from '../../option/PolarSys/index.js';
import { setbarWidth, setStartAngle, handleLegendData } from './handleOption.js';
import { CHART_TYPE } from '../../util/constants.js';

/**
 * Copyright (c) 2024 - present OpenTiny HUICharts Authors.
 * Copyright (c) 2024 - present Huawei Cloud Computing Technologies Co., Ltd.
 *
 * Use of this source code is governed by an MIT-style license.
 *
 * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
 * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
 * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
 *
 */
var JadeJueChart = /*#__PURE__*/function () {
  function JadeJueChart(iChartOption, chartInstance) {
    this.baseOption = {};
    this.iChartOption = {};
    this.baseOption = cloneDeep(BaseOption);
    // 组装 iChartOption, 补全默认值
    this.iChartOption = init(iChartOption);
    // 根据 iChartOption 组装 baseOption
    this.updateOption(chartInstance);
  }
  var _proto = JadeJueChart.prototype;
  _proto.updateOption = function updateOption(chartInstance) {
    var iChartOption = this.iChartOption;
    iChartOption.position = iChartOption.position || iChartOption.chartPosition;
    iChartOption.max = iChartOption.max || iChartOption.calibrationValue;
    // 装载除series之外的其他配置
    PolarCoordSys(this.baseOption, iChartOption, CHART_TYPE.JADGEJUE);
    // 配置玉玦图的标定值和两种data下不同的angleAxis.sum和angleAxis.max
    handleLabelFormatter(iChartOption, this.baseOption);
    // 配置玉玦图的seriesData数据（value,name,color）
    setSeriesData(iChartOption, this.baseOption);
    // 设置默认线宽为(8=>16)
    setbarWidth(iChartOption, this.baseOption);
    // 配置玉玦图的起点角度及文本位置
    setStartAngle(iChartOption, this.baseOption);
    // 为第一种数据类型单独配置legend.data和对应颜色
    handleLegendData(iChartOption, this.baseOption);
  };
  _proto.getOption = function getOption() {
    return this.baseOption;
  };
  _proto.setOption = function setOption() {};
  return JadeJueChart;
}();
_defineProperty(JadeJueChart, "name", CHART_TYPE.JADGEJUE);
export { JadeJueChart as default };
