// @opentiny/vue-image@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/image/vue';
import { defineComponent, Teleport, props, setup } from '@opentiny/vue-common';
import ImageViewer from '@opentiny/vue-image-viewer';
import { iconImageError } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, renderSlot, createElementVNode, createVNode, toDisplayString, mergeProps, createBlock, Teleport as Teleport$1, withCtx, createCommentVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    ImageViewer,
    Teleport,
    TinyIconImageError: iconImageError()
  },
  emits: ["load", "error", "delete", "change-index"],
  props: [].concat(props, ["src", "fit", "lazy", "scrollContainer", "previewSrcList", "zIndex", "keepStyle", "showIndex", "appendToBody"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-image"
};
var _hoisted_2 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "tiny-image__placeholder"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_3 = {
  class: "tiny-image__error"
};
var _hoisted_4 = {
  class: "tiny-image__error-text"
};
var _hoisted_5 = ["src"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_icon_image_error = resolveComponent("tiny-icon-image-error");
  var _component_image_viewer = resolveComponent("image-viewer");
  return openBlock(), createElementBlock("div", _hoisted_1, [_ctx.state.loading ? renderSlot(_ctx.$slots, "placeholder", {
    key: 0
  }, function() {
    return [_hoisted_2];
  }) : _ctx.state.error ? renderSlot(_ctx.$slots, "error", {
    key: 1
  }, function() {
    return [createElementVNode("div", _hoisted_3, [createVNode(_component_tiny_icon_image_error), createElementVNode(
      "div",
      _hoisted_4,
      toDisplayString(_ctx.t("ui.image.loadFail")),
      1
      /* TEXT */
    )])];
  }) : (openBlock(), createElementBlock("img", mergeProps({
    key: 2,
    class: "tiny-image__inner"
  }, _ctx.$attrs, {
    onClick: _cache[0] || (_cache[0] = function() {
      return _ctx.clickHandler && _ctx.clickHandler.apply(_ctx, arguments);
    }),
    src: _ctx.state.src,
    style: _ctx.state.getImageStyle,
    class: {
      "tiny-image__inner-center": _ctx.state.getAlignCenter,
      "tiny-image__preview": _ctx.state.getPreview
    }
  }), null, 16, _hoisted_5)), (openBlock(), createBlock(Teleport$1, {
    disabled: !_ctx.appendToBody,
    to: "body"
  }, [_ctx.state.getPreview && _ctx.state.showViewer ? (openBlock(), createBlock(_component_image_viewer, {
    key: 0,
    ref: "imageViewer",
    "z-index": _ctx.zIndex,
    "on-switch": _ctx.handleSwitch,
    "on-close": _ctx.closeViewer,
    "url-list": _ctx.previewSrcList,
    "keep-style": _ctx.keepStyle,
    "show-index": _ctx.showIndex
  }, {
    count: withCtx(function(slotScoped) {
      return [renderSlot(_ctx.$slots, "count", {
        index: slotScoped.index
      })];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["z-index", "on-switch", "on-close", "url-list", "keep-style", "show-index"])) : createCommentVNode("v-if", true)], 8, ["disabled"]))]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
