// @opentiny/vue-loading@3.20.0/lib/mobile.js
import { defineComponent, $prefix, $props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/loading/vue';
import loadingImg from '@opentiny/vue-theme-mobile/images/loading.png';
import '@opentiny/vue-theme-mobile/loading/index.css';
import { openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, normalizeStyle, createElementBlock, resolveDynamicComponent, toDisplayString, createCommentVNode, vShow } from 'vue';

var classes = {
  "loading-default": "m-0 top-0 right-0 left-0 bottom-0 block transition bg-color-bg-1 duration-1000",
  "loading-fullscreen": "fixed opacity-80",
  "loading-unfullscreen": "absolute",
  "loading-content": "absolute -translate-y-2/4  w-full  top-1/2 text-center  flex justify-center items-center",
  "loading-size-mini": "flex-row",
  "loading-size-unmini": "flex-col",
  "loading-unspinner": "animate-[spin_1.5s_linear_infinite] fill-current",
  "loading-unspinner-size-default": "h-10 w-10 stroke-2",
  "loading-unspinner-size-large": "h-24 w-24 stroke-2",
  "loading-unspinner-size-medium": "h-10 w-10 stroke-2",
  "loading-unspinner-size-small": "h-6 w-6 stroke-1",
  "loading-unspinner-size-mini": "h-3.5 w-3.5 stroke-1",
  "loading-unspinner-svg-circle": "stroke-color-brand tiny-loading-path",
  "loading-spinner": "text-sm fill-color-brand leading-none animate-[spin_2s_linear_infinite]",
  "loading-spinner-text": "tiny-tailwind-text block text-color-brand  text-xs leading-6",
  "loading-spinner-size-mini": "my-0 ml-2",
  "loading-spinner-size-unmini": "mb-1 mt-2.5"
};

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "Loading",
  emits: ["after-leave"],
  props: _extends({}, $props, {
    _constants: Object,
    loadingImg: {
      type: String,
      default: loadingImg
    },
    size: {
      type: String
    }
  }),
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = ["src"];
var _hoisted_2 = {
  key: 2,
  class: "tiny-mobile-loading__text"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  return openBlock(), createBlock(Transition, {
    name: "tiny-loading-fade",
    onAfterLeave: _ctx.handleAfterLeave,
    persisted: ""
  }, {
    default: withCtx(function() {
      var _ref;
      return [withDirectives(createElementVNode(
        "div",
        {
          class: normalizeClass(["tiny-mobile-loading", _ctx.state.customClass]),
          style: normalizeStyle("background-color:" + _ctx.state.background)
        },
        [createElementVNode(
          "div",
          {
            class: normalizeClass(["tiny-mobile-loading__body", (_ref = {}, _ref["tiny-mobile-loading-" + _ctx.size] = _ctx.size, _ref)])
          },
          [!_ctx.state.spinner ? (openBlock(), createElementBlock("img", {
            key: 0,
            class: "tiny-mobile-loading__icon",
            src: _ctx.state.loadingImg,
            style: normalizeStyle(_ctx.state.iconStyle)
          }, null, 12, _hoisted_1)) : (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.spinner), {
            key: 1,
            class: "tiny-mobile-loading__spinner"
          })), _ctx.state.text ? (openBlock(), createElementBlock(
            "span",
            _hoisted_2,
            toDisplayString(_ctx.state.text),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true)],
          2
          /* CLASS */
        )],
        6
        /* CLASS, STYLE */
      ), [[vShow, _ctx.state.visible]])];
    }),
    _: 1
    /* STABLE */
  }, 8, ["onAfterLeave"]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobile as default };
