// @opentiny/vue-loading@3.20.0/lib/mobile-first.js
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/loading/vue';
import { openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, normalizeStyle, createElementBlock, resolveDynamicComponent, toDisplayString, createCommentVNode, vShow } from 'vue';

var classes = {
  "loading-default": "m-0 top-0 right-0 left-0 bottom-0 block transition bg-color-bg-1 duration-1000",
  "loading-fullscreen": "fixed opacity-80",
  "loading-unfullscreen": "absolute",
  "loading-content": "absolute -translate-y-2/4  w-full  top-1/2 text-center  flex justify-center items-center",
  "loading-size-mini": "flex-row",
  "loading-size-unmini": "flex-col",
  "loading-unspinner": "animate-[spin_1.5s_linear_infinite] fill-current",
  "loading-unspinner-size-default": "h-10 w-10 stroke-2",
  "loading-unspinner-size-large": "h-24 w-24 stroke-2",
  "loading-unspinner-size-medium": "h-10 w-10 stroke-2",
  "loading-unspinner-size-small": "h-6 w-6 stroke-1",
  "loading-unspinner-size-mini": "h-3.5 w-3.5 stroke-1",
  "loading-unspinner-svg-circle": "stroke-color-brand tiny-loading-path",
  "loading-spinner": "text-sm fill-color-brand leading-none animate-[spin_2s_linear_infinite]",
  "loading-spinner-text": "tiny-tailwind-text block text-color-brand  text-xs leading-6",
  "loading-spinner-size-mini": "my-0 ml-2",
  "loading-spinner-size-unmini": "mb-1 mt-2.5"
};

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  inheritAttrs: false,
  name: $prefix + "Loading",
  emits: ["after-leave"],
  props: [].concat(props, ["_constants", "loadingImg", "size"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = ["src"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createBlock(Transition, {
    name: "tiny-loading-fade",
    onAfterLeave: _ctx.handleAfterLeave,
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          "data-tag": "tiny-loading",
          class: normalizeClass(_ctx.m(_ctx.state.customClass, _ctx.gcls("loading-default"), _ctx.gcls({
            "loading-fullscreen": _ctx.state.fullscreen
          }), _ctx.gcls({
            "loading-unfullscreen": !_ctx.state.fullscreen
          }))),
          style: normalizeStyle("background-color:" + _ctx.state.background)
        },
        [createElementVNode(
          "div",
          {
            "data-tag": "tiny-loading-body",
            class: normalizeClass(_ctx.m(_ctx.gcls("loading-content"), _ctx.gcls({
              "loading-size-mini": _ctx.size === "mini"
            }), _ctx.gcls({
              "loading-size-unmini": _ctx.size !== "mini"
            })))
          },
          [!_ctx.state.spinner ? (openBlock(), createElementBlock("img", {
            key: 0,
            "data-tag": "tiny-loading-icon",
            class: normalizeClass(_ctx.m(_ctx.gcls({
              "loading-unspinner-size-default": !_ctx.size || !["large", "medium", "small", "mini"].includes(_ctx.size)
            }), _ctx.gcls({
              "loading-unspinner-size-large": _ctx.size === "large"
            }), _ctx.gcls({
              "loading-unspinner-size-medium": _ctx.size === "medium"
            }), _ctx.gcls({
              "loading-unspinner-size-small": _ctx.size === "small"
            }), _ctx.gcls({
              "loading-unspinner-size-mini": _ctx.size === "mini"
            }))),
            src: _ctx.state.loadingImg,
            style: normalizeStyle(_ctx.state.iconStyle)
          }, null, 14, _hoisted_1)) : (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.spinner), {
            key: 1,
            "data-tag": "tiny-loading-spinner",
            class: normalizeClass(_ctx.m(_ctx.gcls("loading-spinner")))
          }, null, 8, ["class"])), _ctx.state.text ? (openBlock(), createElementBlock(
            "span",
            {
              key: 2,
              "data-tag": "tiny-loading-text",
              class: normalizeClass(_ctx.m(_ctx.gcls("loading-spinner-text"), _ctx.gcls({
                "loading-spinner-size-mini": _ctx.size === "mini"
              }), _ctx.gcls({
                "loading-spinner-size-unmini": _ctx.size !== "mini"
              })))
            },
            toDisplayString(_ctx.state.text),
            3
            /* TEXT, CLASS */
          )) : createCommentVNode("v-if", true)],
          2
          /* CLASS */
        )],
        6
        /* CLASS, STYLE */
      ), [[vShow, _ctx.state.visible]])];
    }),
    _: 1
    /* STABLE */
  }, 8, ["onAfterLeave"]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
