// @opentiny/huicharts@1.0.1/feature/linter/lints/modules/title.js
import Theme from '../../../token/index.js';
import { generatedMatchedUiRule } from './util.js';

/**
 * Copyright (c) 2024 - present OpenTiny HUICharts Authors.
 * Copyright (c) 2024 - present Huawei Cloud Computing Technologies Co., Ltd.
 *
 * Use of this source code is governed by an MIT-style license.
 *
 * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
 * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
 * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
 *
 */

function getTitleToken(echartsModule) {
  var token = Theme.getToken();
  var uiRule = {
    itemGap: token.titleItemGap,
    textStyle: {
      color: token.titleTextColor,
      fontSize: token.titleTextFontSize
    },
    subtextStyle: {
      color: token.titleSubTextColor,
      fontSize: token.titleSubtextFontSize
    }
  };
  var commonProperty = ['itemGap'];
  var propertyPermissions = [{
    name: 'text',
    extraCondition: function extraCondition(val) {
      return val || val === 0;
    },
    correlationProperty: ['textStyle']
  }, {
    name: 'subtext',
    extraCondition: function extraCondition(val) {
      return val || val === 0;
    },
    correlationProperty: ['subtextStyle']
  }];
  return generatedMatchedUiRule(echartsModule, uiRule, propertyPermissions, commonProperty);
}
export { getTitleToken as default };
