// @opentiny/vue-huicharts-graph@3.20.0/lib/index.js
import Core, { itemPoint, itemLabel, itemContent } from '@opentiny/vue-huicharts-core';
import { $prefix } from '@opentiny/vue-common';
import { openBlock, createElementBlock, createElementVNode, normalizeStyle, renderSlot } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var getTooltip = function getTooltip2() {
  return {
    trigger: "item",
    formatter: function formatter(item) {
      var tpl = [];
      var name = item.name, value = item.value, color = item.color, dataType = item.dataType;
      color && tpl.push(itemPoint(color));
      tpl.push(itemLabel("" + name, !value));
      if (dataType === "node") {
        value && tpl.push(itemContent(value) + "<br />");
      }
      return tpl.join("");
    }
  };
};
var getGraphSeries = function getGraphSeries2() {
  return [{
    type: "graph",
    label: {
      textBorderWidth: 1,
      color: "#fff",
      textBorderColor: "inherit"
    }
  }];
};
var graph = function graph2(columns, rows, settings, extra) {
  var tooltipVisible = extra.tooltipVisible;
  var tooltip = tooltipVisible && getTooltip();
  var series = getGraphSeries();
  return {
    tooltip,
    series,
    topHtml: tooltip.formatter
  };
};
var _sfc_main = {
  name: $prefix + "ChartGraph",
  mixins: [Core],
  data: function data() {
    return {
      iChartName: "GraphChart"
    };
  },
  methods: {
    updateChart: function updateChart(data2) {
      var _data$columns = data2.columns, columns = _data$columns === void 0 ? [] : _data$columns, _data$rows = data2.rows, rows = _data$rows === void 0 ? [] : _data$rows;
      var extra = {
        tooltipVisible: this.tooltipVisible,
        extend: this.extend
      };
      var option = graph(columns, rows, this.settings, extra);
      this.huiChartOption = _extends({}, option);
    }
  }
};
var _hoisted_1 = {
  class: "hui-chart chart-box"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "div",
    {
      ref: "chartRef",
      style: normalizeStyle({
        width: _ctx.width,
        height: _ctx.height
      })
    },
    null,
    4
    /* STYLE */
  ), renderSlot(_ctx.$slots, "default")]);
}
var HuiChartsGraph = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
HuiChartsGraph.install = function(Vue) {
  Vue.component(HuiChartsGraph.name, HuiChartsGraph);
};

export { HuiChartsGraph, HuiChartsGraph as default };
