// @opentiny/vue-filter-bar@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var FilterBar = defineComponent({
  name: $prefix + "FilterBar",
  props: _extends({}, $props, {
    modelValue: Number,
    data: Array
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
FilterBar.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
FilterBar.install = function(Vue) {
  Vue.component(FilterBar.name, FilterBar);
};
FilterBar.version = version;
export {
  FilterBar as default
};
