// @opentiny/vue-filter-bar@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/filter-bar/vue';
import { defineComponent, setup, $props } from '@opentiny/vue-common';
import { IconUnfilter, IconChevronDown, IconArrowBottom } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, Fragment, renderList, normalizeClass, withModifiers, renderSlot, createElementVNode, toDisplayString, createVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  components: {
    IconUnfilter: IconUnfilter(),
    IconChevronDown: IconChevronDown(),
    IconArrowBottom: IconArrowBottom()
  },
  emits: ["click", "update:modelValue"],
  props: _extends({}, $props, {
    modelValue: Number,
    data: Array
  }),
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-filter-bar",
  class: "relative text-sm text-color-text-primary w-full h-11 flex items-center justify-center px-4 border-b-0.5 border-color-border-separator"
};
var _hoisted_2 = ["onClick"];
var _hoisted_3 = ["title"];
var _hoisted_4 = {
  class: "h-full flex items-center justify-center truncate",
  "data-tag": "tiny-filter-bar-icon"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_IconArrowBottom = resolveComponent("IconArrowBottom");
  return openBlock(), createElementBlock("div", _hoisted_1, [(openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.data, function(item, index) {
      return openBlock(), createElementBlock("div", {
        "data-tag": "tiny-filter-bar-body",
        key: index,
        class: normalizeClass(["flex-1 flex items-center justify-center cursor-pointer overflow-hidden mx-3", {
          "text-color-brand": _ctx.modelValue === index
        }]),
        onClick: withModifiers(function($event) {
          return _ctx.click(index);
        }, ["stop"])
      }, [renderSlot(_ctx.$slots, "default", {
        data: item,
        index,
        active: _ctx.modelValue === index
      }, function() {
        return [createElementVNode("span", {
          "data-tag": "tiny-filter-bar-item",
          class: "truncate",
          title: item
        }, toDisplayString(item), 9, _hoisted_3), createVNode(_component_IconArrowBottom, {
          "custom-class": "flex-none w-3 h-3 ml-2 fill-current transition-transform duration-300",
          class: normalizeClass(_ctx.modelValue !== index ? "text-color-icon-placeholder rotate-0" : "rotate-180")
        }, null, 8, ["class"])];
      })], 10, _hoisted_2);
    }),
    128
    /* KEYED_FRAGMENT */
  )), createElementVNode("div", _hoisted_4, [renderSlot(_ctx.$slots, "icon")])]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
