// @opentiny/vue-card-group@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var cardGroupProps = _extends({}, $props, {
  modelValue: {},
  customClass: String,
  height: String,
  autoWidth: {
    type: Boolean,
    default: true
  },
  size: {
    type: String,
    default: "small"
  },
  status: {
    type: String,
    default: "default"
  },
  checkType: {
    type: String,
    default: ""
  },
  type: {
    type: String,
    default: "text"
  },
  disabled: {
    type: Boolean,
    default: false
  }
});
var CardGroup = defineComponent({
  name: $prefix + "CardGroup",
  props: cardGroupProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
CardGroup.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
CardGroup.install = function(Vue) {
  Vue.component(CardGroup.name, CardGroup);
};
CardGroup.version = version;
export {
  CardGroup as default
};
