// @opentiny/vue-fluent-editor@3.20.2/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/fluent-editor/vue';
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import FluentEditor from '@opentiny/fluent-editor';
import '@opentiny/fluent-editor/style.css';
import { IconEditorAlignCenter, IconEditorAlignLeft, IconEditorAlignRight, IconEditorBackground, IconEditorBold, IconEditorCode, IconEditorDeleteline, IconEditorEraser, IconEditorItalic, IconEditorLeftBorder, IconEditorList, IconEditorListDot, IconEditorListNum, IconEditorMenuLeft, IconEditorMenuRight, IconEditorQuote, IconEditorRedo, IconEditorRightBorder, IconEditorSub, IconEditorSubtitle, IconEditorSuper, IconEditorTable, IconEditorTextcolor, IconEditorTitle, IconEditorUnderline, IconEditorUndo, IconEditorVideo, IconLink, IconPicture, IconCloudUpload, IconFullscreen } from '@opentiny/vue-icon';
import ImageViewer from '@opentiny/vue-image-viewer';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, normalizeStyle, createElementVNode, createBlock, normalizeProps, mergeProps, createCommentVNode, Fragment, renderList, resolveDynamicComponent } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "FluentEditor",
  components: {
    TinyImageViewer: ImageViewer,
    IconEditorAlignCenter: IconEditorAlignCenter(),
    IconEditorAlignLeft: IconEditorAlignLeft(),
    IconEditorAlignRight: IconEditorAlignRight(),
    IconEditorBackground: IconEditorBackground(),
    IconEditorBold: IconEditorBold(),
    IconEditorCode: IconEditorCode(),
    IconEditorDeleteline: IconEditorDeleteline(),
    IconEditorEraser: IconEditorEraser(),
    IconEditorItalic: IconEditorItalic(),
    IconEditorLeftBorder: IconEditorLeftBorder(),
    IconEditorList: IconEditorList(),
    IconEditorListDot: IconEditorListDot(),
    IconEditorListNum: IconEditorListNum(),
    IconEditorMenuLeft: IconEditorMenuLeft(),
    IconEditorMenuRight: IconEditorMenuRight(),
    IconEditorQuote: IconEditorQuote(),
    IconEditorRedo: IconEditorRedo(),
    IconEditorRightBorder: IconEditorRightBorder(),
    IconEditorSub: IconEditorSub(),
    IconEditorSubtitle: IconEditorSubtitle(),
    IconEditorSuper: IconEditorSuper(),
    IconEditorTable: IconEditorTable(),
    IconEditorTextcolor: IconEditorTextcolor(),
    IconEditorTitle: IconEditorTitle(),
    IconEditorUnderline: IconEditorUnderline(),
    IconEditorUndo: IconEditorUndo(),
    IconEditorVideo: IconEditorVideo(),
    IconLink: IconLink(),
    IconPicture: IconPicture(),
    IconCloudUpload: IconCloudUpload(),
    IconFullscreen: IconFullscreen()
  },
  emits: ["ready", "blur", "focus", "update:modelValue", "change"],
  props: [].concat(props, ["_constants", "content", "disabled", "fileUpload", "globalOptions", "imageUpload", "mentionObj", "modelValue", "options", "picPreview", "dataType", "dataUpgrade", "zIndex", "imagePasteFailCallback"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      extendOptions: {
        FluentEditor
      }
    });
  }
});
var _hoisted_1 = {
  class: "toolbar-icon",
  ref: "editor-toolbar-icon"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_image_viewer = resolveComponent("tiny-image-viewer");
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["tiny-editor-wrapper", {
        "fullscreen": _ctx.state.isFullscreen
      }]),
      style: normalizeStyle({
        "z-index": _ctx.state.zIndex
      })
    },
    [createElementVNode(
      "div",
      {
        ref: "editor",
        id: "editor",
        class: "tiny-editor",
        onDblclick: _cache[0] || (_cache[0] = function() {
          return _ctx.handleDblclick && _ctx.handleDblclick.apply(_ctx, arguments);
        })
      },
      null,
      544
      /* NEED_HYDRATION, NEED_PATCH */
    ), _ctx.state.showPreview ? (openBlock(), createBlock(
      _component_tiny_image_viewer,
      normalizeProps(mergeProps({
        key: 0
      }, _ctx.state.previewOptions)),
      null,
      16
      /* FULL_PROPS */
    )) : createCommentVNode("v-if", true), createElementVNode(
      "div",
      _hoisted_1,
      [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.icons, function(icon) {
          return openBlock(), createElementBlock("div", {
            key: icon,
            ref_for: true,
            ref: icon
          }, [(openBlock(), createBlock(resolveDynamicComponent(icon)))]);
        }),
        128
        /* KEYED_FRAGMENT */
      ))],
      512
      /* NEED_PATCH */
    )],
    6
    /* CLASS, STYLE */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
