// @opentiny/vue-action-sheet@3.20.0/lib/mobile.js
import { renderless, api } from '@opentiny/vue-renderless/action-sheet/vue';
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import '@opentiny/vue-theme-mobile/action-sheet/index.css';
import BScroll from '@better-scroll/core';
import { withDirectives, openBlock, createElementBlock, normalizeStyle, createCommentVNode, createElementVNode, normalizeClass, Fragment, renderList, renderSlot, createTextVNode, toDisplayString, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "ActionSheet",
  props: [].concat(props, ["menus", "modelValue", "visible", "ellipsis", "contentPosition", "contentStyle", "height", "valueField", "textField", "contentClass"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      extendOptions: {
        BScroll
      }
    });
  }
});
var _hoisted_1 = ["onClick"];
var _hoisted_2 = {
  key: 1,
  class: "tiny-mobile-action-sheet__action"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return withDirectives((openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-mobile-action-sheet",
      onClick: _cache[1] || (_cache[1] = function() {
        return _ctx.hide && _ctx.hide.apply(_ctx, arguments);
      })
    },
    [!_ctx.contentPosition ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        class: "tiny-mobile-action-sheet__mask",
        style: normalizeStyle(_ctx.state.sheetMaskStyle)
      },
      null,
      4
      /* STYLE */
    )) : createCommentVNode("v-if", true), createElementVNode(
      "div",
      {
        class: normalizeClass(["tiny-mobile-action-sheet__content", _ctx.state.toggle ? "is-toggle" : "", _ctx.contentPosition ? "" : "is-not-content"]),
        style: normalizeStyle([_ctx.state.sheetContentStyle]),
        ref: "scrollMenu"
      },
      [createElementVNode(
        "div",
        {
          class: normalizeClass(["tiny-mobile-action-sheet__menu", _ctx.ellipsis ? "is-ellipsis" : ""])
        },
        [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.menus, function(item, index) {
            return openBlock(), createElementBlock("div", {
              class: normalizeClass(["tiny-mobile-action-sheet__item", item.warn ? "is-warn" : "", item[_ctx.valueField] === _ctx.modelValue ? "is-active" : ""]),
              style: normalizeStyle(_ctx.state.contentStyle),
              key: index,
              onClick: function onClick($event) {
                return _ctx.selectOption(item, index);
              }
            }, [renderSlot(_ctx.$slots, "item", {
              item
            }, function() {
              return [createTextVNode(
                toDisplayString(item[_ctx.textField]),
                1
                /* TEXT */
              )];
            })], 14, _hoisted_1);
          }),
          128
          /* KEYED_FRAGMENT */
        ))],
        2
        /* CLASS */
      )],
      6
      /* CLASS, STYLE */
    ), _ctx.contentPosition ? (openBlock(), createElementBlock("div", _hoisted_2, [renderSlot(_ctx.$slots, "action", {}, function() {
      return [createElementVNode(
        "div",
        {
          class: "tiny-mobile-action-sheet__cancel",
          onClick: _cache[0] || (_cache[0] = function() {
            return _ctx.hide && _ctx.hide.apply(_ctx, arguments);
          })
        },
        toDisplayString(_ctx.t("ui.actionSheet.cancel")),
        1
        /* TEXT */
      )];
    })])) : createCommentVNode("v-if", true)],
    512
    /* NEED_PATCH */
  )), [[vShow, _ctx.visible]]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobile as default };
