// @opentiny/vue-locale@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t2 = arguments[e];
      for (var r in t2)
        ({}).hasOwnProperty.call(t2, r) && (n[r] = t2[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { extend as extend$2 } from "@opentiny/vue-renderless/common/object";
import { formatNumber as _formatNumber, recoverNumber as _recoverNumber } from "@opentiny/vue-renderless/common/decimal";
import { toDate, getDateWithNewTimezone as _getDateWithNewTimezone } from "@opentiny/vue-renderless/common/date";
import { isPlainObject, isNumber, isNull, isDate } from "@opentiny/vue-renderless/common/type";
import { formatDate as _formatDate } from "@opentiny/vue-renderless/common/deps/date-util";
var zhCN$1 = {
  "en-US": "英语",
  "zh-CN": "中文",
  "zh-TW": "中国台湾",
  hello: "你好 {name}",
  code: "zh-CN",
  yes: "是",
  no: "否",
  ui: {
    input: {
      close: "关闭",
      more: "更多",
      detail: "详细信息"
    },
    numeric: {
      equalTo: "等于",
      notEqualTo: "不等于",
      moreThan: "大于",
      moreThanOrEqualTo: "大于等于",
      lessThan: "小于",
      lessThanOrEqualTo: "小于等于",
      empty: "为空",
      nonEmpty: "不为空"
    },
    queryBuilder: {
      addItem: "新增条件",
      addGroup: "新增子条件组",
      removeGroup: "移除条件组"
    },
    wizard: {
      previousStep: "上一步",
      nextStep: "下一步",
      save: "保存",
      submit: "提交"
    },
    linkMenu: {
      title: "消息",
      placeholder: "请输入关键字过滤...",
      sure: "确定",
      cancel: "取消"
    },
    todoList: {
      add: "提交",
      placeholder: "请输入内容..."
    },
    alert: {
      error: "错误",
      info: "消息",
      success: "成功",
      title: "消息提示",
      warning: "警告"
    },
    amount: {
      currency: "币种",
      amount: "金额",
      date: "日期",
      equalTo: "等于",
      notEqualTo: "不等于",
      moreThan: "大于",
      moreThanOrEqualTo: "大于等于",
      lessThan: "小于",
      lessThanOrEqualTo: "小于等于",
      empty: "为空",
      nonEmpty: "不为空"
    },
    actionMenu: {
      moreText: "更多"
    },
    base: {
      all: "全部",
      cancel: "取消",
      confirm: "确定",
      delete: "删除",
      edit: "编辑",
      more: "更多",
      reset: "重置",
      clear: "清空",
      comma: "，"
    },
    button: {
      cancel: "取消",
      confirm: "确定"
    },
    buttonGroup: {
      noData: "暂无数据"
    },
    buttonMessage: {
      cancel: "取消",
      confirm: "确定"
    },
    cell: {
      placeholder: "请选择"
    },
    cascader: {
      noMatch: "无匹配数据",
      loading: "加载中",
      placeholder: "请选择",
      noData: "暂无数据"
    },
    chart: {
      auxiliary: "辅助",
      emptyText: "暂无数据",
      kName: "日K",
      other: "其他",
      summation: "总量",
      total: "总计",
      value: "数值"
    },
    colorSelectPanel: {
      confirm: "选择",
      cancel: "取消",
      predefine: "预定义颜色",
      history: "历史记录",
      empty: "暂无"
    },
    crop: {
      choose: "选择图片",
      zoomOut: "缩小10%",
      zoomIn: "放大10%",
      rotate_45: "逆时针旋转45°",
      rotate45: "顺时针旋转45°",
      closeCropArea: "隐藏选区",
      reset: "重置视图",
      closeCrop: "退出裁剪",
      cropImage: "选择区域"
    },
    datepicker: {
      clear: "清空",
      cancel: "取消",
      endDate: "结束日期",
      confirm: "确定",
      month: "月",
      endTime: "结束时间",
      month2: "2 月",
      month1: "1 月",
      month4: "4 月",
      month3: "3 月",
      month6: "6 月",
      month5: "5 月",
      month8: "8 月",
      month7: "7 月",
      month10: "10 月",
      month9: "9 月",
      month12: "12 月",
      month11: "11 月",
      months: {
        feb: "二月",
        jan: "一月",
        apr: "四月",
        mar: "三月",
        jun: "六月",
        may: "五月",
        aug: "八月",
        jul: "七月",
        oct: "十月",
        sep: "九月",
        dec: "十二月",
        nov: "十一月"
      },
      nextYear: "后一年",
      nextMonth: "下个月",
      prevMonth: "上个月",
      now: "此刻",
      selectDate: "选择日期",
      prevYear: "前一年",
      startDate: "开始日期",
      selectTime: "选择时间",
      today: "今天",
      currentMonth: "本月",
      startTime: "开始时间",
      week: "周次",
      weeks: {
        mon: "一",
        sun: "日",
        wed: "三",
        tue: "二",
        fri: "五",
        thu: "四",
        sat: "六"
      },
      timezone: "选择时区",
      year: "年",
      hour: "时",
      minute: "分",
      second: "秒",
      to: "至",
      yearMonth: "{year}年{month}月",
      yearMonthDay: "{year}年{month}月{day}日"
    },
    richTextEditor: {
      bold: "加粗",
      italic: "斜体",
      link: "链接",
      unlink: "移除链接",
      highlight: "高亮",
      underline: "下划线",
      strike: "中划线",
      subscript: "下标",
      superscript: "上标",
      code: "代码",
      unorderedlist: "无序列表",
      orderedlist: "有序列表",
      taskList: "任务列表",
      quote: "引用",
      codeBlock: "代码块",
      formatClear: "清除标记",
      nodeDelete: "删除节点",
      undo: "回退",
      redo: "前进",
      left: "左对齐",
      center: "居中",
      right: "右对齐",
      fontSize: "字号",
      lineHeight: "行高",
      hBox: "段落标题",
      img: "图片",
      color: "颜色",
      table: "表格",
      backgroundColor: "文字背景色",
      localResources: "本地资源",
      resourceLink: "资源链接"
    },
    calendar: {
      showType: {
        year: "年"
      }
    },
    dept: {
      code: "编码",
      company: "公司",
      dept1: "一级部门",
      dept2: "二级部门",
      dept3: "三级部门",
      dept4: "四级部门",
      dept5: "五级部门",
      dept6: "六级部门",
      dept7: "七级部门",
      dept8: "八级部门",
      input: "可输入部门编码或名称",
      name: "名称",
      search: "辅助查询",
      selected: "已选"
    },
    dialogBox: {
      confirm: "确定",
      cancel: "取消"
    },
    load: {
      dot: "加载中"
    },
    exception: {
      build: "模块正在建设中",
      busy: "系统繁忙，请稍等一下",
      noperm: "茫茫大海，找不到页面",
      weaknet: "网络不给力",
      pcview: "请到PC上查看文件",
      nodata: "休息一下",
      create: "创建",
      provide: "TINY 开发团队提供",
      nodatamf: "暂无数据",
      nopermmf: "无访问权限",
      weaknetmf: "网络异常",
      noresult: "无相关搜索结果",
      nonews: "暂无最新消息",
      pagenoperm: "403:无访问权限",
      pageweaknet: "网络异常",
      pagenothing: "404:你访问的页面不存在",
      pageservererror: "500:服务器异常"
    },
    fileUpload: {
      largefile: "文件过大，将会分片上传，请耐心等待!",
      folder: "文件所在文件夹层数已超过 5 层，将不会上传该文件",
      init: "服务报错，请重试",
      token: "请先进行 EDM 鉴权，获取 token",
      exceed: "文件大小超过限制（{maxSize}）",
      largeFile: "文件大小超出限制 2G ！！",
      fileSize: "文件大小低于限制（{minSize}{sizeUnit}）",
      deleteTip: "按 delete 键可删除",
      downloadFile: "下载文件",
      previewFile: "预览文件",
      updateFile: "更新文件",
      reUploadFile: "重新上传",
      cancelFile: "取消上传",
      deleteFile: "删除文件",
      empty: "是空文件！",
      kiaScanTip: "抱歉，从公网接入下载文档，需要通过KIA检测；当前文档正在KIA检测中，请稍后几分钟后再下载！",
      fileNameExceeds: "超过255个字符，请修改文件名。",
      fileName: "该文件名",
      calcHash: "文档正在计算加密中",
      uploadFile: "文件上传",
      downloadAll: "全部下载",
      onlySupport: "支持{type}格式文件",
      fileNotLessThan: "单个文件不能小于",
      fileNotMoreThan: "单个文件不能超过",
      fileSizeRange: "单个文件大小需在{moreThan}~{lessThan}之间",
      notSupport: "格式（.{format}）暂不支持",
      notSupportNoSuffix: "暂不支持无后缀文件",
      notSupportSpecialCharacters: "文件名包含特殊字符，请重命名后上传",
      attachment: "附件",
      uploadList: "上传列表",
      numberExceed: "批量上传个数超过限制（{number}）",
      numberLimit: "最多上传{number}个文件",
      encryptDialogTitle: "水印及加密设置",
      addWatermark: "添加水印",
      encrypted: "加密",
      docPreview: "文档预览",
      networkError: "网络出错",
      pictureNetworkError: "网络出错，上传失败",
      reUploadTip: "{number}个文件上传失败！"
    },
    uploadList: {
      pictureUploading: "图片上传中",
      uploadFailed: "上传失败",
      uploading: "上传中",
      download: "下载",
      reUpload: "重新上传",
      delete: "删除",
      noAttachments: "暂无附件",
      cancel: "取消",
      preview: "预览",
      releaseAndUpload: "释放鼠标，上传文件",
      dragOrClickImport: "将文件拖到此处，或点击导入",
      shoot: "拍摄",
      selectFromAlbum: "从相册选择",
      uploadFailedAndReupload: "上传失败，点击重新上传"
    },
    upload: {
      addPicture: "添加图片",
      addAudio: "添加音频",
      addVideo: "添加视频"
    },
    grid: {
      dataUnchanged: "数据未改动！",
      deleteSelectRecord: "您确定要删除所选记录吗？",
      emptyText: "暂无数据",
      error: {
        delRevert: "方法 revert 已废弃，请使用 revertData",
        groupFixed: "如果使用分组表头，固定列必须在左右两侧",
        notDelete: "Delete 方法不存在",
        notQuery: "query 方法不存在",
        notResizable: "横向虚拟滚动不支持 resizable",
        notSave: "save 方法不存在",
        reqModule: "缺少 {{name}} 模块",
        rowIdEmpty: "参数 row-id 不允许为空",
        scrollYHeight: "启用虚拟滚动必须要设置 height 或 max-height",
        toolbarId: "工具栏需要设置唯一 id",
        treeFixedExpand: "树结构的固定列与展开行功能有冲突",
        treeInsert: "树结构不支持 insert 操作",
        treeRemove: "树结构不支持 remove 操作",
        unableInsert: "无法插入到指定位置",
        notAllowDragSelf: "不允许自己给自己拖动",
        notAllowDragFixed: "固定列不允许拖动",
        remoteMethod: "个性化模板管理远端存储需要设置 multipleHistory.remoteMethod",
        remoteSelectedMethod: "个性化模板管理远端存储需要设置 multipleHistory.remoteSelectedMethod",
        chainCallError: "列的默认插槽中存在语法错误，请检查。",
        renderParamError: "期望配置一个生成 VNode 的渲染方法。",
        classComponentError: "类组件渲染出错。",
        groupColumnFixedError: "同一个分组内不能设置不同的固定类型。",
        missingValueFormat: "渲染器无法格式化日期字符串，需要提供 valueFormat 源日期格式配置。",
        clipboardWriteError: "剪切板写入错误"
      },
      filter: {
        allSelect: "(全选)",
        endDate: "结束日期",
        startDate: "开始日期",
        dateTips: "请至少输入一个日期",
        clear: "清除当前列筛选",
        clearAll: "清除所有列筛选",
        confirmFilter: "筛选",
        empty: "为空",
        emptyText: "暂无数据",
        equal: "等于",
        include: "包含",
        prefix: "开头是",
        resetFilter: "重置",
        unempty: "不为空"
      },
      individuation: {
        cancelBtn: "取消",
        colConfigs: {
          visible: "显示",
          invisible: "隐藏",
          asc: "正序",
          desc: "倒序",
          unsorted: "未排序",
          frozenLeft: "左冻结",
          frozenRight: "右冻结",
          unfrozen: "未冻结",
          unfreeze: "取消冻结",
          unsort: "取消排序"
        },
        toolbar: {
          set: "设置",
          selected: "已选",
          freeze: "冻结",
          sort: "排序",
          clear: "清空",
          search: "搜索",
          all: "全选"
        },
        columnSet: "列设置",
        overwriteSave: "覆盖保存",
        saveAs: "另存为",
        saveTemplate: "存模板",
        selectTemplate: "选择模板",
        hideMsg: "至少保留一列显示",
        maxFreezeNumMsg: "冻结列不可超过6项",
        defaultTemplateName: "请输入名称，如未填写由系统按时间生成",
        reserveTemplateName: "如未填写名称将保留之前的名称",
        resetBtn: "重置",
        saveBtn: "确定",
        hideAll: "全部隐藏",
        showAll: "全部显示",
        tabs: {
          base: {
            title: "基础设置",
            tips: "点击图标按钮设置个性化"
          },
          other: {
            title: "其他设置",
            tips: "设置服务器排序或客户端排序、每页条数大小。",
            sortType: "排序类型",
            currPageSort: "当前页数据排序",
            allDataSort: "所有数据排序",
            pageSize: "每页条数"
          }
        },
        title: "个性化设置",
        switchtitle: "模板管理",
        switchsave: "保存配置",
        switchlabel: "配置列表：",
        switchapply: "使用",
        switchedit: "编辑",
        switchdel: "删除",
        switchconfirm: "确定",
        switchdelconfirm: "删除确认",
        switchonlytemp: "保存模板",
        switchtempapply: "保存并使用模板",
        switchtempoverwrite: "覆盖并使用模板",
        switchdelcon: "确定要删除这个模板？",
        switchdelyes: "确定",
        switchdelno: "取消",
        switchapplycon: "确定要使用这个模板？"
      },
      removeSelectRecord: "您确定要移除所选记录吗？",
      saveSuccess: "保存成功",
      selectOneRecord: "请至少选择一条记录！",
      isSaveMsg: "有修改的数据，是否要保存？"
    },
    hrapprover: {
      approver: "权签人",
      noselected: "没有选择权签人",
      noapprover: "没有权签人",
      remark: "备注"
    },
    imageViewer: {
      loadErrorAlt: "加载失败",
      save: "保存图片",
      del: "删除图片",
      thumbnail: "缩略图",
      menu: "目录",
      hide: "隐藏侧边栏",
      show: "显示侧边栏"
    },
    navMenu: {
      moreText: "更多"
    },
    logout: {
      in: "登录",
      out: "注销"
    },
    page: {
      goto: "前往",
      item: "条",
      next: "下一页",
      page: "条/页",
      pageClassifier: "页",
      pagesize: "条/页",
      prev: "上一页",
      total: "共",
      totals: "总条数：",
      jump: "跳至",
      hundredThousand: "10万+",
      million: "100万+",
      tenMillion: "1千万+",
      loadingTotals: "加载总条数…"
    },
    popeditor: {
      cancel: "取 消",
      confirm: "确 认",
      historyLists: "历史数据列表",
      reset: "重 置",
      search: "查 询",
      selectionLists: "选择数据列表",
      sourceLists: "所有数据列表",
      title: "选择",
      filterNode: "输入内容进行筛选"
    },
    popupload: {
      fileName: "文件名",
      fileSize: "文件大小",
      fileStatus: "文件状态",
      uploadError: "上传失败",
      dialogTitle: "文件上传",
      cancelButtonText: "取消",
      tipsFileText: "上传提示",
      saveButtonText: "开始上传",
      uploadSuccess: "上传成功！",
      uploadButtonText: "选择文件",
      uploadsButtonText: "选择批量文件",
      errorTypeTips: "上传文件类型不匹配",
      errorNumTips: "上传文件数量超出限制,已取消该操作",
      errorSizeTips: "上传文件大小超出限制",
      confirmDeleteTips: "确定要删除该文件吗？",
      delete: "删除",
      waitUpload: "等待上传",
      operation: "操作",
      success: "上传成功",
      listTip: "共{0}条数据：",
      errorListTip: "其中{0}条出错，请修改后重试",
      limitUploadFileNumber: "上传文件数限制为",
      limitUploadFileType: "上传文件类型限制为",
      limitUploadFileSize: "上传文件大小不超过"
    },
    rate: {
      level: {
        average: "一般",
        excellent: "很好",
        fair: "差",
        good: "好",
        poor: "很差"
      }
    },
    select: {
      loading: "加载中",
      noMatch: "无匹配数据",
      noData: "暂无相关数据",
      placeholder: "请选择",
      pleaseSearch: "请搜索",
      search: "搜索",
      selected: "已选",
      selectedNum: "已选 {num} 个",
      noSearchData: "无相关搜索结果，请重新输入",
      add: "新增",
      collapse: "收起"
    },
    search: {
      placeholder: "搜索"
    },
    signature: {
      confirm: "确认",
      rewrite: "重写",
      cancel: "取消",
      tips: "请手写签名",
      resign: "重新签名",
      placeholder: "请在此签名（必填）"
    },
    tabs: {
      moreItem: "更多"
    },
    tag: {
      add: "添加"
    },
    toggleMenu: {
      placeholder: "请输入内容进行筛选"
    },
    treeMenu: {
      placeholder: "输入关键字搜索"
    },
    transfer: {
      filterPlaceholder: "请输入搜索内容",
      hasCheckedFormat: "已选 {checked}/{total} 项",
      noCheckedFormat: "共 {total} 项",
      noData: "无数据",
      noMatch: "无匹配数据",
      titles: ["列表 1", "列表 2"]
    },
    tree: {
      loading: "加载中",
      emptyText: "暂无数据",
      switchText: "同时勾选下级",
      edit: "编辑",
      delete: "删除",
      addChild: "新增下级",
      newNodeTitle: "新增下级",
      deleteTip1: "删除后数据不可恢复，确定删除吗？",
      deleteTip2: "该节点存在下级节点，是否保留下级节点数据？",
      deleteTip3: "保留下级节点数据"
    },
    usercard: {
      address: "地址",
      collapse: "收起",
      email: "邮箱",
      empno: "工号",
      expand: "展开",
      fax: "传真",
      internal: "内线",
      manager: "主管",
      mobile: "手机",
      other: "其他",
      phone: "固定电话",
      timezone: "时区",
      travelcode: "出差联系信息",
      viop: "VIOP"
    },
    richText: {
      bold: "加粗",
      italic: "倾斜",
      underline: "下划线",
      header: "段落格式",
      strike: "删除线",
      blockquote: "块引用",
      codeBlock: "插入代码段",
      size: "字体大小",
      listOrdered: "编号列表",
      listBullet: "项目列表",
      header1: "h1",
      header2: "h2",
      align: "对齐方式",
      color: "字体颜色",
      background: "背景颜色",
      image: "图像",
      video: "视频",
      link: "添加链接",
      formula: "插入公式",
      clean: "清除格式",
      indent1: "向左缩进",
      indent2: "向右缩进",
      pickerLabel: "标题大小",
      headerPicker1: "标题一",
      headerPicker2: "标题二",
      headerPicker3: "标题三",
      headerPicker4: "标题四",
      headerPicker5: "标题五",
      headerPicker6: "标题六",
      normal: "标准",
      sizeSmall: "小号",
      sizeLarge: "大号",
      sizeHuge: "超大号",
      alignPicker1: "居左对齐",
      alignPicker2: "居中对齐",
      alignPicker3: "居右对齐",
      alignPicker4: "两端对齐",
      subScript: "下标",
      superScript: "上标",
      directionRTL: "从右到左",
      font: "字体",
      file: "文件",
      betterTable: "表格",
      fullscreen: "全屏",
      insertColumnRight: "右插入列",
      insertColumnLeft: "左插入列",
      insertRowUp: "上插入行",
      insertRowDown: "下插入行",
      mergeCells: "合并单元格",
      unmergeCells: "拆分单元格",
      deleteColumn: "删除当前列",
      deleteRow: "删除当前行",
      deleteTable: "删除表格",
      colorPicker: "背景颜色",
      placeholder: "在此处插入文本...",
      maxLength: "文本长度超过限制，支持的最大长度是 "
    },
    fluentEditor: {
      undo: "撤销",
      redo: "重做",
      lineheight: "行高"
    },
    steps: {
      done: "已完成",
      doing: "进行中",
      wait: "等待中"
    },
    actionSheet: {
      cancel: "取消"
    },
    image: {
      loadFail: "加载失败"
    },
    miniPicker: {
      cancel: "取消",
      confirm: "确定"
    },
    pullRefresh: {
      pullingDown: "下拉即可刷新",
      pullingUp: "上拉即可刷新",
      pulling: "下拉即可刷新",
      loosing: "释放即可刷新",
      success: "刷新成功",
      failed: "刷新失败",
      noMore: "没有更多了"
    },
    currency: {
      defaultCurrency: "默认币种",
      setDefault: "设为默认",
      chooseCurrency: "选择币种"
    },
    calendarBar: {
      week: {
        0: "日",
        1: "一",
        2: "二",
        3: "三",
        4: "四",
        5: "五",
        6: "六"
      },
      year: "%s年",
      yearMonth: "%y年%m月",
      month: {
        1: "1月",
        2: "2月",
        3: "3月",
        4: "4月",
        5: "5月",
        6: "6月",
        7: "7月",
        8: "8月",
        9: "9月",
        10: "10月",
        11: "11月",
        12: "12月"
      },
      monthAbbr: {
        1: "1",
        2: "2",
        3: "3",
        4: "4",
        5: "5",
        6: "6",
        7: "7",
        8: "8",
        9: "9",
        10: "10",
        11: "11",
        12: "12"
      }
    },
    calendarView: {
      week: {
        0: "日",
        1: "一",
        2: "二",
        3: "三",
        4: "四",
        5: "五",
        6: "六"
      },
      weekDays: {
        0: "周日",
        1: "周一",
        2: "周二",
        3: "周三",
        4: "周四",
        5: "周五",
        6: "周六"
      },
      backToday: "回今天",
      new: "新增",
      noSchedule: "暂无日程",
      year: "年",
      month: "月",
      dateFormat: "yyyy 年 MM 月"
    },
    selectedBox: {
      select: "已选（%s）",
      allSelect: "已全选（%s）",
      clear: "清空",
      noData: "暂无数据"
    },
    record: {
      record: "录音",
      cancel: "取消",
      confirm: "确定",
      clickToStartRecording: "点击开始录音",
      clickToResumeRecording: "点击继续录音"
    },
    dialogSelect: {
      treeSearch: "请输入关键字并回车"
    },
    loadList: {
      errorText: "出错了",
      loadingText: "加载中...",
      finishedText: "没有更多了"
    }
  },
  validation: {
    array: {
      len: "%s 的长度必须为 %s",
      min: "%s 长度不能小于 %s",
      max: "%s 的长度不能大于 %s",
      range: "%s 的长度必须介于 %s 和 %s 之间"
    },
    date: {
      format: "%s 日期 %s 对于格式 %s 无效",
      invalid: "%s 日期 %s 无效",
      parse: "无法分析 %s 日期， %s 无效"
    },
    default: "%s 字段校验错误",
    enum: "%s 必须是 %s 中的一个",
    number: {
      len: "%s 必须等于 %s",
      min: "%s 不能小于 %s",
      max: "%s 不能大于 %s",
      range: "%s 必须介于 %s 和 %s 之间"
    },
    pattern: {
      mismatch: "%s 值%s 与模式 %s 不匹配"
    },
    required: "必填",
    string: {
      len: "%s 必须是 %s 个字符",
      min: "%s 必须至少为 %s 个字符",
      max: "%s 不能大于 %s 个字符",
      range: "%s 必须介于 %s 和 %s 个字符之间"
    },
    types: {
      acceptFile: "只接受文件",
      acceptImg: "只接受图片格式",
      array: "非法数组",
      boolean: "非法布尔值",
      date: "不符合规则的日期格式",
      dateTime: "不符合规则的日期时间格式",
      dateYM: "不符合规则的日期格式(yyyy-mm)",
      dateYMD: "不符合规则的日期格式(yyyy-MM-dd)",
      digits: "非法纯数字",
      email: "非法邮件地址",
      fileSize: "文件大小的格式不正确,应如 3kb",
      float: "非法浮点数",
      hex: "非法十六进制",
      integer: "非法整数",
      longDateTime: "不符合规则的长日期格式",
      method: "必须是函数（Function）",
      number: "非法数字",
      object: "非法对象",
      regexp: "非法正则表达式",
      specialch: "只能包含数字、字母、下划线、横杠、点号",
      specialch2: "只能包含数字、字母、下划线、横杠",
      speczh: "只能包含数字、字母、下划线、汉",
      string: "非法字符串",
      time: "不符合规则的时间格式",
      url: "非法 URL 地址",
      version: "非法版本格式"
    },
    whitespace: "%s 不能为空"
  }
};
var enUS$1 = {
  "en-US": "English",
  "zh-CN": "Chinese",
  "zh-TW": "中国台湾",
  hello: "Hello {name}",
  code: "en-US",
  yes: "Yes",
  no: "No",
  ui: {
    input: {
      close: "Close",
      more: "More",
      detail: "Detailed Info"
    },
    numeric: {
      equalTo: "Equal to",
      notEqualTo: "Not equal to",
      moreThan: "Greater than",
      moreThanOrEqualTo: "Greater than or equal to",
      lessThan: "Less than",
      lessThanOrEqualTo: "Less than or equal to",
      empty: "Null",
      nonEmpty: "Not null"
    },
    queryBuilder: {
      addItem: "Add Rule",
      addGroup: "Add Rule Group",
      removeGroup: "Del Rule Group"
    },
    wizard: {
      previousStep: "Previous",
      nextStep: "Next",
      save: "Save",
      submit: "Submit"
    },
    linkMenu: {
      title: "",
      placeholder: "Enter a keyword.",
      sure: "OK",
      cancel: "Cancel"
    },
    todoList: {
      add: "Add",
      placeholder: "please input your todo things"
    },
    alert: {
      error: "Error",
      info: "Information",
      success: "Success",
      title: "Information",
      warning: "Warning"
    },
    amount: {
      currency: "Currency",
      amount: "Amount",
      date: "Date",
      equalTo: "Equal to",
      notEqualTo: "Not equal to",
      moreThan: "Greater than",
      moreThanOrEqualTo: "Greater than or equal to",
      lessThan: "Less than",
      lessThanOrEqualTo: "Less than or equal to",
      empty: "Null",
      nonEmpty: "Not null"
    },
    actionMenu: {
      moreText: "More"
    },
    base: {
      all: "All",
      cancel: "Cancel",
      confirm: "OK",
      delete: "Delete",
      edit: "Edit",
      more: "More",
      reset: "Reset",
      clear: "Clear",
      comma: ","
    },
    button: {
      cancel: "Cancel",
      confirm: "Confirm"
    },
    buttonGroup: {
      noData: "No data"
    },
    buttonMessage: {
      cancel: "Cancel",
      confirm: "Confirm"
    },
    cell: {
      placeholder: "--Select--"
    },
    cascader: {
      noMatch: "No matches",
      loading: "Loading",
      placeholder: "--Select--",
      noData: "No data available."
    },
    chart: {
      auxiliary: "Auxiliary",
      emptyText: "No Data",
      kName: "Day K",
      other: "Other",
      summation: "Summation",
      total: "Total",
      value: "Value"
    },
    colorSelectPanel: {
      confirm: "Select",
      cancel: "Cancel",
      predefine: "Default Colors",
      history: "Recent Colors",
      empty: "None"
    },
    crop: {
      choose: "choose image",
      zoomOut: "zoom out 10%",
      zoomIn: "zoom in 10%",
      rotate_45: "rotate -45°",
      rotate45: "rotate 45°",
      closeCropArea: "hide selection",
      reset: "reset",
      cropImage: "crop image",
      croppedImage: "Target image"
    },
    datepicker: {
      clear: "Clear",
      cancel: "Cancel",
      endDate: "End Date",
      confirm: "OK",
      month: "Month",
      endTime: "End Time",
      month2: "Feb",
      month1: "Jan",
      month4: "Apr",
      month3: "Mar",
      month6: "Jun",
      month5: "May",
      month8: "Aug",
      month7: "Jul",
      month10: "Oct",
      month9: "Sep",
      month12: "Dec",
      month11: "Nov",
      months: {
        feb: "February",
        jan: "January",
        apr: "April",
        mar: "March",
        jun: "June",
        may: "May",
        aug: "August",
        jul: "July",
        oct: "October",
        sep: "September",
        dec: "December",
        nov: "November"
      },
      nextYear: "Next year",
      nextMonth: "Next Month",
      prevMonth: "Previous Month",
      now: "Now",
      selectDate: "Select a date.",
      prevYear: "Previous year",
      startDate: "Start Date",
      selectTime: "Select a time.",
      today: "Today",
      currentMonth: "This Month",
      startTime: "Start Time",
      week: "Week",
      weeks: {
        mon: "Mon",
        sun: "Sun",
        wed: "Wed",
        tue: "Tue",
        thu: "Thu",
        sat: "Sat",
        fri: "Fri"
      },
      timezone: "Select a time zone.",
      year: "Year",
      hour: "hour",
      minute: "minute",
      second: "second",
      to: "",
      yearMonth: "{month} {year}",
      yearMonthDay: "{month} {day}, {year}"
    },
    richTextEditor: {
      bold: "bold",
      italic: "Italic",
      link: "Link",
      unlink: "Unlink",
      highlight: "Highlight",
      underline: "Underline",
      strike: "Strikethrough",
      subscript: "Subscript",
      superscript: "Superscript",
      code: "Coding",
      unorderedlist: "Bullet Lists",
      orderedlist: "Numbered Lists",
      taskList: "Multi-level Lists",
      quote: "Reference",
      codeBlock: "Code Block",
      formatClear: "Clear Formatting",
      nodeDelete: "node delete",
      undo: "Undo",
      redo: "Redo",
      left: "Left-Aligned",
      center: "Centered",
      right: "Right-Aligned",
      fontSize: "Font Size",
      lineHeight: "Line Spacing",
      hBox: "Headings",
      img: "Picture",
      color: "Text Color",
      table: "Table",
      backgroundColor: "Background Color",
      localResources: "Upload Local File",
      resourceLink: "Past Link"
    },
    calendar: {
      showType: {
        year: "Year"
      }
    },
    dept: {
      code: "Code",
      company: "Company",
      dept1: "Level-1 Dept.",
      dept2: "Level-2 Dept.",
      dept3: "Level-3 Dept.",
      dept4: "Level-4 Dept.",
      dept5: "Level-5 Dept.",
      dept6: "Level-6 Dept.",
      dept7: "Level-7 Dept.",
      dept8: "Level-8 Dept.",
      input: "Enter a department code ID or name.",
      name: "Name",
      search: "Auxiliary Search",
      selected: "Selected"
    },
    dialogBox: {
      confirm: "OK",
      cancel: "Cancel"
    },
    load: {
      dot: "Loading…"
    },
    exception: {
      build: "Building",
      busy: "The network is busy. Please wait",
      noperm: "Not find the page",
      weaknet: "Poor network performance",
      pcview: "View the file on the PC",
      nodata: "Get some rest",
      create: "Create",
      provide: "Provided by the TINY Team DEV",
      nodatamf: "No data available.",
      nopermmf: "Insufficient permissions.",
      weaknetmf: "Network error.",
      noresult: "No results found.",
      nonews: "No messages.",
      pagenoperm: "403 Forbidden",
      pageweaknet: "Network error.",
      pagenothing: "404 Not Found",
      pageservererror: "500 Internal Server Error"
    },
    fileUpload: {
      largefile: "The file is too large and will be uploaded in segments. Please wait.",
      folder: "The file cannot be uploaded because it is more than five-folder levels deep.",
      init: "Service error. Please try again.",
      token: "Before starting, perform EDM authentication and obtain the token.",
      exceed: "The file size exceeds the upper limit ({maxSize}).",
      largeFile: "The file exceeds 2 GB.",
      fileSize: "The file is less than the minimum size ({minSize} {sizeUnit}).",
      deleteTip: "Press the Delete key.",
      downloadFile: "Download",
      previewFile: "Preview",
      updateFile: "Update",
      reUploadFile: "Upload Again",
      cancelFile: "Cancel Upload",
      deleteFile: "Delete",
      empty: "The file is empty.",
      kiaScanTip: "KIA needs to verify the download of documents from the public network. The current document is being checked by KIA. Try again later.",
      fileNameExceeds: "The file name contains more than 255 characters. Shorten the file name.",
      fileName: "This file name",
      calcHash: "The document is being analyzed and encrypted.",
      uploadFile: "Upload",
      downloadAll: "Download All",
      onlySupport: "Only {type} files are supported.",
      fileNotLessThan: "A single file cannot be less than ",
      fileNotMoreThan: "A single file cannot exceed ",
      fileSizeRange: "The size of a single file must range from {moreThan} to {lessThan}.",
      notSupport: "The format (.{format}) is not supported.",
      notSupportNoSuffix: "Files without file name extensions are not supported.",
      notSupportSpecialCharacters: "Special characters not allowed. Rename the file and upload it again.",
      attachment: "Attachments",
      uploadList: "File List",
      numberExceed: "Too many files in this batch. Max. supported: {number}",
      numberLimit: "Maximum files per batch: {number}",
      encryptDialogTitle: "Watermark and Encryption",
      addWatermark: "Add Watermark",
      encrypted: "Encryption",
      docPreview: "Preview",
      networkError: "Network error.",
      pictureNetworkError: "Network error. File upload failed.",
      reUploadTip: "{number} files failed to be uploaded."
    },
    uploadList: {
      pictureUploading: "Uploading",
      uploadFailed: "Upload failed",
      uploading: "Uploading",
      download: "Download",
      reUpload: "Upload Again",
      delete: "Delete",
      noAttachments: "No attachments available.",
      cancel: "Cancel",
      preview: "Preview",
      releaseAndUpload: "Release mouse capture to upload the file.",
      dragOrClickImport: "Drag a file here or click to import.",
      shoot: "Take Photo or Video",
      selectFromAlbum: "Select from Album",
      uploadFailedAndReupload: "Upload failed. Please try again."
    },
    upload: {
      addPicture: "Add Picture",
      addAudio: "Add Audio",
      addVideo: "Add Video"
    },
    grid: {
      dataUnchanged: "Data remains unchanged.",
      deleteSelectRecord: "Delete the selected records?",
      emptyText: "No data available.",
      error: {
        delRevert: "The <strong>revert</strong> method has been deprecated. Use <strong>revertData</strong> instead.",
        groupFixed: "If you want to use grouping headers, place the fixed columns on the left and right sides.",
        notDelete: "The <strong>delete</strong> method does not exist.",
        notQuery: "The <strong>query</strong> method does not exist.",
        notResizable: "Horizontal virtual scrolling cannot be resized.",
        notSave: "The <strong>save</strong> method does not exist.",
        reqModule: "The module {{name}} is missing.",
        rowIdEmpty: "The <strong>row-id</strong> parameter is required.",
        scrollYHeight: "The height or max-height must be specified before enabling virtual scrolling.",
        toolbarId: "A unique ID is required for the toolbar.",
        treeFixedExpand: "The fixed column conflicts with row expansion in the tree structure.",
        treeInsert: "The tree structure does not support the <strong>insert</strong> operation.",
        treeRemove: "The tree structure does not support the <strong>remove</strong> operation.",
        unableInsert: "Unable to insert data to the specified location.",
        notAllowDragSelf: "Self dragging is not allowed.",
        notAllowDragFixed: "Fixed columns cannot be dragged.",
        remoteMethod: "The <strong>multipleHistory.remoteSelectedMethod</strong> must be set for the remote storage of personalized template management.",
        remoteSelectedMethod: '"remoteSelectedMethod" needs to be set for remote storage for personalized template management.',
        chainCallError: "There are syntax errors in the default slot.",
        renderParamError: "A rendering method is required for generating VNode.",
        classComponentError: "There was an error in the class component rendering.",
        groupColumnFixedError: "Each group must have the same type.",
        missingValueFormat: "The renderer cannot format the date character string. The valueFormat source date format is required.",
        clipboardWriteError: "Clipboard write error."
      },
      filter: {
        allSelect: "All",
        endDate: "End Date",
        startDate: "Start Date",
        dateTips: "Select at least one date.",
        clear: "Clear Filter Criteria from Current Column",
        clearAll: "Clear Filter Criteria from All Columns",
        confirmFilter: "Filter",
        empty: "Empty",
        emptyText: "No data available.",
        equal: "Equal to",
        include: "Contains",
        prefix: "Starts with",
        resetFilter: "Reset",
        unempty: "Not empty"
      },
      individuation: {
        cancelBtn: "Cancel",
        colConfigs: {
          asc: "Ascending",
          desc: "Descending",
          frozenLeft: "Freeze left",
          frozenRight: "Freeze right",
          invisible: "Hide",
          unsorted: "Unsorted",
          unfrozen: "Unfrozen",
          visible: "Show",
          unfreeze: "Unfreeze",
          unsort: "Unsort"
        },
        toolbar: {
          set: "Operation",
          selected: "Selected",
          freeze: "Frozen",
          sort: "Sorting",
          clear: "Clear",
          search: "Search",
          all: "All"
        },
        columnSet: "Set Columns",
        overwriteSave: "Overwrite and Save",
        saveAs: "Save As",
        saveTemplate: "Save",
        selectTemplate: "Select",
        hideMsg: "There must be at least one column.",
        maxFreezeNumMsg: "A maximum of 6 columns can be frozen.",
        defaultTemplateName: "Enter a name. If no name is specified, a name will be automatically generated based on the time.",
        reserveTemplateName: "If no new name is specified, the original name will be kept.",
        resetBtn: "Reset",
        saveBtn: "OK",
        hideAll: "Hide All",
        showAll: "Show All",
        tabs: {
          base: {
            title: "Basic Settings",
            tips: "Click the icons to customize the display style."
          },
          other: {
            allDataSort: "All data",
            currPageSort: "Current page",
            pageSize: "Records per Page",
            title: "Other Settings",
            tips: "Set sorting for servers or clients and the number of records on each page.",
            sortType: "Sorting Data"
          }
        },
        title: "Custom Settings",
        switchtitle: "Manage",
        switchsave: "Save",
        switchlabel: "Configuration List:",
        switchapply: "Use",
        switchedit: "Edit",
        switchdel: "Delete",
        switchdelconfirm: "Deletion Confirmation",
        switchconfirm: "Confirm",
        switchonlytemp: "Save",
        switchtempapply: "Save and Use",
        switchtempoverwrite: "Overwrite and Use",
        switchdelcon: "Delete this template?",
        switchdelyes: "Confirm",
        switchdelno: "Cancel",
        switchapplycon: "Use this template??"
      },
      removeSelectRecord: "Remove the selected records?",
      saveSuccess: "Saved successfully.",
      selectOneRecord: "Select at least one record.",
      isSaveMsg: "Some data has been modified. Save the changes?"
    },
    hrapprover: {
      approver: "Authorized Approver",
      noapprover: "No authorized approver.",
      noselected: "No authorized approver selected.",
      remark: "Remarks"
    },
    imageViewer: {
      loadErrorAlt: "Loading failed.",
      save: "Save",
      del: "Delete",
      thumbnail: "Thumbnail",
      menu: "Directory",
      hide: "Hide Sidebar",
      show: "Show Sidebar"
    },
    navMenu: {
      moreText: "More"
    },
    logout: {
      in: "Logged in",
      out: "Logged out"
    },
    page: {
      goto: "Go to",
      item: "items",
      next: "Next",
      page: "",
      pageClassifier: "",
      pagesize: "",
      prev: "Previous",
      total: "Total:",
      totals: "Total:",
      jump: "Go to",
      hundredThousand: "100,000+",
      million: "1 million+",
      tenMillion: "10 million+",
      loadingTotals: "Loading the total number of items..."
    },
    popeditor: {
      cancel: "Cancel",
      confirm: "OK",
      historyLists: "History",
      reset: "Reset",
      search: "Query",
      selectionLists: "Selected",
      sourceLists: "All",
      title: "Select",
      filterNode: "Enter a keyword."
    },
    popupload: {
      fileName: "Name",
      fileSize: "Size",
      fileStatus: "Status",
      dialogTitle: "Upload Files",
      cancelButtonText: "Cancel",
      tipsFileText: "Info",
      uploadError: "Upload failed.",
      uploadButtonText: "Upload",
      uploadSuccess: "Upload completed.",
      saveButtonText: "Select File",
      uploadsButtonText: "Select Files",
      errorTypeTips: "Invalid file format.",
      errorSizeTips: "File too large.",
      confirmDeleteTips: "Are you sure you want to delete this file?",
      delete: "Delete",
      waitUpload: "Waiting for upload",
      operation: "Operation",
      success: "Upload successful",
      listTip: "There are a total of {0} pieces of data: ",
      errorListTip: "There are {0} errors in them. Please modify them and try again",
      limitUploadFileType: "Supported file formats: ",
      limitUploadFileNumber: "Maximum files allowed: ",
      limitUploadFileSize: "Maximum file size: ",
      errorNumTips: "The number of uploaded files exceeds the limit. The operation has been cancelled"
    },
    rate: {
      level: {
        average: "Acceptable",
        excellent: "Very Good",
        fair: "Poor",
        good: "Good",
        poor: "Very Poor"
      }
    },
    select: {
      loading: "Loading...",
      noData: "No data available.",
      noMatch: "No matches.",
      placeholder: "--Select--",
      pleaseSearch: "Enter a keyword.",
      search: "Search",
      selected: "Selected",
      selectedNum: "{num} selected",
      noSearchData: "No matches. Enter another keyword.",
      add: "Add",
      collapse: "Collapse"
    },
    search: {
      placeholder: "Search"
    },
    signature: {
      confirm: "Done",
      rewrite: "Re-sign",
      cancel: "Cancel",
      tips: "Handwritten signature required.",
      resign: "Re-sign",
      placeholder: "Please sign here (mandatory)."
    },
    tabs: {
      moreItem: "More"
    },
    tag: {
      add: "Add"
    },
    toggleMenu: {
      placeholder: "Enter a keyword."
    },
    treeMenu: {
      placeholder: "Enter a keyword."
    },
    transfer: {
      filterPlaceholder: "Enter a keyword.",
      hasCheckedFormat: "{checked}/{total} checked",
      noData: "No data available.",
      noMatch: "No matches.",
      noCheckedFormat: "{total} items",
      titles: ["List 1", "List 2"]
    },
    tree: {
      loading: "Loading...",
      emptyText: "No data available.",
      switchText: "Select child nodes",
      edit: "Edit",
      delete: "Delete",
      addChild: "Add Child",
      newNodeTitle: "Add Child Node",
      deleteTip1: "Deleted data cannot be restored. Are you sure you want to continue?",
      deleteTip2: "This node has child nodes. Do you want to retain the child nodes?",
      deleteTip3: "Retain child nodes"
    },
    usercard: {
      address: "Address",
      collapse: "Show less",
      email: "Email",
      empno: "Employee ID",
      expand: "Show more",
      fax: "Fax",
      internal: "Extension",
      manager: "Supervisor",
      mobile: "Mobile",
      other: "Additional Info",
      phone: "Landline",
      timezone: "Time Zone",
      travelcode: "Out-of-Office Message for a Business Trip",
      viop: "VIOP"
    },
    richText: {
      bold: "Bold",
      italic: "Tilt",
      underline: "Underline",
      header: "Paragraph Format",
      strike: "Delete Line",
      blockquote: "Block Reference",
      codeBlock: "Insert Code Segment",
      size: "Font Size",
      listOrdered: "No. List",
      listBullet: "Project List",
      header1: "H1",
      header2: "H2",
      align: "Alignment Mode",
      color: "Font Color",
      background: "Background Color",
      image: "Image",
      video: "Video",
      link: "Add Link",
      formula: "Insert Function",
      clean: "Clear Format",
      indent1: "Indent To The Left",
      indent2: "Indent To The Right",
      pickerLabel: "Title Size",
      headerPicker1: "Title 1",
      headerPicker2: "Title 2",
      headerPicker3: "Title 3",
      headerPicker4: "Title 4",
      headerPicker5: "Title 5",
      headerPicker6: "Title 6",
      normal: "Normal",
      sizeSmall: "Small",
      sizeLarge: "Large",
      sizeHuge: "Super Large Size",
      alignPicker1: "Align To The Left",
      alignPicker2: "Align In The Center",
      alignPicker3: "Align To The Right",
      alignPicker4: "Align The Two Ends",
      subScript: "Subscript",
      superScript: "Superscript",
      directionRTL: "Right To Left",
      font: "Font",
      file: "File",
      betterTable: "Table",
      fullscreen: "Fullscreen",
      insertColumnRight: "Insert Column Right",
      insertColumnLeft: "Insert Column Left",
      insertRowUp: "Insert Row Up",
      insertRowDown: "Insert Row Down",
      mergeCells: "Merge Cells",
      unmergeCells: "Split Cells",
      deleteColumn: "Delete Current Column",
      deleteRow: "Delete Current Row",
      deleteTable: "Delete Table",
      colorPicker: "Background Color",
      placeholder: "Insert text here...",
      maxLength: "Text Length exceeds the Limit, max Length config is "
    },
    fluentEditor: {
      undo: "Undo",
      redo: "Redo",
      lineheight: "Line Height"
    },
    steps: {
      done: "Completed",
      doing: "In Progress",
      wait: "Pending"
    },
    actionSheet: {
      cancel: "Cancel"
    },
    image: {
      loadFail: "Loading failed."
    },
    miniPicker: {
      cancel: "Cancel",
      confirm: "OK"
    },
    pullRefresh: {
      pullingDown: "Pull down to refresh.",
      pullingUp: "Pull up to refresh.",
      pulling: "Pull down to refresh.",
      loosing: "Release to refresh.",
      success: "Refreshed.",
      failed: "Refresh failed.",
      noMore: "No more data."
    },
    currency: {
      defaultCurrency: "Default",
      setDefault: "Set as default",
      chooseCurrency: "Select Currency"
    },
    calendarBar: {
      week: {
        0: "Sun",
        1: "Mon",
        2: "Tue",
        3: "Wed",
        4: "Thu",
        5: "Fri",
        6: "Sat"
      },
      year: "%s",
      yearMonth: "%m %y",
      month: {
        1: "Jan",
        2: "Feb",
        3: "Mar",
        4: "Apr",
        5: "May",
        6: "Jun",
        7: "Jul",
        8: "Aug",
        9: "Sep",
        10: "Oct",
        11: "Nov",
        12: "Dec"
      },
      monthAbbr: {
        1: "Jan",
        2: "Feb",
        3: "Mar",
        4: "Apr",
        5: "May",
        6: "Jun",
        7: "Jul",
        8: "Aug",
        9: "Sept",
        10: "Oct",
        11: "Nov",
        12: "Dec"
      }
    },
    calendarView: {
      week: {
        0: "Sun",
        1: "Mon",
        2: "Tue",
        3: "Wed",
        4: "Thu",
        5: "Fri",
        6: "Sat"
      },
      weekDays: {
        0: "SUN",
        1: "MON",
        2: "TUE",
        3: "WED",
        4: "THU",
        5: "FRI",
        6: "SAT"
      },
      backToday: "Today",
      new: "Add",
      noSchedule: "No Schedule",
      year: "",
      month: "",
      dateFormat: "yyyy-MM"
    },
    selectedBox: {
      select: "Selected (%s)",
      allSelect: "All Selected (%s)",
      clear: "Clear",
      noData: "No data"
    },
    record: {
      record: "Recording",
      cancel: "Cancel",
      confirm: "OK",
      clickToStartRecording: "Click to record.",
      clickToResumeRecording: "Click to continue recording."
    },
    dialogSelect: {
      treeSearch: "Enter a keyword and press Enter."
    },
    loadList: {
      errorText: "Something went wrong.",
      loadingText: "Loading...",
      finishedText: "There's no more."
    }
  },
  validation: {
    array: {
      len: "This field must contain %s characters.",
      min: "This field must contain %s characters or more.",
      max: "This field must contain %s characters or less.",
      range: "This field must have %s to %s characters."
    },
    date: {
      format: "The date %s does not match the %s format.",
      invalid: "%s is invalid.",
      parse: "Unable to parse %s, %s is invalid"
    },
    default: "%s field verification error.",
    enum: "Must be one of %s",
    number: {
      len: "The value must be %s.",
      min: "The value must be %s or larger.",
      max: "The value must be %s or less.",
      range: "The value must be between %s and %s."
    },
    pattern: {
      mismatch: "The value %s does not match %s."
    },
    required: "Required field",
    string: {
      len: "This field must contain %s characters.",
      min: "This field must contain at least %s characters.",
      max: "This field can contain a maximum of %s characters.",
      range: "This field must contain %s to %s characters."
    },
    types: {
      acceptFile: "Only files are supported.",
      acceptImg: "Only images are supported.",
      array: "Invalid array.",
      boolean: "Invalid boolean value.",
      date: "Invalid date format.",
      dateTime: "Invalid date and time format.",
      dateYM: "The date format (y-mm) does not comply with the rule.",
      dateYMD: "The date format (y-MM-dd) does not comply with the rule.",
      digits: "Invalid number only.",
      email: "Invalid email address.",
      fileSize: "The file size is not in the correct format, such as 3kb.",
      float: "Invalid floating point number.",
      hex: "Invalid hexadecimal number.",
      integer: "Invalid integer.",
      longDateTime: "Invalid long date format.",
      method: "Must be a function.",
      number: "Invalid number.",
      object: "Invalid object.",
      regexp: "Invalid regular expression.",
      specialch: "Only digits, letters, underscores (_), hyphens (-), and dots (.) are allowed.",
      specialch2: "Only digits, letters, underscores (_), and hyphens (-) are allowed.",
      speczh: "Only digits, letters, and underscores (_) are allowed.",
      string: "Invalid string.",
      time: "Invalid time format.",
      url: "Invalid URL.",
      version: "Invalid version format."
    },
    whitespace: "This field cannot be left blank."
  }
};
var RE_NARGS = /(%|)\{([0-9a-zA-Z_]+)\}/g;
function format(string) {
  for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
    args[_key - 1] = arguments[_key];
  }
  if (args.length === 1 && typeof args[0] === "object") {
    args = args[0];
  }
  if (!args || !args.hasOwnProperty) {
    args = {};
  }
  return string.replace(RE_NARGS, function(match, prefix, i, index2) {
    var result;
    if (string[index2 - 1] === "{" && string[index2 + match.length] === "}") {
      return i;
    } else {
      result = Object.prototype.hasOwnProperty.call(args, i) ? args[i] : null;
      if (result === null || result === void 0) {
        return "";
      }
      return result;
    }
  });
}
var lang = zhCN$1;
var i18nHandler = null;
var t$1 = function t$12(path, options) {
  if (options === void 0) {
    options = void 0;
  }
  if (i18nHandler)
    return i18nHandler.apply(this, arguments);
  var array = path.split(".");
  var value = null;
  var current = lang;
  for (var i = 0, j = array.length; i < j; i++) {
    var property = array[i];
    value = current[property] || "";
    if (i === j - 1)
      return format(value, options);
    if (!value)
      return "";
    current = value;
  }
  return "";
};
var use$1 = function use$12(l) {
  lang = l || lang;
  return lang;
};
var language$1 = function language$12() {
  return lang.code;
};
var i18n$1 = function i18n$12(fn) {
  i18nHandler = fn || t$1;
  return i18nHandler;
};
var extend$1 = extend$2;
var initI18n$1 = function initI18n$12(_ref) {
  var app = _ref.app, createI18n = _ref.createI18n, _ref$messages = _ref.messages, messages = _ref$messages === void 0 ? {} : _ref$messages, _ref$i18n = _ref.i18n, i18n2 = _ref$i18n === void 0 ? {} : _ref$i18n, merge = _ref.merge;
  if (typeof merge !== "function") {
    merge = function merge2(_ref2) {
      var lang3 = _ref2.lang, i18n3 = _ref2.i18n, messages2 = _ref2.messages;
      return extend$1(true, lang3, i18n3.messages, messages2);
    };
  }
  var lang2 = {
    zhCN: zhCN$1,
    enUS: enUS$1
  };
  if (typeof createI18n === "function") {
    var vueI18n = createI18n({
      locale: i18n2.locale || "zhCN",
      messages: merge({
        lang: lang2,
        i18n: i18n2,
        messages
      })
    });
    i18nHandler = function i18nHandler2(key, value) {
      return vueI18n.global.t(key, value);
    };
    return vueI18n;
  }
  if (app && app.config && app.config.globalProperties) {
    app.config.globalProperties.$t = t$1;
  }
  return merge({
    lang: lang2,
    i18n: i18n2,
    messages
  });
};
var isVue2 = true;
var isVue3 = false;
var index = {
  isVue2,
  isVue3,
  use: use$1,
  t: t$1,
  i18n: i18n$1,
  initI18n: initI18n$1,
  extend: extend$1,
  zhCN: zhCN$1,
  enUS: enUS$1
};
var vue = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: index,
  enUS: enUS$1,
  extend: extend$1,
  i18n: i18n$1,
  initI18n: initI18n$1,
  isVue2,
  isVue3,
  language: language$1,
  t: t$1,
  use: use$1,
  zhCN: zhCN$1
}, Symbol.toStringTag, {
  value: "Module"
}));
var getNumberFormat = function getNumberFormat2(config) {
  var groupSize = 3;
  var groupSeparator = ",";
  var decimalSeparator = ".";
  if (isPlainObject(config)) {
    return config;
  }
  if (typeof config === "string") {
    var match = config.match(/\d{2}([^\d]?)\d{3}([^\d]?)\d{2}/);
    if (match && match.length === 3) {
      groupSeparator = match[1];
      decimalSeparator = match[2];
    }
  }
  return {
    groupSeparator,
    groupSize,
    decimalSeparator
  };
};
var getDateFormat = function getDateFormat2(config) {
  var _config$DateFormat = config.DateFormat, DateFormat = _config$DateFormat === void 0 ? "yyyy-MM-dd" : _config$DateFormat, _config$TimeFormat = config.TimeFormat, TimeFormat = _config$TimeFormat === void 0 ? "HH:mm:ss" : _config$TimeFormat;
  return {
    DateFormat,
    DateTimeFormat: DateFormat + " " + TimeFormat,
    TimeFormat
  };
};
var TZRE = /(-|\+)(\d{2}):?(\d{2})$/;
var getStrTimezone = function getStrTimezone2(value) {
  var localTimeZone = 0 - (/* @__PURE__ */ new Date()).getTimezoneOffset() / 60;
  var match = typeof value === "string" && value.match(TZRE);
  if (match) {
    var minoffset = match[2] * 1 + match[3] * 1 / 60;
    value = minoffset * (match[1] + "1");
  }
  if (isNumber(value) && value >= -12 && value <= 12) {
    return value;
  }
  return localTimeZone;
};
function glob(t2) {
  return function(config) {
    var opt = _extends({}, getDateFormat(config), {
      NumberFormat: getNumberFormat(config.NumberFormat),
      DbTimezone: getStrTimezone(config.DbTimezone),
      Timezone: getStrTimezone(config.Timezone),
      TimezoneOffset: config.TimezoneOffset
    });
    var tools = {
      getFormatConfig: function getFormatConfig() {
        return opt;
      },
      setFormatConfig: function setFormatConfig(obj) {
        Object.assign(opt, obj);
      },
      getNumberFormat: function getNumberFormat3() {
        return opt.NumberFormat;
      },
      getDateFormat: function getDateFormat3() {
        return {
          DateTimeFormat: opt.DateTimeFormat,
          TimeFormat: opt.TimeFormat,
          Timezone: opt.Timezone,
          DateFormat: opt.DateFormat,
          DbTimezone: opt.DbTimezone,
          TimezoneOffset: opt.TimezoneOffset
        };
      },
      /**
       *
       * @param {Date|String} value 日期或日期字符串
       * @param {String} format 格式化模式
       * @returns {String}
       */
      formatDate: function formatDate(value, format2) {
        if (isNull(value)) {
          return value;
        }
        var date = isDate(value) ? value : toDate(value);
        var dbtimezone = opt.DbTimezone;
        var includeTz = value.match && value.match(TZRE);
        var convers = format2 === false || arguments[2] === false;
        if (includeTz) {
          dbtimezone = getStrTimezone(value);
          date = toDate(value.replace("T", " ").slice(0, -5));
        }
        if (!convers) {
          date = this.getDateWithNewTimezone(date, dbtimezone, opt.Timezone, opt.TimezoneOffset);
        }
        return isDate(date) ? _formatDate(date, format2 || opt.DateFormat, t2) : null;
      },
      /**
       *
       * @param {Number} value 数字
       * @param {Object} format 格式化选项
       * @returns {String}
       */
      formatNumber: function formatNumber(value, format2) {
        return _formatNumber(value, _extends({}, opt.NumberFormat, format2));
      },
      /**
       *
       * @param {String} value 格式化后的字符串
       * @param {Object} format 格式化选项
       * @returns {Number}
       */
      recoverNumber: function recoverNumber(value, format2) {
        return _recoverNumber(value, _extends({}, opt.NumberFormat, format2));
      },
      /**
       *
       * @param {Date} value Date
       * @param {Number} from
       * @param {Number} to
       * @returns {String}
       */
      getDateWithNewTimezone: function getDateWithNewTimezone(value, from, to, timezoneOffset) {
        from = from === 0 ? from : from || opt.DbTimezone;
        to = to === 0 ? to : to || opt.Timezone;
        timezoneOffset = timezoneOffset === 0 ? timezoneOffset : timezoneOffset || opt.TimezoneOffset;
        return _getDateWithNewTimezone(value, from, to, timezoneOffset);
      }
    };
    return tools;
  };
}
var version = "3.20.0";
var use = vue.use, t = vue.t, i18n = vue.i18n, initI18n = vue.initI18n, extend = vue.extend, zhCN = vue.zhCN, enUS = vue.enUS, language = vue.language;
var globalization = /* @__PURE__ */ glob(t);
var locale = /* @__PURE__ */ Object.assign({}, index, {
  language,
  globalization
});
export {
  locale as default,
  enUS,
  extend,
  globalization,
  i18n,
  initI18n,
  language,
  t,
  use,
  version,
  zhCN
};
