// @opentiny/vue-mask@3.20.0/lib/index.js
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/mask/vue';
import '@opentiny/vue-theme-mobile/mask/index.css';
import { openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeStyle, renderSlot, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "Mask",
  props: _extends({}, props, {
    visible: {
      type: Boolean,
      default: false
    },
    zIndex: {
      type: Number,
      default: 99
    },
    cancelTouch: {
      type: Boolean,
      default: false
    }
  }),
  emits: ["update:visible", "click"],
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createBlock(Transition, {
    name: "mask-fade",
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          class: "tiny-mobile-mask",
          style: normalizeStyle(_ctx.state.calcStyle),
          onMousedown: _cache[0] || (_cache[0] = function() {
            return _ctx.handleTouch && _ctx.handleTouch.apply(_ctx, arguments);
          }),
          onTouchmove: _cache[1] || (_cache[1] = function() {
            return _ctx.handleTouch && _ctx.handleTouch.apply(_ctx, arguments);
          }),
          onTouchstart: _cache[2] || (_cache[2] = function() {
            return _ctx.handleTouch && _ctx.handleTouch.apply(_ctx, arguments);
          })
        },
        [renderSlot(_ctx.$slots, "default")],
        36
        /* STYLE, NEED_HYDRATION */
      ), [[vShow, _ctx.visible]])];
    }),
    _: 3
    /* FORWARDED */
  });
}
var Mask = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.20.0";
Mask.install = function(Vue) {
  Vue.component(Mask.name, Mask);
};
Mask.version = version;

export { Mask as default };
