// @opentiny/vue-date-table@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
import { isDate } from "@opentiny/vue-renderless/common/deps/date-util";
import "@opentiny/vue-theme/date-table/index.css";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var DateTable = defineComponent({
  name: $prefix + "DateTable",
  componentName: "DateTable",
  props: _extends({}, $props, {
    cellClassName: {},
    date: {},
    defaultValue: {
      validator: function validator(val) {
        return val === null || isDate(val) || Array.isArray(val) && val.every(isDate);
      }
    },
    disabledDate: {},
    firstDayOfWeek: {
      default: 7,
      type: Number,
      validator: function validator2(val) {
        return val >= 1 && val <= 7;
      }
    },
    maxDate: {},
    minDate: {},
    rangeState: {
      default: function _default() {
        return {
          endDate: null,
          selecting: false
        };
      }
    },
    selectionMode: {
      default: function _default2() {
        return "day";
      }
    },
    showWeekNumber: {
      type: Boolean,
      default: function _default3() {
        return false;
      }
    },
    value: {},
    formatWeeks: Function
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
DateTable.install = function(Vue) {
  Vue.component(DateTable.name, DateTable);
};
DateTable.version = version;
export {
  DateTable as default
};
