// @opentiny/vue-date-table@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/date-table/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { openBlock, createElementBlock, normalizeClass, createElementVNode, toDisplayString, createCommentVNode, Fragment, renderList } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var common = {
  "next-month": "text-color-text-disabled cursor-pointer group-hover:bg-inherit",
  "current": "group-hover:text-color-text-inverse group-hover:bg-color-brand text-color-text-inverse bg-color-brand",
  "start-date": "text-color-text-inverse bg-color-brand group-hover:bg-color-brand"
};
var classes = _extends({}, common, {
  "today": "border border-solid border-color-border-focus",
  "pre-month": common["next-month"],
  "selected": common["current"],
  "th": "px-1 py-4 text-color-text-primary font-normal whitespace-nowrap",
  "td": "box-border text-center cursor-pointer relative group",
  "cell": "h-8 mb-2.5 box-border",
  "in-range": "",
  "cell-in-range": "bg-color-brand-hover-subtle text-color-brand",
  "disabled": "text-color-text-disabled group-hover:text-color-text-disabled cursor-pointer cursor-not-allowed bg-color-bg-3 group-hover:bg-color-bg-3",
  "cell-disabled": "bg-color-bg-3 text-color-text-disabled",
  "text": "w-8 h-8 leading-8 block absolute left-1/2 box-border -translate-x-2/4 rounded group-hover:bg-color-brand-hover-subtle",
  "end-date": common["start-date"],
  "token": ""
});
var _sfc_main = defineComponent({
  props: [].concat(props, ["cellClassName", "date", "defaultValue", "disabledDate", "firstDayOfWeek", "maxDate", "minDate", "rangeState", "selectionMode", "showWeekNumber", "value", "formatWeeks"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = {
  key: 0
};
var _hoisted_2 = ["data-tag"];
var _hoisted_3 = ["textContent"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "table",
    {
      cellpadding: "0",
      cellspacing: "0",
      "data-tag": "tiny-date-table",
      class: normalizeClass(["table-fixed w-full"]),
      onMousemove: _cache[0] || (_cache[0] = function() {
        return _ctx.handleMouseMove && _ctx.handleMouseMove.apply(_ctx, arguments);
      }),
      onClick: _cache[1] || (_cache[1] = function() {
        return _ctx.handleClick && _ctx.handleClick.apply(_ctx, arguments);
      })
    },
    [createElementVNode("tbody", null, [createElementVNode("tr", null, [_ctx.showWeekNumber ? (openBlock(), createElementBlock("th", _hoisted_1, [createElementVNode(
      "div",
      {
        class: normalizeClass(_ctx.gcls("th"))
      },
      toDisplayString(_ctx.t("ui.datepicker.week")),
      3
      /* TEXT, CLASS */
    )])) : createCommentVNode("v-if", true), (openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.weeks, function(week, weekIdx) {
        return openBlock(), createElementBlock("th", {
          key: weekIdx
        }, [createElementVNode(
          "div",
          {
            class: normalizeClass(_ctx.gcls("th"))
          },
          toDisplayString(_ctx.t("ui.datepicker.weeks." + week)),
          3
          /* TEXT, CLASS */
        )]);
      }),
      128
      /* KEYED_FRAGMENT */
    ))]), (openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.rows, function(row, trIdx) {
        return openBlock(), createElementBlock(
          "tr",
          {
            "data-tag": "tiny-date-table__row",
            key: trIdx,
            class: normalizeClass({
              current: _ctx.isWeekActive(row[1])
            })
          },
          [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(row, function(cell, tdIdx) {
              return openBlock(), createElementBlock("td", {
                key: tdIdx,
                "data-tag": _ctx.getCellClasses(cell),
                class: normalizeClass(_ctx.gcls("td"))
              }, [createElementVNode(
                "div",
                {
                  class: normalizeClass(_ctx.m(_ctx.gcls("cell"), _ctx.gcls(_ctx.getCssToken(cell, "cell-")), {
                    "rounded-l": tdIdx === 0
                  }, {
                    "rounded-r": tdIdx === 6
                  }))
                },
                [createElementVNode("span", {
                  textContent: toDisplayString(cell.text),
                  class: normalizeClass(_ctx.m(_ctx.gcls("text"), _ctx.gcls(_ctx.getCellClasses(cell).split(" "))))
                }, null, 10, _hoisted_3)],
                2
                /* CLASS */
              )], 10, _hoisted_2);
            }),
            128
            /* KEYED_FRAGMENT */
          ))],
          2
          /* CLASS */
        );
      }),
      128
      /* KEYED_FRAGMENT */
    ))])],
    32
    /* NEED_HYDRATION */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
