// @opentiny/vue-alert@3.20.0/lib/mobile.js
import { renderless, api } from '@opentiny/vue-renderless/alert/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { iconClose, iconSuccess, iconError, iconHelp, iconWarningTriangle } from '@opentiny/vue-icon';
import '@opentiny/vue-theme-mobile/alert/index.css';
import { resolveComponent, openBlock, createBlock, Transition, withCtx, createElementBlock, normalizeClass, normalizeStyle, resolveDynamicComponent, createCommentVNode, createElementVNode, renderSlot, createTextVNode, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["icon", "type", "size", "description", "closable", "showIcon", "closeText", "duration", "offset", "autoHide", "target", "center"]),
  components: {
    IconClose: iconClose(),
    IconSuccess: iconSuccess(),
    IconError: iconError(),
    IconHelp: iconHelp(),
    IconWarning: iconWarningTriangle()
    // key值在 $constants 中已定义
  },
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_close = resolveComponent("icon-close");
  return openBlock(), createBlock(Transition, {
    name: "tiny-mobile-alert-fade"
  }, {
    default: withCtx(function() {
      return [_ctx.state.show ? (openBlock(), createElementBlock(
        "div",
        {
          key: 0,
          class: normalizeClass(_ctx.state.alertClass),
          style: normalizeStyle(_ctx.state.alertStyle)
        },
        [_ctx.showIcon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.getIcon), {
          key: 0,
          class: "tiny-mobile-alert__icon"
        })) : createCommentVNode("v-if", true), createElementVNode(
          "div",
          {
            class: normalizeClass(["tiny-mobile-alert__content", {
              "is-hideicon": !_ctx.showIcon
            }])
          },
          [renderSlot(_ctx.$slots, "default", {}, function() {
            return [createTextVNode(
              toDisplayString(_ctx.description),
              1
              /* TEXT */
            )];
          })],
          2
          /* CLASS */
        ), !_ctx.closeText && _ctx.closable ? (openBlock(), createBlock(_component_icon_close, {
          key: 1,
          onClick: _ctx.handleClose,
          class: "tiny-mobile-alert__icon tiny-mobile-alert__close"
        }, null, 8, ["onClick"])) : _ctx.closeText && _ctx.closable ? (openBlock(), createElementBlock(
          "span",
          {
            key: 2,
            onClick: _cache[0] || (_cache[0] = function() {
              return _ctx.handleClose && _ctx.handleClose.apply(_ctx, arguments);
            }),
            class: "is-custom"
          },
          toDisplayString(_ctx.closeText),
          1
          /* TEXT */
        )) : createCommentVNode("v-if", true)],
        6
        /* CLASS, STYLE */
      )) : createCommentVNode("v-if", true)];
    }),
    _: 3
    /* FORWARDED */
  });
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobile as default };
