// @opentiny/vue-breadcrumb@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/breadcrumb/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  EVENT_NAME: {
    breadcrumbItemSelect: "breadcrumbItemSelect"
  }
};
var breadcrumbProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  separator: {
    type: String,
    default: ""
  },
  separatorIcon: {
    type: Object
  },
  textField: {
    type: String,
    default: "label"
  },
  size: String,
  options: {
    type: Array,
    default: function _default2() {
      return [];
    }
  }
});
var Breadcrumb = defineComponent({
  name: $prefix + "Breadcrumb",
  props: breadcrumbProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Breadcrumb.install = function(Vue) {
  Vue.component(Breadcrumb.name, Breadcrumb);
};
Breadcrumb.version = version;
export {
  Breadcrumb as default
};
