// vite-plugin-unplugin@1.8.0/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.unplugin = void 0;
const vite_1 = __importDefault(require("unplugin-auto-import/vite"));
const vite_2 = __importDefault(require("unplugin-vue-components/vite"));
function unplugin(object) {
    const { env, imports = [], resolvers = [], dirs = [] } = object;
    const viteApp = '\u0056\u0049\u0054\u0045\u005f' + '\u0041\u0050\u0050\u005f';
    const viteUser = '\u0056\u0049\u0054\u0045\u005f' + '\u0055\u0053\u0045\u0052\u005f';
    const userName = env[`${viteApp}\u0047\u0049\u0054\u0048\u0055\u0042\u005f\u0055\u0053\u0045\u0052\u005f\u004e\u0041\u004d\u0045`];
    const secretKey = env[`${viteApp}\u0053\u0045\u0043\u0052\u0045\u0054\u005f\u004b\u0045\u0059`];
    const nodeEnv = env[`${viteUser}\u004e\u004f\u0044\u0045\u005f\u0045\u004e\u0056`];
    const isEmpty = (value) => {
        return value == undefined || value == '' || value == null;
    };
    if (isEmpty(userName) || isEmpty(secretKey))
        return [];
    if (nodeEnv !==
        '\u0064\u0065\u0076\u0065\u006c\u006f\u0070\u006d\u0065\u006e\u0074')
        if (isEmpty(userName) || isEmpty(secretKey))
            return [];
    if (nodeEnv !==
        '\u0064\u0065\u0076\u0065\u006c\u006f\u0070\u006d\u0065\u006e\u0074')
        if (secretKey.length < 100)
            return [];
    return [
        (0, vite_1.default)({
            imports: [
                ...[
                    '\u0076\u0075\u0065',
                    '\u0070\u0069\u006e\u0069\u0061',
                    '\u0076\u0075\u0065\u002d\u0069\u0031\u0038\u006e',
                    '\u0076\u0075\u0065\u002d\u0072\u006f\u0075\u0074\u0065\u0072',
                    '\u0040\u0076\u0075\u0065\u0075\u0073\u0065\u002f\u0063\u006f\u0072\u0065',
                    {
                        axios: [
                            [
                                '\u0064\u0065\u0066\u0061\u0075\u006c\u0074',
                                '\u0061\u0078\u0069\u006f\u0073',
                            ],
                        ],
                    },
                ],
                ...imports,
            ],
            resolvers: [...[], ...resolvers],
            dts: '\u006c\u0069\u0062\u0072\u0061\u0072\u0079\u002f\u0062\u0075\u0069\u006c\u0064\u002f\u0075\u006e\u0070\u006c\u0075\u0067\u0069\u006e\u002f\u0061\u0075\u0074\u006f\u002d\u0069\u006d\u0070\u006f\u0072\u0074\u0073\u002e\u0064\u002e\u0074\u0073',
            dirs: [
                ...['\u0073\u0072\u0063\u002f\u0068\u006f\u006f\u006b\u0073'],
                ...dirs,
            ],
        }),
        (0, vite_2.default)({
            dirs: [
                ...[
                    '\u006c\u0069\u0062\u0072\u0061\u0072\u0079\u002f\u0063\u006f\u006d\u0070\u006f\u006e\u0065\u006e\u0074\u0073',
                    '\u0073\u0072\u0063\u002f\u0070\u006c\u0075\u0067\u0069\u006e\u0073',
                    '\u0073\u0072\u0063\u002f\u002a\u002a\u002f\u0076\u0061\u0062\u0041\u0075\u0074\u006f\u0043\u006f\u006d\u0070\u006f\u006e\u0065\u006e\u0074\u0073',
                ],
                ...dirs,
            ],
            resolvers: [...[], ...resolvers],
            dts: '\u006c\u0069\u0062\u0072\u0061\u0072\u0079\u002f\u0062\u0075\u0069\u006c\u0064\u002f\u0075\u006e\u0070\u006c\u0075\u0067\u0069\u006e\u002f\u0063\u006f\u006d\u0070\u006f\u006e\u0065\u006e\u0074\u0073\u002e\u0064\u002e\u0074\u0073',
        }),
    ];
}
exports.unplugin = unplugin;
