// @opentiny/vue-dialog-select@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PCTemplate from "./pc.js";
var template = function template2() {
  return PCTemplate;
};
var $constants = {
  INPUT_HEIGHT: "32px",
  INPUT_MARGIN_BOTTOM: "8px"
};
var DialogSelect = defineComponent({
  name: $prefix + "DialogSelect",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    dialogOp: {
      type: Object,
      default: function _default2() {
        return {};
      }
    },
    visible: Boolean,
    popseletor: {
      type: String,
      default: "grid",
      validator: function validator(value) {
        return ~["grid", "tree"].indexOf(value);
      }
    },
    multi: {
      type: Boolean,
      default: false
    },
    gridOp: {
      type: Object,
      default: function _default3() {
        return {
          columns: [],
          data: []
        };
      }
    },
    textField: {
      type: String,
      default: "label"
    },
    textSplit: {
      type: String,
      default: "/"
    },
    valueField: {
      type: String,
      default: "id"
    },
    valueSplit: {
      type: String,
      default: ";"
    },
    remoteSearch: Function,
    showPager: {
      type: Boolean,
      default: true
    },
    pagerOp: {
      type: Object,
      default: function _default4() {
        return {};
      }
    },
    showSelectedBox: {
      type: Boolean,
      default: true
    },
    selectedBoxOp: {
      type: Object,
      default: function _default5() {
        return {};
      }
    },
    treeOp: {
      type: Object,
      default: function _default6() {
        return {};
      }
    },
    beforeClose: Function,
    autoLookup: {
      type: Boolean,
      default: true
    },
    lookupMethod: Function,
    mainHeight: {
      type: Number,
      default: 290
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
DialogSelect.install = function(Vue) {
  Vue.component(DialogSelect.name, DialogSelect);
};
DialogSelect.version = version;
export {
  DialogSelect as default
};
