// @opentiny/vue-toggle-menu@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/toggle-menu/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  LOCALE_PLACEHOLDER: "ui.toggleMenu.placeholder"
};
var ToggleMenu = defineComponent({
  name: $prefix + "ToggleMenu",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    data: Array,
    placeholder: {
      type: String,
      default: ""
    },
    icon: Object,
    defaultExpandAll: {
      type: Boolean,
      default: false
    },
    draggable: {
      type: Boolean,
      default: false
    },
    expandOnClickNode: {
      type: Boolean,
      default: true
    },
    ellipsis: {
      type: Boolean,
      default: false
    },
    wrap: {
      type: Boolean,
      default: false
    },
    showFilter: {
      type: Boolean,
      default: true
    },
    automaticFiltering: {
      type: Boolean,
      default: true
    },
    props: {
      default: function _default2() {
        return {
          children: "children",
          disabled: "disabled",
          label: "label"
        };
      }
    },
    getMenuDataSync: Function,
    // deprecated v3.4.0废弃, v3.16.0移除；移除原因：如果是同步数据则和:data功能重复
    getMenuDataAsync: Function,
    accordion: Boolean,
    defaultCheckedKeys: Array,
    showCheckbox: Boolean
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
ToggleMenu.install = function(Vue) {
  Vue.component(ToggleMenu.name, ToggleMenu);
};
ToggleMenu.version = version;
export {
  ToggleMenu as default
};
