// @opentiny/vue-fall-menu@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/fall-menu/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var fallMenuProps = _extends({}, $props, {
  data: {
    type: Array,
    default: function _default() {
      return [];
    }
  }
});
var FallMenu = defineComponent({
  name: $prefix + "FallMenu",
  props: fallMenuProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
FallMenu.install = function(Vue) {
  Vue.component(FallMenu.name, FallMenu);
};
FallMenu.version = version;
export {
  FallMenu as default
};
