// @opentiny/vue-time-panel@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/time-panel/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import Scrollbar from '@opentiny/vue-scrollbar';
import { resolveComponent, openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeStyle, normalizeClass, createVNode, createElementBlock, Fragment, renderList, toDisplayString, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["dodestroy", "pick"],
  components: {
    TinyScrollbar: Scrollbar
  },
  props: [].concat(props, ["emitter"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["disabled", "onClick"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_scrollbar = resolveComponent("tiny-scrollbar");
  return openBlock(), createBlock(Transition, {
    name: "tiny-zoom-in-top",
    onBeforeEnter: _ctx.handleMenuEnter,
    onAfterLeave: _ctx.emitDestroy,
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          ref: "popper",
          style: normalizeStyle({
            width: _ctx.state.width + "px"
          }),
          class: normalizeClass([_ctx.state.popperClass, "tiny-picker-panel tiny-time-select tiny-popper"])
        },
        [createVNode(_component_tiny_scrollbar, {
          noresize: "",
          "wrap-class": "tiny-picker-panel__content"
        }, {
          default: withCtx(function() {
            return [(openBlock(true), createElementBlock(
              Fragment,
              null,
              renderList(_ctx.state.items, function(item) {
                return openBlock(), createElementBlock("div", {
                  class: normalizeClass(["tiny-time-select__item", {
                    selected: _ctx.state.value === item.value,
                    disabled: item.disabled,
                    default: item.value === _ctx.state.default
                  }]),
                  disabled: item.disabled,
                  key: item.value,
                  onClick: function onClick($event) {
                    return _ctx.handleClick(item);
                  }
                }, toDisplayString(item.value), 11, _hoisted_1);
              }),
              128
              /* KEYED_FRAGMENT */
            ))];
          }),
          _: 1
          /* STABLE */
        })],
        6
        /* CLASS, STYLE */
      ), [[vShow, _ctx.state.visible]])];
    }),
    _: 1
    /* STABLE */
  }, 8, ["onBeforeEnter", "onAfterLeave"]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
