// @opentiny/vue-picker-column@3.20.0/lib/index.js
import { renderless, api } from '@opentiny/vue-renderless/picker-column/vue';
import { defineComponent, $prefix, setup } from '@opentiny/vue-common';
import { openBlock, createElementBlock, createElementVNode, normalizeStyle, Fragment, renderList, normalizeClass, withModifiers, toDisplayString } from 'vue';
import '@opentiny/vue-theme-mobile/picker-column/index.css';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "PickerColumn",
  emits: ["change"],
  props: {
    columnsItem: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    defaultIndex: Number,
    itemHeight: Number,
    visibleItemCount: Number,
    swipeDuration: Number,
    valueKey: String
  },
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  class: "tiny-mobile-picker-column"
};
var _hoisted_2 = ["onClick"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "div",
    {
      class: "tiny-mobile-picker-column__mask",
      style: normalizeStyle(_ctx.state.maskStyle)
    },
    null,
    4
    /* STYLE */
  ), createElementVNode(
    "div",
    {
      class: "tiny-mobile-picker-column__indicator",
      style: normalizeStyle({
        height: _ctx.itemHeight + "px"
      })
    },
    null,
    4
    /* STYLE */
  ), createElementVNode(
    "ul",
    {
      class: "tiny-mobile-picker-column__wrapper",
      ref: "track",
      style: normalizeStyle(_ctx.state.wrapperStyle),
      onClick: _cache[1] || (_cache[1] = function($event) {
        return _ctx.onClickItem(1);
      })
    },
    [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.columnsItem.values, function(item, index) {
        return openBlock(), createElementBlock("li", {
          class: normalizeClass({
            "tiny-mobile-picker-column__item": true,
            "is-select": index === _ctx.state.currentIndex
          }),
          key: index,
          onClick: withModifiers(function($event) {
            return _ctx.onClickItem(index);
          }, ["stop"]),
          style: normalizeStyle({
            height: _ctx.itemHeight + "px",
            lineHeight: _ctx.itemHeight + "px"
          }),
          onTransitionend: _cache[0] || (_cache[0] = function() {
            return _ctx.onTransitionEnd && _ctx.onTransitionEnd.apply(_ctx, arguments);
          })
        }, toDisplayString(item), 47, _hoisted_2);
      }),
      128
      /* KEYED_FRAGMENT */
    ))],
    4
    /* STYLE */
  )]);
}
var PickerColumn = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.20.0";
PickerColumn.install = function(Vue) {
  Vue.component(PickerColumn.name, PickerColumn);
};
PickerColumn.version = version;

export { PickerColumn as default };
