// @opentiny/vue-select-dropdown@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/select-dropdown/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import Search from '@opentiny/vue-search';
import Button from '@opentiny/vue-button';
import { iconClose, iconChevronUp, iconError } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, withModifiers, normalizeStyle, renderSlot, createElementVNode, toDisplayString, createCommentVNode, createVNode, createBlock, withCtx, createTextVNode, Fragment, renderList } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    TinySearch: Search,
    TinyButton: Button,
    IconClose: iconClose(),
    IconChevronUp: iconChevronUp(),
    IconError: iconError()
  },
  props: [].concat(props, ["placement", "boundariesPadding", "popperOptions", "visibleArrow", "appendToBody", "transformOrigin", "reference", "popper", "offset", "modelValue", "arrowOffset", "isDropInheritWidth", "title", "closeByMask", "searchPlaceholder"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 1,
  "data-tag": "tiny-select-dropdown tiny-popper",
  class: "text-sm overflow-hidden"
};
var _hoisted_2 = {
  "data-tag": "tiny-select-dropdown-header",
  class: "px-4 pt-4"
};
var _hoisted_3 = {
  class: "flex mb-3"
};
var _hoisted_4 = {
  class: "flex-1 text-base text-center leading-5"
};
var _hoisted_5 = {
  key: 0,
  class: "flex mb-3"
};
var _hoisted_6 = {
  "data-tag": "tiny-select-dropdown-body",
  class: "px-4 pb-4 overflow-auto flex-1 scrollbar-size-0"
};
var _hoisted_7 = {
  key: 0,
  "data-tag": "tiny-select-dropdown-footer",
  class: "px-4 h-18 flex justify-between items-center bg-color-bg-1 shadow-[0_-2px_6px_rgba(0,0,0,0.08)] shrink-0"
};
var _hoisted_8 = {
  "data-tag": "tiny-selected-box-header",
  class: "flex h-10 justify-between items-center"
};
var _hoisted_9 = {
  "data-tag": "tiny-selected-box-body",
  class: "flex-1 overflow-auto mb-3"
};
var _hoisted_10 = {
  class: "mr-4 flex-1 leading-5 line-clamp-2"
};
var _hoisted_11 = {
  "data-tag": "tiny-selected-box-footer",
  class: "text-center"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_close = resolveComponent("icon-close");
  var _component_tiny_search = resolveComponent("tiny-search");
  var _component_icon_chevron_up = resolveComponent("icon-chevron-up");
  var _component_tiny_button = resolveComponent("tiny-button");
  var _component_icon_error = resolveComponent("icon-error");
  return _ctx.state.device === "pc" ? (openBlock(), createElementBlock(
    "div",
    {
      key: 0,
      "data-tag": "tiny-select-dropdown tiny-popper",
      class: normalizeClass(["absolute z-[1001] border-none rounded bg-color-bg-1 shadow-md mt-1 box-border transition-[transform,opacity] ease-in-out duration-300 origin-top", [_ctx.state.popperClass, _ctx.state.visible ? "scale-y-100 opacity-100" : "scale-y-0 opacity-0"]]),
      onMousedown: _cache[0] || (_cache[0] = withModifiers(function() {
      }, ["stop"])),
      style: normalizeStyle({
        minWidth: _ctx.state.minWidth,
        width: _ctx.isDropInheritWidth ? _ctx.state.minWidth : "auto"
      })
    },
    [renderSlot(_ctx.$slots, "default")],
    38
    /* CLASS, STYLE, NEED_HYDRATION */
  )) : (openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "div",
    {
      ref: "mask",
      class: normalizeClass(_ctx.m("fixed left-0 right-0 top-0 bottom-0 w-full h-full bg-color-bg-7 transition-[opacity] ease-linear duration-100 opacity-0", {
        "opacity-100": _ctx.state.visible
      })),
      onClick: _cache[1] || (_cache[1] = function($event) {
        return _ctx.closeModal($event, true);
      }),
      style: normalizeStyle({
        "z-index": _ctx.state.zIndex
      })
    },
    null,
    6
    /* CLASS, STYLE */
  ), createElementVNode(
    "div",
    {
      "data-tag": "tiny-select-dropdown-main",
      ref: "drawerBox",
      onMousedown: _cache[9] || (_cache[9] = withModifiers(function() {
      }, ["stop"])),
      style: normalizeStyle({
        "z-index": _ctx.state.zIndex
      }),
      class: normalizeClass(_ctx.m("fixed w-full max-w-full flex flex-col justify-between bg-color-bg-1 shadow-sm border-color-border-separator max-h-[90%]", "min-h-[256px] left-0 bottom-0 border-t-0.5 rounded-t-lg transition-translate ease-linear duration-200 translate-y-full", {
        "translate-y-0": _ctx.state.visible
      }))
    },
    [createElementVNode("div", _hoisted_2, [createElementVNode("div", _hoisted_3, [_ctx.state.showClose ? (openBlock(), createElementBlock(
      "span",
      {
        key: 0,
        class: "inline-block h-6 leading-6 text-color-brand",
        onClick: _cache[2] || (_cache[2] = function() {
          return _ctx.handleClear && _ctx.handleClear.apply(_ctx, arguments);
        })
      },
      toDisplayString(_ctx.t("ui.base.clear")),
      1
      /* TEXT */
    )) : createCommentVNode("v-if", true), createElementVNode(
      "p",
      _hoisted_4,
      toDisplayString(_ctx.title),
      1
      /* TEXT */
    ), createVNode(_component_icon_close, {
      "custom-class": "w-5 h-5 ml-4  cursor-pointer",
      onClick: _cache[3] || (_cache[3] = function($event) {
        return _ctx.closeModal($event, false);
      })
    })]), _ctx.state.filterable ? (openBlock(), createElementBlock("div", _hoisted_5, [createVNode(_component_tiny_search, {
      modelValue: _ctx.state.query,
      "onUpdate:modelValue": _cache[4] || (_cache[4] = function($event) {
        return _ctx.state.query = $event;
      }),
      placeholder: _ctx.searchPlaceholder,
      class: "flex-1 px-0",
      onInput: _ctx.debouncedQueryChange,
      clearable: "",
      big: ""
    }, null, 8, ["modelValue", "placeholder", "onInput"]), createElementVNode(
      "span",
      {
        class: "inline-block h-8 leading-8 text-color-text-secondary shrink-0 ml-3 text-base",
        onClick: _cache[5] || (_cache[5] = function() {
          return _ctx.cancelSearch && _ctx.cancelSearch.apply(_ctx, arguments);
        })
      },
      toDisplayString(_ctx.t("ui.base.cancel")),
      1
      /* TEXT */
    )])) : createCommentVNode("v-if", true)]), createElementVNode("div", _hoisted_6, [renderSlot(_ctx.$slots, "default")]), _ctx.state.multiple ? (openBlock(), createElementBlock("div", _hoisted_7, [createElementVNode("div", null, [createElementVNode(
      "span",
      {
        onClick: _cache[6] || (_cache[6] = function($event) {
          return _ctx.toggleSelectedBox(true);
        }),
        class: "mr-2"
      },
      toDisplayString(_ctx.t("ui.select.selectedNum", {
        num: _ctx.state.selected.length
      })),
      1
      /* TEXT */
    ), _ctx.state.selected.length !== 0 ? (openBlock(), createBlock(_component_icon_chevron_up, {
      key: 0,
      "custom-class": "w-5 h-5 cursor-pointer"
    })) : createCommentVNode("v-if", true)]), createVNode(_component_tiny_button, {
      type: "primary",
      "custom-class": "w-28",
      onClick: _ctx.selectDropdownConfirm
    }, {
      default: withCtx(function() {
        return [createTextVNode(
          toDisplayString(_ctx.t("ui.base.confirm")),
          1
          /* TEXT */
        )];
      }),
      _: 1
      /* STABLE */
    }, 8, ["onClick"])])) : createCommentVNode("v-if", true), _ctx.state.multiple ? (openBlock(), createElementBlock(
      "div",
      {
        key: 1,
        "data-tag": "tiny-selected-box",
        class: normalizeClass(["absolute left-0 bottom-0 flex flex-col w-full bg-color-bg-1 transition-all ease-linear duration-200 overflow-hidden z-[2001]", [_ctx.state.showSelectedBox ? "h-full p-4" : "h-0 p-0"]])
      },
      [createElementVNode("div", _hoisted_8, [createElementVNode(
        "span",
        {
          class: "inline-block h-6 py-1 text-color-brand",
          onClick: _cache[7] || (_cache[7] = function($event) {
            return _ctx.deleteSelected("", true);
          })
        },
        toDisplayString(_ctx.t("ui.base.clear")),
        1
        /* TEXT */
      ), createVNode(_component_icon_close, {
        "custom-class": "w-5 h-5 ml-4 cursor-pointer",
        onClick: _cache[8] || (_cache[8] = function($event) {
          return _ctx.toggleSelectedBox(false);
        })
      })]), createElementVNode("div", _hoisted_9, [createElementVNode("ul", null, [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.selectedArr, function(item, index) {
          return openBlock(), createElementBlock("li", {
            key: (item.value || "") + index,
            class: "flex h-12 justify-between items-center"
          }, [createElementVNode(
            "span",
            _hoisted_10,
            toDisplayString(item.label),
            1
            /* TEXT */
          ), createVNode(_component_icon_error, {
            "custom-class": "w-5 h-5 cursor-pointer fill-color-none-hover",
            onClick: function onClick($event) {
              return _ctx.deleteSelected(item);
            }
          }, null, 8, ["onClick"])]);
        }),
        128
        /* KEYED_FRAGMENT */
      ))])]), createElementVNode("div", _hoisted_11, [createVNode(_component_tiny_button, {
        type: "primary",
        "custom-class": "w-full",
        onClick: _ctx.selectedBoxConfirm
      }, {
        default: withCtx(function() {
          return [createTextVNode(
            toDisplayString(_ctx.t("ui.base.confirm")),
            1
            /* TEXT */
          )];
        }),
        _: 1
        /* STABLE */
      }, 8, ["onClick"])])],
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true)],
    38
    /* CLASS, STYLE, NEED_HYDRATION */
  )]));
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
