// @opentiny/vue-tag-group@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var tagGroupProps = _extends({}, $props, {
  size: {
    type: String,
    default: "medium",
    validator: function validator(value) {
      return ["medium", "small", "mini"].includes(value);
    }
  },
  data: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  effect: {
    type: String,
    default: "light",
    validator: function validator2(value) {
      return ["dark", "light", "plain"].includes(value);
    }
  }
});
var TagGroup = defineComponent({
  name: $prefix + "TagGroup",
  props: tagGroupProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
TagGroup.install = function(Vue) {
  Vue.component(TagGroup.name, TagGroup);
};
TagGroup.version = version;
export {
  TagGroup as default
};
