// @opentiny/vue-tag-group@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/tag-group/vue';
import { defineComponent, $prefix, $props, setup } from '@opentiny/vue-common';
import { IconEllipsis } from '@opentiny/vue-icon';
import Popover from '@opentiny/vue-popover';
import Tag from '@opentiny/vue-tag';
import { resolveComponent, openBlock, createElementBlock, Fragment, renderList, createBlock, withCtx, createTextVNode, toDisplayString, createVNode, createElementVNode, normalizeClass } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "TagGroup",
  emits: ["item-click"],
  props: _extends({}, $props, {
    size: {
      type: String,
      default: "medium",
      validator: function validator(value) {
        return ~["medium", "small", "mini"].indexOf(value);
      }
    },
    data: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    effect: {
      type: String,
      default: "light",
      validator: function validator2(value) {
        return ~["dark", "light", "plain"].indexOf(value);
      }
    }
  }),
  components: {
    TinyTag: Tag,
    TinyPopover: Popover,
    IconEllipsis: IconEllipsis()
  },
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  ref: "tagGroup",
  "data-tag": "tiny-tag-group",
  class: "overflow-hidden relative pr-8"
};
var _hoisted_2 = {
  "data-tag": "tiny-tag-group-popover"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_tiny_tag = resolveComponent("tiny-tag");
  var _component_icon_ellipsis = resolveComponent("icon-ellipsis");
  var _component_tiny_popover = resolveComponent("tiny-popover");
  return openBlock(), createElementBlock(
    "div",
    _hoisted_1,
    [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.data, function(item, i) {
        return openBlock(), createBlock(_component_tiny_tag, {
          key: item.name + i,
          size: _ctx.size,
          type: item.type || "info",
          effect: _ctx.effect,
          onClick: function onClick($event) {
            return _ctx.handelItemClick(item, i, $event);
          }
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(item.name),
              1
              /* TEXT */
            )];
          }),
          _: 2
          /* DYNAMIC */
        }, 1032, ["size", "type", "effect", "onClick"]);
      }),
      128
      /* KEYED_FRAGMENT */
    )), createVNode(_component_tiny_popover, {
      placement: "top-start",
      trigger: "hover",
      class: "absolute top-0 right-0"
    }, {
      reference: withCtx(function() {
        return [createElementVNode(
          "div",
          {
            class: normalizeClass(["absolute top-0 right-0 pointer-events-none sm:pointer-events-auto", [_ctx.state.showMore ? "block" : "invisible"]])
          },
          [createVNode(_component_tiny_tag, {
            ref: "more",
            size: _ctx.size,
            type: "info",
            customClass: "px-1.5 mr-0"
          }, {
            default: withCtx(function() {
              return [createVNode(_component_icon_ellipsis, {
                class: "fill-color-icon-secondary"
              })];
            }),
            _: 1
            /* STABLE */
          }, 8, ["size"])],
          2
          /* CLASS */
        )];
      }),
      default: withCtx(function() {
        return [createElementVNode("div", _hoisted_2, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.hiddenTags, function(tag, i) {
            return openBlock(), createBlock(_component_tiny_tag, {
              key: tag.name + i,
              size: _ctx.size,
              type: tag.type || "info",
              customClass: "mb-2",
              onClick: function onClick($event) {
                return _ctx.handelItemClick(tag, _ctx.data.length - _ctx.state.hiddenTags.length + i, $event);
              }
            }, {
              default: withCtx(function() {
                return [createTextVNode(
                  toDisplayString(tag.name),
                  1
                  /* TEXT */
                )];
              }),
              _: 2
              /* DYNAMIC */
            }, 1032, ["size", "type", "onClick"]);
          }),
          128
          /* KEYED_FRAGMENT */
        ))])];
      }),
      _: 1
      /* STABLE */
    })],
    512
    /* NEED_PATCH */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
