// @opentiny/vue-tag-group@3.20.0/lib/pc.js
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/tag-group/vue';
import { IconEllipsis } from '@opentiny/vue-icon';
import Popover from '@opentiny/vue-popover';
import Tag from '@opentiny/vue-tag';
import '@opentiny/vue-theme/tag-group/index.css';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, Fragment, renderList, createBlock, withCtx, createTextVNode, toDisplayString, createVNode, createElementVNode, createCommentVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "TagGroup",
  components: {
    TinyTag: Tag,
    TinyPopover: Popover,
    IconEllipsis: IconEllipsis()
  },
  props: [].concat(props, ["size", "data", "effect"]),
  emits: ["item-click"],
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_tag = resolveComponent("tiny-tag");
  var _component_icon_ellipsis = resolveComponent("icon-ellipsis");
  var _component_tiny_popover = resolveComponent("tiny-popover");
  return openBlock(), createElementBlock(
    "div",
    {
      ref: "tagGroup",
      class: normalizeClass(["tiny-tag-group", {
        "tiny-tag-group-has-more": _ctx.state.showMore
      }])
    },
    [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.data, function(item, i) {
        return openBlock(), createBlock(_component_tiny_tag, {
          key: item.name + i,
          size: _ctx.size,
          type: item.type,
          effect: _ctx.effect,
          onClick: function onClick($event) {
            return _ctx.handelItemClick(item, i, $event);
          }
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(item.name),
              1
              /* TEXT */
            )];
          }),
          _: 2
          /* DYNAMIC */
        }, 1032, ["size", "type", "effect", "onClick"]);
      }),
      128
      /* KEYED_FRAGMENT */
    )), _ctx.state.showMore ? (openBlock(), createBlock(_component_tiny_popover, {
      key: 0,
      placement: "top-start",
      trigger: "hover",
      width: "auto",
      "popper-class": "tiny-tag-group-popper"
    }, {
      reference: withCtx(function() {
        return [createVNode(_component_tiny_tag, {
          ref: "more",
          size: _ctx.size,
          class: "tiny-tag-group_icon"
        }, {
          default: withCtx(function() {
            return [createVNode(_component_icon_ellipsis)];
          }),
          _: 1
          /* STABLE */
        }, 8, ["size"])];
      }),
      default: withCtx(function() {
        return [createElementVNode("div", null, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.hiddenTags, function(tag, i) {
            return openBlock(), createBlock(_component_tiny_tag, {
              key: tag.name + i,
              size: _ctx.size,
              type: tag.type,
              onClick: function onClick($event) {
                return _ctx.handelItemClick(tag, _ctx.data.length - _ctx.state.hiddenTags.length + i, $event);
              }
            }, {
              default: withCtx(function() {
                return [createTextVNode(
                  toDisplayString(tag.name),
                  1
                  /* TEXT */
                )];
              }),
              _: 2
              /* DYNAMIC */
            }, 1032, ["size", "type", "onClick"]);
          }),
          128
          /* KEYED_FRAGMENT */
        ))])];
      }),
      _: 1
      /* STABLE */
    })) : createCommentVNode("v-if", true)],
    2
    /* CLASS */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
