// @opentiny/vue-card@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
import { IconEllipsis } from "@opentiny/vue-icon";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var $constants = {
  CARD_GROUP: "CardGroup"
};
var cardProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  modelValue: {},
  src: String,
  title: String,
  customClass: String,
  height: String,
  label: {},
  autoWidth: {
    type: Boolean,
    default: false
  },
  options: {
    type: Array,
    default: function _default2() {
      return [];
    }
  },
  iconMore: {
    type: [Object, String],
    default: function _default3() {
      return IconEllipsis();
    }
  },
  size: {
    type: String,
    default: "medium"
  },
  status: {
    type: String,
    default: "default"
  },
  checkType: {
    type: String,
    default: ""
  },
  type: {
    type: String,
    default: "text"
  },
  disabled: {
    type: Boolean,
    default: false
  },
  checkMode: {
    type: String,
    default: "normal"
  }
});
var Card = defineComponent({
  name: $prefix + "Card",
  props: cardProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Card.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Card.install = function(Vue) {
  Vue.component(Card.name, Card);
};
Card.version = version;
export {
  Card as default
};
