// @opentiny/vue-statistic@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/statistic/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  PREFIX: "tiny-statistic"
};
var statisticProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  precision: {
    type: Number,
    default: 0
  },
  formatter: Function,
  value: {
    type: Number,
    default: 0
  },
  prefix: String,
  suffix: String,
  title: [String, Object],
  valueStyle: {
    type: [Object, Array]
  },
  groupSeparator: {
    type: String,
    default: ","
  }
});
var Statistic = defineComponent({
  name: $prefix + "Statistic",
  props: statisticProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Statistic.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Statistic.install = function(Vue) {
  Vue.component(Statistic.name, Statistic);
};
Statistic.version = version;
export {
  Statistic as default
};
