// @opentiny/vue-statistic@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/statistic/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { openBlock, createElementBlock, renderSlot, toDisplayString, createCommentVNode, createElementVNode, normalizeClass, normalizeStyle } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {},
  emits: [],
  props: [].concat(props, ["formatter", "precision", "prefix", "suffix", "title", "value", "valueStyle", "groupSeparator"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-statistic"
};
var _hoisted_2 = {
  key: 0,
  class: "tiny-statistic__title"
};
var _hoisted_3 = {
  key: 0
};
var _hoisted_4 = {
  key: 1
};
var _hoisted_5 = {
  key: 2
};
var _hoisted_6 = {
  key: 0,
  class: "tiny-statistic__prefix"
};
var _hoisted_7 = {
  key: 1,
  class: "tiny-statistic__suffix"
};
var _hoisted_8 = {
  key: 1,
  class: /* @__PURE__ */ normalizeClass(["tiny-statistic__title"])
};
var _hoisted_9 = {
  key: 0
};
var _hoisted_10 = {
  key: 1
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [_ctx.title && (!(_ctx.title instanceof Object) || _ctx.title instanceof Object && _ctx.title.position !== "bottom") ? (openBlock(), createElementBlock("div", _hoisted_2, [_ctx.$slots.title ? (openBlock(), createElementBlock("div", _hoisted_3, [renderSlot(_ctx.$slots, "title")])) : typeof _ctx.title === "string" ? (openBlock(), createElementBlock(
    "div",
    _hoisted_4,
    toDisplayString(_ctx.title),
    1
    /* TEXT */
  )) : _ctx.title instanceof Object ? (openBlock(), createElementBlock(
    "div",
    _hoisted_5,
    toDisplayString(_ctx.title.value),
    1
    /* TEXT */
  )) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true), createElementVNode(
    "div",
    {
      class: normalizeClass(["tiny-statistic__slots", _ctx.title && _ctx.title.position === "bottom" ? "tiny-statistic__description-margin" : ""])
    },
    [_ctx.$slots.prefix || _ctx.prefix ? (openBlock(), createElementBlock("div", _hoisted_6, [renderSlot(_ctx.$slots, "prefix", {}, function() {
      return [createElementVNode(
        "span",
        null,
        toDisplayString(_ctx.prefix),
        1
        /* TEXT */
      )];
    })])) : createCommentVNode("v-if", true), createElementVNode(
      "div",
      {
        class: normalizeClass(["tiny-statistic__description"]),
        style: normalizeStyle(_ctx.valueStyle)
      },
      toDisplayString(_ctx.state.value),
      5
      /* TEXT, STYLE */
    ), _ctx.$slots.suffix || _ctx.suffix ? (openBlock(), createElementBlock("div", _hoisted_7, [renderSlot(_ctx.$slots, "suffix", {}, function() {
      return [createElementVNode(
        "span",
        null,
        toDisplayString(_ctx.suffix),
        1
        /* TEXT */
      )];
    })])) : createCommentVNode("v-if", true)],
    2
    /* CLASS */
  ), _ctx.title && _ctx.title instanceof Object && _ctx.title.position === "bottom" ? (openBlock(), createElementBlock("div", _hoisted_8, [_ctx.$slots.title ? (openBlock(), createElementBlock("div", _hoisted_9, [renderSlot(_ctx.$slots, "title")])) : (openBlock(), createElementBlock("div", _hoisted_10, [createElementVNode(
    "span",
    null,
    toDisplayString(_ctx.title.value),
    1
    /* TEXT */
  )]))])) : createCommentVNode("v-if", true)]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
