// @opentiny/vue-drawer@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/drawer/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { IconClose } from '@opentiny/vue-icon';
import Button from '@opentiny/vue-button';
import { resolveComponent, withDirectives, openBlock, createElementBlock, normalizeClass, createCommentVNode, createElementVNode, normalizeStyle, renderSlot, toDisplayString, createVNode, withCtx, createTextVNode, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    TinyButton: Button,
    IconClose: IconClose()
  },
  props: [].concat(props, ["visible", "title", "showHeader", "showFooter", "customClass", "placement", "width", "mask", "dragable", "maskClosable", "lockScroll", "flex", "beforeClose"]),
  emits: ["update:visible", "open", "close", "confirm"],
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-drawer",
  class: "text-sm"
};
var _hoisted_2 = {
  class: /* @__PURE__ */ normalizeClass(["flex-auto flex-col flex max-h-full overflow-hidden"])
};
var _hoisted_3 = {
  key: 0,
  "data-tag": "drawer-header",
  ref: "header",
  class: "flex-none flex leading-6 p-4 text-base items-center"
};
var _hoisted_4 = {
  key: 0,
  class: "max-w-[80%] pr-4 text-left truncate text-color-text-primary"
};
var _hoisted_5 = {
  class: "flex-1 flex items-center justify-end"
};
var _hoisted_6 = {
  key: 1,
  "data-tag": "drawer-footer",
  ref: "footer",
  class: "px-4 py-3"
};
var _hoisted_7 = {
  class: "flex-1 text-right"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_IconClose = resolveComponent("IconClose");
  var _component_tiny_button = resolveComponent("tiny-button");
  return withDirectives((openBlock(), createElementBlock(
    "div",
    _hoisted_1,
    [_ctx.mask ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        ref: "mask",
        class: normalizeClass(_ctx.m("fixed z-50 left-0 right-0 top-0 bottom-0 w-full h-full transition ease-linear duration-200", {
          "bg-color-bg-7": _ctx.state.visible
        })),
        onClick: _cache[0] || (_cache[0] = function($event) {
          return _ctx.handleClose("mask");
        })
      },
      null,
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true), createElementVNode(
      "div",
      {
        "data-tag": "tiny-drawer-main",
        ref: "drawerBox",
        class: normalizeClass(_ctx.m("fixed w-full max-w-full flex flex-col bg-color-bg-1 z-50 shadow-sm border-color-border-separator", {
          "transition-all ease-linear duration-200": !_ctx.state.dragEvent.isDrag
        }, {
          "h-full": ["left", "right"].includes(_ctx.placement)
        }, {
          "max-h-full": ["top", "bottom"].includes(_ctx.placement)
        }, {
          "left-0 bottom-0 translate-y-full border-t-0.5 rounded-t-lg": _ctx.placement === "bottom"
        }, {
          "left-0 top-0 -translate-y-full border-b-0.5 rounded-b-lg": _ctx.placement === "top"
        }, {
          "translate-y-0": ["top", "bottom"].includes(_ctx.placement) && _ctx.state.visible
        }, {
          "left-0 top-0 -translate-x-full border-r-0.5 rounded-r-lg": _ctx.placement === "left"
        }, {
          "right-0 top-0 translate-x-full border-l-0.5 rounded-l-lg": _ctx.placement === "right"
        }, {
          "translate-x-0": ["left", "right"].includes(_ctx.placement) && _ctx.state.visible
        }, _ctx.customClass)),
        style: normalizeStyle({
          width: ["left", "right"].includes(_ctx.placement) ? _ctx.state.computedWidth : null,
          height: ["top", "bottom"].includes(_ctx.placement) && _ctx.dragable && _ctx.state.height ? _ctx.state.height + "px" : null
        })
      },
      [_ctx.dragable ? (openBlock(), createElementBlock(
        "div",
        {
          key: 0,
          "data-tag": "drawer-drag-bar",
          ref: "dragBar",
          class: normalizeClass(["absolute", ["left", "right"].includes(_ctx.placement) && "cursor-e-resize h-full top-0 w-2", ["top", "bottom"].includes(_ctx.placement) && "cursor-n-resize w-full h-2 left-0", _ctx.placement === "left" && "-right-1", _ctx.placement === "right" && "-left-1", _ctx.placement === "top" && "-bottom-1", _ctx.placement === "bottom" && "-top-1"])
        },
        null,
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_2, [_ctx.showHeader ? (openBlock(), createElementBlock(
        "div",
        _hoisted_3,
        [renderSlot(_ctx.$slots, "header", {}, function() {
          return [_ctx.title ? (openBlock(), createElementBlock(
            "div",
            _hoisted_4,
            toDisplayString(_ctx.title),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_5, [renderSlot(_ctx.$slots, "header-right", {}, function() {
            return [createVNode(_component_IconClose, {
              "custom-class": "h-5 w-5 cursor-pointer",
              class: "fill-color-icon-primary",
              onClick: _cache[1] || (_cache[1] = function($event) {
                return _ctx.handleClose("close");
              })
            })];
          })])];
        })],
        512
        /* NEED_PATCH */
      )) : createCommentVNode("v-if", true), createElementVNode(
        "div",
        {
          "data-tag": "drawer-body",
          ref: "body",
          class: normalizeClass(["flex-auto overflow-auto", {
            "flex flex-col": _ctx.flex
          }])
        },
        [renderSlot(_ctx.$slots, "default")],
        2
        /* CLASS */
      ), _ctx.showFooter ? (openBlock(), createElementBlock(
        "div",
        _hoisted_6,
        [createElementVNode("div", _hoisted_7, [renderSlot(_ctx.$slots, "footer", {}, function() {
          return [createVNode(_component_tiny_button, {
            tiny_mode: "mobile-first",
            onClick: _cache[2] || (_cache[2] = function($event) {
              return _ctx.handleClose("cancel");
            })
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.button.cancel")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          }), createVNode(_component_tiny_button, {
            tiny_mode: "mobile-first",
            class: "ml-2",
            type: "primary",
            onClick: _cache[3] || (_cache[3] = function($event) {
              return _ctx.handleClose("confirm");
            })
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.button.confirm")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          })];
        })])],
        512
        /* NEED_PATCH */
      )) : createCommentVNode("v-if", true)])],
      6
      /* CLASS, STYLE */
    )],
    512
    /* NEED_PATCH */
  )), [[vShow, _ctx.visible]]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
