// @opentiny/vue-month-range@3.20.0/lib/index.js
import { renderless, api } from '@opentiny/vue-renderless/month-range/vue';
import { defineComponent, $prefix, directive, setup, $props } from '@opentiny/vue-common';
import Clickoutside from '@opentiny/vue-renderless/common/deps/clickoutside';
import MonthTable from '@opentiny/vue-month-table';
import { iconDoubleRight, iconDoubleLeft } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, renderSlot, createElementBlock, Fragment, renderList, toDisplayString, createCommentVNode, createVNode, vShow } from 'vue';
import '@opentiny/vue-theme/month-range/index.css';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "MonthRange",
  directives: directive({
    Clickoutside
  }),
  components: {
    MonthTable,
    IconDoubleRight: iconDoubleRight(),
    IconDoubleLeft: iconDoubleLeft()
  },
  props: _extends({}, $props, {
    emitter: Object
  }),
  emits: ["dodestroy", "pick"],
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  class: "tiny-picker-panel__body-wrapper"
};
var _hoisted_2 = {
  key: 0,
  class: "tiny-picker-panel__sidebar"
};
var _hoisted_3 = ["onClick", "title"];
var _hoisted_4 = {
  class: "tiny-picker-panel__body"
};
var _hoisted_5 = {
  class: "tiny-picker-panel__content tiny-date-range-picker__content is-left"
};
var _hoisted_6 = {
  class: "tiny-date-range-picker__header"
};
var _hoisted_7 = ["disabled"];
var _hoisted_8 = {
  class: "tiny-date-range-picker__table"
};
var _hoisted_9 = {
  class: "tiny-picker-panel__content tiny-date-range-picker__content is-right"
};
var _hoisted_10 = {
  class: "tiny-date-range-picker__header"
};
var _hoisted_11 = ["disabled"];
var _hoisted_12 = {
  class: "tiny-date-range-picker__table"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_icon_double_left = resolveComponent("icon-double-left");
  var _component_icon_double_right = resolveComponent("icon-double-right");
  var _component_month_table = resolveComponent("month-table");
  return openBlock(), createBlock(Transition, {
    name: "tiny-zoom-in-top",
    onAfterLeave: _cache[4] || (_cache[4] = function($event) {
      return _ctx.$emit("dodestroy");
    }),
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          class: normalizeClass(["tiny-month-range tiny-picker-panel tiny-date-range-picker tiny-popper", [{
            "has-sidebar": _ctx.slots.sidebar || _ctx.state.shortcuts
          }, _ctx.state.popperClass]])
        },
        [createElementVNode("div", _hoisted_1, [renderSlot(_ctx.$slots, "sidebar", {
          class: "tiny-picker-panel__sidebar"
        }), _ctx.state.shortcuts ? (openBlock(), createElementBlock("div", _hoisted_2, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.shortcuts, function(shortcut, key) {
            return openBlock(), createElementBlock("button", {
              type: "button",
              class: "tiny-picker-panel__shortcut",
              key,
              onClick: function onClick($event) {
                return _ctx.handleShortcutClick(shortcut);
              },
              title: shortcut.text
            }, toDisplayString(shortcut.text), 9, _hoisted_3);
          }),
          128
          /* KEYED_FRAGMENT */
        ))])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_4, [createElementVNode("div", _hoisted_5, [createElementVNode("div", _hoisted_6, [createElementVNode("button", {
          type: "button",
          onClick: _cache[0] || (_cache[0] = function() {
            return _ctx.leftPrevYear && _ctx.leftPrevYear.apply(_ctx, arguments);
          }),
          class: "tiny-picker-panel__icon-btn tiny-icon-d-arrow-left"
        }, [createVNode(_component_icon_double_left)]), _ctx.state.unlinkPanels ? (openBlock(), createElementBlock("button", {
          key: 0,
          type: "button",
          onClick: _cache[1] || (_cache[1] = function() {
            return _ctx.leftNextYear && _ctx.leftNextYear.apply(_ctx, arguments);
          }),
          disabled: !_ctx.state.enableYearArrow,
          class: normalizeClass([{
            "is-disabled": !_ctx.state.enableYearArrow
          }, "tiny-picker-panel__icon-btn tiny-icon-d-arrow-right"])
        }, [createVNode(_component_icon_double_right)], 10, _hoisted_7)) : createCommentVNode("v-if", true), createElementVNode(
          "div",
          null,
          toDisplayString(_ctx.state.leftLabel),
          1
          /* TEXT */
        )]), createElementVNode("div", _hoisted_8, [createVNode(_component_month_table, {
          "selection-mode": "range",
          date: _ctx.state.leftDate,
          "default-value": _ctx.state.defaultValue,
          "min-date": _ctx.state.minDate,
          "max-date": _ctx.state.maxDate,
          "range-state": _ctx.state.rangeState,
          "disabled-date": _ctx.state.disabledDate,
          onChangerange: _ctx.handleChangeRange,
          onPick: _ctx.handleRangePick
        }, null, 8, ["date", "default-value", "min-date", "max-date", "range-state", "disabled-date", "onChangerange", "onPick"])])]), createElementVNode("div", _hoisted_9, [createElementVNode("div", _hoisted_10, [_ctx.state.unlinkPanels ? (openBlock(), createElementBlock("button", {
          key: 0,
          type: "button",
          onClick: _cache[2] || (_cache[2] = function() {
            return _ctx.rightPrevYear && _ctx.rightPrevYear.apply(_ctx, arguments);
          }),
          disabled: !_ctx.state.enableYearArrow,
          class: normalizeClass([{
            "is-disabled": !_ctx.state.enableYearArrow
          }, "tiny-picker-panel__icon-btn tiny-icon-d-arrow-left"])
        }, [createVNode(_component_icon_double_left)], 10, _hoisted_11)) : createCommentVNode("v-if", true), createElementVNode("button", {
          type: "button",
          onClick: _cache[3] || (_cache[3] = function() {
            return _ctx.rightNextYear && _ctx.rightNextYear.apply(_ctx, arguments);
          }),
          class: "tiny-picker-panel__icon-btn tiny-icon-d-arrow-right"
        }, [createVNode(_component_icon_double_right)]), createElementVNode(
          "div",
          null,
          toDisplayString(_ctx.state.rightLabel),
          1
          /* TEXT */
        )]), createElementVNode("div", _hoisted_12, [createVNode(_component_month_table, {
          "selection-mode": "range",
          date: _ctx.state.rightDate,
          "default-value": _ctx.state.defaultValue,
          "min-date": _ctx.state.minDate,
          "max-date": _ctx.state.maxDate,
          "range-state": _ctx.state.rangeState,
          "disabled-date": _ctx.state.disabledDate,
          onChangerange: _ctx.handleChangeRange,
          onPick: _ctx.handleRangePick
        }, null, 8, ["date", "default-value", "min-date", "max-date", "range-state", "disabled-date", "onChangerange", "onPick"])])])])])],
        2
        /* CLASS */
      ), [[vShow, _ctx.state.visible]])];
    }),
    _: 3
    /* FORWARDED */
  });
}
var MonthRange = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.20.0";
MonthRange.install = function(Vue) {
  Vue.component(MonthRange.name, MonthRange);
};
MonthRange.version = version;

export { MonthRange as default };
