// @opentiny/vue-record@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var $constants = {
  STATUS: {
    READY: "ready",
    RECORDING: "recording",
    PAUSE: "pause"
  },
  HWH5_STATUS: {
    START: "start",
    PAUSE: "pause",
    CONTINUE: "continue",
    CANCEL: "cancel",
    END: "end"
  }
};
var Record = defineComponent({
  name: $prefix + "Record",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    modelValue: Boolean,
    isHwh5: {
      type: Boolean,
      default: false
    },
    lockScroll: {
      type: Boolean,
      default: true
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Record.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Record.install = function(Vue) {
  Vue.component(Record.name, Record);
};
Record.version = version;
export {
  Record as default
};
