// @opentiny/vue-floating-button@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/floating-button/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { iconPlus } from '@opentiny/vue-icon';
import { openBlock, createElementBlock, normalizeClass, createCommentVNode, createElementVNode, normalizeStyle, createBlock, resolveDynamicComponent, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["click"],
  props: [].concat(props, ["position", "backgroundColor", "icon", "resetTime", "animated", "expandList", "isExpand", "vague", "mask", "elementSelector"]),
  components: {
    IconPlus: iconPlus()
  },
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-floating-button-body"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [_ctx.vague || _ctx.mask ? (openBlock(), createElementBlock(
    "div",
    {
      key: 0,
      "data-tag": "tiny-floating-button-mask",
      class: normalizeClass(["w-screen h-screen fixed top-0 bottom-0 left-0 right-0", {
        "bg-color-fill-5 backdrop-blur-md": _ctx.vague
      }, {
        "bg-black opacity-30": _ctx.mask
      }])
    },
    null,
    2
    /* CLASS */
  )) : createCommentVNode("v-if", true), createElementVNode(
    "div",
    {
      "data-tag": "tiny-floating-button",
      class: normalizeClass(["rounded-full fixed bottom-11 leading-none", _ctx.position === "center" ? "" : _ctx.position === "left" ? !_ctx.state.specialHiddenSpace && "left-3" : "right-3"]),
      style: normalizeStyle(_ctx.state.style)
    },
    [createElementVNode(
      "button",
      {
        "data-tag": "tiny-floating-button-main",
        onClick: _cache[0] || (_cache[0] = function($event) {
          return _ctx.handleClick($event, 0);
        }),
        ref: "floatingButton",
        class: normalizeClass(_ctx.m("w-11 h-11 p-3 rounded-full", _ctx.state.disabled ? "text-color-text-disabled bg-color-bg-3 border-transparent active:cursor-not-allowed" : _ctx.backgroundColor === "white" ? "bg-color-bg-1" : "text-color-bg-1 bg-color-brand"))
      },
      [(openBlock(), createBlock(resolveDynamicComponent(_ctx.icon), {
        "data-tag": "tiny-floating-button-icon",
        class: normalizeClass(_ctx.m("w-5 h-5 rotate-0 transition-transform duration-300 will-change-transform", _ctx.state.disabled ? "fill-color-icon-disabled active:cursor-not-allowed" : _ctx.backgroundColor === "white" ? "fill-color-icon-secondary" : "fill-color-bg-1", {
          "rotate-45": _ctx.state.isExpand
        }))
      }, null, 8, ["class"]))],
      2
      /* CLASS */
    ), _ctx.state.expandList && _ctx.state.expandList.length >= 1 ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        "data-tag": "tiny-floating-button-expand-first",
        onClick: _cache[1] || (_cache[1] = function($event) {
          return _ctx.handleClick($event, 1);
        }),
        class: normalizeClass(_ctx.m("absolute -bottom-5 -mb-0.5 -left-0.5 -z-10 transition-bottom ease-linear duration-100 cursor-pointer", {
          "bottom-18 [&_span]:opacity-100 [&_span]:visible": _ctx.state.isExpand && _ctx.position === "right"
        }, {
          "bottom-16 -mb-1 [&_span]:opacity-100 [&_span]:visible": _ctx.state.isExpand && _ctx.position === "center"
        }, {
          "bottom-18 [&_span]:opacity-100 [&_span]:visible": _ctx.state.isExpand && _ctx.position === "left"
        }))
      },
      [createElementVNode(
        "button",
        {
          class: normalizeClass(_ctx.m("w-11 h-11 rounded-full mx-0.5", _ctx.state.disabled ? "text-color-text-disabled bg-color-bg-3 border-transparent active:cursor-not-allowed" : "bg-color-info-primary-subtler"))
        },
        [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.expandList && _ctx.state.expandList[0].icon), {
          "data-tag": "tiny-floating-button-icon",
          class: normalizeClass(["w-5 h-5", _ctx.state.disabled ? "fill-color-icon-disabled active:cursor-not-allowed" : "fill-color-brand"])
        }, null, 8, ["class"]))],
        2
        /* CLASS */
      ), createElementVNode(
        "span",
        {
          class: normalizeClass(["text-center mt-0.5 text-xs w-12 leading-5 block opacity-0 transition-opacity ease-linear duration-100 invisible", !_ctx.mask ? "text-color-fill-primary" : "text-color-bg-1"])
        },
        toDisplayString(_ctx.state.expandList && _ctx.state.expandList[0].title),
        3
        /* TEXT, CLASS */
      )],
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true), _ctx.state.expandList && _ctx.state.expandList.length >= 2 ? (openBlock(), createElementBlock(
      "div",
      {
        key: 1,
        "data-tag": "tiny-floating-button-expand-second",
        onClick: _cache[2] || (_cache[2] = function($event) {
          return _ctx.handleClick($event, 2);
        }),
        class: normalizeClass(_ctx.m("absolute -bottom-5 -mb-0.5 -left-0.5 -z-10 transition-bottom ease-linear duration-100 cursor-pointer", {
          "-left-18 bottom-11 mb-0 [&_span]:opacity-100 [&_span]:visible": _ctx.state.isExpand && _ctx.position === "right"
        }, {
          "-left-18 right-auto bottom-4.5 mb-0 [&_span]:opacity-100 [&_span]:visible": _ctx.state.isExpand && _ctx.position === "center"
        }, {
          "left-18 right-auto bottom-11 mb-0 [&_span]:opacity-100 [&_span]:visible": _ctx.state.isExpand && _ctx.position === "left"
        }))
      },
      [createElementVNode(
        "button",
        {
          class: normalizeClass(_ctx.m("w-11 h-11 rounded-full mx-0.5", _ctx.state.disabled ? "text-color-text-disabled bg-color-bg-3 border-transparent active:cursor-not-allowed" : "bg-color-success-subtler"))
        },
        [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.expandList && _ctx.state.expandList[1].icon), {
          "data-tag": "tiny-floating-button-icon",
          class: normalizeClass(["w-5 h-5", _ctx.state.disabled ? "fill-color-icon-disabled active:cursor-not-allowed" : "fill-color-success-active"])
        }, null, 8, ["class"]))],
        2
        /* CLASS */
      ), createElementVNode(
        "span",
        {
          class: normalizeClass(["text-center text-xs mt-0.5 w-12 leading-5 block opacity-0 transition-opacity ease-linear duration-100 invisible", !_ctx.mask ? "text-color-fill-primary" : "text-color-bg-1"])
        },
        toDisplayString(_ctx.state.expandList && _ctx.state.expandList[1].title),
        3
        /* TEXT, CLASS */
      )],
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true), _ctx.state.expandList && _ctx.state.expandList.length >= 3 ? (openBlock(), createElementBlock(
      "div",
      {
        key: 2,
        "data-tag": "tiny-floating-button-expand-third",
        onClick: _cache[3] || (_cache[3] = function($event) {
          return _ctx.handleClick($event, 3);
        }),
        class: normalizeClass(_ctx.m("absolute -bottom-5 -mb-0.5 -left-0.5 -z-10 transition-bottom ease-linear duration-100 cursor-pointer", {
          "-left-18 -bottom-9 -mb-1 [&_span]:opacity-100 [&_span]:visible": _ctx.state.isExpand && _ctx.position === "right"
        }, {
          "left-18 right-auto bottom-4.5 mb-0 [&_span]:opacity-100 [&_span]:visible": _ctx.state.isExpand && _ctx.position === "center"
        }, {
          "left-18 -bottom-9 -mb-1 right-auto [&_span]:opacity-100 [&_span]:visible": _ctx.state.isExpand && _ctx.position === "left"
        }))
      },
      [createElementVNode(
        "button",
        {
          class: normalizeClass(_ctx.m("w-11 h-11 rounded-full mx-0.5", _ctx.state.disabled ? "text-color-text-disabled bg-color-bg-3 border-transparent active:cursor-not-allowed" : "bg-color-warning-subtler"))
        },
        [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.expandList && _ctx.state.expandList[2].icon), {
          "data-tag": "tiny-floating-button-icon",
          class: normalizeClass(["w-5 h-5", _ctx.state.disabled ? "fill-color-icon-disabled active:cursor-not-allowed" : "fill-color-alert-hover"])
        }, null, 8, ["class"]))],
        2
        /* CLASS */
      ), createElementVNode(
        "span",
        {
          class: normalizeClass(["text-center text-xs mt-0.5 w-12 leading-5 block opacity-0 transition-opacity ease-linear duration-100 invisible", !_ctx.mask ? "text-color-fill-primary" : "text-color-bg-1"])
        },
        toDisplayString(_ctx.state.expandList && _ctx.state.expandList[2].title),
        3
        /* TEXT, CLASS */
      )],
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true)],
    6
    /* CLASS, STYLE */
  )]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
