// @opentiny/fluent-editor@3.24.0/lib/global-link/formats/wiki-link.cjs.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const Quill = require("quill");
const constants = require("../constants.cjs.js");
const Embed = Quill.imports["blots/embed"];
class WikiLink extends Embed {
  static create(value) {
    const node = super.create(value);
    node.setAttribute("href", value.link);
    node.setAttribute("target", "_blank");
    node.setAttribute("title", value.text);
    node.textContent = value.text;
    return node;
  }
  static value(domNode) {
    return {
      link: domNode.getAttribute("href"),
      text: domNode.textContent
    };
  }
  constructor(scroll, domNode, data) {
    super(scroll, domNode, data);
    this.wikiData = data;
  }
  remove() {
    this.scroll.emitter.emit(constants.ON_WIKI_LINK_REMOVE, this.wikiData);
    return super.remove();
  }
}
WikiLink.blotName = "wiki-link";
WikiLink.tagName = "A";
WikiLink.className = "ql-wiki-link";
exports.default = WikiLink;
