// @opentiny/vue-year-table@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/year-table/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import '@opentiny/vue-theme/year-table/index.css';
import { openBlock, createElementBlock, createElementVNode, Fragment, renderList, normalizeClass, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["pick", "changerange"],
  props: [].concat(props, ["disabledDate", "value", "defaultValue", "date", "selectionMode", "startYear", "maxDate", "minDate", "rangeState"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["aria-disabled"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "table",
    {
      class: "tiny-year-table",
      onMousemove: _cache[0] || (_cache[0] = function() {
        return _ctx.handleMouseMove && _ctx.handleMouseMove.apply(_ctx, arguments);
      }),
      onClick: _cache[1] || (_cache[1] = function() {
        return _ctx.handleYearTableClick && _ctx.handleYearTableClick.apply(_ctx, arguments);
      })
    },
    [createElementVNode("tbody", null, [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.rows, function(row, trKey) {
        return openBlock(), createElementBlock("tr", {
          key: trKey
        }, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(row, function(cell, tdKey) {
            return openBlock(), createElementBlock(
              "td",
              {
                class: normalizeClass(["available", [{
                  "disabled": _ctx.getIsDisabled(cell.text),
                  "current": _ctx.getIsCurrent(cell.text),
                  "default": _ctx.getIsDefault(cell.text),
                  "today": _ctx.state.currentYear === cell.text,
                  "in-range": cell.inRange,
                  "start-date": cell.start,
                  "end-date": cell.end
                }]]),
                key: tdKey
              },
              [createElementVNode("div", null, [createElementVNode("a", {
                class: "cell",
                "aria-disabled": _ctx.getIsDisabled(cell.text) ? true : void 0
              }, toDisplayString(cell.text), 9, _hoisted_1)])],
              2
              /* CLASS */
            );
          }),
          128
          /* KEYED_FRAGMENT */
        ))]);
      }),
      128
      /* KEYED_FRAGMENT */
    ))])],
    32
    /* NEED_HYDRATION */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
