// @opentiny/vue-user-head-group@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var UserHeadGroup = defineComponent({
  name: $prefix + "UserHeadGroup",
  componentName: "TinyUserHeadGroup",
  props: _extends({}, $props, {
    groupSize: {
      type: Number,
      default: 40
    },
    list: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    showMax: {
      type: Boolean,
      default: false
    },
    horizontal: {
      type: Boolean,
      default: false
    },
    color: String,
    backgroundColor: String
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
UserHeadGroup.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
UserHeadGroup.install = function(Vue) {
  Vue.component(UserHeadGroup.name, UserHeadGroup);
};
UserHeadGroup.version = version;
export {
  UserHeadGroup as default
};
