// @opentiny/vue-rate@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/rate/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { IconStarActive, IconStarDisable, IconSmile, IconFrown, IconMeh } from '@opentiny/vue-icon';
import { openBlock, createElementBlock, normalizeClass, Fragment, renderList, createBlock, resolveDynamicComponent, normalizeStyle, createCommentVNode, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var classes = {
  "rate": "flex justify-start items-center text-center",
  "rate-item": "align-bottom flex-initial flex justify-center box-content [&>svg:nth-child(1)]:w-6 [&>svg:nth-child(1)]:h-6  px-2 sm:[&>svg:nth-child(1)]:w-4 sm:[&>svg:nth-child(1)]:h-4 sm:px-1",
  "rate-item-textOnBottom": "px-2 sm:px-1 inline-block",
  "cursor-disabled": "cursor-auto",
  "cursor-undisabled": "cursor-pointer",
  "svg-hover": "scale-[1.15]",
  "text-Bottom": "text-sm mt-2",
  "text-left": "text-color-text-primary text-sm  ml-6 leading-6 sm:ml-3 sm:leading-4",
  "allowhalf-item": "relative inline-block leading-none",
  "allowhalf-active": "inline-block  absolute top-0 left-1/2 -translate-x-1/2 w-6 h-6 sm:w-4 sm:h-4",
  "big-text-bottom": "text-base",
  "big-text-left": "text-color-text-primary sm:ml-5 sm:text-base sm:leading-6",
  "type-big": "sm:[&>svg:nth-child(1)]:w-6 sm:[&>svg:nth-child(1)]:h-6  sm:px-2"
};
var _sfc_main = defineComponent({
  emits: ["update:modelValue", "change"],
  props: [].concat(props, ["modelValue", "radio", "lowThreshold", "highThreshold", "max", "colors", "voidColor", "disabledVoidColor", "iconClasses", "voidIconClass", "disabledVoidIconClass", "disabled", "allowHalf", "showText", "showScore", "textColor", "texts", "textOnBottom", "type", "scoreTemplate"]),
  components: {
    IconStarActive: IconStarActive(),
    IconStarDisable: IconStarDisable(),
    IconSmile: IconSmile(),
    IconFrown: IconFrown(),
    IconMeh: IconMeh()
  },
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = ["onMousemove", "onClick"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-rate",
      class: normalizeClass(_ctx.m(_ctx.gcls("rate"))),
      tabindex: "0",
      onKeydown: _cache[1] || (_cache[1] = function() {
        return _ctx.handelKey && _ctx.handelKey.apply(_ctx, arguments);
      })
    },
    [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.max, function(item) {
        return openBlock(), createElementBlock("span", {
          "data-tag": "tiny-rate-content",
          key: item,
          class: normalizeClass(_ctx.m(_ctx.gcls({
            "allowhalf-item": _ctx.allowHalf
          }), _ctx.gcls("rate-item"), _ctx.gcls(_ctx.disabled ? "cursor-disabled" : "cursor-undisabled"), _ctx.gcls({
            "rate-item-textOnBottom": _ctx.textOnBottom
          }), _ctx.gcls({
            "type-big": _ctx.type === "large"
          }))),
          onMousemove: function onMousemove($event) {
            return _ctx.setCurrentValue({
              item,
              event: $event
            });
          },
          onMouseleave: _cache[0] || (_cache[0] = function() {
            return _ctx.resetCurrentValue && _ctx.resetCurrentValue.apply(_ctx, arguments);
          }),
          onClick: function onClick($event) {
            return _ctx.selectValue(item);
          }
        }, [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.classes[item - 1]), {
          style: normalizeStyle(_ctx.getIconStyle(item)),
          class: normalizeClass(_ctx.m(_ctx.gcls({
            "svg-hover": _ctx.state.hoverIndex === item
          })))
        }, null, 8, ["style", "class"])), _ctx.showDecimalIcon(item) ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.decimalIconClass), {
          key: 0,
          style: normalizeStyle(_ctx.state.decimalStyle),
          class: normalizeClass(_ctx.m(_ctx.gcls({
            "allowhalf-active": _ctx.allowHalf
          })))
        }, null, 8, ["style", "class"])) : createCommentVNode("v-if", true), _ctx.textOnBottom && !_ctx.radio ? (openBlock(), createElementBlock(
          "div",
          {
            key: 1,
            "data-tag": "tiny-rate-text-bottom",
            style: normalizeStyle({
              color: _ctx.textColor
            }),
            class: normalizeClass(_ctx.m(_ctx.gcls("text-Bottom"), _ctx.gcls({
              "big-text-bottom": _ctx.type === "large"
            })))
          },
          toDisplayString(_ctx.texts[item - 1]),
          7
          /* TEXT, CLASS, STYLE */
        )) : createCommentVNode("v-if", true), _ctx.textOnBottom && _ctx.radio ? (openBlock(), createElementBlock(
          "div",
          {
            key: 2,
            "data-tag": "tiny-rate-radio-bottom",
            style: normalizeStyle(_ctx.getTextStyle()),
            class: normalizeClass(_ctx.m(_ctx.gcls("text-Bottom"), _ctx.gcls({
              "big-text-bottom": _ctx.type === "large"
            })))
          },
          toDisplayString(_ctx.texts[item - 1]),
          7
          /* TEXT, CLASS, STYLE */
        )) : createCommentVNode("v-if", true)], 42, _hoisted_1);
      }),
      128
      /* KEYED_FRAGMENT */
    )), !_ctx.textOnBottom && _ctx.showText || _ctx.showScore ? (openBlock(), createElementBlock(
      "span",
      {
        key: 0,
        "data-tag": "tiny-rate-score",
        style: normalizeStyle({
          color: _ctx.textColor
        }),
        class: normalizeClass(_ctx.m(_ctx.gcls("text-left"), _ctx.gcls({
          "big-text-left": _ctx.type === "large"
        })))
      },
      toDisplayString(_ctx.state.text),
      7
      /* TEXT, CLASS, STYLE */
    )) : createCommentVNode("v-if", true)],
    34
    /* CLASS, NEED_HYDRATION */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
