// @opentiny/vue-panel@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Panel = defineComponent({
  name: $prefix + "Panel",
  props: _extends({}, $props, {
    expand: {
      type: Boolean,
      default: true
    },
    foot: {
      type: String
    },
    head: {
      type: String
    },
    iconOpen: {
      type: String,
      default: "icon-chevron-down"
    },
    iconClose: {
      type: String,
      default: "icon-chevron-right"
    },
    isToggle: {
      type: Boolean,
      default: false
    },
    transition: {
      type: String,
      default: "panel"
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Panel.install = function(Vue) {
  Vue.component(Panel.name, Panel);
};
Panel.version = version;
export {
  Panel as default
};
