// @opentiny/vue-amount@3.20.0/lib/pc.js
import Popover from '@opentiny/vue-popover';
import Input from '@opentiny/vue-input';
import Currency from '@opentiny/vue-currency';
import Button from '@opentiny/vue-button';
import DatePicker from '@opentiny/vue-date-picker';
import { IconCoin } from '@opentiny/vue-icon';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/amount/vue';
import FilterBox from '@opentiny/vue-filter-box';
import Radio from '@opentiny/vue-radio';
import RadioGroup from '@opentiny/vue-radio-group';
import '@opentiny/vue-theme/amount/index.css';
import { resolveComponent, openBlock, createElementBlock, mergeProps, createBlock, withCtx, createVNode, createCommentVNode, createElementVNode, Fragment, renderList, createTextVNode, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  inheritAttrs: false,
  components: {
    TinyPopover: Popover,
    TinyInput: Input,
    TinyButton: Button,
    TinyCurrency: Currency,
    TinyDatePicker: DatePicker,
    IconCoin: IconCoin(),
    TinyFilterBox: FilterBox,
    TinyRadio: Radio,
    TinyRadioGroup: RadioGroup
  },
  emits: ["update:modelValue", "update:currency", "update:date", "change", "clear", "filter-change"],
  props: [].concat(props, ["modelValue", "tabindex", "size", "placeholder", "currency", "date", "dateAllowEmpty", "digits", "stringMode", "rounding", "maxLen", "negative", "disabled", "fetchCurrency", "fields", "popperClass", "popperAppendToBody", "format", "type", "holdZero", "modelTruncation", "strictInput", "plugin", "popUp", "hideCurrency", "displayOnly", "hideIcon", "numAllowEmpty", "label", "tip", "shape", "clearable", "filter", "blank"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "popover-con",
  ref: "popover"
};
var _hoisted_2 = {
  key: 0,
  class: "popover-filter-content"
};
var _hoisted_3 = {
  key: 1,
  class: "module"
};
var _hoisted_4 = {
  class: "popover-left"
};
var _hoisted_5 = {
  key: 2,
  class: "module"
};
var _hoisted_6 = {
  class: "popover-left"
};
var _hoisted_7 = {
  class: "module"
};
var _hoisted_8 = {
  class: "popover-left"
};
var _hoisted_9 = {
  key: 3,
  class: "module"
};
var _hoisted_10 = {
  key: 4,
  class: "module"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_filter_box = resolveComponent("tiny-filter-box");
  var _component_icon_coin = resolveComponent("icon-coin");
  var _component_tiny_input = resolveComponent("tiny-input");
  var _component_tiny_radio = resolveComponent("tiny-radio");
  var _component_tiny_radio_group = resolveComponent("tiny-radio-group");
  var _component_tiny_currency = resolveComponent("tiny-currency");
  var _component_tiny_date_picker = resolveComponent("tiny-date-picker");
  var _component_tiny_button = resolveComponent("tiny-button");
  var _component_tiny_popover = resolveComponent("tiny-popover");
  return openBlock(), createElementBlock(
    "div",
    mergeProps({
      ref: "root",
      class: ["tiny-amount", {
        "hide-icon": _ctx.hideIcon
      }]
    }, _ctx.a(_ctx.$attrs, ["^on[A-Z]"])),
    [!(_ctx.displayOnly || _ctx.hideIcon) ? (openBlock(), createBlock(_component_tiny_popover, {
      key: 0,
      modelValue: _ctx.state.visible,
      "onUpdate:modelValue": _cache[5] || (_cache[5] = function($event) {
        return _ctx.state.visible = $event;
      }),
      placement: "bottom-start",
      "popper-class": "tiny-amount-popper" + (_ctx.popperClass ? " " + _ctx.popperClass : "") + (_ctx.shape === "filter" ? " tiny-amount-filter-box" : ""),
      trigger: "manual",
      "append-to-body": _ctx.popperAppendToBody
    }, {
      reference: withCtx(function() {
        return [_ctx.shape === "filter" ? (openBlock(), createBlock(_component_tiny_filter_box, {
          key: 0,
          ref: "reference",
          label: _ctx.label,
          value: _ctx.state.amountValue,
          tip: _ctx.tip,
          disabled: _ctx.disabled,
          onClick: _ctx.toggleVisible,
          "show-close": _ctx.clearable,
          blank: _ctx.blank
        }, null, 8, ["label", "value", "tip", "disabled", "onClick", "show-close", "blank"])) : (openBlock(), createBlock(_component_tiny_input, {
          key: 1,
          tabindex: _ctx.tabindex,
          size: _ctx.size,
          "model-value": _ctx.state.amountValue,
          maxlength: _ctx.maxLen,
          "display-only": _ctx.displayOnly,
          placeholder: _ctx.placeholder,
          disabled: _ctx.disabled,
          onFocus: _ctx.inputFocus,
          onBlur: _ctx.inputBlur,
          "onUpdate:modelValue": _ctx.onInput
        }, {
          suffix: withCtx(function() {
            return [!_ctx.hideIcon ? (openBlock(), createElementBlock("div", {
              key: 0,
              onClick: _cache[0] || (_cache[0] = function() {
                return _ctx.toggleVisible && _ctx.toggleVisible.apply(_ctx, arguments);
              }),
              class: "tiny-amount-input-icon"
            }, [createVNode(_component_icon_coin, {
              class: "tiny-svg-size"
            })])) : createCommentVNode("v-if", true)];
          }),
          _: 1
          /* STABLE */
        }, 8, ["tabindex", "size", "model-value", "maxlength", "display-only", "placeholder", "disabled", "onFocus", "onBlur", "onUpdate:modelValue"]))];
      }),
      default: withCtx(function() {
        return [createElementVNode(
          "div",
          _hoisted_1,
          [_ctx.shape === "filter" && _ctx.filter ? (openBlock(), createElementBlock("div", _hoisted_2, [createVNode(_component_tiny_radio_group, {
            modelValue: _ctx.state.radioVal,
            "onUpdate:modelValue": _cache[1] || (_cache[1] = function($event) {
              return _ctx.state.radioVal = $event;
            }),
            onChange: _ctx.handleChange
          }, {
            default: withCtx(function() {
              return [(openBlock(true), createElementBlock(
                Fragment,
                null,
                renderList(_ctx.state.filterMenu, function(item, index) {
                  return openBlock(), createBlock(_component_tiny_radio, {
                    key: index,
                    label: _ctx.t(item),
                    class: "popover-filter-botton"
                  }, {
                    default: withCtx(function() {
                      return [createTextVNode(
                        toDisplayString(_ctx.t(item)),
                        1
                        /* TEXT */
                      )];
                    }),
                    _: 2
                    /* DYNAMIC */
                  }, 1032, ["label"]);
                }),
                128
                /* KEYED_FRAGMENT */
              ))];
            }),
            _: 1
            /* STABLE */
          }, 8, ["modelValue", "onChange"])])) : createCommentVNode("v-if", true), _ctx.popUp ? (openBlock(), createElementBlock("div", _hoisted_3, [createElementVNode(
            "div",
            _hoisted_4,
            toDisplayString(_ctx.t("ui.amount.currency")),
            1
            /* TEXT */
          ), createVNode(_component_tiny_currency, {
            tiny_mode: "pc",
            tiny_mode_root: true,
            class: "popover-right",
            modelValue: _ctx.editorState.currency,
            "onUpdate:modelValue": _cache[2] || (_cache[2] = function($event) {
              return _ctx.editorState.currency = $event;
            }),
            clearable: false,
            fields: _ctx.fields,
            "fetch-currency": _ctx.fetchCurrency,
            "popper-class": _ctx.popperClass,
            "popper-append-to-body": _ctx.popperAppendToBody,
            currency: _ctx.currency
          }, null, 8, ["modelValue", "fields", "fetch-currency", "popper-class", "popper-append-to-body", "currency"])])) : createCommentVNode("v-if", true), _ctx.date || _ctx.dateAllowEmpty ? (openBlock(), createElementBlock("div", _hoisted_5, [createElementVNode(
            "div",
            _hoisted_6,
            toDisplayString(_ctx.t("ui.amount.date")),
            1
            /* TEXT */
          ), createVNode(_component_tiny_date_picker, {
            class: "popover-right",
            modelValue: _ctx.editorState.date,
            "onUpdate:modelValue": _cache[3] || (_cache[3] = function($event) {
              return _ctx.editorState.date = $event;
            }),
            "popper-class": "tiny-amount-popper" + (_ctx.popperClass ? " " + _ctx.popperClass : ""),
            "popper-append-to-body": _ctx.popperAppendToBody
          }, null, 8, ["modelValue", "popper-class", "popper-append-to-body"])])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_7, [createElementVNode(
            "div",
            _hoisted_8,
            toDisplayString(_ctx.t("ui.amount.amount")),
            1
            /* TEXT */
          ), createVNode(_component_tiny_input, {
            class: "popover-right",
            modelValue: _ctx.editorState.amount,
            "onUpdate:modelValue": [_cache[4] || (_cache[4] = function($event) {
              return _ctx.editorState.amount = $event;
            }), _ctx.popInput],
            maxlength: _ctx.maxLen
          }, null, 8, ["modelValue", "onUpdate:modelValue", "maxlength"])]), _ctx.state.theme === "saas" ? (openBlock(), createElementBlock("div", _hoisted_9, [createVNode(_component_tiny_button, {
            onClick: _ctx.reset
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.base.reset")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          }, 8, ["onClick"]), createVNode(_component_tiny_button, {
            type: "primary",
            onClick: _ctx.save
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.base.confirm")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          }, 8, ["onClick"])])) : (openBlock(), createElementBlock("div", _hoisted_10, [createVNode(_component_tiny_button, {
            type: "primary",
            onClick: _ctx.save
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.base.confirm")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          }, 8, ["onClick"]), createVNode(_component_tiny_button, {
            onClick: _ctx.reset
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.base.reset")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          }, 8, ["onClick"])]))],
          512
          /* NEED_PATCH */
        )];
      }),
      _: 1
      /* STABLE */
    }, 8, ["modelValue", "popper-class", "append-to-body"])) : (openBlock(), createBlock(_component_tiny_input, {
      key: 1,
      tabindex: _ctx.tabindex,
      size: _ctx.size,
      "model-value": _ctx.state.amountValue,
      maxlength: _ctx.maxLen,
      "display-only": _ctx.displayOnly,
      placeholder: _ctx.placeholder,
      disabled: _ctx.disabled,
      onFocus: _ctx.inputFocus,
      onBlur: _ctx.inputBlur,
      "onUpdate:modelValue": _ctx.onInput
    }, {
      suffix: withCtx(function() {
        return [!_ctx.hideIcon ? (openBlock(), createElementBlock("div", {
          key: 0,
          onClick: _cache[6] || (_cache[6] = function() {
            return _ctx.toggleVisible && _ctx.toggleVisible.apply(_ctx, arguments);
          }),
          class: "tiny-amount-input-icon"
        }, [createVNode(_component_icon_coin, {
          class: "tiny-svg-size"
        })])) : createCommentVNode("v-if", true)];
      }),
      _: 1
      /* STABLE */
    }, 8, ["tabindex", "size", "model-value", "maxlength", "display-only", "placeholder", "disabled", "onFocus", "onBlur", "onUpdate:modelValue"]))],
    16
    /* FULL_PROPS */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
