// @opentiny/vue-currency@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
import "@opentiny/vue-theme/currency/index.css";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var Currency = defineComponent({
  name: $prefix + "Currency",
  props: _extends({}, $props, {
    clearable: {
      type: Boolean,
      default: true
    },
    currency: {
      type: String,
      default: "CNY"
    },
    disabled: {
      type: Boolean,
      default: false
    },
    fetchCurrency: Function,
    fields: Object,
    modelValue: {},
    placeholder: {
      type: String,
      default: ""
    },
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    popperClass: String,
    size: String,
    setDefault: {
      type: Boolean,
      default: false
    },
    fetchDefaultCurrency: Function,
    setDefaultCurrency: Function,
    visible: {
      type: Boolean,
      default: false
    },
    title: String,
    mode: String,
    multiple: {
      type: Boolean,
      default: false
    },
    searchConfig: Object,
    dropStyle: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    label: {
      type: String,
      default: ""
    },
    tip: String,
    shape: String
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Currency.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Currency.install = function(Vue) {
  Vue.component(Currency.name, Currency);
};
Currency.version = version;
export {
  Currency as default
};
