// @opentiny/vue-slider@3.20.0/lib/mobile.js
import { renderless, api } from '@opentiny/vue-renderless/slider/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import '@opentiny/vue-theme-mobile/slider/index.css';
import { openBlock, createElementBlock, createElementVNode, normalizeClass, normalizeStyle, createCommentVNode, withDirectives, toDisplayString, vShow, renderSlot, vModelText } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["update:modelValue", "change", "start", "stop"],
  props: [].concat(props, ["modelValue", "disabled", "max", "min", "vertical", "step", "numPages", "showTip", "showInput", "height", "range", "formatTooltip", "changeCompat"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  style: {
    "position": "relative"
  }
};
var _hoisted_2 = {
  key: 0,
  class: "tiny-mobile-slider__input"
};
var _hoisted_3 = ["disabled"];
var _hoisted_4 = /* @__PURE__ */ createElementVNode(
  "span",
  {
    class: "tiny-mobile-slider__per"
  },
  "%",
  -1
  /* HOISTED */
);
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "div",
    {
      class: normalizeClass(["tiny-mobile-slider", {
        "tiny-mobile-slider__vertical": _ctx.vertical,
        disabled: _ctx.disabled
      }]),
      style: normalizeStyle({
        height: _ctx.vertical ? _ctx.height : ""
      }),
      onTouchstart: _cache[6] || (_cache[6] = function() {
        return _ctx.bindMouseDown && _ctx.bindMouseDown.apply(_ctx, arguments);
      })
    },
    [createElementVNode(
      "div",
      {
        class: normalizeClass(["tiny-mobile-slider__range", _ctx.vertical ? "tiny-mobile-slider__down" : "tiny-mobile-slider__left"]),
        style: normalizeStyle(_ctx.state.barStyle)
      },
      null,
      6
      /* CLASS, STYLE */
    ), _ctx.state.leftBtnShow ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        class: normalizeClass(["tiny-mobile-slider__handle", {
          "handle-focus": _ctx.state.activeIndex === 0
        }]),
        tabindex: "0",
        style: normalizeStyle(_ctx.state.leftBtnStyle),
        onMouseenter: _cache[0] || (_cache[0] = function() {
          return _ctx.displayTip && _ctx.displayTip.apply(_ctx, arguments);
        }),
        onMouseleave: _cache[1] || (_cache[1] = function() {
          return _ctx.hideTip && _ctx.hideTip.apply(_ctx, arguments);
        }),
        onKeydown: _cache[2] || (_cache[2] = function() {
          return _ctx.bindKeyDown && _ctx.bindKeyDown.apply(_ctx, arguments);
        })
      },
      null,
      38
      /* CLASS, STYLE, NEED_HYDRATION */
    )) : createCommentVNode("v-if", true), _ctx.state.rightBtnShow ? (openBlock(), createElementBlock(
      "div",
      {
        key: 1,
        class: normalizeClass(["tiny-mobile-slider__handle", {
          "handle-focus": _ctx.state.activeIndex === 1
        }]),
        tabindex: "0",
        style: normalizeStyle(_ctx.state.rightBtnStyle),
        onMouseenter: _cache[3] || (_cache[3] = function() {
          return _ctx.displayTip && _ctx.displayTip.apply(_ctx, arguments);
        }),
        onMouseleave: _cache[4] || (_cache[4] = function() {
          return _ctx.hideTip && _ctx.hideTip.apply(_ctx, arguments);
        }),
        onKeydown: _cache[5] || (_cache[5] = function() {
          return _ctx.bindKeyDown && _ctx.bindKeyDown.apply(_ctx, arguments);
        })
      },
      null,
      38
      /* CLASS, STYLE, NEED_HYDRATION */
    )) : createCommentVNode("v-if", true), withDirectives(createElementVNode(
      "div",
      {
        class: "tiny-mobile-slider__tips",
        style: normalizeStyle(_ctx.state.tipStyle)
      },
      toDisplayString(_ctx.state.tipValue),
      5
      /* TEXT, STYLE */
    ), [[vShow, _ctx.showTip && _ctx.state.showTip]])],
    38
    /* CLASS, STYLE, NEED_HYDRATION */
  ), _ctx.showInput && !_ctx.state.isDouble ? (openBlock(), createElementBlock("div", _hoisted_2, [renderSlot(_ctx.$slots, "default", {
    slotScope: _ctx.state.activeValue
  }, function() {
    return [withDirectives(createElementVNode("input", {
      type: "text",
      "onUpdate:modelValue": _cache[7] || (_cache[7] = function($event) {
        return _ctx.state.activeValue = $event;
      }),
      disabled: _ctx.disabled,
      onChange: _cache[8] || (_cache[8] = function() {
        return _ctx.inputOnChange && _ctx.inputOnChange.apply(_ctx, arguments);
      })
    }, null, 40, _hoisted_3), [[vModelText, _ctx.state.activeValue]]), _hoisted_4];
  })])) : createCommentVNode("v-if", true)]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobile as default };
