// @opentiny/vue-collapse-item@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
import "@opentiny/vue-theme/collapse-item/index.css";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var $constants = {
  INTERVAL: 50
};
var collapseItemProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  title: String,
  titleRight: String,
  name: {
    type: [String, Number],
    default: function _default2() {
      return this._uid;
    }
  },
  expandIcon: {
    type: [Object, String]
  },
  disabled: Boolean
});
var CollapseItem = defineComponent({
  name: $prefix + "CollapseItem",
  componentName: "CollapseItem",
  inject: ["collapse"],
  props: collapseItemProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
CollapseItem.install = function(Vue) {
  Vue.component(CollapseItem.name, CollapseItem);
};
CollapseItem.version = version;
export {
  CollapseItem as default
};
