// @opentiny/huicharts@1.0.1/feature/token/factory/chartsToken/BoxplotChart.js
/**
 * Copyright (c) 2024 - present OpenTiny HUICharts Authors.
 * Copyright (c) 2024 - present Huawei Cloud Computing Technologies Co., Ltd.
 *
 * Use of this source code is governed by an MIT-style license.
 *
 * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
 * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
 * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
 *
 */
function BoxplotChart(aliasToken) {
  var lineWidth = aliasToken.lineWidth,
    symbolSize = aliasToken.symbolSize,
    symbolBorderWidthNone = aliasToken.symbolBorderWidthNone,
    colorBorderTransparent = aliasToken.colorBorderTransparent,
    symbolBorderWidth = aliasToken.symbolBorderWidth,
    colorFillSecondary = aliasToken.colorFillSecondary;
  return {
    boxplotItemStyleBorderWidth: lineWidth,
    boxplotEmphasisItemStyleBorderWidth: lineWidth,
    scatterSymbolSize: symbolSize,
    scatterItemBorderWidth: symbolBorderWidthNone,
    scatterItemBorderColor: colorBorderTransparent,
    scatterEmphasisItemBorderWidth: symbolBorderWidth,
    scatterEmphasisItemColor: colorFillSecondary
  };
}
export { BoxplotChart as default };
