// @opentiny/huicharts@1.0.1/feature/token/factory/chartsToken/ChartCard.js
/**
 * Copyright (c) 2024 - present OpenTiny HUICharts Authors.
 * Copyright (c) 2024 - present Huawei Cloud Computing Technologies Co., Ltd.
 *
 * Use of this source code is governed by an MIT-style license.
 *
 * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
 * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
 * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
 *
 */
function ChartCard(aliasToken) {
  var colorTextPrimary = aliasToken.colorTextPrimary,
    colorFill = aliasToken.colorFill,
    colorTextSecondary = aliasToken.colorTextSecondary,
    colorBorderInverse = aliasToken.colorBorderInverse,
    colorBgPrimary = aliasToken.colorBgPrimary,
    colorTextDisabled = aliasToken.colorTextDisabled,
    colorBgTertiary = aliasToken.colorBgTertiary;
  return {
    // 卡片背景
    cardBg: colorBgPrimary,
    // 标题的颜色
    titleColor: colorTextPrimary,
    // 文本的颜色
    labelColor: colorTextSecondary,
    // 选中背景色
    selectBgActive: colorBgPrimary,
    // 未选中背景色
    selectBgAInactive: colorBgTertiary,
    // 未选中文本色
    selectTextDisabled: colorTextDisabled,
    borderColor: colorBorderInverse,
    // 分隔线
    selectSplitLineColor: colorFill
  };
}
export { ChartCard as default };
