// @opentiny/huicharts@1.0.1/feature/token/factory/chartsToken/RadarChart.js
/**
 * Copyright (c) 2024 - present OpenTiny HUICharts Authors.
 * Copyright (c) 2024 - present Huawei Cloud Computing Technologies Co., Ltd.
 *
 * Use of this source code is governed by an MIT-style license.
 *
 * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
 * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
 * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
 *
 */
function RadarChart(aliasToken) {
  var symbolSize = aliasToken.symbolSize,
    lineWidth = aliasToken.lineWidth,
    lineWidthNone = aliasToken.lineWidthNone,
    colorBgTransparent = aliasToken.colorBgTransparent,
    symbolBorderWidth = aliasToken.symbolBorderWidth,
    symbolBorderWidthNone = aliasToken.symbolBorderWidthNone,
    colorFillSecondary = aliasToken.colorFillSecondary,
    colorBorderTransparent = aliasToken.colorBorderTransparent;
  return {
    itemBorderWidth: symbolBorderWidthNone,
    emphasisItemBorderWidth: symbolBorderWidth,
    areaColor: colorBgTransparent,
    symbolSize: symbolSize,
    lineWidth: lineWidth,
    thresholdLineWidth: lineWidthNone,
    itemBorderColor: colorBorderTransparent,
    emphasisItemColor: colorFillSecondary,
    gradientItemBorderColor: colorFillSecondary
  };
}
export { RadarChart as default };
