// @opentiny/vue-autocomplete@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/autocomplete/vue';
import { defineComponent, directive, props, setup } from '@opentiny/vue-common';
import Scrollbar from '@opentiny/vue-scrollbar';
import Input from '@opentiny/vue-input';
import Clickoutside from '@opentiny/vue-renderless/common/deps/clickoutside';
import { IconLoadingShadow } from '@opentiny/vue-icon';
import { resolveComponent, resolveDirective, withDirectives, openBlock, createElementBlock, normalizeClass, createBlock, mergeProps, withKeys, withModifiers, createSlots, withCtx, renderSlot, createVNode, Transition, createElementVNode, normalizeStyle, Fragment, renderList, createTextVNode, toDisplayString, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    TinyInput: Input,
    TinyScrollbar: Scrollbar,
    IconLoadingShadow: IconLoadingShadow()
  },
  directives: directive({
    Clickoutside
  }),
  props: [].concat(props, ["valueKey", "popperClass", "tabindex", "popperOptions", "placeholder", "clearable", "disabled", "name", "size", "modelValue", "maxlength", "minlength", "autofocus", "fetchSuggestions", "triggerOnFocus", "customItem", "selectWhenUnmatched", "prefixIcon", "suffixIcon", "label", "debounce", "placement", "hideLoading", "popperAppendToBody", "highlightFirstItem", "displayOnly", "validateEvent"]),
  emits: ["update:modelValue", "focus", "blur", "clear", "select", "created"],
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["aria-expanded", "aria-owns"];
var _hoisted_2 = {
  key: 0,
  class: "tiny-autocomplete-suggestion__list-item tiny-autocomplete-suggestion__list-loading",
  role: "loading"
};
var _hoisted_3 = ["onClick", "id", "aria-selected"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_input = resolveComponent("tiny-input");
  var _component_icon_loading_shadow = resolveComponent("icon-loading-shadow");
  var _component_tiny_scrollbar = resolveComponent("tiny-scrollbar");
  var _directive_clickoutside = resolveDirective("clickoutside");
  return withDirectives((openBlock(), createElementBlock("div", {
    class: normalizeClass(["tiny-autocomplete", {
      "show-auto-width": _ctx.state.showAutoWidth
    }]),
    "aria-haspopup": "listbox",
    role: "combobox",
    "aria-expanded": _ctx.state.suggestionVisible,
    "aria-owns": _ctx.state.id
  }, [withDirectives((openBlock(), createBlock(_component_tiny_input, mergeProps({
    tabindex: _ctx.tabindex,
    "display-only": _ctx.displayOnly,
    ref: "input"
  }, _ctx.f(_ctx.$props, _ctx.$attrs), {
    "front-clear-icon": "",
    "onUpdate:modelValue": _ctx.handleChange,
    "validate-event": false,
    onFocus: _ctx.handleFocus,
    onClear: _ctx.handleClear,
    onKeydown: [_cache[0] || (_cache[0] = withKeys(withModifiers(function($event) {
      return _ctx.highlight(_ctx.state.highlightedIndex - 1);
    }, ["prevent"]), ["up"])), _cache[1] || (_cache[1] = withKeys(withModifiers(function($event) {
      return _ctx.highlight(_ctx.state.highlightedIndex + 1);
    }, ["prevent"]), ["down"])), withKeys(_ctx.handleKeyEnter, ["enter"]), withKeys(_ctx.close, ["tab"])]
  }), createSlots({
    _: 2
    /* DYNAMIC */
  }, [_ctx.slots.prepend ? {
    name: "prepend",
    fn: withCtx(function() {
      return [renderSlot(_ctx.$slots, "prepend")];
    }),
    key: "0"
  } : void 0, _ctx.slots.append ? {
    name: "append",
    fn: withCtx(function() {
      return [renderSlot(_ctx.$slots, "append")];
    }),
    key: "1"
  } : void 0, _ctx.slots.prefix ? {
    name: "prefix",
    fn: withCtx(function() {
      return [renderSlot(_ctx.$slots, "prefix")];
    }),
    key: "2"
  } : void 0, _ctx.slots.suffix ? {
    name: "suffix",
    fn: withCtx(function() {
      return [renderSlot(_ctx.$slots, "suffix")];
    }),
    key: "3"
  } : void 0]), 1040, ["tabindex", "display-only", "onUpdate:modelValue", "onFocus", "onClear", "onKeydown"])), [[_directive_clickoutside, _ctx.handleBlur]]), createVNode(Transition, {
    name: "tiny-zoom-in-top",
    onAfterLeave: _ctx.doDestroy,
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          ref: "popper",
          class: normalizeClass(["tiny-autocomplete-suggestion tiny-popper", [_ctx.popperClass ? _ctx.popperClass : "", {
            "is-loading": !_ctx.hideLoading && _ctx.state.loading
          }]]),
          style: normalizeStyle({
            width: _ctx.suggestionState.dropdownWidth
          }),
          role: "region"
        },
        [createVNode(_component_tiny_scrollbar, {
          tag: "ul",
          "wrap-class": "tiny-autocomplete-suggestion__wrap",
          "view-class": "tiny-autocomplete-suggestion__list"
        }, {
          default: withCtx(function() {
            return [renderSlot(_ctx.$slots, "panel", {
              loading: !_ctx.hideLoading && _ctx.state.loading,
              list: _ctx.state.suggestions,
              highlighted: _ctx.state.highlightedIndex,
              id: _ctx.state.id,
              select: _ctx.select
            }, function() {
              return [!_ctx.hideLoading && _ctx.state.loading ? (openBlock(), createElementBlock("li", _hoisted_2, [createVNode(_component_icon_loading_shadow)])) : (openBlock(true), createElementBlock(
                Fragment,
                {
                  key: 1
                },
                renderList(_ctx.state.suggestions, function(item, index) {
                  return openBlock(), createElementBlock("li", {
                    key: index,
                    class: normalizeClass(["tiny-autocomplete-suggestion__list-item", {
                      highlighted: _ctx.state.highlightedIndex === index,
                      selected: _ctx.modelValue === item[_ctx.valueKey]
                    }]),
                    onClick: function onClick($event) {
                      return _ctx.select(item);
                    },
                    id: _ctx.state.id + "-item-" + index,
                    role: "option",
                    "aria-selected": _ctx.state.highlightedIndex === index
                  }, [renderSlot(_ctx.$slots, "default", {
                    slotScope: item
                  }, function() {
                    return [createTextVNode(
                      toDisplayString(item[_ctx.valueKey]),
                      1
                      /* TEXT */
                    )];
                  })], 10, _hoisted_3);
                }),
                128
                /* KEYED_FRAGMENT */
              ))];
            })];
          }),
          _: 3
          /* FORWARDED */
        })],
        6
        /* CLASS, STYLE */
      ), [[vShow, _ctx.hideLoading ? _ctx.suggestionState.showPopper && !_ctx.state.loading : _ctx.suggestionState.showPopper]])];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["onAfterLeave"])], 10, _hoisted_1)), [[_directive_clickoutside, _ctx.close]]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
