// @opentiny/vue-color-select-panel@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {};
var ColorSelectPanel = defineComponent({
  name: $prefix + "ColorSelectPanel",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    modelValue: String,
    visible: Boolean,
    alpha: Boolean,
    history: Array,
    predefine: Array,
    enableHistory: {
      type: Boolean,
      default: false
    },
    enablePredefineColor: {
      type: Boolean,
      default: false
    },
    format: {
      type: Array,
      default: function _default2() {
        return [];
      },
      validator: function validator(formats) {
        formats.forEach(function(formatValue) {
          if (formatValue[formatValue.length - 1] === "a") {
            console.warn("If you want enable alpha, You should set `alpha` prop to true");
          }
        });
        return formats.every(function(formatValue) {
          return ["hsv", "hsl", "rgb", "hex"].includes(formatValue);
        });
      }
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
ColorSelectPanel.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
ColorSelectPanel.install = function(Vue) {
  Vue.component(ColorSelectPanel.name, ColorSelectPanel);
};
ColorSelectPanel.version = version;
export {
  ColorSelectPanel as default
};
