// @opentiny/vue-guide@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import browser from "@opentiny/vue-renderless/common/browser";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Guide = defineComponent({
  name: $prefix + "Guide",
  props: _extends({}, $props, {
    showStep: {
      type: Boolean,
      default: false
    },
    domData: Array,
    mainAxis: {
      type: Number,
      default: 18
    },
    crossAxis: {
      type: Number,
      default: null
    },
    alignmentAxis: {
      type: Number,
      default: null
    },
    popPosition: {
      type: String,
      default: "bottom"
    },
    arrow: {
      type: Boolean,
      default: true
    },
    modalOverlayOpeningPadding: {
      type: Number,
      default: 0
    },
    modalOverlayOpeningRadius: {
      type: Number,
      default: 0
    },
    lightClass: {
      type: String,
      default: ""
    },
    width: {
      type: String,
      default: browser.isMobile ? "350" : "510"
    },
    height: {
      type: String,
      default: ""
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Guide.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Guide.install = function(Vue) {
  Vue.component(Guide.name, Guide);
};
Guide.version = version;
export {
  Guide as default
};
