// @opentiny/vue-slider-button@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var $constants = {
  SLIDER_BUTTON_GROUP: "SliderButtonGroup"
};
var SliderButton = defineComponent({
  name: $prefix + "SliderButton",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    text: [Number, String],
    events: {
      type: Object,
      default: function _default2() {
        return {};
      }
    },
    label: [Number, String],
    disabled: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
SliderButton.install = function(Vue) {
  Vue.component(SliderButton.name, SliderButton);
};
SliderButton.version = version;
export {
  SliderButton as default
};
