// @opentiny/vue-slider-button@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/slider-button/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { openBlock, createElementBlock, normalizeClass, withDirectives, createElementVNode, vModelRadio, withModifiers, renderSlot, Fragment, createTextVNode, toDisplayString, createCommentVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["label", "events", "text", "disabled"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["aria-checked", "tabindex"];
var _hoisted_2 = ["value"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("label", {
    "data-tag": "tiny-slider-button",
    ref: "sliderButton",
    class: normalizeClass(["inline-flex relative outline-0 cursor-pointer items-center", _ctx.state.size === "large" ? "m-1" : "m-0.5"]),
    role: "radio",
    "aria-checked": _ctx.state.value === (_ctx.label || _ctx.text),
    tabindex: _ctx.state.tabIndex
  }, [withDirectives(createElementVNode("input", {
    class: "opacity-0 outline-0 absolute -z-10",
    value: _ctx.label || _ctx.text,
    type: "radio",
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.value = $event;
    }),
    onChange: _cache[1] || (_cache[1] = function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }),
    tabindex: "-1",
    ref: "sliderInput"
  }, null, 40, _hoisted_2), [[vModelRadio, _ctx.state.value]]), createElementVNode(
    "span",
    {
      "data-tag": "tiny-slider-button-text",
      class: normalizeClass(_ctx.m("relative items-center text-center overflow-hidden text-ellipsis whitespace-nowrap cursor-pointer box-border", _ctx.state.value !== (_ctx.label || _ctx.text) ? !_ctx.state.disabled && "shadow-none text-color-text-secondary hover:text-color-brand" : !_ctx.state.disabled && "text-color-brand active:text-color-brand", _ctx.state.type === "icon" ? "flex items-center justify-center w-6 h-6 p-1 [&_svg]:w-6 [&_svg]:h-6" : "inline-block", _ctx.state.size === "large" ? _ctx.state.type === "icon" ? "w-10 h-10 p-2" : "h-10 px-4 min-w-[theme(spacing.12)] max-w-[theme(spacing.32)] text-sm leading-10" : _ctx.state.type !== "icon" && "h-6 px-2 min-w-[theme(spacing.10)] max-w-[theme(spacing.28)] text-xs leading-6", _ctx.state.size === "medium" ? _ctx.state.type === "icon" ? "p-1 w-7 h-7 [&_svg]:w-4.5 [&_svg]:h-4.5" : "text-sm h-7 leading-7" : "", _ctx.state.type === "icon" && _ctx.state.value === (_ctx.label || _ctx.text) ? "fill-color-brand active:fill-color-brand" : "fill-color-none-hover active:fill-color-brand hover:fill-color-brand", _ctx.state.value === (_ctx.label || _ctx.text) && _ctx.state.disabled ? _ctx.state.type === "icon" ? "fill-color-brand-disabled cursor-not-allowed" : "text-color-brand-disabled cursor-not-allowed" : "", _ctx.state.value !== (_ctx.label || _ctx.text) && _ctx.state.disabled ? _ctx.state.type === "icon" ? "fill-color-icon-disabled cursor-not-allowed" : "text-color-text-disabled cursor-not-allowed" : "")),
      onKeydown: _cache[2] || (_cache[2] = withModifiers(function() {
      }, ["stop"]))
    },
    [renderSlot(_ctx.$slots, "default"), !_ctx.slots.default ? (openBlock(), createElementBlock(
      Fragment,
      {
        key: 0
      },
      [createTextVNode(
        toDisplayString(_ctx.text || _ctx.label),
        1
        /* TEXT */
      )],
      64
      /* STABLE_FRAGMENT */
    )) : createCommentVNode("v-if", true)],
    34
    /* CLASS, NEED_HYDRATION */
  )], 10, _hoisted_1);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
