// @opentiny/vue-link@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/link/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var linkProps = _extends({}, $props, {
  /** 文本链接类型：'primary' | 'success' | 'warning' | 'danger' | 'info' */
  type: {
    type: String,
    default: "default"
  },
  /** 显示内容 */
  value: String,
  /** 是否下划线 */
  underline: {
    type: Boolean,
    default: true
  },
  /** 原生 href 属性 */
  href: String,
  /** 自定义图标 */
  icon: [Object, String],
  /** 是否禁用状态 */
  disabled: Boolean
});
var Link = defineComponent({
  name: $prefix + "Link",
  props: linkProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Link.install = function(Vue) {
  Vue.component(Link.name, Link);
};
Link.version = version;
export {
  Link as default
};
