// @opentiny/vue-split@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/split/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  PREFIX: "tiny-split",
  HORIZONTAL: "horizontal",
  VERTICAL: "vertical"
};
var Split = defineComponent({
  name: $prefix + "Split",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    modelValue: {
      type: [Number, String],
      default: 0.5
    },
    mode: {
      validator: function validator(value) {
        var arr = [$constants.HORIZONTAL, $constants.VERTICAL];
        for (var i = 0; i < arr.length; i++) {
          if (value === arr[i]) {
            return true;
          }
        }
        return false;
      },
      default: $constants.HORIZONTAL
    },
    leftTopMin: {
      type: [Number, String],
      default: "40px"
    },
    rightBottomMin: {
      type: [Number, String],
      default: "40px"
    },
    disabled: {
      type: Boolean,
      default: false
    },
    "trigger-simple": {
      type: Boolean,
      default: false
    },
    "collapse-left-top": {
      type: Boolean,
      default: false
    },
    "collapse-right-bottom": {
      type: Boolean,
      default: false
    },
    "three-areas": {
      type: Boolean,
      default: false
    },
    scrollable: {
      type: Boolean,
      default: false
    },
    border: {
      type: Boolean,
      default: true
    },
    rightBottomValue: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Split.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Split.install = function(Vue) {
  Vue.component(Split.name, Split);
};
Split.version = version;
export {
  Split as default
};
