// @opentiny/vue-split@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/split/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { IconLeftWard, IconRightward, IconStretchUpright, IconStretchCrosswise } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createElementVNode, normalizeStyle, renderSlot, Fragment, setBlockTracking, renderList, createCommentVNode, createVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    IconLeftWard: IconLeftWard(),
    IconRightWard: IconRightward(),
    IconStretchUpright: IconStretchUpright(),
    // 纵向
    IconStretchCrosswise: IconStretchCrosswise()
    // 横向
  },
  emits: ["moving", "mousemove", "mouseup", "moveend", "movestart", "update:modelValue", "left-top-click", "right-bottom-click"],
  props: [].concat(props, ["modelValue", "mode", "leftTopMin", "rightBottomMin", "disabled", "triggerSimple", "collapseLeftTop", "collapseRightBottom", "threeAreas", "scrollable", "border", "rightBottomValue"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_IconStretchUpright = resolveComponent("IconStretchUpright");
  var _component_icon_left_ward = resolveComponent("icon-left-ward");
  var _component_icon_right_ward = resolveComponent("icon-right-ward");
  var _component_IconStretchCrosswise = resolveComponent("IconStretchCrosswise");
  return openBlock(), createElementBlock(
    "div",
    {
      ref: "outerWrapper",
      class: normalizeClass(_ctx.state.wrapperClasses)
    },
    [_ctx.state.isHorizontal ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        class: normalizeClass(_ctx.state.prefix + "-horizontal")
      },
      [createElementVNode(
        "div",
        {
          style: normalizeStyle({
            right: _ctx.state.anotherOffset + "%",
            width: _ctx.state.anotherOffset !== 100 && _ctx.threeAreas ? _ctx.state.leftTopPane + "px" : ""
          }),
          class: normalizeClass([_ctx.state.paneClasses, "left-pane"])
        },
        [renderSlot(_ctx.$slots, "left")],
        6
        /* CLASS, STYLE */
      ), createElementVNode(
        "div",
        {
          class: normalizeClass([_ctx.state.prefix + "-trigger-con", _ctx.state.prefix + "-trigger-con-vertical", _ctx.disabled ? _ctx.state.prefix + "-trigger-con-disabled" : "", _ctx.state.isMoving ? _ctx.state.prefix + "-trigger-con-drag" : "", _ctx.state.offset === 0 && _ctx.triggerSimple ? _ctx.state.prefix + "-trigger-con-left-active" : "", _ctx.triggerSimple ? _ctx.state.prefix + "-trigger-con-simple" : ""]),
          style: normalizeStyle({
            left: _ctx.threeAreas ? _ctx.state.leftTopPane + "px" : _ctx.state.offset + "%"
          }),
          onMousedown: _cache[5] || (_cache[5] = function() {
            return _ctx.handleMousedown && _ctx.handleMousedown.apply(_ctx, arguments);
          })
        },
        [renderSlot(_ctx.$slots, "trigger", {}, function() {
          return [createElementVNode(
            "div",
            {
              class: normalizeClass([_ctx.state.prefix + "-trigger", _ctx.state.prefix + "-trigger-vertical"])
            },
            [!_ctx.state.triggerBarConWithLine ? (openBlock(), createElementBlock(
              Fragment,
              {
                key: 0
              },
              [!_ctx.triggerSimple ? (openBlock(), createElementBlock(
                "div",
                {
                  key: 0,
                  class: normalizeClass([_ctx.state.prefix + "-trigger-bar-con", "vertical"])
                },
                [_cache[0] || (setBlockTracking(-1), _cache[0] = (openBlock(), createElementBlock(
                  Fragment,
                  null,
                  renderList(3, function(i) {
                    return createElementVNode(
                      "i",
                      {
                        class: normalizeClass(_ctx.state.prefix + "-trigger-bar"),
                        key: "trigger-" + i
                      },
                      null,
                      2
                      /* CLASS */
                    );
                  }),
                  64
                  /* STABLE_FRAGMENT */
                )), setBlockTracking(1), _cache[0])],
                2
                /* CLASS */
              )) : createCommentVNode("v-if", true)],
              64
              /* STABLE_FRAGMENT */
            )) : (openBlock(), createElementBlock(
              Fragment,
              {
                key: 1
              },
              [!_ctx.triggerSimple ? (openBlock(), createElementBlock(
                "div",
                {
                  key: 0,
                  class: normalizeClass([_ctx.state.prefix + "-trigger-bar-con", "vertical"])
                },
                [createVNode(_component_IconStretchUpright)],
                2
                /* CLASS */
              )) : createCommentVNode("v-if", true)],
              64
              /* STABLE_FRAGMENT */
            ))],
            2
            /* CLASS */
          )];
        }), _ctx.triggerSimple ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            class: normalizeClass([_ctx.state.prefix + "-trigger-button", _ctx.state.prefix + "-trigger-left-button", _ctx.state.offset === 100 ? _ctx.state.prefix + "-trigger-button-active" : "", _ctx.state.offset === 100 || _ctx.collapseLeftTop ? _ctx.state.prefix + "-trigger-button-show" : ""]),
            onMousedown: _cache[1] || (_cache[1] = function() {
              return _ctx.buttonMousedown && _ctx.buttonMousedown.apply(_ctx, arguments);
            }),
            onClick: _cache[2] || (_cache[2] = function() {
              return _ctx.buttonLeftTopClick && _ctx.buttonLeftTopClick.apply(_ctx, arguments);
            })
          },
          [createVNode(_component_icon_left_ward, {
            class: "tiny-svg-size"
          })],
          34
          /* CLASS, NEED_HYDRATION */
        )) : createCommentVNode("v-if", true), _ctx.triggerSimple ? (openBlock(), createElementBlock(
          "div",
          {
            key: 1,
            class: normalizeClass([_ctx.state.prefix + "-trigger-button", _ctx.state.prefix + "-trigger-right-button", _ctx.state.offset === 0 ? _ctx.state.prefix + "-trigger-button-active" : "", _ctx.state.offset === 0 || _ctx.collapseRightBottom ? _ctx.state.prefix + "-trigger-button-show" : ""]),
            onMousedown: _cache[3] || (_cache[3] = function() {
              return _ctx.buttonMousedown && _ctx.buttonMousedown.apply(_ctx, arguments);
            }),
            onClick: _cache[4] || (_cache[4] = function() {
              return _ctx.buttonRightBottomClick && _ctx.buttonRightBottomClick.apply(_ctx, arguments);
            })
          },
          [createVNode(_component_icon_right_ward, {
            class: "tiny-svg-size"
          })],
          34
          /* CLASS, NEED_HYDRATION */
        )) : createCommentVNode("v-if", true)],
        38
        /* CLASS, STYLE, NEED_HYDRATION */
      ), createElementVNode(
        "div",
        {
          style: normalizeStyle({
            left: _ctx.threeAreas ? _ctx.state.leftTopPane + "px" : _ctx.state.offset + "%",
            width: _ctx.threeAreas ? "calc(100% - " + _ctx.state.leftTopPane + "px)" : ""
          }),
          class: normalizeClass([_ctx.state.paneClasses, "right-pane"])
        },
        [renderSlot(_ctx.$slots, "right")],
        6
        /* CLASS, STYLE */
      )],
      2
      /* CLASS */
    )) : (openBlock(), createElementBlock(
      "div",
      {
        key: 1,
        class: normalizeClass(_ctx.state.prefix + "-vertical")
      },
      [createElementVNode(
        "div",
        {
          style: normalizeStyle({
            bottom: _ctx.state.anotherOffset + "%",
            height: _ctx.state.anotherOffset !== 100 && _ctx.threeAreas ? _ctx.state.leftTopPane + "px" : ""
          }),
          class: normalizeClass([_ctx.state.paneClasses, "top-pane"])
        },
        [renderSlot(_ctx.$slots, "top")],
        6
        /* CLASS, STYLE */
      ), createElementVNode(
        "div",
        {
          class: normalizeClass([_ctx.state.prefix + "-trigger-con", _ctx.state.prefix + "-trigger-con-horizontal", _ctx.disabled ? _ctx.state.prefix + "-trigger-con-disabled" : "", _ctx.state.isMoving ? _ctx.state.prefix + "-trigger-con-drag" : "", _ctx.state.offset === 0 && _ctx.triggerSimple ? _ctx.state.prefix + "-trigger-con-top-active" : "", _ctx.triggerSimple ? _ctx.state.prefix + "-trigger-con-simple" : ""]),
          style: normalizeStyle({
            top: _ctx.threeAreas ? _ctx.state.leftTopPane + "px" : _ctx.state.offset + "%"
          }),
          onMousedown: _cache[11] || (_cache[11] = function() {
            return _ctx.handleMousedown && _ctx.handleMousedown.apply(_ctx, arguments);
          })
        },
        [renderSlot(_ctx.$slots, "trigger", {}, function() {
          return [createElementVNode(
            "div",
            {
              class: normalizeClass([_ctx.state.prefix + "-trigger", _ctx.state.prefix + "-trigger-horizontal"])
            },
            [!_ctx.state.triggerBarConWithLine ? (openBlock(), createElementBlock(
              Fragment,
              {
                key: 0
              },
              [!_ctx.triggerSimple ? (openBlock(), createElementBlock(
                "div",
                {
                  key: 0,
                  class: normalizeClass([_ctx.state.prefix + "-trigger-bar-con", "horizontal"])
                },
                [_cache[6] || (setBlockTracking(-1), _cache[6] = (openBlock(), createElementBlock(
                  Fragment,
                  null,
                  renderList(3, function(i) {
                    return createElementVNode(
                      "i",
                      {
                        class: normalizeClass(_ctx.state.prefix + "-trigger-bar"),
                        key: "trigger-" + i
                      },
                      null,
                      2
                      /* CLASS */
                    );
                  }),
                  64
                  /* STABLE_FRAGMENT */
                )), setBlockTracking(1), _cache[6])],
                2
                /* CLASS */
              )) : createCommentVNode("v-if", true)],
              64
              /* STABLE_FRAGMENT */
            )) : (openBlock(), createElementBlock(
              Fragment,
              {
                key: 1
              },
              [!_ctx.triggerSimple ? (openBlock(), createElementBlock(
                "div",
                {
                  key: 0,
                  class: normalizeClass([_ctx.state.prefix + "-trigger-bar-con", "horizontal"])
                },
                [createVNode(_component_IconStretchCrosswise)],
                2
                /* CLASS */
              )) : createCommentVNode("v-if", true)],
              64
              /* STABLE_FRAGMENT */
            ))],
            2
            /* CLASS */
          )];
        }), _ctx.triggerSimple ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            class: normalizeClass([_ctx.state.prefix + "-trigger-button", _ctx.state.prefix + "-trigger-top-button", _ctx.state.offset === 100 ? _ctx.state.prefix + "-trigger-button-active" : "", _ctx.state.offset === 100 || _ctx.collapseLeftTop ? _ctx.state.prefix + "-trigger-button-show" : ""]),
            onMousedown: _cache[7] || (_cache[7] = function() {
              return _ctx.buttonMousedown && _ctx.buttonMousedown.apply(_ctx, arguments);
            }),
            onClick: _cache[8] || (_cache[8] = function() {
              return _ctx.buttonLeftTopClick && _ctx.buttonLeftTopClick.apply(_ctx, arguments);
            })
          },
          [createVNode(_component_icon_left_ward, {
            class: "tiny-svg-size"
          })],
          34
          /* CLASS, NEED_HYDRATION */
        )) : createCommentVNode("v-if", true), _ctx.triggerSimple ? (openBlock(), createElementBlock(
          "div",
          {
            key: 1,
            class: normalizeClass([_ctx.state.prefix + "-trigger-button", _ctx.state.prefix + "-trigger-bottom-button", _ctx.state.offset === 0 ? _ctx.state.prefix + "-trigger-button-active" : "", _ctx.state.offset === 0 || _ctx.collapseRightBottom ? _ctx.state.prefix + "-trigger-button-show" : ""]),
            onMousedown: _cache[9] || (_cache[9] = function() {
              return _ctx.buttonMousedown && _ctx.buttonMousedown.apply(_ctx, arguments);
            }),
            onClick: _cache[10] || (_cache[10] = function() {
              return _ctx.buttonRightBottomClick && _ctx.buttonRightBottomClick.apply(_ctx, arguments);
            })
          },
          [createVNode(_component_icon_right_ward, {
            class: "tiny-svg-size"
          })],
          34
          /* CLASS, NEED_HYDRATION */
        )) : createCommentVNode("v-if", true)],
        38
        /* CLASS, STYLE, NEED_HYDRATION */
      ), createElementVNode(
        "div",
        {
          style: normalizeStyle({
            top: _ctx.threeAreas ? _ctx.state.leftTopPane + "px" : _ctx.state.offset + "%",
            height: _ctx.threeAreas ? "calc(100% - " + _ctx.state.leftTopPane + "px)" : ""
          }),
          class: normalizeClass([_ctx.state.paneClasses, "bottom-pane"])
        },
        [renderSlot(_ctx.$slots, "bottom")],
        6
        /* CLASS, STYLE */
      )],
      2
      /* CLASS */
    ))],
    2
    /* CLASS */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
