// @opentiny/vue-wizard@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/wizard/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  DOING_STATUS: "doing",
  READY_STATUS: "ready",
  WAIT_STATUS: "wait"
};
var wizardProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  data: {
    type: Array,
    default: function _default2() {
      return [];
    }
  },
  pageGuide: {
    type: Boolean,
    default: false
  },
  vertical: {
    type: Boolean,
    default: false
  },
  timeLineFlow: {
    type: Boolean,
    default: false
  }
});
var Wizard = defineComponent({
  name: $prefix + "Wizard",
  props: wizardProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Wizard.install = function(Vue) {
  Vue.component(Wizard.name, Wizard);
};
Wizard.version = version;
export {
  Wizard as default
};
