// @opentiny/vue-user-contact@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/user-contact/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var userContactProps = _extends({}, $props, {
  showName: {
    type: Boolean,
    default: true
  },
  showArrow: {
    type: Boolean,
    default: false
  },
  /**
   * 是否显示 roleNumber, colType 为 false 时生效
   */
  showNumber: {
    type: Boolean,
    default: true
  },
  /**
   * 是否显示 头像
   */
  showImg: {
    type: Boolean,
    default: true
  },
  placement: String,
  /**
   * 数据
   */
  data: {
    type: Object
  },
  espace: Array,
  popperClass: String,
  popperAppendToBody: {
    type: Boolean,
    default: true
  },
  isNewImMode: {
    type: Boolean,
    default: true
  }
});
var UserContact = defineComponent({
  name: $prefix + "UserContact",
  props: userContactProps,
  setup: function setup(props, context) {
    return $setup({
      context,
      props,
      template
    });
  }
});
var version = "3.20.0";
UserContact.install = function(Vue) {
  Vue.component(UserContact.name, UserContact);
};
UserContact.version = version;
export {
  UserContact as default
};
