// @opentiny/vue-drop-times@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/drop-times/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var DropTimes = defineComponent({
  name: $prefix + "DropTimes",
  props: _extends({}, $props, {
    modelValue: {},
    placeholder: {
      type: String,
      default: ""
    },
    tabindex: {
      type: String,
      default: "1"
    },
    size: String,
    popperClass: String,
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    start: {
      type: Number,
      default: 0
    },
    end: {
      type: Number,
      default: 24 * 60
    },
    step: {
      type: Number,
      default: 15
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
DropTimes.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
DropTimes.install = function(Vue) {
  Vue.component(DropTimes.name, DropTimes);
};
DropTimes.version = version;
export {
  DropTimes as default
};
