// @opentiny/vue-carousel@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
import "@opentiny/vue-theme/carousel/index.css";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var $constants = {
  CHILD_NAME: $prefix + "CarouselItem"
};
var Carousel = defineComponent({
  name: $prefix + "Carousel",
  componentName: "Carousel",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    initialIndex: {
      type: Number,
      default: 0
    },
    height: String,
    trigger: {
      type: String,
      default: "hover"
    },
    autoplay: {
      type: Boolean,
      default: false
    },
    interval: {
      type: Number,
      default: 3e3
    },
    loop: {
      type: Boolean,
      default: true
    },
    disabled: {
      type: Boolean,
      default: false
    },
    indicatorPosition: String,
    arrow: {
      type: String,
      default: "hover"
    },
    type: {
      type: String,
      default: "horizontal",
      validator: function validator(value) {
        return Boolean(!value || ~["card", "vertical", "horizontal"].indexOf(value));
      }
    },
    showTitle: {
      type: Boolean,
      default: false
    },
    aspectRatio: {
      type: String,
      default: "16:2"
    },
    indicatorStyle: {
      type: String,
      default: "light",
      validator: function validator2(value) {
        return Boolean(!value || ~["light", "dark"].indexOf(value));
      }
    },
    swipeable: Boolean,
    lite: Boolean,
    beforeSwipe: Function
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Carousel.install = function(Vue) {
  Vue.component(Carousel.name, Carousel);
};
Carousel.version = version;
export {
  Carousel as default
};
