// @opentiny/vue-calendar-bar@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var $constants = {
  POSITON: {
    START: "start",
    CENTER: "center",
    END: "end"
  }
};
var CalendarBar = defineComponent({
  name: $prefix + "CalendarBar",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    modelValue: String,
    config: {
      type: Object,
      default: function _default2() {
        return {};
      }
    },
    cascaderPosition: {
      type: String,
      default: "center",
      validator: function validator(value) {
        return Boolean(~Object.values($constants.POSITON).indexOf(value));
      }
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
CalendarBar.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
CalendarBar.install = function(Vue) {
  Vue.component(CalendarBar.name, CalendarBar);
};
CalendarBar.version = version;
export {
  CalendarBar as default
};
