// @opentiny/vue-popconfirm@3.20.0/lib/mobile-first.js
import { defineComponent, $prefix, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/popconfirm/vue';
import Popover from '@opentiny/vue-popover';
import Button from '@opentiny/vue-button';
import { iconSuccess, iconInfoSolid, iconWarning, iconError } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, createVNode, withCtx, createElementVNode, createBlock, resolveDynamicComponent, normalizeClass, createCommentVNode, toDisplayString, renderSlot, createTextVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var $constants = {
  ICON_MAP: {
    success: "icon-success",
    error: "icon-error",
    info: "icon-info-solid",
    warning: "icon-warning"
  }
};
var _sfc_main = defineComponent({
  name: $prefix + "Popconfirm",
  components: {
    TinyPopover: Popover,
    TinyButton: Button,
    IconSuccess: iconSuccess(),
    IconInfoSolid: iconInfoSolid(),
    IconWarning: iconWarning(),
    IconError: iconError()
  },
  props: {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    message: String,
    customClass: String,
    trigger: {
      type: String,
      default: "hover",
      validator: function validator(value) {
        return ~["click", "hover"].indexOf(value);
      }
    },
    cancelButton: {
      type: Boolean,
      default: true
    },
    closeOnClickOutside: {
      type: Boolean,
      default: true
    },
    title: String,
    placement: {
      type: String,
      default: "top"
    },
    width: {
      type: [String, Number],
      default: ""
    },
    type: [String, Object],
    reference: {},
    events: Object,
    popperAppendToBody: {
      type: Boolean,
      default: true
    }
  },
  emits: ["hide", "show", "confirm", "handleEmit"],
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-popconfirm",
  class: "inline-block"
};
var _hoisted_2 = {
  "data-tag": "tiny-popconfirm-header",
  class: "flex items-center leading-6 sm:pt-1"
};
var _hoisted_3 = {
  class: "flex-auto"
};
var _hoisted_4 = {
  "data-tag": "tiny-popconfirm-content"
};
var _hoisted_5 = {
  class: "sm:leading-6 sm:text-xs sm:text-color-text-secondary"
};
var _hoisted_6 = {
  "data-tag": "tiny-popconfirm-footer",
  class: "sm:pb-1"
};
var _hoisted_7 = {
  class: "hidden sm:flex mt-4 justify-end"
};
var _hoisted_8 = {
  class: "flex sm:hidden text-base justify-center"
};
var _hoisted_9 = {
  key: 1,
  class: "border-r border-r-color-border-separator"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_button = resolveComponent("tiny-button");
  var _component_tiny_popover = resolveComponent("tiny-popover");
  return openBlock(), createElementBlock("div", _hoisted_1, [createVNode(_component_tiny_popover, {
    ref: "popover",
    tiny_mode: "mobile-first",
    modelValue: _ctx.state.showPopover,
    "onUpdate:modelValue": _cache[2] || (_cache[2] = function($event) {
      return _ctx.state.showPopover = $event;
    }),
    placement: _ctx.placement,
    trigger: "manual",
    width: _ctx.width,
    title: _ctx.title,
    "popper-class": _ctx.m("min-w-[theme(spacing.44)] max-w-[theme(spacing.96)]", _ctx.customClass),
    "append-to-body": _ctx.popperAppendToBody,
    reference: _ctx.reference,
    onShow: _cache[3] || (_cache[3] = function($event) {
      return _ctx.handleEmit("show");
    }),
    onHide: _cache[4] || (_cache[4] = function($event) {
      return _ctx.handleEmit("hide");
    })
  }, {
    header: withCtx(function() {
      return [createElementVNode("div", _hoisted_2, [_ctx.type ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.getIcon), {
        key: 0,
        class: normalizeClass(_ctx.m("h-6 w-6 mr-2", {
          "fill-color-info-secondary": _ctx.type === "info"
        }, {
          "fill-color-error": _ctx.type === "error"
        }, {
          "fill-color-warning": _ctx.type === "warning"
        }, {
          "fill-color-success": _ctx.type === "success"
        }))
      }, null, 8, ["class"])) : createCommentVNode("v-if", true), createElementVNode(
        "div",
        _hoisted_3,
        toDisplayString(_ctx.title),
        1
        /* TEXT */
      )])];
    }),
    default: withCtx(function() {
      return [createElementVNode("div", _hoisted_4, [createElementVNode(
        "div",
        _hoisted_5,
        toDisplayString(_ctx.message),
        1
        /* TEXT */
      )])];
    }),
    footer: withCtx(function() {
      return [createElementVNode("div", _hoisted_6, [createElementVNode("div", _hoisted_7, [renderSlot(_ctx.$slots, "footer", {}, function() {
        return [_ctx.cancelButton ? (openBlock(), createBlock(_component_tiny_button, {
          key: 0,
          class: "mr-2 w-16",
          size: "mini",
          tiny_mode: "mobile-first",
          onClick: _ctx.hide
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.buttonMessage.cancel")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        }, 8, ["onClick"])) : createCommentVNode("v-if", true), createVNode(_component_tiny_button, {
          class: "w-16",
          size: "mini",
          tiny_mode: "mobile-first",
          type: "primary",
          onClick: _ctx.confirm
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.buttonMessage.confirm")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        }, 8, ["onClick"])];
      })]), createElementVNode("div", _hoisted_8, [renderSlot(_ctx.$slots, "footer", {}, function() {
        return [_ctx.cancelButton ? (openBlock(), createBlock(_component_tiny_button, {
          key: 0,
          "custom-class": "flex-1 text-color-text-secondary",
          tiny_mode: "mobile-first",
          type: "text",
          onClick: _ctx.hide
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.buttonMessage.cancel")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        }, 8, ["onClick"])) : createCommentVNode("v-if", true), _ctx.cancelButton ? (openBlock(), createElementBlock("span", _hoisted_9)) : createCommentVNode("v-if", true), createVNode(_component_tiny_button, {
          "custom-class": "flex-1 text-color-brand tiny-modal-mf-button",
          tiny_mode: "mobile-first",
          type: "text",
          onClick: _ctx.confirm
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.buttonMessage.confirm")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        }, 8, ["onClick"])];
      })])])];
    }),
    reference: withCtx(function() {
      return [createElementVNode(
        "div",
        {
          class: "inline-block",
          onClick: _cache[0] || (_cache[0] = function($event) {
            return _ctx.show("click");
          }),
          onMouseover: _cache[1] || (_cache[1] = function($event) {
            return _ctx.show("hover");
          })
        },
        [renderSlot(_ctx.$slots, "reference")],
        32
        /* NEED_HYDRATION */
      )];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["modelValue", "placement", "width", "title", "popper-class", "append-to-body", "reference"])]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
