// @opentiny/vue-float-button@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/float-button/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { iconLoading, iconClose } from '@opentiny/vue-icon';
import '@opentiny/vue-theme/float-button/index.css';
import { resolveComponent, withDirectives, openBlock, createElementBlock, normalizeClass, normalizeStyle, createBlock, createCommentVNode, resolveDynamicComponent, toDisplayString, renderSlot, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["click", "mouseout", "mouseover"],
  props: [].concat(props, ["type", "open", "description", "size", "icon", "resetTime", "tooltip", "loading", "disabled", "href", "target", "top", "bottom", "right", "left", "trigger", "shape", "backTop", "visibilityHeight", "element"]),
  components: {
    IconLoading: iconLoading(),
    IconClose: iconClose()
  },
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 3
};
var _hoisted_2 = {
  key: 4,
  ref: "tinyFloatButtonOpen",
  class: "tiny-float-button--open"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_loading = resolveComponent("icon-loading");
  var _component_icon_close = resolveComponent("icon-close");
  return withDirectives((openBlock(), createElementBlock(
    "button",
    {
      ref: "tinyFloatButton",
      class: normalizeClass(["tiny-float-button", [_ctx.type ? "tiny-float-button--" + _ctx.type : "", _ctx.size ? "tiny-float-button--" + _ctx.size : "", {
        "is-disabled": _ctx.state.buttonDisabled,
        "is-loading": _ctx.loading,
        "is-icon": _ctx.icon && !_ctx.loading && (_ctx.description || _ctx.slots.default),
        "is-only-icon": _ctx.icon && !_ctx.loading && !(_ctx.description || _ctx.slots.default),
        "is-circle": _ctx.shape === "circle",
        "is-square": _ctx.shape === "square"
      }]]),
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.handleClick && _ctx.handleClick.apply(_ctx, arguments);
      }),
      style: normalizeStyle({
        top: _ctx.top,
        bottom: _ctx.bottom,
        right: _ctx.right,
        left: _ctx.left
      })
    },
    [_ctx.loading ? (openBlock(), createBlock(_component_icon_loading, {
      key: 0,
      class: "tiny-icon-loading tiny-svg-size"
    })) : createCommentVNode("v-if", true), _ctx.icon && !_ctx.loading && !_ctx.state.open ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.icon), {
      key: 1,
      class: normalizeClass({
        "is-text": _ctx.description || _ctx.slots.default
      })
    }, null, 8, ["class"])) : createCommentVNode("v-if", true), !_ctx.loading && _ctx.state.open ? (openBlock(), createBlock(_component_icon_close, {
      key: 2
    })) : createCommentVNode("v-if", true), _ctx.description && !_ctx.state.open ? (openBlock(), createElementBlock(
      "div",
      _hoisted_1,
      toDisplayString(_ctx.description),
      1
      /* TEXT */
    )) : createCommentVNode("v-if", true), _ctx.state.open ? (openBlock(), createElementBlock(
      "div",
      _hoisted_2,
      [renderSlot(_ctx.$slots, "default")],
      512
      /* NEED_PATCH */
    )) : createCommentVNode("v-if", true)],
    6
    /* CLASS, STYLE */
  )), [[vShow, _ctx.state.show]]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
