// @opentiny/vue-tabbar-item@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileTemplate from "./mobile.js";
import MobileFirstTemplate from "./mobile-first.js";
import "@opentiny/vue-theme-mobile/tabbar-item/index.css";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("mobile" === (tinyMode || mode)) {
    return MobileTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return MobileTemplate;
};
var TabbarItem = defineComponent({
  name: $prefix + "TabbarItem",
  componentName: "TabbarItem",
  props: _extends({}, $props, {
    url: String,
    replace: Boolean,
    to: [String, Object],
    dot: Boolean,
    icon: Object,
    name: [Number, String],
    text: String,
    info: [Number, String],
    badge: [Number, String],
    customIcon: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
TabbarItem.install = function(Vue) {
  Vue.component(TabbarItem.name, TabbarItem);
};
TabbarItem.version = version;
export {
  TabbarItem as default
};
