// @opentiny/vue-tabbar-item@3.20.0/lib/mobile-first.js
import { defineComponent, $prefix, $props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/tabbar-item/vue';
import Badge from '@opentiny/vue-badge';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, normalizeStyle, createElementVNode, renderSlot, createBlock, withCtx, resolveDynamicComponent, createTextVNode, toDisplayString, createCommentVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "TabbarItem",
  componentName: "TabbarItem",
  props: _extends({}, $props, {
    dot: Boolean,
    icon: Object,
    name: [Number, String],
    badge: [Number, String],
    text: String,
    customIcon: {
      type: Boolean,
      default: false
    }
  }),
  components: {
    TinyBadge: Badge
  },
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_tiny_badge = resolveComponent("tiny-badge");
  return _ctx.state.showVm ? (openBlock(), createElementBlock(
    "div",
    {
      key: 0,
      "data-tag": "tiny-tabbar-item",
      class: normalizeClass(_ctx.m("flex flex-1 flex-col items-center justify-center cursor-pointer mx-0.5 [&:first-of-type]:ml-1 [&:last-of-type]:mr-1", _ctx.state.customIcon ? "my-1.5" : "my-1")),
      style: normalizeStyle({
        color: _ctx.state.renderColor
      }),
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.onClick && _ctx.onClick.apply(_ctx, arguments);
      })
    },
    [createElementVNode(
      "div",
      {
        "data-tag": "tiny-tabbar-item-icon",
        class: normalizeClass(["relative pointer-events-none", _ctx.state.customIcon ? "w-11 h-11 [&_svg]:w-11 [&_svg]:h-11" : "w-6 h-6 [&_svg]:w-6 [&_svg]:h-6 my-1"])
      },
      [renderSlot(_ctx.$slots, "icon", {
        active: _ctx.state.renderActive
      }, function() {
        return [_ctx.badge ? (openBlock(), createBlock(_component_tiny_badge, {
          key: 0,
          type: "icon",
          value: _ctx.badge
        }, {
          default: withCtx(function() {
            return [(openBlock(), createBlock(resolveDynamicComponent(_ctx.icon), {
              class: normalizeClass({
                "fill-color-brand": _ctx.state.renderActive || _ctx.state.customIcon
              }),
              style: normalizeStyle({
                fill: _ctx.state.renderColor
              })
            }, null, 8, ["class", "style"]))];
          }),
          _: 1
          /* STABLE */
        }, 8, ["value"])) : _ctx.dot ? (openBlock(), createBlock(_component_tiny_badge, {
          key: 1,
          type: "icon",
          "is-dot": _ctx.dot
        }, {
          default: withCtx(function() {
            return [(openBlock(), createBlock(resolveDynamicComponent(_ctx.icon), {
              class: normalizeClass({
                "fill-color-brand": _ctx.state.renderActive || _ctx.state.customIcon
              }),
              style: normalizeStyle({
                fill: _ctx.state.renderColor
              })
            }, null, 8, ["class", "style"]))];
          }),
          _: 1
          /* STABLE */
        }, 8, ["is-dot"])) : (openBlock(), createBlock(resolveDynamicComponent(_ctx.icon), {
          key: 2,
          class: normalizeClass({
            "fill-color-brand": _ctx.state.renderActive || _ctx.state.customIcon
          }),
          style: normalizeStyle({
            fill: _ctx.state.renderColor
          })
        }, null, 8, ["class", "style"]))];
      })],
      2
      /* CLASS */
    ), !_ctx.state.customIcon ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        "data-tag": "tiny-tabbar-item-title",
        class: normalizeClass(["pointer-events-none overflow-hidden whitespace-nowrap text-ellipsis text-xs text-center leading-3", {
          "text-color-brand": _ctx.state.renderActive
        }]),
        style: normalizeStyle({
          color: _ctx.state.renderColor,
          width: _ctx.state.itemWidth
        })
      },
      [renderSlot(_ctx.$slots, "default", {}, function() {
        return [createTextVNode(
          toDisplayString(_ctx.text || _ctx.name),
          1
          /* TEXT */
        )];
      })],
      6
      /* CLASS, STYLE */
    )) : createCommentVNode("v-if", true)],
    6
    /* CLASS, STYLE */
  )) : createCommentVNode("v-if", true);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
