// @opentiny/vue-cascader-menu@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/cascader-menu/vue';
import { defineComponent, $prefix, $props, setup } from '@opentiny/vue-common';
import Scrollbar from '@opentiny/vue-scrollbar';
import CascaderNode from '@opentiny/vue-cascader-node';
import { resolveComponent, openBlock, createBlock, normalizeClass, withCtx, createElementBlock, toDisplayString, Fragment, renderList } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "CascaderMenu",
  components: {
    TinyScrollbar: Scrollbar,
    CascaderNode
  },
  props: _extends({}, $props, {
    nodes: {
      type: Array,
      required: true
    },
    index: Number
  }),
  inject: ["panel", "cascaderRoot"],
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 0,
  "data-tag": "tiny-cascader-menu__empty-text",
  class: "absolute top-1/2 left-1/2 -translate-x-2/4 -translate-y-2/4 text-center text-color-text-placeholder"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_cascader_node = resolveComponent("cascader-node");
  var _component_tiny_scrollbar = resolveComponent("tiny-scrollbar");
  return openBlock(), createBlock(_component_tiny_scrollbar, {
    "data-tag": "tiny-cascader-menu",
    id: _ctx.state.menuId,
    role: "menu",
    ref: "cascaderMenu",
    class: normalizeClass(_ctx.m("min-w-[theme(spacing.54)] box-border text-color-text-primary pt-1")),
    tag: "div",
    "wrap-class": "overflow-auto h-auto pb-1.5 max-h-[theme(spacing.72)]",
    "view-class": "relative min-h-full m-0  list-none box-border pb-1 "
  }, {
    default: withCtx(function() {
      return [_ctx.state.isEmpty ? (openBlock(), createElementBlock(
        "div",
        _hoisted_1,
        toDisplayString(_ctx.t("ui.cascader.noData")),
        1
        /* TEXT */
      )) : (openBlock(true), createElementBlock(
        Fragment,
        {
          key: 1
        },
        renderList(_ctx.nodes, function(node, nodeIndex) {
          return openBlock(), createBlock(_component_cascader_node, {
            tiny_mode: "mobile-first",
            key: node.uid,
            node,
            "node-id": _ctx.state.menuId + "-" + nodeIndex,
            "data-haspopup": node.hasChildren,
            "data-owns": node.hasChildren ? _ctx.state.menuId : null
          }, null, 8, ["node", "node-id", "data-haspopup", "data-owns"]);
        }),
        128
        /* KEYED_FRAGMENT */
      ))];
    }),
    _: 1
    /* STABLE */
  }, 8, ["id", "class"]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
