// @opentiny/vue-floatbar@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/floatbar/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import { openBlock, createElementBlock, normalizeStyle, renderSlot, createElementVNode, Fragment, renderList, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["position", "data"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "navs"
};
var _hoisted_2 = ["href", "target"];
var _hoisted_3 = {
  key: 1,
  class: "nav-text"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-float-bar tiny-toolbar",
      style: normalizeStyle(_ctx.position)
    },
    [renderSlot(_ctx.$slots, "default", {
      slotScope: {
        data: _ctx.state.data
      }
    }, function() {
      return [createElementVNode("ul", _hoisted_1, [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.data, function(item, index) {
          return openBlock(), createElementBlock("li", {
            key: index,
            class: "nav-item"
          }, [item.url ? (openBlock(), createElementBlock("a", {
            key: 0,
            href: item.url,
            target: item.target,
            rel: "noopener noreferrer",
            class: "nav-link"
          }, toDisplayString(item.title), 9, _hoisted_2)) : (openBlock(), createElementBlock(
            "span",
            _hoisted_3,
            toDisplayString(item.title),
            1
            /* TEXT */
          ))]);
        }),
        128
        /* KEYED_FRAGMENT */
      ))])];
    })],
    4
    /* STYLE */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
