// @opentiny/vue-text-popup@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/text-popup/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var TextPopup = defineComponent({
  name: $prefix + "TextPopup",
  props: _extends({}, $props, {
    modelValue: {
      default: ""
    },
    placeholder: {
      type: String,
      required: false
    },
    readonly: {
      type: Boolean,
      default: false
    },
    width: {
      type: [Number, String],
      default: 150,
      validator: function validator(value) {
        return parseInt(value) >= 0;
      }
    },
    separtor: {
      type: String,
      default: ","
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
TextPopup.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
TextPopup.install = function(Vue) {
  Vue.component(TextPopup.name, TextPopup);
};
TextPopup.version = version;
export {
  TextPopup as default
};
