// @opentiny/vue-upload-dragger@3.20.0/lib/mobile-first.js
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/upload-dragger/vue';
import { openBlock, createElementBlock, normalizeClass, withModifiers, renderSlot } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "UploadDragger",
  emits: ["file"],
  props: [].concat(props, ["disabled", "customClass"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-upload-dragger",
      class: normalizeClass(_ctx.m("min-w-[theme(spacing.72)] min-h-[theme(spacing.40)] border border-dashed rounded", _ctx.state.dragover ? "border-color-brand" : "border-color-none-hover", _ctx.customClass)),
      onDrop: _cache[0] || (_cache[0] = withModifiers(function() {
        return _ctx.onDrop && _ctx.onDrop.apply(_ctx, arguments);
      }, ["prevent"])),
      onDragover: _cache[1] || (_cache[1] = withModifiers(function() {
        return _ctx.onDragOver && _ctx.onDragOver.apply(_ctx, arguments);
      }, ["prevent"])),
      onDragleave: _cache[2] || (_cache[2] = withModifiers(function($event) {
        return _ctx.state.dragover = false;
      }, ["prevent"]))
    },
    [renderSlot(_ctx.$slots, "default")],
    34
    /* CLASS, NEED_HYDRATION */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
