// @opentiny/vue-form-item@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var $constants = {
  FORM_NAME: "Form",
  FORM_ITEM_NAME: "FormItem"
};
var formItemProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  appendToBody: {
    type: Boolean,
    default: void 0
  },
  error: {
    type: String,
    default: ""
  },
  for: String,
  inlineMessage: {
    type: Boolean,
    default: void 0
  },
  messageType: String,
  label: String,
  labelWidth: String,
  manual: Boolean,
  popperOptions: {
    type: Object,
    default: function _default2() {
      return {};
    }
  },
  prop: String,
  required: {
    type: Boolean,
    default: void 0
  },
  rules: [Object, Array],
  showMessage: {
    type: Boolean,
    default: true
  },
  size: String,
  tipContent: String,
  validateDisabled: Boolean,
  validateDebounce: Boolean,
  validatePosition: String,
  validateStatus: String,
  validateType: String,
  validateIcon: Object,
  ellipsis: {
    type: Boolean,
    default: false
  },
  vertical: {
    type: Boolean,
    default: false
  },
  extra: String
});
var FormItem = defineComponent({
  name: $prefix + "FormItem",
  componentName: "FormItem",
  props: formItemProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
FormItem.install = function(Vue) {
  Vue.component(FormItem.name, FormItem);
};
export {
  FormItem as default
};
