// @opentiny/vue-skeleton@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/skeleton/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {};
var skeletonProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  modelValue: String,
  loading: {
    type: Boolean,
    default: true
  },
  rows: {
    type: Number,
    default: 3
  },
  avatar: {
    type: Boolean,
    default: false
  },
  rowsWidth: {
    type: Array,
    default: function _default2() {
      return [];
    }
  },
  animated: {
    type: Boolean,
    default: true
  }
});
var Skeleton = defineComponent({
  name: $prefix + "Skeleton",
  props: skeletonProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
Skeleton.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Skeleton.install = function(Vue) {
  Vue.component(Skeleton.name, Skeleton);
};
Skeleton.version = version;
export {
  Skeleton as default
};
