// @opentiny/vue-virtual-scroll-box@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import { numericProp, makeNumericProp, makeArrayProp } from "@opentiny/vue-renderless/common/prop-util";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var VirtualScrollBox = defineComponent({
  name: $prefix + "VirtualScrollBox",
  props: _extends({}, $props, {
    width: numericProp,
    // 区域的宽度
    height: numericProp,
    // 区域的高度
    rowBuffer: makeNumericProp(200),
    // 行缓冲区高度
    columnBuffer: makeNumericProp(200),
    // 列缓冲区宽度
    scrollbarSize: numericProp,
    // 滚动条尺寸
    columns: makeArrayProp(),
    // 所有的列
    columnSizes: makeArrayProp(),
    // 每个列对应的宽度
    rows: makeArrayProp(),
    // 所有的行
    rowSizes: makeArrayProp(),
    // 每个行对应的高度
    fixedRows: makeArrayProp(),
    // 行固定配置
    fixedColumns: makeArrayProp(),
    // 列固定配置
    spanConfig: makeArrayProp()
    // 单元格合并配置
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
VirtualScrollBox.install = function(Vue) {
  Vue.component(VirtualScrollBox.name, VirtualScrollBox);
};
VirtualScrollBox.version = version;
export {
  VirtualScrollBox as default
};
