// @opentiny/vue-multi-select-item@3.20.0/lib/index.js
function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileTemplate from "./mobile.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileTemplate;
};
var MultiSelectItem = defineComponent({
  name: $prefix + "MultiSelectItem",
  inject: {},
  props: _extends({}, $props, {
    option: {
      type: Object,
      default: function _default() {
        return {
          label: ""
        };
      }
    },
    divided: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.20.0";
MultiSelectItem.install = function(Vue) {
  Vue.component(MultiSelectItem.name, MultiSelectItem);
};
MultiSelectItem.version = version;
export {
  MultiSelectItem as default
};
