// @opentiny/vue-multi-select-item@3.20.0/lib/mobile.js
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/multi-select-item/vue';
import { iconChevronDown } from '@opentiny/vue-icon';
import Button from '@opentiny/vue-button';
import '@opentiny/vue-theme-mobile/multi-select-item/index.css';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, withModifiers, createElementVNode, Fragment, renderList, createBlock, resolveDynamicComponent, createCommentVNode, renderSlot, toDisplayString, normalizeStyle, createVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "MultiSelectItem",
  components: {
    TinyButton: Button,
    IconChevronDown: iconChevronDown()
  },
  props: [].concat(props, ["option", "divided"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["aria-disabled", "tabindex"];
var _hoisted_2 = {
  class: "tiny-mobile-multi-select-item__wrap"
};
var _hoisted_3 = {
  class: "tiny-mobile-multi-select-item__content"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _ctx$option$children;
  var _component_IconChevronDown = resolveComponent("IconChevronDown");
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(["tiny-mobile-multi-select-item", {
      "is-active": _ctx.option.active,
      "is-disabled": _ctx.option.disabled,
      "tiny-mobile-multi-select-item--divided": _ctx.divided,
      "has-children": (_ctx$option$children = _ctx.option.children) == null ? void 0 : _ctx$option$children.length
    }]),
    ref: "multiSelectItem",
    onClick: _cache[1] || (_cache[1] = withModifiers(function() {
      return _ctx.handleClick && _ctx.handleClick.apply(_ctx, arguments);
    }, ["stop"])),
    onMousedown: _cache[2] || (_cache[2] = withModifiers(function() {
    }, ["stop"])),
    "aria-disabled": _ctx.option.disabled,
    tabindex: _ctx.option.disabled ? null : -1
  }, [createElementVNode("div", _hoisted_2, [createElementVNode("div", _hoisted_3, [(openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.option.level, function(i) {
      return openBlock(), createElementBlock("span", {
        key: i,
        class: "tiny-mobile-multi-select-item__content-indent"
      });
    }),
    128
    /* KEYED_FRAGMENT */
  )), _ctx.option.icon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.option.icon), {
    key: 0,
    class: "tiny-svg-size"
  })) : createCommentVNode("v-if", true), renderSlot(_ctx.$slots, "default", {
    itemData: _ctx.option
  }, function() {
    return [createElementVNode(
      "span",
      null,
      toDisplayString(_ctx.option.label),
      1
      /* TEXT */
    )];
  })]), _ctx.option.hasChild ? (openBlock(), createElementBlock(
    "span",
    {
      key: 0,
      class: normalizeClass(["tiny-mobile-multi-select-item__suffix-icon"]),
      style: normalizeStyle({
        transform: _ctx.option.expanded && !_ctx.slots.suffix ? "rotate(180deg)" : "none"
      }),
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.toggleExpand && _ctx.toggleExpand.apply(_ctx, arguments);
      })
    },
    [renderSlot(_ctx.$slots, "suffix", {}, function() {
      return [createVNode(_component_IconChevronDown)];
    })],
    4
    /* STYLE */
  )) : createCommentVNode("v-if", true)])], 42, _hoisted_1);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobile as default };
