// @opentiny/vue-column-list-group@3.20.0/lib/index.js
import { renderless, api } from '@opentiny/vue-renderless/column-list-group/vue';
import { defineComponent, $prefix, $props, setup } from '@opentiny/vue-common';
import { openBlock, createElementBlock, normalizeClass, renderSlot } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "ColumnListGroup",
  componentName: "ColumnListGroup",
  emits: ["update:modelValue", "change"],
  props: _extends({}, $props, {
    modelValue: {},
    title: String,
    disabled: Boolean,
    size: {
      type: String,
      default: "medium"
    },
    type: {
      type: String,
      default: ""
    },
    showCheckbox: {
      type: Boolean,
      default: false
    },
    checkboxDisabled: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-column-list-group",
      class: normalizeClass(["w-full", _ctx.type !== "list" ? "[&_[role=column-list-item]]:mb-3 [&_[role=column-list-item]:last-child]:mb-0" : ""])
    },
    [renderSlot(_ctx.$slots, "default")],
    2
    /* CLASS */
  );
}
var ColumnListGroup = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.20.0";
ColumnListGroup.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
ColumnListGroup.install = function(Vue) {
  Vue.component(ColumnListGroup.name, ColumnListGroup);
};
ColumnListGroup.version = version;

export { ColumnListGroup as default };
