// @opentiny/vue-popover@3.20.0/lib/mobile-first.js
import { defineComponent, $prefix, setup, $props } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/popover/vue';
import { iconClose } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createVNode, Transition, withCtx, withDirectives, createElementVNode, normalizeStyle, renderSlot, toDisplayString, createCommentVNode, createTextVNode, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var classes = {
  "title": "text-color-icon-primary font-bold text-xs",
  "popper": "sm:leading-5 sm:absolute sm:px-4 sm:py-3 sm:z-[2000] sm:text-justify sm:text-xs break-all sm:max-w-[theme(spacing.96)] sm:shadow-md sm:bg-color-bg-1 sm:rounded sm:[&[x-placement^=top]]:mb-2.5 sm:[&[x-placement^=bottom]]:mt-2.5 sm:[&[x-placement^=right]]:ml-2.5 sm:[&[x-placement^=left]]:mr-2.5",
  "arrow": 'absolute ml-0.5 border-b-color-bg-1 border-l-color-bg-1 rounded-bl-sm block w-0 h-0 border-4 border-transparent border-solid after:absolute after:block after:border-transparent after:border-solid after:content-[""]',
  "placement-top": "-bottom-1 left-1/2 -rotate-45",
  "placement-bottom": "-top-1 left-1/2 rotate-[135deg]",
  "placement-right": "-left-1.5 top-1/2 rotate-45",
  "placement-left": "top-1/2 -right-1 rotate-[225deg]"
};
var _sfc_main = defineComponent({
  inheritAttrs: false,
  name: $prefix + "Popover",
  components: {
    IconClose: iconClose()
  },
  emits: ["update:modelValue", "hide", "show", "after-enter", "after-leave", "created"],
  props: _extends({}, $props, {
    appendToBody: {
      type: Boolean,
      default: true
    },
    arrowOffset: {
      type: Number,
      default: 0
    },
    boundariesPadding: {
      type: Number,
      default: 5
    },
    closeDelay: {
      type: Number,
      default: 200
    },
    content: String,
    disabled: Boolean,
    modelValue: Boolean,
    offset: {
      default: 0
    },
    openDelay: {
      type: Number,
      default: 0
    },
    placement: {
      type: String,
      default: "bottom"
    },
    popper: {},
    popperClass: String,
    popperOptions: {
      type: Object,
      default: function _default() {
        return {
          gpuAcceleration: false
        };
      }
    },
    reference: {},
    tabindex: {
      type: Number,
      default: 0
    },
    title: String,
    transformOrigin: {
      type: [Boolean, String],
      default: true
    },
    transition: {
      type: String,
      default: "fade-in-linear"
    },
    trigger: {
      type: String,
      default: "click",
      validator: function validator(value) {
        return ~["click", "focus", "hover", "manual"].indexOf(value);
      }
    },
    visibleArrow: {
      default: true
    },
    genArrowByHtml: {
      type: Boolean,
      default: function _default2() {
        return true;
      }
    },
    width: {}
  }),
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = ["id", "aria-hidden"];
var _hoisted_2 = {
  class: "hidden sm:block leading-5.5",
  "data-tag": "tiny-popover-arrow"
};
var _hoisted_3 = ["textContent"];
var _hoisted_4 = {
  "data-tag": "tiny-popover-body",
  class: /* @__PURE__ */ normalizeClass(["sm:hidden block fixed left-0 right-0 top-0 bottom-0 text-sm text-color-text-primary transition-top ease-out duration-300", "before:content-[''] before:fixed before:left-0 before:right-0 before:top-0 before:bottom-0 before:w-full before:h-full before:bg-color-bg-7"])
};
var _hoisted_5 = {
  class: /* @__PURE__ */ normalizeClass(["absolute bottom-0 top-auto left-1/2 -translate-x-1/2 w-full max-h-full p-3 pointer-events-auto"])
};
var _hoisted_6 = {
  class: "flex flex-col w-full h-full rounded-lg overflow-hidden shadow-xl bg-color-bg-1"
};
var _hoisted_7 = {
  "data-tag": "tiny-popover-header",
  class: "flex items-center leading-6 text-sm px-6 pt-5"
};
var _hoisted_8 = {
  class: "flex-auto text-base truncate font-semibold"
};
var _hoisted_9 = {
  "data-tag": "tiny-popover-content",
  class: "flex flex-auto leading-5 overflow-auto mx-6 mt-4 mb-5"
};
var _hoisted_10 = {
  key: 0,
  "data-tag": "tiny-popover-footer",
  class: "w-full py-2.5 px-6 border-t-0.5 border-color-border-separator text-center"
};
var _hoisted_11 = {
  "data-tag": "tiny-popover-reference",
  ref: "wrapper"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_icon_close = resolveComponent("icon-close");
  return openBlock(), createElementBlock(
    "span",
    {
      ref: "root",
      class: normalizeClass(_ctx.$attrs.class)
    },
    [createVNode(Transition, {
      name: _ctx.transition,
      onAfterEnter: _ctx.handleAfterEnter,
      onAfterLeave: _ctx.handleAfterLeave,
      persisted: ""
    }, {
      default: withCtx(function() {
        return [withDirectives(createElementVNode("div", {
          "data-tag": "tiny-popover tiny-popper",
          class: normalizeClass([_ctx.m(_ctx.gcls("popper"), _ctx.popperClass)]),
          ref: "popper",
          style: normalizeStyle({
            width: _ctx.width + "px"
          }),
          role: "tooltip",
          id: _ctx.state.tooltipId,
          "aria-hidden": _ctx.disabled || !_ctx.state.showPopper ? "true" : "false"
        }, [createElementVNode("div", _hoisted_2, [_ctx.slots.header ? renderSlot(_ctx.$slots, "header", {
          key: 0
        }) : _ctx.title ? (openBlock(), createElementBlock("div", {
          key: 1,
          class: normalizeClass([_ctx.gcls("title")]),
          textContent: toDisplayString(_ctx.title)
        }, null, 10, _hoisted_3)) : createCommentVNode("v-if", true), renderSlot(_ctx.$slots, "default", {}, function() {
          return [createTextVNode(
            toDisplayString(_ctx.content),
            1
            /* TEXT */
          )];
        }), renderSlot(_ctx.$slots, "footer"), _ctx.visibleArrow ? (openBlock(), createElementBlock(
          "div",
          {
            key: 2,
            "data-tag": "tiny-popover-visible",
            "x-arrow": "",
            class: normalizeClass([_ctx.gcls("arrow"), _ctx.gcls("placement-" + _ctx.state.xPlacement.split("-")[0])])
          },
          null,
          2
          /* CLASS */
        )) : createCommentVNode("v-if", true)]), createElementVNode("div", _hoisted_4, [createElementVNode("div", _hoisted_5, [createElementVNode("div", _hoisted_6, [createElementVNode("div", _hoisted_7, [createElementVNode(
          "span",
          _hoisted_8,
          toDisplayString(_ctx.title || _ctx.t("ui.alert.title")),
          1
          /* TEXT */
        ), createVNode(_component_icon_close, {
          onClick: _cache[0] || (_cache[0] = function($event) {
            return _ctx.state.showPopper = false;
          })
        })]), createElementVNode("div", _hoisted_9, [renderSlot(_ctx.$slots, "default", {}, function() {
          return [createTextVNode(
            toDisplayString(_ctx.content),
            1
            /* TEXT */
          )];
        })]), _ctx.slots.footer ? (openBlock(), createElementBlock("div", _hoisted_10, [renderSlot(_ctx.$slots, "footer")])) : createCommentVNode("v-if", true)])])])], 14, _hoisted_1), [[vShow, !_ctx.disabled && _ctx.state.showPopper]])];
      }),
      _: 3
      /* FORWARDED */
    }, 8, ["name", "onAfterEnter", "onAfterLeave"]), createElementVNode(
      "span",
      _hoisted_11,
      [renderSlot(_ctx.$slots, "reference")],
      512
      /* NEED_PATCH */
    )],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
