// @opentiny/vue@3.20.0/index.js
import ActionMenu from "@opentiny/vue-action-menu";
import { default as default2, default as default3 } from "@opentiny/vue-action-menu";
import ActionSheet from "@opentiny/vue-action-sheet";
import { default as default4, default as default5 } from "@opentiny/vue-action-sheet";
import Alert from "@opentiny/vue-alert";
import { default as default6, default as default7 } from "@opentiny/vue-alert";
import Amount from "@opentiny/vue-amount";
import { default as default8, default as default9 } from "@opentiny/vue-amount";
import Anchor from "@opentiny/vue-anchor";
import { default as default10, default as default11 } from "@opentiny/vue-anchor";
import Area from "@opentiny/vue-area";
import { default as default12, default as default13 } from "@opentiny/vue-area";
import AsyncFlowchart from "@opentiny/vue-async-flowchart";
import { default as default14, default as default15 } from "@opentiny/vue-async-flowchart";
import Autocomplete from "@opentiny/vue-autocomplete";
import { default as default16, default as default17 } from "@opentiny/vue-autocomplete";
import Avatar from "@opentiny/vue-avatar";
import { default as default18, default as default19 } from "@opentiny/vue-avatar";
import Badge from "@opentiny/vue-badge";
import { default as default20, default as default21 } from "@opentiny/vue-badge";
import BaseSelect from "@opentiny/vue-base-select";
import { default as default22, default as default23 } from "@opentiny/vue-base-select";
import Breadcrumb from "@opentiny/vue-breadcrumb";
import { default as default24, default as default25 } from "@opentiny/vue-breadcrumb";
import BreadcrumbItem from "@opentiny/vue-breadcrumb-item";
import { default as default26, default as default27 } from "@opentiny/vue-breadcrumb-item";
import BulletinBoard from "@opentiny/vue-bulletin-board";
import { default as default28, default as default29 } from "@opentiny/vue-bulletin-board";
import Button from "@opentiny/vue-button";
import { default as default30, default as default31 } from "@opentiny/vue-button";
import ButtonGroup from "@opentiny/vue-button-group";
import { default as default32, default as default33 } from "@opentiny/vue-button-group";
import Calendar from "@opentiny/vue-calendar";
import { default as default34, default as default35 } from "@opentiny/vue-calendar";
import CalendarBar from "@opentiny/vue-calendar-bar";
import { default as default36, default as default37 } from "@opentiny/vue-calendar-bar";
import CalendarView from "@opentiny/vue-calendar-view";
import { default as default38, default as default39 } from "@opentiny/vue-calendar-view";
import Card from "@opentiny/vue-card";
import { default as default40, default as default41 } from "@opentiny/vue-card";
import CardGroup from "@opentiny/vue-card-group";
import { default as default42, default as default43 } from "@opentiny/vue-card-group";
import CardTemplate from "@opentiny/vue-card-template";
import { default as default44, default as default45 } from "@opentiny/vue-card-template";
import Carousel from "@opentiny/vue-carousel";
import { default as default46, default as default47 } from "@opentiny/vue-carousel";
import CarouselItem from "@opentiny/vue-carousel-item";
import { default as default48, default as default49 } from "@opentiny/vue-carousel-item";
import Cascader from "@opentiny/vue-cascader";
import { default as default50, default as default51 } from "@opentiny/vue-cascader";
import CascaderMenu from "@opentiny/vue-cascader-menu";
import { default as default52, default as default53 } from "@opentiny/vue-cascader-menu";
import CascaderMobile from "@opentiny/vue-cascader-mobile";
import { default as default54, default as default55 } from "@opentiny/vue-cascader-mobile";
import CascaderNode from "@opentiny/vue-cascader-node";
import { default as default56, default as default57 } from "@opentiny/vue-cascader-node";
import CascaderPanel from "@opentiny/vue-cascader-panel";
import { default as default58, default as default59 } from "@opentiny/vue-cascader-panel";
import CascaderSelect from "@opentiny/vue-cascader-select";
import { default as default60, default as default61 } from "@opentiny/vue-cascader-select";
import CascaderView from "@opentiny/vue-cascader-view";
import { default as default62, default as default63 } from "@opentiny/vue-cascader-view";
import Cell from "@opentiny/vue-cell";
import { default as default64, default as default65 } from "@opentiny/vue-cell";
import Checkbox from "@opentiny/vue-checkbox";
import { default as default66, default as default67 } from "@opentiny/vue-checkbox";
import CheckboxButton from "@opentiny/vue-checkbox-button";
import { default as default68, default as default69 } from "@opentiny/vue-checkbox-button";
import CheckboxGroup from "@opentiny/vue-checkbox-group";
import { default as default70, default as default71 } from "@opentiny/vue-checkbox-group";
import Col from "@opentiny/vue-col";
import { default as default72, default as default73 } from "@opentiny/vue-col";
import Collapse from "@opentiny/vue-collapse";
import { default as default74, default as default75 } from "@opentiny/vue-collapse";
import CollapseItem from "@opentiny/vue-collapse-item";
import { default as default76, default as default77 } from "@opentiny/vue-collapse-item";
import CollapseTransition from "@opentiny/vue-collapse-transition";
import { default as default78, default as default79 } from "@opentiny/vue-collapse-transition";
import ColorPicker from "@opentiny/vue-color-picker";
import { default as default80, default as default81 } from "@opentiny/vue-color-picker";
import ColorSelectPanel from "@opentiny/vue-color-select-panel";
import { default as default82, default as default83 } from "@opentiny/vue-color-select-panel";
import ColumnListGroup from "@opentiny/vue-column-list-group";
import { default as default84, default as default85 } from "@opentiny/vue-column-list-group";
import ColumnListItem from "@opentiny/vue-column-list-item";
import { default as default86, default as default87 } from "@opentiny/vue-column-list-item";
import Company from "@opentiny/vue-company";
import { default as default88, default as default89 } from "@opentiny/vue-company";
import ConfigProvider from "@opentiny/vue-config-provider";
import { default as default90, default as default91 } from "@opentiny/vue-config-provider";
import Container from "@opentiny/vue-container";
import { default as default92, default as default93 } from "@opentiny/vue-container";
import Country from "@opentiny/vue-country";
import { default as default94, default as default95 } from "@opentiny/vue-country";
import Crop from "@opentiny/vue-crop";
import { default as default96, default as default97 } from "@opentiny/vue-crop";
import Currency from "@opentiny/vue-currency";
import { default as default98, default as default99 } from "@opentiny/vue-currency";
import DatePanel from "@opentiny/vue-date-panel";
import { default as default100, default as default101 } from "@opentiny/vue-date-panel";
import DatePicker from "@opentiny/vue-date-picker";
import { default as default102, default as default103 } from "@opentiny/vue-date-picker";
import DatePickerMobileFirst from "@opentiny/vue-date-picker-mobile-first";
import { default as default104, default as default105 } from "@opentiny/vue-date-picker-mobile-first";
import DateRange from "@opentiny/vue-date-range";
import { default as default106, default as default107 } from "@opentiny/vue-date-range";
import DateTable from "@opentiny/vue-date-table";
import { default as default108, default as default109 } from "@opentiny/vue-date-table";
import Dept from "@opentiny/vue-dept";
import { default as default110, default as default111 } from "@opentiny/vue-dept";
import DialogBox from "@opentiny/vue-dialog-box";
import { default as default112, default as default113 } from "@opentiny/vue-dialog-box";
import DialogSelect from "@opentiny/vue-dialog-select";
import { default as default114, default as default115 } from "@opentiny/vue-dialog-select";
import Divider from "@opentiny/vue-divider";
import { default as default116, default as default117 } from "@opentiny/vue-divider";
import Drawer from "@opentiny/vue-drawer";
import { default as default118, default as default119 } from "@opentiny/vue-drawer";
import DropRoles from "@opentiny/vue-drop-roles";
import { default as default120, default as default121 } from "@opentiny/vue-drop-roles";
import DropTimes from "@opentiny/vue-drop-times";
import { default as default122, default as default123 } from "@opentiny/vue-drop-times";
import Dropdown from "@opentiny/vue-dropdown";
import { default as default124, default as default125 } from "@opentiny/vue-dropdown";
import DropdownItem from "@opentiny/vue-dropdown-item";
import { default as default126, default as default127 } from "@opentiny/vue-dropdown-item";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import { default as default128, default as default129 } from "@opentiny/vue-dropdown-menu";
import DynamicScroller from "@opentiny/vue-dynamic-scroller";
import { default as default130, default as default131 } from "@opentiny/vue-dynamic-scroller";
import DynamicScrollerItem from "@opentiny/vue-dynamic-scroller-item";
import { default as default132, default as default133 } from "@opentiny/vue-dynamic-scroller-item";
import Espace from "@opentiny/vue-espace";
import { default as default134, default as default135 } from "@opentiny/vue-espace";
import Exception from "@opentiny/vue-exception";
import { default as default136, default as default137 } from "@opentiny/vue-exception";
import FallMenu from "@opentiny/vue-fall-menu";
import { default as default138, default as default139 } from "@opentiny/vue-fall-menu";
import FileUpload from "@opentiny/vue-file-upload";
import { default as default140, default as default141 } from "@opentiny/vue-file-upload";
import Filter from "@opentiny/vue-filter";
import { default as default142, default as default143 } from "@opentiny/vue-filter";
import FilterBar from "@opentiny/vue-filter-bar";
import { default as default144, default as default145 } from "@opentiny/vue-filter-bar";
import FilterBox from "@opentiny/vue-filter-box";
import { default as default146, default as default147 } from "@opentiny/vue-filter-box";
import FilterPanel from "@opentiny/vue-filter-panel";
import { default as default148, default as default149 } from "@opentiny/vue-filter-panel";
import FloatButton from "@opentiny/vue-float-button";
import { default as default150, default as default151 } from "@opentiny/vue-float-button";
import Floatbar from "@opentiny/vue-floatbar";
import { default as default152, default as default153 } from "@opentiny/vue-floatbar";
import FloatingButton from "@opentiny/vue-floating-button";
import { default as default154, default as default155 } from "@opentiny/vue-floating-button";
import Flowchart from "@opentiny/vue-flowchart";
import { default as default156, default as default157 } from "@opentiny/vue-flowchart";
import FluentEditor from "@opentiny/vue-fluent-editor";
import { default as default158, default as default159 } from "@opentiny/vue-fluent-editor";
import Form from "@opentiny/vue-form";
import { default as default160, default as default161 } from "@opentiny/vue-form";
import FormItem from "@opentiny/vue-form-item";
import { default as default162, default as default163 } from "@opentiny/vue-form-item";
import Fullscreen from "@opentiny/vue-fullscreen";
import { default as default164, default as default165 } from "@opentiny/vue-fullscreen";
import Grid from "@opentiny/vue-grid";
import { default as default166, default as default167 } from "@opentiny/vue-grid";
import GridColumn from "@opentiny/vue-grid-column";
import { default as default168, default as default169 } from "@opentiny/vue-grid-column";
import GridManager from "@opentiny/vue-grid-manager";
import { default as default170, default as default171 } from "@opentiny/vue-grid-manager";
import GridSelect from "@opentiny/vue-grid-select";
import { default as default172, default as default173 } from "@opentiny/vue-grid-select";
import GridToolbar from "@opentiny/vue-grid-toolbar";
import { default as default174, default as default175 } from "@opentiny/vue-grid-toolbar";
import Guide from "@opentiny/vue-guide";
import { default as default176, default as default177 } from "@opentiny/vue-guide";
import Hrapprover from "@opentiny/vue-hrapprover";
import { default as default178, default as default179 } from "@opentiny/vue-hrapprover";
import Huicharts from "@opentiny/vue-huicharts";
import { default as default180, default as default181, default as default182 } from "@opentiny/vue-huicharts";
import HuichartsAmap from "@opentiny/vue-huicharts-amap";
import { default as default183, default as default184, default as default185 } from "@opentiny/vue-huicharts-amap";
import HuichartsBar from "@opentiny/vue-huicharts-bar";
import { default as default186, default as default187, default as default188 } from "@opentiny/vue-huicharts-bar";
import HuichartsBmap from "@opentiny/vue-huicharts-bmap";
import { default as default189, default as default190, default as default191 } from "@opentiny/vue-huicharts-bmap";
import HuichartsBoxplot from "@opentiny/vue-huicharts-boxplot";
import { default as default192, default as default193, default as default194 } from "@opentiny/vue-huicharts-boxplot";
import HuichartsCandle from "@opentiny/vue-huicharts-candle";
import { default as default195, default as default196, default as default197 } from "@opentiny/vue-huicharts-candle";
import Core from "@opentiny/vue-huicharts-core";
import { default as default198, default as default199, default as default200 } from "@opentiny/vue-huicharts-core";
import HuichartsFunnel from "@opentiny/vue-huicharts-funnel";
import { default as default201, default as default202, default as default203 } from "@opentiny/vue-huicharts-funnel";
import HuichartsGauge from "@opentiny/vue-huicharts-gauge";
import { default as default204, default as default205, default as default206 } from "@opentiny/vue-huicharts-gauge";
import HuichartsGraph from "@opentiny/vue-huicharts-graph";
import { default as default207, default as default208, default as default209 } from "@opentiny/vue-huicharts-graph";
import HuichartsHeatmap from "@opentiny/vue-huicharts-heatmap";
import { default as default210, default as default211, default as default212 } from "@opentiny/vue-huicharts-heatmap";
import HuichartsHistogram from "@opentiny/vue-huicharts-histogram";
import { default as default213, default as default214, default as default215 } from "@opentiny/vue-huicharts-histogram";
import HuichartsLine from "@opentiny/vue-huicharts-line";
import { default as default216, default as default217, default as default218 } from "@opentiny/vue-huicharts-line";
import HuichartsLiquidfill from "@opentiny/vue-huicharts-liquidfill";
import { default as default219, default as default220, default as default221 } from "@opentiny/vue-huicharts-liquidfill";
import HuichartsMap from "@opentiny/vue-huicharts-map";
import { default as default222, default as default223, default as default224 } from "@opentiny/vue-huicharts-map";
import HuichartsPie from "@opentiny/vue-huicharts-pie";
import { default as default225, default as default226, default as default227 } from "@opentiny/vue-huicharts-pie";
import HuichartsProcess from "@opentiny/vue-huicharts-process";
import { default as default228, default as default229, default as default230 } from "@opentiny/vue-huicharts-process";
import HuichartsRadar from "@opentiny/vue-huicharts-radar";
import { default as default231, default as default232, default as default233 } from "@opentiny/vue-huicharts-radar";
import HuichartsRing from "@opentiny/vue-huicharts-ring";
import { default as default234, default as default235, default as default236 } from "@opentiny/vue-huicharts-ring";
import HuichartsSankey from "@opentiny/vue-huicharts-sankey";
import { default as default237, default as default238, default as default239 } from "@opentiny/vue-huicharts-sankey";
import HuichartsScatter from "@opentiny/vue-huicharts-scatter";
import { default as default240, default as default241, default as default242 } from "@opentiny/vue-huicharts-scatter";
import HuichartsSunburst from "@opentiny/vue-huicharts-sunburst";
import { default as default243, default as default244, default as default245 } from "@opentiny/vue-huicharts-sunburst";
import HuichartsTree from "@opentiny/vue-huicharts-tree";
import { default as default246, default as default247, default as default248 } from "@opentiny/vue-huicharts-tree";
import HuichartsWaterfall from "@opentiny/vue-huicharts-waterfall";
import { default as default249, default as default250, default as default251 } from "@opentiny/vue-huicharts-waterfall";
import HuichartsWordcloud from "@opentiny/vue-huicharts-wordcloud";
import { default as default252, default as default253, default as default254 } from "@opentiny/vue-huicharts-wordcloud";
import Image from "@opentiny/vue-image";
import { default as default255, default as default256 } from "@opentiny/vue-image";
import ImageViewer from "@opentiny/vue-image-viewer";
import { default as default257, default as default258 } from "@opentiny/vue-image-viewer";
import IndexBar from "@opentiny/vue-index-bar";
import { default as default259, default as default260 } from "@opentiny/vue-index-bar";
import IndexBarAnchor from "@opentiny/vue-index-bar-anchor";
import { default as default261, default as default262 } from "@opentiny/vue-index-bar-anchor";
import Input from "@opentiny/vue-input";
import { default as default263, default as default264 } from "@opentiny/vue-input";
import IpAddress from "@opentiny/vue-ip-address";
import { default as default265, default as default266 } from "@opentiny/vue-ip-address";
import Label from "@opentiny/vue-label";
import { default as default267, default as default268 } from "@opentiny/vue-label";
import Layout from "@opentiny/vue-layout";
import { default as default269, default as default270 } from "@opentiny/vue-layout";
import Link from "@opentiny/vue-link";
import { default as default271, default as default272 } from "@opentiny/vue-link";
import LinkMenu from "@opentiny/vue-link-menu";
import { default as default273, default as default274 } from "@opentiny/vue-link-menu";
import List from "@opentiny/vue-list";
import { default as default275, default as default276 } from "@opentiny/vue-list";
import LoadList from "@opentiny/vue-load-list";
import { default as default277, default as default278 } from "@opentiny/vue-load-list";
import Loading from "@opentiny/vue-loading";
import { default as default279, default as default280 } from "@opentiny/vue-loading";
import Locales from "@opentiny/vue-locales";
import { default as default281, default as default282 } from "@opentiny/vue-locales";
import LogonUser from "@opentiny/vue-logon-user";
import { default as default283, default as default284 } from "@opentiny/vue-logon-user";
import Logout from "@opentiny/vue-logout";
import { default as default285, default as default286 } from "@opentiny/vue-logout";
import Mask from "@opentiny/vue-mask";
import { default as default287, default as default288 } from "@opentiny/vue-mask";
import Menu from "@opentiny/vue-menu";
import { default as default289, default as default290 } from "@opentiny/vue-menu";
import Message from "@opentiny/vue-message";
import { default as default291, default as default292 } from "@opentiny/vue-message";
import Milestone from "@opentiny/vue-milestone";
import { default as default293, default as default294 } from "@opentiny/vue-milestone";
import MindMap from "@opentiny/vue-mind-map";
import { default as default295, default as default296 } from "@opentiny/vue-mind-map";
import MiniPicker from "@opentiny/vue-mini-picker";
import { default as default297, default as default298 } from "@opentiny/vue-mini-picker";
import Modal from "@opentiny/vue-modal";
import { default as default299, default as default300 } from "@opentiny/vue-modal";
import MonthRange from "@opentiny/vue-month-range";
import { default as default301, default as default302 } from "@opentiny/vue-month-range";
import MonthTable from "@opentiny/vue-month-table";
import { default as default303, default as default304 } from "@opentiny/vue-month-table";
import MultiSelect from "@opentiny/vue-multi-select";
import { default as default305, default as default306 } from "@opentiny/vue-multi-select";
import MultiSelectItem from "@opentiny/vue-multi-select-item";
import { default as default307, default as default308 } from "@opentiny/vue-multi-select-item";
import NavBar from "@opentiny/vue-nav-bar";
import { default as default309, default as default310 } from "@opentiny/vue-nav-bar";
import NavMenu from "@opentiny/vue-nav-menu";
import { default as default311, default as default312 } from "@opentiny/vue-nav-menu";
import Notify from "@opentiny/vue-notify";
import { default as default313, default as default314 } from "@opentiny/vue-notify";
import Numeric from "@opentiny/vue-numeric";
import { default as default315, default as default316 } from "@opentiny/vue-numeric";
import Option from "@opentiny/vue-option";
import { default as default317, default as default318 } from "@opentiny/vue-option";
import OptionGroup from "@opentiny/vue-option-group";
import { default as default319, default as default320 } from "@opentiny/vue-option-group";
import Pager from "@opentiny/vue-pager";
import { default as default321, default as default322 } from "@opentiny/vue-pager";
import PagerItem from "@opentiny/vue-pager-item";
import { default as default323, default as default324 } from "@opentiny/vue-pager-item";
import Panel from "@opentiny/vue-panel";
import { default as default325, default as default326 } from "@opentiny/vue-panel";
import Picker from "@opentiny/vue-picker";
import { default as default327, default as default328 } from "@opentiny/vue-picker";
import PickerColumn from "@opentiny/vue-picker-column";
import { default as default329, default as default330 } from "@opentiny/vue-picker-column";
import PopUpload from "@opentiny/vue-pop-upload";
import { default as default331, default as default332 } from "@opentiny/vue-pop-upload";
import Popconfirm from "@opentiny/vue-popconfirm";
import { default as default333, default as default334 } from "@opentiny/vue-popconfirm";
import Popeditor from "@opentiny/vue-popeditor";
import { default as default335, default as default336 } from "@opentiny/vue-popeditor";
import Popover from "@opentiny/vue-popover";
import { default as default337, default as default338 } from "@opentiny/vue-popover";
import Popup from "@opentiny/vue-popup";
import { default as default339, default as default340 } from "@opentiny/vue-popup";
import Progress from "@opentiny/vue-progress";
import { default as default341, default as default342 } from "@opentiny/vue-progress";
import PullRefresh from "@opentiny/vue-pull-refresh";
import { default as default343, default as default344 } from "@opentiny/vue-pull-refresh";
import QrCode from "@opentiny/vue-qr-code";
import { default as default345, default as default346 } from "@opentiny/vue-qr-code";
import QuarterPanel from "@opentiny/vue-quarter-panel";
import { default as default347, default as default348 } from "@opentiny/vue-quarter-panel";
import QueryBuilder from "@opentiny/vue-query-builder";
import { default as default349, default as default350 } from "@opentiny/vue-query-builder";
import Radio from "@opentiny/vue-radio";
import { default as default351, default as default352 } from "@opentiny/vue-radio";
import RadioButton from "@opentiny/vue-radio-button";
import { default as default353, default as default354 } from "@opentiny/vue-radio-button";
import RadioGroup from "@opentiny/vue-radio-group";
import { default as default355, default as default356 } from "@opentiny/vue-radio-group";
import Rate from "@opentiny/vue-rate";
import { default as default357, default as default358 } from "@opentiny/vue-rate";
import Record from "@opentiny/vue-record";
import { default as default359, default as default360 } from "@opentiny/vue-record";
import RecycleScroller from "@opentiny/vue-recycle-scroller";
import { default as default361, default as default362 } from "@opentiny/vue-recycle-scroller";
import RichTextEditor from "@opentiny/vue-rich-text-editor";
import { default as default363, default as default364 } from "@opentiny/vue-rich-text-editor";
import River from "@opentiny/vue-river";
import { default as default365, default as default366 } from "@opentiny/vue-river";
import Roles from "@opentiny/vue-roles";
import { default as default367, default as default368 } from "@opentiny/vue-roles";
import Row from "@opentiny/vue-row";
import { default as default369, default as default370 } from "@opentiny/vue-row";
import ScrollText from "@opentiny/vue-scroll-text";
import { default as default371, default as default372 } from "@opentiny/vue-scroll-text";
import Scrollbar from "@opentiny/vue-scrollbar";
import { default as default373, default as default374 } from "@opentiny/vue-scrollbar";
import Search from "@opentiny/vue-search";
import { default as default375, default as default376 } from "@opentiny/vue-search";
import Select from "@opentiny/vue-select";
import { default as default377, default as default378 } from "@opentiny/vue-select";
import SelectDropdown from "@opentiny/vue-select-dropdown";
import { default as default379, default as default380 } from "@opentiny/vue-select-dropdown";
import SelectMobile from "@opentiny/vue-select-mobile";
import { default as default381, default as default382 } from "@opentiny/vue-select-mobile";
import SelectView from "@opentiny/vue-select-view";
import { default as default383, default as default384 } from "@opentiny/vue-select-view";
import SelectedBox from "@opentiny/vue-selected-box";
import { default as default385, default as default386 } from "@opentiny/vue-selected-box";
import Signature from "@opentiny/vue-signature";
import { default as default387, default as default388 } from "@opentiny/vue-signature";
import Skeleton from "@opentiny/vue-skeleton";
import { default as default389, default as default390 } from "@opentiny/vue-skeleton";
import SkeletonItem from "@opentiny/vue-skeleton-item";
import { default as default391, default as default392 } from "@opentiny/vue-skeleton-item";
import Slider from "@opentiny/vue-slider";
import { default as default393, default as default394 } from "@opentiny/vue-slider";
import SliderButton from "@opentiny/vue-slider-button";
import { default as default395, default as default396 } from "@opentiny/vue-slider-button";
import SliderButtonGroup from "@opentiny/vue-slider-button-group";
import { default as default397, default as default398 } from "@opentiny/vue-slider-button-group";
import Split from "@opentiny/vue-split";
import { default as default399, default as default400 } from "@opentiny/vue-split";
import StandardListItem from "@opentiny/vue-standard-list-item";
import { default as default401, default as default402 } from "@opentiny/vue-standard-list-item";
import Statistic from "@opentiny/vue-statistic";
import { default as default403, default as default404 } from "@opentiny/vue-statistic";
import Steps from "@opentiny/vue-steps";
import { default as default405, default as default406 } from "@opentiny/vue-steps";
import Sticky from "@opentiny/vue-sticky";
import { default as default407, default as default408 } from "@opentiny/vue-sticky";
import Switch from "@opentiny/vue-switch";
import { default as default409, default as default410 } from "@opentiny/vue-switch";
import TabItem from "@opentiny/vue-tab-item";
import { default as default411, default as default412 } from "@opentiny/vue-tab-item";
import Tabbar from "@opentiny/vue-tabbar";
import { default as default413, default as default414 } from "@opentiny/vue-tabbar";
import TabbarItem from "@opentiny/vue-tabbar-item";
import { default as default415, default as default416 } from "@opentiny/vue-tabbar-item";
import Table from "@opentiny/vue-table";
import { default as default417, default as default418 } from "@opentiny/vue-table";
import Tabs from "@opentiny/vue-tabs";
import { default as default419, default as default420 } from "@opentiny/vue-tabs";
import Tag from "@opentiny/vue-tag";
import { default as default421, default as default422 } from "@opentiny/vue-tag";
import TagGroup from "@opentiny/vue-tag-group";
import { default as default423, default as default424 } from "@opentiny/vue-tag-group";
import TextPopup from "@opentiny/vue-text-popup";
import { default as default425, default as default426 } from "@opentiny/vue-text-popup";
import Time from "@opentiny/vue-time";
import { default as default427, default as default428 } from "@opentiny/vue-time";
import TimeLine from "@opentiny/vue-time-line";
import { default as default429, default as default430 } from "@opentiny/vue-time-line";
import TimeLineNew from "@opentiny/vue-time-line-new";
import { default as default431, default as default432 } from "@opentiny/vue-time-line-new";
import TimePanel from "@opentiny/vue-time-panel";
import { default as default433, default as default434 } from "@opentiny/vue-time-panel";
import TimePicker from "@opentiny/vue-time-picker";
import { default as default435, default as default436 } from "@opentiny/vue-time-picker";
import TimePickerMobile from "@opentiny/vue-time-picker-mobile";
import { default as default437, default as default438 } from "@opentiny/vue-time-picker-mobile";
import TimeRange from "@opentiny/vue-time-range";
import { default as default439, default as default440 } from "@opentiny/vue-time-range";
import TimeSelect from "@opentiny/vue-time-select";
import { default as default441, default as default442 } from "@opentiny/vue-time-select";
import TimeSpinner from "@opentiny/vue-time-spinner";
import { default as default443, default as default444 } from "@opentiny/vue-time-spinner";
import TimelineItem from "@opentiny/vue-timeline-item";
import { default as default445, default as default446 } from "@opentiny/vue-timeline-item";
import Toast from "@opentiny/vue-toast";
import { default as default447, default as default448 } from "@opentiny/vue-toast";
import ToggleMenu from "@opentiny/vue-toggle-menu";
import { default as default449, default as default450 } from "@opentiny/vue-toggle-menu";
import Tooltip from "@opentiny/vue-tooltip";
import { default as default451, default as default452 } from "@opentiny/vue-tooltip";
import TopBox from "@opentiny/vue-top-box";
import { default as default453, default as default454 } from "@opentiny/vue-top-box";
import Transfer from "@opentiny/vue-transfer";
import { default as default455, default as default456 } from "@opentiny/vue-transfer";
import TransferPanel from "@opentiny/vue-transfer-panel";
import { default as default457, default as default458 } from "@opentiny/vue-transfer-panel";
import Tree from "@opentiny/vue-tree";
import { default as default459, default as default460 } from "@opentiny/vue-tree";
import TreeMenu from "@opentiny/vue-tree-menu";
import { default as default461, default as default462 } from "@opentiny/vue-tree-menu";
import TreeSelect from "@opentiny/vue-tree-select";
import { default as default463, default as default464 } from "@opentiny/vue-tree-select";
import Upload from "@opentiny/vue-upload";
import { default as default465, default as default466 } from "@opentiny/vue-upload";
import UploadDragger from "@opentiny/vue-upload-dragger";
import { default as default467, default as default468 } from "@opentiny/vue-upload-dragger";
import UploadList from "@opentiny/vue-upload-list";
import { default as default469, default as default470 } from "@opentiny/vue-upload-list";
import User from "@opentiny/vue-user";
import { default as default471, default as default472 } from "@opentiny/vue-user";
import UserAccount from "@opentiny/vue-user-account";
import { default as default473, default as default474 } from "@opentiny/vue-user-account";
import UserContact from "@opentiny/vue-user-contact";
import { default as default475, default as default476 } from "@opentiny/vue-user-contact";
import UserHead from "@opentiny/vue-user-head";
import { default as default477, default as default478 } from "@opentiny/vue-user-head";
import UserHeadGroup from "@opentiny/vue-user-head-group";
import { default as default479, default as default480 } from "@opentiny/vue-user-head-group";
import UserLink from "@opentiny/vue-user-link";
import { default as default481, default as default482 } from "@opentiny/vue-user-link";
import VirtualScrollBox from "@opentiny/vue-virtual-scroll-box";
import { default as default483, default as default484 } from "@opentiny/vue-virtual-scroll-box";
import VirtualTree from "@opentiny/vue-virtual-tree";
import { default as default485, default as default486 } from "@opentiny/vue-virtual-tree";
import Watermark from "@opentiny/vue-watermark";
import { default as default487, default as default488 } from "@opentiny/vue-watermark";
import Wheel from "@opentiny/vue-wheel";
import { default as default489, default as default490 } from "@opentiny/vue-wheel";
import Wizard from "@opentiny/vue-wizard";
import { default as default491, default as default492 } from "@opentiny/vue-wizard";
import YearRange from "@opentiny/vue-year-range";
import { default as default493, default as default494 } from "@opentiny/vue-year-range";
import YearTable from "@opentiny/vue-year-table";
import { default as default495, default as default496 } from "@opentiny/vue-year-table";
import { $prefix } from "@opentiny/vue-common";
var components = [ActionMenu, ActionSheet, Alert, Amount, Anchor, Area, AsyncFlowchart, Autocomplete, Avatar, Badge, BaseSelect, Breadcrumb, BreadcrumbItem, BulletinBoard, Button, ButtonGroup, Calendar, CalendarBar, CalendarView, Card, CardGroup, CardTemplate, Carousel, CarouselItem, Cascader, CascaderMenu, CascaderMobile, CascaderNode, CascaderPanel, CascaderSelect, CascaderView, Cell, Checkbox, CheckboxButton, CheckboxGroup, Col, Collapse, CollapseItem, CollapseTransition, ColorPicker, ColorSelectPanel, ColumnListGroup, ColumnListItem, Company, ConfigProvider, Container, Country, Crop, Currency, DatePanel, DatePicker, DatePickerMobileFirst, DateRange, DateTable, Dept, DialogBox, DialogSelect, Divider, Drawer, DropRoles, DropTimes, Dropdown, DropdownItem, DropdownMenu, DynamicScroller, DynamicScrollerItem, Espace, Exception, FallMenu, FileUpload, Filter, FilterBar, FilterBox, FilterPanel, FloatButton, Floatbar, FloatingButton, Flowchart, FluentEditor, Form, FormItem, Fullscreen, Grid, GridColumn, GridManager, GridSelect, GridToolbar, Guide, Hrapprover, Huicharts, HuichartsAmap, HuichartsBar, HuichartsBmap, HuichartsBoxplot, HuichartsCandle, Core, HuichartsFunnel, HuichartsGauge, HuichartsGraph, HuichartsHeatmap, HuichartsHistogram, HuichartsLine, HuichartsLiquidfill, HuichartsMap, HuichartsPie, HuichartsProcess, HuichartsRadar, HuichartsRing, HuichartsSankey, HuichartsScatter, HuichartsSunburst, HuichartsTree, HuichartsWaterfall, HuichartsWordcloud, Image, ImageViewer, IndexBar, IndexBarAnchor, Input, IpAddress, Label, Layout, Link, LinkMenu, List, LoadList, Loading, Locales, LogonUser, Logout, Mask, Menu, Message, Milestone, MindMap, MiniPicker, Modal, MonthRange, MonthTable, MultiSelect, MultiSelectItem, NavBar, NavMenu, Notify, Numeric, Option, OptionGroup, Pager, PagerItem, Panel, Picker, PickerColumn, PopUpload, Popconfirm, Popeditor, Popover, Popup, Progress, PullRefresh, QrCode, QuarterPanel, QueryBuilder, Radio, RadioButton, RadioGroup, Rate, Record, RecycleScroller, RichTextEditor, River, Roles, Row, ScrollText, Scrollbar, Search, Select, SelectDropdown, SelectMobile, SelectView, SelectedBox, Signature, Skeleton, SkeletonItem, Slider, SliderButton, SliderButtonGroup, Split, StandardListItem, Statistic, Steps, Sticky, Switch, TabItem, Tabbar, TabbarItem, Table, Tabs, Tag, TagGroup, TextPopup, Time, TimeLine, TimeLineNew, TimePanel, TimePicker, TimePickerMobile, TimeRange, TimeSelect, TimeSpinner, TimelineItem, Toast, ToggleMenu, Tooltip, TopBox, Transfer, TransferPanel, Tree, TreeMenu, TreeSelect, Upload, UploadDragger, UploadList, User, UserAccount, UserContact, UserHead, UserHeadGroup, UserLink, VirtualScrollBox, VirtualTree, Watermark, Wheel, Wizard, YearRange, YearTable];
var install = function install2(app, opts) {
  if (opts === void 0) {
    opts = {};
  }
  var regex = new RegExp("^" + $prefix);
  components.forEach(function(component) {
    var name = component.name;
    var alias = opts.alias || opts.prefix;
    if (typeof component.install !== "function")
      return;
    if (name && alias) {
      app.component(name.replace(regex, alias), component);
    } else {
      component.install(app);
    }
  });
};
var version = "3.20.0";
var index = {
  ActionMenu,
  TinyActionMenu: ActionMenu,
  ActionSheet,
  TinyActionSheet: ActionSheet,
  Alert,
  TinyAlert: Alert,
  Amount,
  TinyAmount: Amount,
  Anchor,
  TinyAnchor: Anchor,
  Area,
  TinyArea: Area,
  AsyncFlowchart,
  TinyAsyncFlowchart: AsyncFlowchart,
  Autocomplete,
  TinyAutocomplete: Autocomplete,
  Avatar,
  TinyAvatar: Avatar,
  Badge,
  TinyBadge: Badge,
  BaseSelect,
  TinyBaseSelect: BaseSelect,
  Breadcrumb,
  TinyBreadcrumb: Breadcrumb,
  BreadcrumbItem,
  TinyBreadcrumbItem: BreadcrumbItem,
  BulletinBoard,
  TinyBulletinBoard: BulletinBoard,
  Button,
  TinyButton: Button,
  ButtonGroup,
  TinyButtonGroup: ButtonGroup,
  Calendar,
  TinyCalendar: Calendar,
  CalendarBar,
  TinyCalendarBar: CalendarBar,
  CalendarView,
  TinyCalendarView: CalendarView,
  Card,
  TinyCard: Card,
  CardGroup,
  TinyCardGroup: CardGroup,
  CardTemplate,
  TinyCardTemplate: CardTemplate,
  Carousel,
  TinyCarousel: Carousel,
  CarouselItem,
  TinyCarouselItem: CarouselItem,
  Cascader,
  TinyCascader: Cascader,
  CascaderMenu,
  TinyCascaderMenu: CascaderMenu,
  CascaderMobile,
  TinyCascaderMobile: CascaderMobile,
  CascaderNode,
  TinyCascaderNode: CascaderNode,
  CascaderPanel,
  TinyCascaderPanel: CascaderPanel,
  CascaderSelect,
  TinyCascaderSelect: CascaderSelect,
  CascaderView,
  TinyCascaderView: CascaderView,
  Cell,
  TinyCell: Cell,
  Checkbox,
  TinyCheckbox: Checkbox,
  CheckboxButton,
  TinyCheckboxButton: CheckboxButton,
  CheckboxGroup,
  TinyCheckboxGroup: CheckboxGroup,
  Col,
  TinyCol: Col,
  Collapse,
  TinyCollapse: Collapse,
  CollapseItem,
  TinyCollapseItem: CollapseItem,
  CollapseTransition,
  TinyCollapseTransition: CollapseTransition,
  ColorPicker,
  TinyColorPicker: ColorPicker,
  ColorSelectPanel,
  TinyColorSelectPanel: ColorSelectPanel,
  ColumnListGroup,
  TinyColumnListGroup: ColumnListGroup,
  ColumnListItem,
  TinyColumnListItem: ColumnListItem,
  Company,
  TinyCompany: Company,
  ConfigProvider,
  TinyConfigProvider: ConfigProvider,
  Container,
  TinyContainer: Container,
  Country,
  TinyCountry: Country,
  Crop,
  TinyCrop: Crop,
  Currency,
  TinyCurrency: Currency,
  DatePanel,
  TinyDatePanel: DatePanel,
  DatePicker,
  TinyDatePicker: DatePicker,
  DatePickerMobileFirst,
  TinyDatePickerMobileFirst: DatePickerMobileFirst,
  DateRange,
  TinyDateRange: DateRange,
  DateTable,
  TinyDateTable: DateTable,
  Dept,
  TinyDept: Dept,
  DialogBox,
  TinyDialogBox: DialogBox,
  DialogSelect,
  TinyDialogSelect: DialogSelect,
  Divider,
  TinyDivider: Divider,
  Drawer,
  TinyDrawer: Drawer,
  DropRoles,
  TinyDropRoles: DropRoles,
  DropTimes,
  TinyDropTimes: DropTimes,
  Dropdown,
  TinyDropdown: Dropdown,
  DropdownItem,
  TinyDropdownItem: DropdownItem,
  DropdownMenu,
  TinyDropdownMenu: DropdownMenu,
  DynamicScroller,
  TinyDynamicScroller: DynamicScroller,
  DynamicScrollerItem,
  TinyDynamicScrollerItem: DynamicScrollerItem,
  Espace,
  TinyEspace: Espace,
  Exception,
  TinyException: Exception,
  FallMenu,
  TinyFallMenu: FallMenu,
  FileUpload,
  TinyFileUpload: FileUpload,
  Filter,
  TinyFilter: Filter,
  FilterBar,
  TinyFilterBar: FilterBar,
  FilterBox,
  TinyFilterBox: FilterBox,
  FilterPanel,
  TinyFilterPanel: FilterPanel,
  FloatButton,
  TinyFloatButton: FloatButton,
  Floatbar,
  TinyFloatbar: Floatbar,
  FloatingButton,
  TinyFloatingButton: FloatingButton,
  Flowchart,
  TinyFlowchart: Flowchart,
  FluentEditor,
  TinyFluentEditor: FluentEditor,
  Form,
  TinyForm: Form,
  FormItem,
  TinyFormItem: FormItem,
  Fullscreen,
  TinyFullscreen: Fullscreen,
  Grid,
  TinyGrid: Grid,
  GridColumn,
  TinyGridColumn: GridColumn,
  GridManager,
  TinyGridManager: GridManager,
  GridSelect,
  TinyGridSelect: GridSelect,
  GridToolbar,
  TinyGridToolbar: GridToolbar,
  Guide,
  TinyGuide: Guide,
  Hrapprover,
  TinyHrapprover: Hrapprover,
  Huicharts,
  Charts: Huicharts,
  TinyHuicharts: Huicharts,
  HuichartsAmap,
  ChartsAmap: HuichartsAmap,
  TinyHuichartsAmap: HuichartsAmap,
  HuichartsBar,
  ChartsBar: HuichartsBar,
  TinyHuichartsBar: HuichartsBar,
  HuichartsBmap,
  ChartsBmap: HuichartsBmap,
  TinyHuichartsBmap: HuichartsBmap,
  HuichartsBoxplot,
  ChartsBoxplot: HuichartsBoxplot,
  TinyHuichartsBoxplot: HuichartsBoxplot,
  HuichartsCandle,
  ChartsCandle: HuichartsCandle,
  TinyHuichartsCandle: HuichartsCandle,
  HuichartsCore: Core,
  ChartsCore: Core,
  TinyHuichartsCore: Core,
  HuichartsFunnel,
  ChartsFunnel: HuichartsFunnel,
  TinyHuichartsFunnel: HuichartsFunnel,
  HuichartsGauge,
  ChartsGauge: HuichartsGauge,
  TinyHuichartsGauge: HuichartsGauge,
  HuichartsGraph,
  ChartsGraph: HuichartsGraph,
  TinyHuichartsGraph: HuichartsGraph,
  HuichartsHeatmap,
  ChartsHeatmap: HuichartsHeatmap,
  TinyHuichartsHeatmap: HuichartsHeatmap,
  HuichartsHistogram,
  ChartsHistogram: HuichartsHistogram,
  TinyHuichartsHistogram: HuichartsHistogram,
  HuichartsLine,
  ChartsLine: HuichartsLine,
  TinyHuichartsLine: HuichartsLine,
  HuichartsLiquidfill,
  ChartsLiquidfill: HuichartsLiquidfill,
  TinyHuichartsLiquidfill: HuichartsLiquidfill,
  HuichartsMap,
  ChartsMap: HuichartsMap,
  TinyHuichartsMap: HuichartsMap,
  HuichartsPie,
  ChartsPie: HuichartsPie,
  TinyHuichartsPie: HuichartsPie,
  HuichartsProcess,
  ChartsProcess: HuichartsProcess,
  TinyHuichartsProcess: HuichartsProcess,
  HuichartsRadar,
  ChartsRadar: HuichartsRadar,
  TinyHuichartsRadar: HuichartsRadar,
  HuichartsRing,
  ChartsRing: HuichartsRing,
  TinyHuichartsRing: HuichartsRing,
  HuichartsSankey,
  ChartsSankey: HuichartsSankey,
  TinyHuichartsSankey: HuichartsSankey,
  HuichartsScatter,
  ChartsScatter: HuichartsScatter,
  TinyHuichartsScatter: HuichartsScatter,
  HuichartsSunburst,
  ChartsSunburst: HuichartsSunburst,
  TinyHuichartsSunburst: HuichartsSunburst,
  HuichartsTree,
  ChartsTree: HuichartsTree,
  TinyHuichartsTree: HuichartsTree,
  HuichartsWaterfall,
  ChartsWaterfall: HuichartsWaterfall,
  TinyHuichartsWaterfall: HuichartsWaterfall,
  HuichartsWordcloud,
  ChartsWordcloud: HuichartsWordcloud,
  TinyHuichartsWordcloud: HuichartsWordcloud,
  Image,
  TinyImage: Image,
  ImageViewer,
  TinyImageViewer: ImageViewer,
  IndexBar,
  TinyIndexBar: IndexBar,
  IndexBarAnchor,
  TinyIndexBarAnchor: IndexBarAnchor,
  Input,
  TinyInput: Input,
  IpAddress,
  TinyIpAddress: IpAddress,
  Label,
  TinyLabel: Label,
  Layout,
  TinyLayout: Layout,
  Link,
  TinyLink: Link,
  LinkMenu,
  TinyLinkMenu: LinkMenu,
  List,
  TinyList: List,
  LoadList,
  TinyLoadList: LoadList,
  Loading,
  TinyLoading: Loading,
  Locales,
  TinyLocales: Locales,
  LogonUser,
  TinyLogonUser: LogonUser,
  Logout,
  TinyLogout: Logout,
  Mask,
  TinyMask: Mask,
  Menu,
  TinyMenu: Menu,
  Message,
  TinyMessage: Message,
  Milestone,
  TinyMilestone: Milestone,
  MindMap,
  TinyMindMap: MindMap,
  MiniPicker,
  TinyMiniPicker: MiniPicker,
  Modal,
  TinyModal: Modal,
  MonthRange,
  TinyMonthRange: MonthRange,
  MonthTable,
  TinyMonthTable: MonthTable,
  MultiSelect,
  TinyMultiSelect: MultiSelect,
  MultiSelectItem,
  TinyMultiSelectItem: MultiSelectItem,
  NavBar,
  TinyNavBar: NavBar,
  NavMenu,
  TinyNavMenu: NavMenu,
  Notify,
  TinyNotify: Notify,
  Numeric,
  TinyNumeric: Numeric,
  Option,
  TinyOption: Option,
  OptionGroup,
  TinyOptionGroup: OptionGroup,
  Pager,
  TinyPager: Pager,
  PagerItem,
  TinyPagerItem: PagerItem,
  Panel,
  TinyPanel: Panel,
  Picker,
  TinyPicker: Picker,
  PickerColumn,
  TinyPickerColumn: PickerColumn,
  PopUpload,
  TinyPopUpload: PopUpload,
  Popconfirm,
  TinyPopconfirm: Popconfirm,
  Popeditor,
  TinyPopeditor: Popeditor,
  Popover,
  TinyPopover: Popover,
  Popup,
  TinyPopup: Popup,
  Progress,
  TinyProgress: Progress,
  PullRefresh,
  TinyPullRefresh: PullRefresh,
  QrCode,
  TinyQrCode: QrCode,
  QuarterPanel,
  TinyQuarterPanel: QuarterPanel,
  QueryBuilder,
  TinyQueryBuilder: QueryBuilder,
  Radio,
  TinyRadio: Radio,
  RadioButton,
  TinyRadioButton: RadioButton,
  RadioGroup,
  TinyRadioGroup: RadioGroup,
  Rate,
  TinyRate: Rate,
  Record,
  TinyRecord: Record,
  RecycleScroller,
  TinyRecycleScroller: RecycleScroller,
  RichTextEditor,
  TinyRichTextEditor: RichTextEditor,
  River,
  TinyRiver: River,
  Roles,
  TinyRoles: Roles,
  Row,
  TinyRow: Row,
  ScrollText,
  TinyScrollText: ScrollText,
  Scrollbar,
  TinyScrollbar: Scrollbar,
  Search,
  TinySearch: Search,
  Select,
  TinySelect: Select,
  SelectDropdown,
  TinySelectDropdown: SelectDropdown,
  SelectMobile,
  TinySelectMobile: SelectMobile,
  SelectView,
  TinySelectView: SelectView,
  SelectedBox,
  TinySelectedBox: SelectedBox,
  Signature,
  TinySignature: Signature,
  Skeleton,
  TinySkeleton: Skeleton,
  SkeletonItem,
  TinySkeletonItem: SkeletonItem,
  Slider,
  TinySlider: Slider,
  SliderButton,
  TinySliderButton: SliderButton,
  SliderButtonGroup,
  TinySliderButtonGroup: SliderButtonGroup,
  Split,
  TinySplit: Split,
  StandardListItem,
  TinyStandardListItem: StandardListItem,
  Statistic,
  TinyStatistic: Statistic,
  Steps,
  TinySteps: Steps,
  Sticky,
  TinySticky: Sticky,
  Switch,
  TinySwitch: Switch,
  TabItem,
  TinyTabItem: TabItem,
  Tabbar,
  TinyTabbar: Tabbar,
  TabbarItem,
  TinyTabbarItem: TabbarItem,
  Table,
  TinyTable: Table,
  Tabs,
  TinyTabs: Tabs,
  Tag,
  TinyTag: Tag,
  TagGroup,
  TinyTagGroup: TagGroup,
  TextPopup,
  TinyTextPopup: TextPopup,
  Time,
  TinyTime: Time,
  TimeLine,
  TinyTimeLine: TimeLine,
  TimeLineNew,
  TinyTimeLineNew: TimeLineNew,
  TimePanel,
  TinyTimePanel: TimePanel,
  TimePicker,
  TinyTimePicker: TimePicker,
  TimePickerMobile,
  TinyTimePickerMobile: TimePickerMobile,
  TimeRange,
  TinyTimeRange: TimeRange,
  TimeSelect,
  TinyTimeSelect: TimeSelect,
  TimeSpinner,
  TinyTimeSpinner: TimeSpinner,
  TimelineItem,
  TinyTimelineItem: TimelineItem,
  Toast,
  TinyToast: Toast,
  ToggleMenu,
  TinyToggleMenu: ToggleMenu,
  Tooltip,
  TinyTooltip: Tooltip,
  TopBox,
  TinyTopBox: TopBox,
  Transfer,
  TinyTransfer: Transfer,
  TransferPanel,
  TinyTransferPanel: TransferPanel,
  Tree,
  TinyTree: Tree,
  TreeMenu,
  TinyTreeMenu: TreeMenu,
  TreeSelect,
  TinyTreeSelect: TreeSelect,
  Upload,
  TinyUpload: Upload,
  UploadDragger,
  TinyUploadDragger: UploadDragger,
  UploadList,
  TinyUploadList: UploadList,
  User,
  TinyUser: User,
  UserAccount,
  TinyUserAccount: UserAccount,
  UserContact,
  TinyUserContact: UserContact,
  UserHead,
  TinyUserHead: UserHead,
  UserHeadGroup,
  TinyUserHeadGroup: UserHeadGroup,
  UserLink,
  TinyUserLink: UserLink,
  VirtualScrollBox,
  TinyVirtualScrollBox: VirtualScrollBox,
  VirtualTree,
  TinyVirtualTree: VirtualTree,
  Watermark,
  TinyWatermark: Watermark,
  Wheel,
  TinyWheel: Wheel,
  Wizard,
  TinyWizard: Wizard,
  YearRange,
  TinyYearRange: YearRange,
  YearTable,
  TinyYearTable: YearTable,
  install
};
export {
  default2 as ActionMenu,
  default4 as ActionSheet,
  default6 as Alert,
  default8 as Amount,
  default10 as Anchor,
  default12 as Area,
  default14 as AsyncFlowchart,
  default16 as Autocomplete,
  default18 as Avatar,
  default20 as Badge,
  default22 as BaseSelect,
  default24 as Breadcrumb,
  default26 as BreadcrumbItem,
  default28 as BulletinBoard,
  default30 as Button,
  default32 as ButtonGroup,
  default34 as Calendar,
  default36 as CalendarBar,
  default38 as CalendarView,
  default40 as Card,
  default42 as CardGroup,
  default44 as CardTemplate,
  default46 as Carousel,
  default48 as CarouselItem,
  default50 as Cascader,
  default52 as CascaderMenu,
  default54 as CascaderMobile,
  default56 as CascaderNode,
  default58 as CascaderPanel,
  default60 as CascaderSelect,
  default62 as CascaderView,
  default64 as Cell,
  default180 as Charts,
  default183 as ChartsAmap,
  default186 as ChartsBar,
  default189 as ChartsBmap,
  default192 as ChartsBoxplot,
  default195 as ChartsCandle,
  default198 as ChartsCore,
  default201 as ChartsFunnel,
  default204 as ChartsGauge,
  default207 as ChartsGraph,
  default210 as ChartsHeatmap,
  default213 as ChartsHistogram,
  default216 as ChartsLine,
  default219 as ChartsLiquidfill,
  default222 as ChartsMap,
  default225 as ChartsPie,
  default228 as ChartsProcess,
  default231 as ChartsRadar,
  default234 as ChartsRing,
  default237 as ChartsSankey,
  default240 as ChartsScatter,
  default243 as ChartsSunburst,
  default246 as ChartsTree,
  default249 as ChartsWaterfall,
  default252 as ChartsWordcloud,
  default66 as Checkbox,
  default68 as CheckboxButton,
  default70 as CheckboxGroup,
  default72 as Col,
  default74 as Collapse,
  default76 as CollapseItem,
  default78 as CollapseTransition,
  default80 as ColorPicker,
  default82 as ColorSelectPanel,
  default84 as ColumnListGroup,
  default86 as ColumnListItem,
  default88 as Company,
  default90 as ConfigProvider,
  default92 as Container,
  default94 as Country,
  default96 as Crop,
  default98 as Currency,
  default100 as DatePanel,
  default102 as DatePicker,
  default104 as DatePickerMobileFirst,
  default106 as DateRange,
  default108 as DateTable,
  default110 as Dept,
  default112 as DialogBox,
  default114 as DialogSelect,
  default116 as Divider,
  default118 as Drawer,
  default120 as DropRoles,
  default122 as DropTimes,
  default124 as Dropdown,
  default126 as DropdownItem,
  default128 as DropdownMenu,
  default130 as DynamicScroller,
  default132 as DynamicScrollerItem,
  default134 as Espace,
  default136 as Exception,
  default138 as FallMenu,
  default140 as FileUpload,
  default142 as Filter,
  default144 as FilterBar,
  default146 as FilterBox,
  default148 as FilterPanel,
  default150 as FloatButton,
  default152 as Floatbar,
  default154 as FloatingButton,
  default156 as Flowchart,
  default158 as FluentEditor,
  default160 as Form,
  default162 as FormItem,
  default164 as Fullscreen,
  default166 as Grid,
  default168 as GridColumn,
  default170 as GridManager,
  default172 as GridSelect,
  default174 as GridToolbar,
  default176 as Guide,
  default178 as Hrapprover,
  default181 as Huicharts,
  default184 as HuichartsAmap,
  default187 as HuichartsBar,
  default190 as HuichartsBmap,
  default193 as HuichartsBoxplot,
  default196 as HuichartsCandle,
  default199 as HuichartsCore,
  default202 as HuichartsFunnel,
  default205 as HuichartsGauge,
  default208 as HuichartsGraph,
  default211 as HuichartsHeatmap,
  default214 as HuichartsHistogram,
  default217 as HuichartsLine,
  default220 as HuichartsLiquidfill,
  default223 as HuichartsMap,
  default226 as HuichartsPie,
  default229 as HuichartsProcess,
  default232 as HuichartsRadar,
  default235 as HuichartsRing,
  default238 as HuichartsSankey,
  default241 as HuichartsScatter,
  default244 as HuichartsSunburst,
  default247 as HuichartsTree,
  default250 as HuichartsWaterfall,
  default253 as HuichartsWordcloud,
  default255 as Image,
  default257 as ImageViewer,
  default259 as IndexBar,
  default261 as IndexBarAnchor,
  default263 as Input,
  default265 as IpAddress,
  default267 as Label,
  default269 as Layout,
  default271 as Link,
  default273 as LinkMenu,
  default275 as List,
  default277 as LoadList,
  default279 as Loading,
  default281 as Locales,
  default283 as LogonUser,
  default285 as Logout,
  default287 as Mask,
  default289 as Menu,
  default291 as Message,
  default293 as Milestone,
  default295 as MindMap,
  default297 as MiniPicker,
  default299 as Modal,
  default301 as MonthRange,
  default303 as MonthTable,
  default305 as MultiSelect,
  default307 as MultiSelectItem,
  default309 as NavBar,
  default311 as NavMenu,
  default313 as Notify,
  default315 as Numeric,
  default317 as Option,
  default319 as OptionGroup,
  default321 as Pager,
  default323 as PagerItem,
  default325 as Panel,
  default327 as Picker,
  default329 as PickerColumn,
  default331 as PopUpload,
  default333 as Popconfirm,
  default335 as Popeditor,
  default337 as Popover,
  default339 as Popup,
  default341 as Progress,
  default343 as PullRefresh,
  default345 as QrCode,
  default347 as QuarterPanel,
  default349 as QueryBuilder,
  default351 as Radio,
  default353 as RadioButton,
  default355 as RadioGroup,
  default357 as Rate,
  default359 as Record,
  default361 as RecycleScroller,
  default363 as RichTextEditor,
  default365 as River,
  default367 as Roles,
  default369 as Row,
  default371 as ScrollText,
  default373 as Scrollbar,
  default375 as Search,
  default377 as Select,
  default379 as SelectDropdown,
  default381 as SelectMobile,
  default383 as SelectView,
  default385 as SelectedBox,
  default387 as Signature,
  default389 as Skeleton,
  default391 as SkeletonItem,
  default393 as Slider,
  default395 as SliderButton,
  default397 as SliderButtonGroup,
  default399 as Split,
  default401 as StandardListItem,
  default403 as Statistic,
  default405 as Steps,
  default407 as Sticky,
  default409 as Switch,
  default411 as TabItem,
  default413 as Tabbar,
  default415 as TabbarItem,
  default417 as Table,
  default419 as Tabs,
  default421 as Tag,
  default423 as TagGroup,
  default425 as TextPopup,
  default427 as Time,
  default429 as TimeLine,
  default431 as TimeLineNew,
  default433 as TimePanel,
  default435 as TimePicker,
  default437 as TimePickerMobile,
  default439 as TimeRange,
  default441 as TimeSelect,
  default443 as TimeSpinner,
  default445 as TimelineItem,
  default3 as TinyActionMenu,
  default5 as TinyActionSheet,
  default7 as TinyAlert,
  default9 as TinyAmount,
  default11 as TinyAnchor,
  default13 as TinyArea,
  default15 as TinyAsyncFlowchart,
  default17 as TinyAutocomplete,
  default19 as TinyAvatar,
  default21 as TinyBadge,
  default23 as TinyBaseSelect,
  default25 as TinyBreadcrumb,
  default27 as TinyBreadcrumbItem,
  default29 as TinyBulletinBoard,
  default31 as TinyButton,
  default33 as TinyButtonGroup,
  default35 as TinyCalendar,
  default37 as TinyCalendarBar,
  default39 as TinyCalendarView,
  default41 as TinyCard,
  default43 as TinyCardGroup,
  default45 as TinyCardTemplate,
  default47 as TinyCarousel,
  default49 as TinyCarouselItem,
  default51 as TinyCascader,
  default53 as TinyCascaderMenu,
  default55 as TinyCascaderMobile,
  default57 as TinyCascaderNode,
  default59 as TinyCascaderPanel,
  default61 as TinyCascaderSelect,
  default63 as TinyCascaderView,
  default65 as TinyCell,
  default67 as TinyCheckbox,
  default69 as TinyCheckboxButton,
  default71 as TinyCheckboxGroup,
  default73 as TinyCol,
  default75 as TinyCollapse,
  default77 as TinyCollapseItem,
  default79 as TinyCollapseTransition,
  default81 as TinyColorPicker,
  default83 as TinyColorSelectPanel,
  default85 as TinyColumnListGroup,
  default87 as TinyColumnListItem,
  default89 as TinyCompany,
  default91 as TinyConfigProvider,
  default93 as TinyContainer,
  default95 as TinyCountry,
  default97 as TinyCrop,
  default99 as TinyCurrency,
  default101 as TinyDatePanel,
  default103 as TinyDatePicker,
  default105 as TinyDatePickerMobileFirst,
  default107 as TinyDateRange,
  default109 as TinyDateTable,
  default111 as TinyDept,
  default113 as TinyDialogBox,
  default115 as TinyDialogSelect,
  default117 as TinyDivider,
  default119 as TinyDrawer,
  default121 as TinyDropRoles,
  default123 as TinyDropTimes,
  default125 as TinyDropdown,
  default127 as TinyDropdownItem,
  default129 as TinyDropdownMenu,
  default131 as TinyDynamicScroller,
  default133 as TinyDynamicScrollerItem,
  default135 as TinyEspace,
  default137 as TinyException,
  default139 as TinyFallMenu,
  default141 as TinyFileUpload,
  default143 as TinyFilter,
  default145 as TinyFilterBar,
  default147 as TinyFilterBox,
  default149 as TinyFilterPanel,
  default151 as TinyFloatButton,
  default153 as TinyFloatbar,
  default155 as TinyFloatingButton,
  default157 as TinyFlowchart,
  default159 as TinyFluentEditor,
  default161 as TinyForm,
  default163 as TinyFormItem,
  default165 as TinyFullscreen,
  default167 as TinyGrid,
  default169 as TinyGridColumn,
  default171 as TinyGridManager,
  default173 as TinyGridSelect,
  default175 as TinyGridToolbar,
  default177 as TinyGuide,
  default179 as TinyHrapprover,
  default182 as TinyHuicharts,
  default185 as TinyHuichartsAmap,
  default188 as TinyHuichartsBar,
  default191 as TinyHuichartsBmap,
  default194 as TinyHuichartsBoxplot,
  default197 as TinyHuichartsCandle,
  default200 as TinyHuichartsCore,
  default203 as TinyHuichartsFunnel,
  default206 as TinyHuichartsGauge,
  default209 as TinyHuichartsGraph,
  default212 as TinyHuichartsHeatmap,
  default215 as TinyHuichartsHistogram,
  default218 as TinyHuichartsLine,
  default221 as TinyHuichartsLiquidfill,
  default224 as TinyHuichartsMap,
  default227 as TinyHuichartsPie,
  default230 as TinyHuichartsProcess,
  default233 as TinyHuichartsRadar,
  default236 as TinyHuichartsRing,
  default239 as TinyHuichartsSankey,
  default242 as TinyHuichartsScatter,
  default245 as TinyHuichartsSunburst,
  default248 as TinyHuichartsTree,
  default251 as TinyHuichartsWaterfall,
  default254 as TinyHuichartsWordcloud,
  default256 as TinyImage,
  default258 as TinyImageViewer,
  default260 as TinyIndexBar,
  default262 as TinyIndexBarAnchor,
  default264 as TinyInput,
  default266 as TinyIpAddress,
  default268 as TinyLabel,
  default270 as TinyLayout,
  default272 as TinyLink,
  default274 as TinyLinkMenu,
  default276 as TinyList,
  default278 as TinyLoadList,
  default280 as TinyLoading,
  default282 as TinyLocales,
  default284 as TinyLogonUser,
  default286 as TinyLogout,
  default288 as TinyMask,
  default290 as TinyMenu,
  default292 as TinyMessage,
  default294 as TinyMilestone,
  default296 as TinyMindMap,
  default298 as TinyMiniPicker,
  default300 as TinyModal,
  default302 as TinyMonthRange,
  default304 as TinyMonthTable,
  default306 as TinyMultiSelect,
  default308 as TinyMultiSelectItem,
  default310 as TinyNavBar,
  default312 as TinyNavMenu,
  default314 as TinyNotify,
  default316 as TinyNumeric,
  default318 as TinyOption,
  default320 as TinyOptionGroup,
  default322 as TinyPager,
  default324 as TinyPagerItem,
  default326 as TinyPanel,
  default328 as TinyPicker,
  default330 as TinyPickerColumn,
  default332 as TinyPopUpload,
  default334 as TinyPopconfirm,
  default336 as TinyPopeditor,
  default338 as TinyPopover,
  default340 as TinyPopup,
  default342 as TinyProgress,
  default344 as TinyPullRefresh,
  default346 as TinyQrCode,
  default348 as TinyQuarterPanel,
  default350 as TinyQueryBuilder,
  default352 as TinyRadio,
  default354 as TinyRadioButton,
  default356 as TinyRadioGroup,
  default358 as TinyRate,
  default360 as TinyRecord,
  default362 as TinyRecycleScroller,
  default364 as TinyRichTextEditor,
  default366 as TinyRiver,
  default368 as TinyRoles,
  default370 as TinyRow,
  default372 as TinyScrollText,
  default374 as TinyScrollbar,
  default376 as TinySearch,
  default378 as TinySelect,
  default380 as TinySelectDropdown,
  default382 as TinySelectMobile,
  default384 as TinySelectView,
  default386 as TinySelectedBox,
  default388 as TinySignature,
  default390 as TinySkeleton,
  default392 as TinySkeletonItem,
  default394 as TinySlider,
  default396 as TinySliderButton,
  default398 as TinySliderButtonGroup,
  default400 as TinySplit,
  default402 as TinyStandardListItem,
  default404 as TinyStatistic,
  default406 as TinySteps,
  default408 as TinySticky,
  default410 as TinySwitch,
  default412 as TinyTabItem,
  default414 as TinyTabbar,
  default416 as TinyTabbarItem,
  default418 as TinyTable,
  default420 as TinyTabs,
  default422 as TinyTag,
  default424 as TinyTagGroup,
  default426 as TinyTextPopup,
  default428 as TinyTime,
  default430 as TinyTimeLine,
  default432 as TinyTimeLineNew,
  default434 as TinyTimePanel,
  default436 as TinyTimePicker,
  default438 as TinyTimePickerMobile,
  default440 as TinyTimeRange,
  default442 as TinyTimeSelect,
  default444 as TinyTimeSpinner,
  default446 as TinyTimelineItem,
  default447 as TinyToast,
  default449 as TinyToggleMenu,
  default451 as TinyTooltip,
  default453 as TinyTopBox,
  default455 as TinyTransfer,
  default457 as TinyTransferPanel,
  default459 as TinyTree,
  default461 as TinyTreeMenu,
  default463 as TinyTreeSelect,
  default465 as TinyUpload,
  default467 as TinyUploadDragger,
  default469 as TinyUploadList,
  default471 as TinyUser,
  default473 as TinyUserAccount,
  default475 as TinyUserContact,
  default477 as TinyUserHead,
  default479 as TinyUserHeadGroup,
  default481 as TinyUserLink,
  default483 as TinyVirtualScrollBox,
  default485 as TinyVirtualTree,
  default487 as TinyWatermark,
  default489 as TinyWheel,
  default491 as TinyWizard,
  default493 as TinyYearRange,
  default495 as TinyYearTable,
  default448 as Toast,
  default450 as ToggleMenu,
  default452 as Tooltip,
  default454 as TopBox,
  default456 as Transfer,
  default458 as TransferPanel,
  default460 as Tree,
  default462 as TreeMenu,
  default464 as TreeSelect,
  default466 as Upload,
  default468 as UploadDragger,
  default470 as UploadList,
  default472 as User,
  default474 as UserAccount,
  default476 as UserContact,
  default478 as UserHead,
  default480 as UserHeadGroup,
  default482 as UserLink,
  default484 as VirtualScrollBox,
  default486 as VirtualTree,
  default488 as Watermark,
  default490 as Wheel,
  default492 as Wizard,
  default494 as YearRange,
  default496 as YearTable,
  index as default,
  install,
  version
};
