// @opentiny/vue@3.20.0/mobile.js
import ActionSheet from "@opentiny/vue-action-sheet";
import { default as default2, default as default3 } from "@opentiny/vue-action-sheet";
import Alert from "@opentiny/vue-alert";
import { default as default4, default as default5 } from "@opentiny/vue-alert";
import Avatar from "@opentiny/vue-avatar";
import { default as default6, default as default7 } from "@opentiny/vue-avatar";
import Badge from "@opentiny/vue-badge";
import { default as default8, default as default9 } from "@opentiny/vue-badge";
import Button from "@opentiny/vue-button";
import { default as default10, default as default11 } from "@opentiny/vue-button";
import Checkbox from "@opentiny/vue-checkbox";
import { default as default12, default as default13 } from "@opentiny/vue-checkbox";
import CheckboxGroup from "@opentiny/vue-checkbox-group";
import { default as default14, default as default15 } from "@opentiny/vue-checkbox-group";
import ColorPicker from "@opentiny/vue-color-picker";
import { default as default16, default as default17 } from "@opentiny/vue-color-picker";
import Container from "@opentiny/vue-container";
import { default as default18, default as default19 } from "@opentiny/vue-container";
import DatePicker from "@opentiny/vue-date-picker";
import { default as default20, default as default21 } from "@opentiny/vue-date-picker";
import DialogBox from "@opentiny/vue-dialog-box";
import { default as default22, default as default23 } from "@opentiny/vue-dialog-box";
import DropdownItem from "@opentiny/vue-dropdown-item";
import { default as default24, default as default25 } from "@opentiny/vue-dropdown-item";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import { default as default26, default as default27 } from "@opentiny/vue-dropdown-menu";
import Exception from "@opentiny/vue-exception";
import { default as default28, default as default29 } from "@opentiny/vue-exception";
import FileUpload from "@opentiny/vue-file-upload";
import { default as default30, default as default31 } from "@opentiny/vue-file-upload";
import Form from "@opentiny/vue-form";
import { default as default32, default as default33 } from "@opentiny/vue-form";
import ImageViewer from "@opentiny/vue-image-viewer";
import { default as default34, default as default35 } from "@opentiny/vue-image-viewer";
import IndexBar from "@opentiny/vue-index-bar";
import { default as default36, default as default37 } from "@opentiny/vue-index-bar";
import Input from "@opentiny/vue-input";
import { default as default38, default as default39 } from "@opentiny/vue-input";
import Label from "@opentiny/vue-label";
import { default as default40, default as default41 } from "@opentiny/vue-label";
import List from "@opentiny/vue-list";
import { default as default42, default as default43 } from "@opentiny/vue-list";
import Loading from "@opentiny/vue-loading";
import { default as default44, default as default45 } from "@opentiny/vue-loading";
import Mask from "@opentiny/vue-mask";
import { default as default46, default as default47 } from "@opentiny/vue-mask";
import MiniPicker from "@opentiny/vue-mini-picker";
import { default as default48, default as default49 } from "@opentiny/vue-mini-picker";
import Modal from "@opentiny/vue-modal";
import { default as default50, default as default51 } from "@opentiny/vue-modal";
import MultiSelect from "@opentiny/vue-multi-select";
import { default as default52, default as default53 } from "@opentiny/vue-multi-select";
import MultiSelectItem from "@opentiny/vue-multi-select-item";
import { default as default54, default as default55 } from "@opentiny/vue-multi-select-item";
import NavBar from "@opentiny/vue-nav-bar";
import { default as default56, default as default57 } from "@opentiny/vue-nav-bar";
import Numeric from "@opentiny/vue-numeric";
import { default as default58, default as default59 } from "@opentiny/vue-numeric";
import PickerColumn from "@opentiny/vue-picker-column";
import { default as default60, default as default61 } from "@opentiny/vue-picker-column";
import Popover from "@opentiny/vue-popover";
import { default as default62, default as default63 } from "@opentiny/vue-popover";
import Progress from "@opentiny/vue-progress";
import { default as default64, default as default65 } from "@opentiny/vue-progress";
import PullRefresh from "@opentiny/vue-pull-refresh";
import { default as default66, default as default67 } from "@opentiny/vue-pull-refresh";
import Radio from "@opentiny/vue-radio";
import { default as default68, default as default69 } from "@opentiny/vue-radio";
import Search from "@opentiny/vue-search";
import { default as default70, default as default71 } from "@opentiny/vue-search";
import Slider from "@opentiny/vue-slider";
import { default as default72, default as default73 } from "@opentiny/vue-slider";
import Switch from "@opentiny/vue-switch";
import { default as default74, default as default75 } from "@opentiny/vue-switch";
import Tabbar from "@opentiny/vue-tabbar";
import { default as default76, default as default77 } from "@opentiny/vue-tabbar";
import TabbarItem from "@opentiny/vue-tabbar-item";
import { default as default78, default as default79 } from "@opentiny/vue-tabbar-item";
import Table from "@opentiny/vue-table";
import { default as default80, default as default81 } from "@opentiny/vue-table";
import Tabs from "@opentiny/vue-tabs";
import { default as default82, default as default83 } from "@opentiny/vue-tabs";
import Tag from "@opentiny/vue-tag";
import { default as default84, default as default85 } from "@opentiny/vue-tag";
import TimeLine from "@opentiny/vue-time-line";
import { default as default86, default as default87 } from "@opentiny/vue-time-line";
import TimeLineNew from "@opentiny/vue-time-line-new";
import { default as default88, default as default89 } from "@opentiny/vue-time-line-new";
import Toast from "@opentiny/vue-toast";
import { default as default90, default as default91 } from "@opentiny/vue-toast";
import UploadList from "@opentiny/vue-upload-list";
import { default as default92, default as default93 } from "@opentiny/vue-upload-list";
import UserHead from "@opentiny/vue-user-head";
import { default as default94, default as default95 } from "@opentiny/vue-user-head";
import Wheel from "@opentiny/vue-wheel";
import { default as default96, default as default97 } from "@opentiny/vue-wheel";
import { $prefix } from "@opentiny/vue-common";
var components = [ActionSheet, Alert, Avatar, Badge, Button, Checkbox, CheckboxGroup, ColorPicker, Container, DatePicker, DialogBox, DropdownItem, DropdownMenu, Exception, FileUpload, Form, ImageViewer, IndexBar, Input, Label, List, Loading, Mask, MiniPicker, Modal, MultiSelect, MultiSelectItem, NavBar, Numeric, PickerColumn, Popover, Progress, PullRefresh, Radio, Search, Slider, Switch, Tabbar, TabbarItem, Table, Tabs, Tag, TimeLine, TimeLineNew, Toast, UploadList, UserHead, Wheel];
var install = function install2(app, opts) {
  var _app$config;
  if (opts === void 0) {
    opts = {};
  }
  var regex = new RegExp("^" + $prefix);
  var appProperties = ((_app$config = app.config) == null ? void 0 : _app$config.globalProperties) || app.prototype;
  appProperties.tiny_mode = {
    value: "mobile"
  };
  components.forEach(function(component) {
    var name = component.name;
    var alias = opts.alias || opts.prefix;
    if (typeof component.install !== "function")
      return;
    if (name && alias) {
      app.component(name.replace(regex, alias), component);
    } else {
      component.install(app);
    }
  });
};
var version = "3.20.0";
var mobile = {
  ActionSheet,
  TinyActionSheet: ActionSheet,
  Alert,
  TinyAlert: Alert,
  Avatar,
  TinyAvatar: Avatar,
  Badge,
  TinyBadge: Badge,
  Button,
  TinyButton: Button,
  Checkbox,
  TinyCheckbox: Checkbox,
  CheckboxGroup,
  TinyCheckboxGroup: CheckboxGroup,
  ColorPicker,
  TinyColorPicker: ColorPicker,
  Container,
  TinyContainer: Container,
  DatePicker,
  TinyDatePicker: DatePicker,
  DialogBox,
  TinyDialogBox: DialogBox,
  DropdownItem,
  TinyDropdownItem: DropdownItem,
  DropdownMenu,
  TinyDropdownMenu: DropdownMenu,
  Exception,
  TinyException: Exception,
  FileUpload,
  TinyFileUpload: FileUpload,
  Form,
  TinyForm: Form,
  ImageViewer,
  TinyImageViewer: ImageViewer,
  IndexBar,
  TinyIndexBar: IndexBar,
  Input,
  TinyInput: Input,
  Label,
  TinyLabel: Label,
  List,
  TinyList: List,
  Loading,
  TinyLoading: Loading,
  Mask,
  TinyMask: Mask,
  MiniPicker,
  TinyMiniPicker: MiniPicker,
  Modal,
  TinyModal: Modal,
  MultiSelect,
  TinyMultiSelect: MultiSelect,
  MultiSelectItem,
  TinyMultiSelectItem: MultiSelectItem,
  NavBar,
  TinyNavBar: NavBar,
  Numeric,
  TinyNumeric: Numeric,
  PickerColumn,
  TinyPickerColumn: PickerColumn,
  Popover,
  TinyPopover: Popover,
  Progress,
  TinyProgress: Progress,
  PullRefresh,
  TinyPullRefresh: PullRefresh,
  Radio,
  TinyRadio: Radio,
  Search,
  TinySearch: Search,
  Slider,
  TinySlider: Slider,
  Switch,
  TinySwitch: Switch,
  Tabbar,
  TinyTabbar: Tabbar,
  TabbarItem,
  TinyTabbarItem: TabbarItem,
  Table,
  TinyTable: Table,
  Tabs,
  TinyTabs: Tabs,
  Tag,
  TinyTag: Tag,
  TimeLine,
  TinyTimeLine: TimeLine,
  TimeLineNew,
  TinyTimeLineNew: TimeLineNew,
  Toast,
  TinyToast: Toast,
  UploadList,
  TinyUploadList: UploadList,
  UserHead,
  TinyUserHead: UserHead,
  Wheel,
  TinyWheel: Wheel,
  install
};
export {
  default2 as ActionSheet,
  default4 as Alert,
  default6 as Avatar,
  default8 as Badge,
  default10 as Button,
  default12 as Checkbox,
  default14 as CheckboxGroup,
  default16 as ColorPicker,
  default18 as Container,
  default20 as DatePicker,
  default22 as DialogBox,
  default24 as DropdownItem,
  default26 as DropdownMenu,
  default28 as Exception,
  default30 as FileUpload,
  default32 as Form,
  default34 as ImageViewer,
  default36 as IndexBar,
  default38 as Input,
  default40 as Label,
  default42 as List,
  default44 as Loading,
  default46 as Mask,
  default48 as MiniPicker,
  default50 as Modal,
  default52 as MultiSelect,
  default54 as MultiSelectItem,
  default56 as NavBar,
  default58 as Numeric,
  default60 as PickerColumn,
  default62 as Popover,
  default64 as Progress,
  default66 as PullRefresh,
  default68 as Radio,
  default70 as Search,
  default72 as Slider,
  default74 as Switch,
  default76 as Tabbar,
  default78 as TabbarItem,
  default80 as Table,
  default82 as Tabs,
  default84 as Tag,
  default86 as TimeLine,
  default88 as TimeLineNew,
  default3 as TinyActionSheet,
  default5 as TinyAlert,
  default7 as TinyAvatar,
  default9 as TinyBadge,
  default11 as TinyButton,
  default13 as TinyCheckbox,
  default15 as TinyCheckboxGroup,
  default17 as TinyColorPicker,
  default19 as TinyContainer,
  default21 as TinyDatePicker,
  default23 as TinyDialogBox,
  default25 as TinyDropdownItem,
  default27 as TinyDropdownMenu,
  default29 as TinyException,
  default31 as TinyFileUpload,
  default33 as TinyForm,
  default35 as TinyImageViewer,
  default37 as TinyIndexBar,
  default39 as TinyInput,
  default41 as TinyLabel,
  default43 as TinyList,
  default45 as TinyLoading,
  default47 as TinyMask,
  default49 as TinyMiniPicker,
  default51 as TinyModal,
  default53 as TinyMultiSelect,
  default55 as TinyMultiSelectItem,
  default57 as TinyNavBar,
  default59 as TinyNumeric,
  default61 as TinyPickerColumn,
  default63 as TinyPopover,
  default65 as TinyProgress,
  default67 as TinyPullRefresh,
  default69 as TinyRadio,
  default71 as TinySearch,
  default73 as TinySlider,
  default75 as TinySwitch,
  default77 as TinyTabbar,
  default79 as TinyTabbarItem,
  default81 as TinyTable,
  default83 as TinyTabs,
  default85 as TinyTag,
  default87 as TinyTimeLine,
  default89 as TinyTimeLineNew,
  default90 as TinyToast,
  default92 as TinyUploadList,
  default94 as TinyUserHead,
  default96 as TinyWheel,
  default91 as Toast,
  default93 as UploadList,
  default95 as UserHead,
  default97 as Wheel,
  mobile as default,
  install,
  version
};
