// @opentiny/vue@3.20.0/pc.js
import ActionMenu from "@opentiny/vue-action-menu";
import { default as default2, default as default3 } from "@opentiny/vue-action-menu";
import Alert from "@opentiny/vue-alert";
import { default as default4, default as default5 } from "@opentiny/vue-alert";
import Amount from "@opentiny/vue-amount";
import { default as default6, default as default7 } from "@opentiny/vue-amount";
import Anchor from "@opentiny/vue-anchor";
import { default as default8, default as default9 } from "@opentiny/vue-anchor";
import Area from "@opentiny/vue-area";
import { default as default10, default as default11 } from "@opentiny/vue-area";
import Autocomplete from "@opentiny/vue-autocomplete";
import { default as default12, default as default13 } from "@opentiny/vue-autocomplete";
import Badge from "@opentiny/vue-badge";
import { default as default14, default as default15 } from "@opentiny/vue-badge";
import BaseSelect from "@opentiny/vue-base-select";
import { default as default16, default as default17 } from "@opentiny/vue-base-select";
import Breadcrumb from "@opentiny/vue-breadcrumb";
import { default as default18, default as default19 } from "@opentiny/vue-breadcrumb";
import BreadcrumbItem from "@opentiny/vue-breadcrumb-item";
import { default as default20, default as default21 } from "@opentiny/vue-breadcrumb-item";
import BulletinBoard from "@opentiny/vue-bulletin-board";
import { default as default22, default as default23 } from "@opentiny/vue-bulletin-board";
import Button from "@opentiny/vue-button";
import { default as default24, default as default25 } from "@opentiny/vue-button";
import ButtonGroup from "@opentiny/vue-button-group";
import { default as default26, default as default27 } from "@opentiny/vue-button-group";
import Calendar from "@opentiny/vue-calendar";
import { default as default28, default as default29 } from "@opentiny/vue-calendar";
import CalendarView from "@opentiny/vue-calendar-view";
import { default as default30, default as default31 } from "@opentiny/vue-calendar-view";
import Card from "@opentiny/vue-card";
import { default as default32, default as default33 } from "@opentiny/vue-card";
import CardGroup from "@opentiny/vue-card-group";
import { default as default34, default as default35 } from "@opentiny/vue-card-group";
import CardTemplate from "@opentiny/vue-card-template";
import { default as default36, default as default37 } from "@opentiny/vue-card-template";
import Carousel from "@opentiny/vue-carousel";
import { default as default38, default as default39 } from "@opentiny/vue-carousel";
import CarouselItem from "@opentiny/vue-carousel-item";
import { default as default40, default as default41 } from "@opentiny/vue-carousel-item";
import Cascader from "@opentiny/vue-cascader";
import { default as default42, default as default43 } from "@opentiny/vue-cascader";
import CascaderMenu from "@opentiny/vue-cascader-menu";
import { default as default44, default as default45 } from "@opentiny/vue-cascader-menu";
import CascaderNode from "@opentiny/vue-cascader-node";
import { default as default46, default as default47 } from "@opentiny/vue-cascader-node";
import CascaderPanel from "@opentiny/vue-cascader-panel";
import { default as default48, default as default49 } from "@opentiny/vue-cascader-panel";
import Checkbox from "@opentiny/vue-checkbox";
import { default as default50, default as default51 } from "@opentiny/vue-checkbox";
import CheckboxButton from "@opentiny/vue-checkbox-button";
import { default as default52, default as default53 } from "@opentiny/vue-checkbox-button";
import CheckboxGroup from "@opentiny/vue-checkbox-group";
import { default as default54, default as default55 } from "@opentiny/vue-checkbox-group";
import Col from "@opentiny/vue-col";
import { default as default56, default as default57 } from "@opentiny/vue-col";
import Collapse from "@opentiny/vue-collapse";
import { default as default58, default as default59 } from "@opentiny/vue-collapse";
import CollapseItem from "@opentiny/vue-collapse-item";
import { default as default60, default as default61 } from "@opentiny/vue-collapse-item";
import CollapseTransition from "@opentiny/vue-collapse-transition";
import { default as default62, default as default63 } from "@opentiny/vue-collapse-transition";
import ColorPicker from "@opentiny/vue-color-picker";
import { default as default64, default as default65 } from "@opentiny/vue-color-picker";
import ColorSelectPanel from "@opentiny/vue-color-select-panel";
import { default as default66, default as default67 } from "@opentiny/vue-color-select-panel";
import Company from "@opentiny/vue-company";
import { default as default68, default as default69 } from "@opentiny/vue-company";
import ConfigProvider from "@opentiny/vue-config-provider";
import { default as default70, default as default71 } from "@opentiny/vue-config-provider";
import Container from "@opentiny/vue-container";
import { default as default72, default as default73 } from "@opentiny/vue-container";
import Country from "@opentiny/vue-country";
import { default as default74, default as default75 } from "@opentiny/vue-country";
import Crop from "@opentiny/vue-crop";
import { default as default76, default as default77 } from "@opentiny/vue-crop";
import Currency from "@opentiny/vue-currency";
import { default as default78, default as default79 } from "@opentiny/vue-currency";
import DatePanel from "@opentiny/vue-date-panel";
import { default as default80, default as default81 } from "@opentiny/vue-date-panel";
import DatePicker from "@opentiny/vue-date-picker";
import { default as default82, default as default83 } from "@opentiny/vue-date-picker";
import DatePickerMobileFirst from "@opentiny/vue-date-picker-mobile-first";
import { default as default84, default as default85 } from "@opentiny/vue-date-picker-mobile-first";
import DateRange from "@opentiny/vue-date-range";
import { default as default86, default as default87 } from "@opentiny/vue-date-range";
import DateTable from "@opentiny/vue-date-table";
import { default as default88, default as default89 } from "@opentiny/vue-date-table";
import Dept from "@opentiny/vue-dept";
import { default as default90, default as default91 } from "@opentiny/vue-dept";
import DialogBox from "@opentiny/vue-dialog-box";
import { default as default92, default as default93 } from "@opentiny/vue-dialog-box";
import DialogSelect from "@opentiny/vue-dialog-select";
import { default as default94, default as default95 } from "@opentiny/vue-dialog-select";
import Divider from "@opentiny/vue-divider";
import { default as default96, default as default97 } from "@opentiny/vue-divider";
import Drawer from "@opentiny/vue-drawer";
import { default as default98, default as default99 } from "@opentiny/vue-drawer";
import DropRoles from "@opentiny/vue-drop-roles";
import { default as default100, default as default101 } from "@opentiny/vue-drop-roles";
import DropTimes from "@opentiny/vue-drop-times";
import { default as default102, default as default103 } from "@opentiny/vue-drop-times";
import Dropdown from "@opentiny/vue-dropdown";
import { default as default104, default as default105 } from "@opentiny/vue-dropdown";
import DropdownItem from "@opentiny/vue-dropdown-item";
import { default as default106, default as default107 } from "@opentiny/vue-dropdown-item";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import { default as default108, default as default109 } from "@opentiny/vue-dropdown-menu";
import DynamicScroller from "@opentiny/vue-dynamic-scroller";
import { default as default110, default as default111 } from "@opentiny/vue-dynamic-scroller";
import DynamicScrollerItem from "@opentiny/vue-dynamic-scroller-item";
import { default as default112, default as default113 } from "@opentiny/vue-dynamic-scroller-item";
import Espace from "@opentiny/vue-espace";
import { default as default114, default as default115 } from "@opentiny/vue-espace";
import FallMenu from "@opentiny/vue-fall-menu";
import { default as default116, default as default117 } from "@opentiny/vue-fall-menu";
import FileUpload from "@opentiny/vue-file-upload";
import { default as default118, default as default119 } from "@opentiny/vue-file-upload";
import FilterBox from "@opentiny/vue-filter-box";
import { default as default120, default as default121 } from "@opentiny/vue-filter-box";
import FilterPanel from "@opentiny/vue-filter-panel";
import { default as default122, default as default123 } from "@opentiny/vue-filter-panel";
import FloatButton from "@opentiny/vue-float-button";
import { default as default124, default as default125 } from "@opentiny/vue-float-button";
import Floatbar from "@opentiny/vue-floatbar";
import { default as default126, default as default127 } from "@opentiny/vue-floatbar";
import Flowchart from "@opentiny/vue-flowchart";
import { default as default128, default as default129 } from "@opentiny/vue-flowchart";
import FluentEditor from "@opentiny/vue-fluent-editor";
import { default as default130, default as default131 } from "@opentiny/vue-fluent-editor";
import Form from "@opentiny/vue-form";
import { default as default132, default as default133 } from "@opentiny/vue-form";
import FormItem from "@opentiny/vue-form-item";
import { default as default134, default as default135 } from "@opentiny/vue-form-item";
import Fullscreen from "@opentiny/vue-fullscreen";
import { default as default136, default as default137 } from "@opentiny/vue-fullscreen";
import Grid from "@opentiny/vue-grid";
import { default as default138, default as default139 } from "@opentiny/vue-grid";
import GridColumn from "@opentiny/vue-grid-column";
import { default as default140, default as default141 } from "@opentiny/vue-grid-column";
import GridManager from "@opentiny/vue-grid-manager";
import { default as default142, default as default143 } from "@opentiny/vue-grid-manager";
import GridSelect from "@opentiny/vue-grid-select";
import { default as default144, default as default145 } from "@opentiny/vue-grid-select";
import GridToolbar from "@opentiny/vue-grid-toolbar";
import { default as default146, default as default147 } from "@opentiny/vue-grid-toolbar";
import Guide from "@opentiny/vue-guide";
import { default as default148, default as default149 } from "@opentiny/vue-guide";
import Hrapprover from "@opentiny/vue-hrapprover";
import { default as default150, default as default151 } from "@opentiny/vue-hrapprover";
import Huicharts from "@opentiny/vue-huicharts";
import { default as default152, default as default153, default as default154 } from "@opentiny/vue-huicharts";
import HuichartsAmap from "@opentiny/vue-huicharts-amap";
import { default as default155, default as default156, default as default157 } from "@opentiny/vue-huicharts-amap";
import HuichartsBar from "@opentiny/vue-huicharts-bar";
import { default as default158, default as default159, default as default160 } from "@opentiny/vue-huicharts-bar";
import HuichartsBmap from "@opentiny/vue-huicharts-bmap";
import { default as default161, default as default162, default as default163 } from "@opentiny/vue-huicharts-bmap";
import HuichartsBoxplot from "@opentiny/vue-huicharts-boxplot";
import { default as default164, default as default165, default as default166 } from "@opentiny/vue-huicharts-boxplot";
import HuichartsCandle from "@opentiny/vue-huicharts-candle";
import { default as default167, default as default168, default as default169 } from "@opentiny/vue-huicharts-candle";
import Core from "@opentiny/vue-huicharts-core";
import { default as default170, default as default171, default as default172 } from "@opentiny/vue-huicharts-core";
import HuichartsFunnel from "@opentiny/vue-huicharts-funnel";
import { default as default173, default as default174, default as default175 } from "@opentiny/vue-huicharts-funnel";
import HuichartsGauge from "@opentiny/vue-huicharts-gauge";
import { default as default176, default as default177, default as default178 } from "@opentiny/vue-huicharts-gauge";
import HuichartsGraph from "@opentiny/vue-huicharts-graph";
import { default as default179, default as default180, default as default181 } from "@opentiny/vue-huicharts-graph";
import HuichartsHeatmap from "@opentiny/vue-huicharts-heatmap";
import { default as default182, default as default183, default as default184 } from "@opentiny/vue-huicharts-heatmap";
import HuichartsHistogram from "@opentiny/vue-huicharts-histogram";
import { default as default185, default as default186, default as default187 } from "@opentiny/vue-huicharts-histogram";
import HuichartsLine from "@opentiny/vue-huicharts-line";
import { default as default188, default as default189, default as default190 } from "@opentiny/vue-huicharts-line";
import HuichartsLiquidfill from "@opentiny/vue-huicharts-liquidfill";
import { default as default191, default as default192, default as default193 } from "@opentiny/vue-huicharts-liquidfill";
import HuichartsMap from "@opentiny/vue-huicharts-map";
import { default as default194, default as default195, default as default196 } from "@opentiny/vue-huicharts-map";
import HuichartsPie from "@opentiny/vue-huicharts-pie";
import { default as default197, default as default198, default as default199 } from "@opentiny/vue-huicharts-pie";
import HuichartsProcess from "@opentiny/vue-huicharts-process";
import { default as default200, default as default201, default as default202 } from "@opentiny/vue-huicharts-process";
import HuichartsRadar from "@opentiny/vue-huicharts-radar";
import { default as default203, default as default204, default as default205 } from "@opentiny/vue-huicharts-radar";
import HuichartsRing from "@opentiny/vue-huicharts-ring";
import { default as default206, default as default207, default as default208 } from "@opentiny/vue-huicharts-ring";
import HuichartsSankey from "@opentiny/vue-huicharts-sankey";
import { default as default209, default as default210, default as default211 } from "@opentiny/vue-huicharts-sankey";
import HuichartsScatter from "@opentiny/vue-huicharts-scatter";
import { default as default212, default as default213, default as default214 } from "@opentiny/vue-huicharts-scatter";
import HuichartsSunburst from "@opentiny/vue-huicharts-sunburst";
import { default as default215, default as default216, default as default217 } from "@opentiny/vue-huicharts-sunburst";
import HuichartsTree from "@opentiny/vue-huicharts-tree";
import { default as default218, default as default219, default as default220 } from "@opentiny/vue-huicharts-tree";
import HuichartsWaterfall from "@opentiny/vue-huicharts-waterfall";
import { default as default221, default as default222, default as default223 } from "@opentiny/vue-huicharts-waterfall";
import HuichartsWordcloud from "@opentiny/vue-huicharts-wordcloud";
import { default as default224, default as default225, default as default226 } from "@opentiny/vue-huicharts-wordcloud";
import Image from "@opentiny/vue-image";
import { default as default227, default as default228 } from "@opentiny/vue-image";
import ImageViewer from "@opentiny/vue-image-viewer";
import { default as default229, default as default230 } from "@opentiny/vue-image-viewer";
import IndexBarAnchor from "@opentiny/vue-index-bar-anchor";
import { default as default231, default as default232 } from "@opentiny/vue-index-bar-anchor";
import Input from "@opentiny/vue-input";
import { default as default233, default as default234 } from "@opentiny/vue-input";
import IpAddress from "@opentiny/vue-ip-address";
import { default as default235, default as default236 } from "@opentiny/vue-ip-address";
import Layout from "@opentiny/vue-layout";
import { default as default237, default as default238 } from "@opentiny/vue-layout";
import Link from "@opentiny/vue-link";
import { default as default239, default as default240 } from "@opentiny/vue-link";
import LinkMenu from "@opentiny/vue-link-menu";
import { default as default241, default as default242 } from "@opentiny/vue-link-menu";
import Loading from "@opentiny/vue-loading";
import { default as default243, default as default244 } from "@opentiny/vue-loading";
import Locales from "@opentiny/vue-locales";
import { default as default245, default as default246 } from "@opentiny/vue-locales";
import LogonUser from "@opentiny/vue-logon-user";
import { default as default247, default as default248 } from "@opentiny/vue-logon-user";
import Logout from "@opentiny/vue-logout";
import { default as default249, default as default250 } from "@opentiny/vue-logout";
import Message from "@opentiny/vue-message";
import { default as default251, default as default252 } from "@opentiny/vue-message";
import Milestone from "@opentiny/vue-milestone";
import { default as default253, default as default254 } from "@opentiny/vue-milestone";
import MindMap from "@opentiny/vue-mind-map";
import { default as default255, default as default256 } from "@opentiny/vue-mind-map";
import Modal from "@opentiny/vue-modal";
import { default as default257, default as default258 } from "@opentiny/vue-modal";
import MonthRange from "@opentiny/vue-month-range";
import { default as default259, default as default260 } from "@opentiny/vue-month-range";
import MonthTable from "@opentiny/vue-month-table";
import { default as default261, default as default262 } from "@opentiny/vue-month-table";
import NavMenu from "@opentiny/vue-nav-menu";
import { default as default263, default as default264 } from "@opentiny/vue-nav-menu";
import Notify from "@opentiny/vue-notify";
import { default as default265, default as default266 } from "@opentiny/vue-notify";
import Numeric from "@opentiny/vue-numeric";
import { default as default267, default as default268 } from "@opentiny/vue-numeric";
import Option from "@opentiny/vue-option";
import { default as default269, default as default270 } from "@opentiny/vue-option";
import OptionGroup from "@opentiny/vue-option-group";
import { default as default271, default as default272 } from "@opentiny/vue-option-group";
import Pager from "@opentiny/vue-pager";
import { default as default273, default as default274 } from "@opentiny/vue-pager";
import PagerItem from "@opentiny/vue-pager-item";
import { default as default275, default as default276 } from "@opentiny/vue-pager-item";
import Panel from "@opentiny/vue-panel";
import { default as default277, default as default278 } from "@opentiny/vue-panel";
import Picker from "@opentiny/vue-picker";
import { default as default279, default as default280 } from "@opentiny/vue-picker";
import PopUpload from "@opentiny/vue-pop-upload";
import { default as default281, default as default282 } from "@opentiny/vue-pop-upload";
import Popconfirm from "@opentiny/vue-popconfirm";
import { default as default283, default as default284 } from "@opentiny/vue-popconfirm";
import Popeditor from "@opentiny/vue-popeditor";
import { default as default285, default as default286 } from "@opentiny/vue-popeditor";
import Popover from "@opentiny/vue-popover";
import { default as default287, default as default288 } from "@opentiny/vue-popover";
import Popup from "@opentiny/vue-popup";
import { default as default289, default as default290 } from "@opentiny/vue-popup";
import Progress from "@opentiny/vue-progress";
import { default as default291, default as default292 } from "@opentiny/vue-progress";
import QrCode from "@opentiny/vue-qr-code";
import { default as default293, default as default294 } from "@opentiny/vue-qr-code";
import QuarterPanel from "@opentiny/vue-quarter-panel";
import { default as default295, default as default296 } from "@opentiny/vue-quarter-panel";
import QueryBuilder from "@opentiny/vue-query-builder";
import { default as default297, default as default298 } from "@opentiny/vue-query-builder";
import Radio from "@opentiny/vue-radio";
import { default as default299, default as default300 } from "@opentiny/vue-radio";
import RadioButton from "@opentiny/vue-radio-button";
import { default as default301, default as default302 } from "@opentiny/vue-radio-button";
import RadioGroup from "@opentiny/vue-radio-group";
import { default as default303, default as default304 } from "@opentiny/vue-radio-group";
import Rate from "@opentiny/vue-rate";
import { default as default305, default as default306 } from "@opentiny/vue-rate";
import RecycleScroller from "@opentiny/vue-recycle-scroller";
import { default as default307, default as default308 } from "@opentiny/vue-recycle-scroller";
import RichTextEditor from "@opentiny/vue-rich-text-editor";
import { default as default309, default as default310 } from "@opentiny/vue-rich-text-editor";
import River from "@opentiny/vue-river";
import { default as default311, default as default312 } from "@opentiny/vue-river";
import Roles from "@opentiny/vue-roles";
import { default as default313, default as default314 } from "@opentiny/vue-roles";
import Row from "@opentiny/vue-row";
import { default as default315, default as default316 } from "@opentiny/vue-row";
import ScrollText from "@opentiny/vue-scroll-text";
import { default as default317, default as default318 } from "@opentiny/vue-scroll-text";
import Scrollbar from "@opentiny/vue-scrollbar";
import { default as default319, default as default320 } from "@opentiny/vue-scrollbar";
import Search from "@opentiny/vue-search";
import { default as default321, default as default322 } from "@opentiny/vue-search";
import Select from "@opentiny/vue-select";
import { default as default323, default as default324 } from "@opentiny/vue-select";
import SelectDropdown from "@opentiny/vue-select-dropdown";
import { default as default325, default as default326 } from "@opentiny/vue-select-dropdown";
import SelectedBox from "@opentiny/vue-selected-box";
import { default as default327, default as default328 } from "@opentiny/vue-selected-box";
import Skeleton from "@opentiny/vue-skeleton";
import { default as default329, default as default330 } from "@opentiny/vue-skeleton";
import SkeletonItem from "@opentiny/vue-skeleton-item";
import { default as default331, default as default332 } from "@opentiny/vue-skeleton-item";
import Slider from "@opentiny/vue-slider";
import { default as default333, default as default334 } from "@opentiny/vue-slider";
import Split from "@opentiny/vue-split";
import { default as default335, default as default336 } from "@opentiny/vue-split";
import Statistic from "@opentiny/vue-statistic";
import { default as default337, default as default338 } from "@opentiny/vue-statistic";
import Steps from "@opentiny/vue-steps";
import { default as default339, default as default340 } from "@opentiny/vue-steps";
import Sticky from "@opentiny/vue-sticky";
import { default as default341, default as default342 } from "@opentiny/vue-sticky";
import Switch from "@opentiny/vue-switch";
import { default as default343, default as default344 } from "@opentiny/vue-switch";
import TabItem from "@opentiny/vue-tab-item";
import { default as default345, default as default346 } from "@opentiny/vue-tab-item";
import Table from "@opentiny/vue-table";
import { default as default347, default as default348 } from "@opentiny/vue-table";
import Tabs from "@opentiny/vue-tabs";
import { default as default349, default as default350 } from "@opentiny/vue-tabs";
import Tag from "@opentiny/vue-tag";
import { default as default351, default as default352 } from "@opentiny/vue-tag";
import TagGroup from "@opentiny/vue-tag-group";
import { default as default353, default as default354 } from "@opentiny/vue-tag-group";
import TextPopup from "@opentiny/vue-text-popup";
import { default as default355, default as default356 } from "@opentiny/vue-text-popup";
import Time from "@opentiny/vue-time";
import { default as default357, default as default358 } from "@opentiny/vue-time";
import TimeLine from "@opentiny/vue-time-line";
import { default as default359, default as default360 } from "@opentiny/vue-time-line";
import TimeLineNew from "@opentiny/vue-time-line-new";
import { default as default361, default as default362 } from "@opentiny/vue-time-line-new";
import TimePanel from "@opentiny/vue-time-panel";
import { default as default363, default as default364 } from "@opentiny/vue-time-panel";
import TimePicker from "@opentiny/vue-time-picker";
import { default as default365, default as default366 } from "@opentiny/vue-time-picker";
import TimeRange from "@opentiny/vue-time-range";
import { default as default367, default as default368 } from "@opentiny/vue-time-range";
import TimeSelect from "@opentiny/vue-time-select";
import { default as default369, default as default370 } from "@opentiny/vue-time-select";
import TimeSpinner from "@opentiny/vue-time-spinner";
import { default as default371, default as default372 } from "@opentiny/vue-time-spinner";
import TimelineItem from "@opentiny/vue-timeline-item";
import { default as default373, default as default374 } from "@opentiny/vue-timeline-item";
import ToggleMenu from "@opentiny/vue-toggle-menu";
import { default as default375, default as default376 } from "@opentiny/vue-toggle-menu";
import Tooltip from "@opentiny/vue-tooltip";
import { default as default377, default as default378 } from "@opentiny/vue-tooltip";
import TopBox from "@opentiny/vue-top-box";
import { default as default379, default as default380 } from "@opentiny/vue-top-box";
import Transfer from "@opentiny/vue-transfer";
import { default as default381, default as default382 } from "@opentiny/vue-transfer";
import TransferPanel from "@opentiny/vue-transfer-panel";
import { default as default383, default as default384 } from "@opentiny/vue-transfer-panel";
import Tree from "@opentiny/vue-tree";
import { default as default385, default as default386 } from "@opentiny/vue-tree";
import TreeMenu from "@opentiny/vue-tree-menu";
import { default as default387, default as default388 } from "@opentiny/vue-tree-menu";
import TreeSelect from "@opentiny/vue-tree-select";
import { default as default389, default as default390 } from "@opentiny/vue-tree-select";
import Upload from "@opentiny/vue-upload";
import { default as default391, default as default392 } from "@opentiny/vue-upload";
import UploadDragger from "@opentiny/vue-upload-dragger";
import { default as default393, default as default394 } from "@opentiny/vue-upload-dragger";
import UploadList from "@opentiny/vue-upload-list";
import { default as default395, default as default396 } from "@opentiny/vue-upload-list";
import User from "@opentiny/vue-user";
import { default as default397, default as default398 } from "@opentiny/vue-user";
import UserAccount from "@opentiny/vue-user-account";
import { default as default399, default as default400 } from "@opentiny/vue-user-account";
import UserContact from "@opentiny/vue-user-contact";
import { default as default401, default as default402 } from "@opentiny/vue-user-contact";
import UserHead from "@opentiny/vue-user-head";
import { default as default403, default as default404 } from "@opentiny/vue-user-head";
import UserLink from "@opentiny/vue-user-link";
import { default as default405, default as default406 } from "@opentiny/vue-user-link";
import VirtualScrollBox from "@opentiny/vue-virtual-scroll-box";
import { default as default407, default as default408 } from "@opentiny/vue-virtual-scroll-box";
import VirtualTree from "@opentiny/vue-virtual-tree";
import { default as default409, default as default410 } from "@opentiny/vue-virtual-tree";
import Watermark from "@opentiny/vue-watermark";
import { default as default411, default as default412 } from "@opentiny/vue-watermark";
import Wizard from "@opentiny/vue-wizard";
import { default as default413, default as default414 } from "@opentiny/vue-wizard";
import YearRange from "@opentiny/vue-year-range";
import { default as default415, default as default416 } from "@opentiny/vue-year-range";
import YearTable from "@opentiny/vue-year-table";
import { default as default417, default as default418 } from "@opentiny/vue-year-table";
import { $prefix } from "@opentiny/vue-common";
var components = [ActionMenu, Alert, Amount, Anchor, Area, Autocomplete, Badge, BaseSelect, Breadcrumb, BreadcrumbItem, BulletinBoard, Button, ButtonGroup, Calendar, CalendarView, Card, CardGroup, CardTemplate, Carousel, CarouselItem, Cascader, CascaderMenu, CascaderNode, CascaderPanel, Checkbox, CheckboxButton, CheckboxGroup, Col, Collapse, CollapseItem, CollapseTransition, ColorPicker, ColorSelectPanel, Company, ConfigProvider, Container, Country, Crop, Currency, DatePanel, DatePicker, DatePickerMobileFirst, DateRange, DateTable, Dept, DialogBox, DialogSelect, Divider, Drawer, DropRoles, DropTimes, Dropdown, DropdownItem, DropdownMenu, DynamicScroller, DynamicScrollerItem, Espace, FallMenu, FileUpload, FilterBox, FilterPanel, FloatButton, Floatbar, Flowchart, FluentEditor, Form, FormItem, Fullscreen, Grid, GridColumn, GridManager, GridSelect, GridToolbar, Guide, Hrapprover, Huicharts, HuichartsAmap, HuichartsBar, HuichartsBmap, HuichartsBoxplot, HuichartsCandle, Core, HuichartsFunnel, HuichartsGauge, HuichartsGraph, HuichartsHeatmap, HuichartsHistogram, HuichartsLine, HuichartsLiquidfill, HuichartsMap, HuichartsPie, HuichartsProcess, HuichartsRadar, HuichartsRing, HuichartsSankey, HuichartsScatter, HuichartsSunburst, HuichartsTree, HuichartsWaterfall, HuichartsWordcloud, Image, ImageViewer, IndexBarAnchor, Input, IpAddress, Layout, Link, LinkMenu, Loading, Locales, LogonUser, Logout, Message, Milestone, MindMap, Modal, MonthRange, MonthTable, NavMenu, Notify, Numeric, Option, OptionGroup, Pager, PagerItem, Panel, Picker, PopUpload, Popconfirm, Popeditor, Popover, Popup, Progress, QrCode, QuarterPanel, QueryBuilder, Radio, RadioButton, RadioGroup, Rate, RecycleScroller, RichTextEditor, River, Roles, Row, ScrollText, Scrollbar, Search, Select, SelectDropdown, SelectedBox, Skeleton, SkeletonItem, Slider, Split, Statistic, Steps, Sticky, Switch, TabItem, Table, Tabs, Tag, TagGroup, TextPopup, Time, TimeLine, TimeLineNew, TimePanel, TimePicker, TimeRange, TimeSelect, TimeSpinner, TimelineItem, ToggleMenu, Tooltip, TopBox, Transfer, TransferPanel, Tree, TreeMenu, TreeSelect, Upload, UploadDragger, UploadList, User, UserAccount, UserContact, UserHead, UserLink, VirtualScrollBox, VirtualTree, Watermark, Wizard, YearRange, YearTable];
var install = function install2(app, opts) {
  var _app$config;
  if (opts === void 0) {
    opts = {};
  }
  var regex = new RegExp("^" + $prefix);
  var appProperties = ((_app$config = app.config) == null ? void 0 : _app$config.globalProperties) || app.prototype;
  appProperties.tiny_mode = {
    value: "pc"
  };
  components.forEach(function(component) {
    var name = component.name;
    var alias = opts.alias || opts.prefix;
    if (typeof component.install !== "function")
      return;
    if (name && alias) {
      app.component(name.replace(regex, alias), component);
    } else {
      component.install(app);
    }
  });
};
var version = "3.20.0";
var pc = {
  ActionMenu,
  TinyActionMenu: ActionMenu,
  Alert,
  TinyAlert: Alert,
  Amount,
  TinyAmount: Amount,
  Anchor,
  TinyAnchor: Anchor,
  Area,
  TinyArea: Area,
  Autocomplete,
  TinyAutocomplete: Autocomplete,
  Badge,
  TinyBadge: Badge,
  BaseSelect,
  TinyBaseSelect: BaseSelect,
  Breadcrumb,
  TinyBreadcrumb: Breadcrumb,
  BreadcrumbItem,
  TinyBreadcrumbItem: BreadcrumbItem,
  BulletinBoard,
  TinyBulletinBoard: BulletinBoard,
  Button,
  TinyButton: Button,
  ButtonGroup,
  TinyButtonGroup: ButtonGroup,
  Calendar,
  TinyCalendar: Calendar,
  CalendarView,
  TinyCalendarView: CalendarView,
  Card,
  TinyCard: Card,
  CardGroup,
  TinyCardGroup: CardGroup,
  CardTemplate,
  TinyCardTemplate: CardTemplate,
  Carousel,
  TinyCarousel: Carousel,
  CarouselItem,
  TinyCarouselItem: CarouselItem,
  Cascader,
  TinyCascader: Cascader,
  CascaderMenu,
  TinyCascaderMenu: CascaderMenu,
  CascaderNode,
  TinyCascaderNode: CascaderNode,
  CascaderPanel,
  TinyCascaderPanel: CascaderPanel,
  Checkbox,
  TinyCheckbox: Checkbox,
  CheckboxButton,
  TinyCheckboxButton: CheckboxButton,
  CheckboxGroup,
  TinyCheckboxGroup: CheckboxGroup,
  Col,
  TinyCol: Col,
  Collapse,
  TinyCollapse: Collapse,
  CollapseItem,
  TinyCollapseItem: CollapseItem,
  CollapseTransition,
  TinyCollapseTransition: CollapseTransition,
  ColorPicker,
  TinyColorPicker: ColorPicker,
  ColorSelectPanel,
  TinyColorSelectPanel: ColorSelectPanel,
  Company,
  TinyCompany: Company,
  ConfigProvider,
  TinyConfigProvider: ConfigProvider,
  Container,
  TinyContainer: Container,
  Country,
  TinyCountry: Country,
  Crop,
  TinyCrop: Crop,
  Currency,
  TinyCurrency: Currency,
  DatePanel,
  TinyDatePanel: DatePanel,
  DatePicker,
  TinyDatePicker: DatePicker,
  DatePickerMobileFirst,
  TinyDatePickerMobileFirst: DatePickerMobileFirst,
  DateRange,
  TinyDateRange: DateRange,
  DateTable,
  TinyDateTable: DateTable,
  Dept,
  TinyDept: Dept,
  DialogBox,
  TinyDialogBox: DialogBox,
  DialogSelect,
  TinyDialogSelect: DialogSelect,
  Divider,
  TinyDivider: Divider,
  Drawer,
  TinyDrawer: Drawer,
  DropRoles,
  TinyDropRoles: DropRoles,
  DropTimes,
  TinyDropTimes: DropTimes,
  Dropdown,
  TinyDropdown: Dropdown,
  DropdownItem,
  TinyDropdownItem: DropdownItem,
  DropdownMenu,
  TinyDropdownMenu: DropdownMenu,
  DynamicScroller,
  TinyDynamicScroller: DynamicScroller,
  DynamicScrollerItem,
  TinyDynamicScrollerItem: DynamicScrollerItem,
  Espace,
  TinyEspace: Espace,
  FallMenu,
  TinyFallMenu: FallMenu,
  FileUpload,
  TinyFileUpload: FileUpload,
  FilterBox,
  TinyFilterBox: FilterBox,
  FilterPanel,
  TinyFilterPanel: FilterPanel,
  FloatButton,
  TinyFloatButton: FloatButton,
  Floatbar,
  TinyFloatbar: Floatbar,
  Flowchart,
  TinyFlowchart: Flowchart,
  FluentEditor,
  TinyFluentEditor: FluentEditor,
  Form,
  TinyForm: Form,
  FormItem,
  TinyFormItem: FormItem,
  Fullscreen,
  TinyFullscreen: Fullscreen,
  Grid,
  TinyGrid: Grid,
  GridColumn,
  TinyGridColumn: GridColumn,
  GridManager,
  TinyGridManager: GridManager,
  GridSelect,
  TinyGridSelect: GridSelect,
  GridToolbar,
  TinyGridToolbar: GridToolbar,
  Guide,
  TinyGuide: Guide,
  Hrapprover,
  TinyHrapprover: Hrapprover,
  Huicharts,
  Charts: Huicharts,
  TinyHuicharts: Huicharts,
  HuichartsAmap,
  ChartsAmap: HuichartsAmap,
  TinyHuichartsAmap: HuichartsAmap,
  HuichartsBar,
  ChartsBar: HuichartsBar,
  TinyHuichartsBar: HuichartsBar,
  HuichartsBmap,
  ChartsBmap: HuichartsBmap,
  TinyHuichartsBmap: HuichartsBmap,
  HuichartsBoxplot,
  ChartsBoxplot: HuichartsBoxplot,
  TinyHuichartsBoxplot: HuichartsBoxplot,
  HuichartsCandle,
  ChartsCandle: HuichartsCandle,
  TinyHuichartsCandle: HuichartsCandle,
  HuichartsCore: Core,
  ChartsCore: Core,
  TinyHuichartsCore: Core,
  HuichartsFunnel,
  ChartsFunnel: HuichartsFunnel,
  TinyHuichartsFunnel: HuichartsFunnel,
  HuichartsGauge,
  ChartsGauge: HuichartsGauge,
  TinyHuichartsGauge: HuichartsGauge,
  HuichartsGraph,
  ChartsGraph: HuichartsGraph,
  TinyHuichartsGraph: HuichartsGraph,
  HuichartsHeatmap,
  ChartsHeatmap: HuichartsHeatmap,
  TinyHuichartsHeatmap: HuichartsHeatmap,
  HuichartsHistogram,
  ChartsHistogram: HuichartsHistogram,
  TinyHuichartsHistogram: HuichartsHistogram,
  HuichartsLine,
  ChartsLine: HuichartsLine,
  TinyHuichartsLine: HuichartsLine,
  HuichartsLiquidfill,
  ChartsLiquidfill: HuichartsLiquidfill,
  TinyHuichartsLiquidfill: HuichartsLiquidfill,
  HuichartsMap,
  ChartsMap: HuichartsMap,
  TinyHuichartsMap: HuichartsMap,
  HuichartsPie,
  ChartsPie: HuichartsPie,
  TinyHuichartsPie: HuichartsPie,
  HuichartsProcess,
  ChartsProcess: HuichartsProcess,
  TinyHuichartsProcess: HuichartsProcess,
  HuichartsRadar,
  ChartsRadar: HuichartsRadar,
  TinyHuichartsRadar: HuichartsRadar,
  HuichartsRing,
  ChartsRing: HuichartsRing,
  TinyHuichartsRing: HuichartsRing,
  HuichartsSankey,
  ChartsSankey: HuichartsSankey,
  TinyHuichartsSankey: HuichartsSankey,
  HuichartsScatter,
  ChartsScatter: HuichartsScatter,
  TinyHuichartsScatter: HuichartsScatter,
  HuichartsSunburst,
  ChartsSunburst: HuichartsSunburst,
  TinyHuichartsSunburst: HuichartsSunburst,
  HuichartsTree,
  ChartsTree: HuichartsTree,
  TinyHuichartsTree: HuichartsTree,
  HuichartsWaterfall,
  ChartsWaterfall: HuichartsWaterfall,
  TinyHuichartsWaterfall: HuichartsWaterfall,
  HuichartsWordcloud,
  ChartsWordcloud: HuichartsWordcloud,
  TinyHuichartsWordcloud: HuichartsWordcloud,
  Image,
  TinyImage: Image,
  ImageViewer,
  TinyImageViewer: ImageViewer,
  IndexBarAnchor,
  TinyIndexBarAnchor: IndexBarAnchor,
  Input,
  TinyInput: Input,
  IpAddress,
  TinyIpAddress: IpAddress,
  Layout,
  TinyLayout: Layout,
  Link,
  TinyLink: Link,
  LinkMenu,
  TinyLinkMenu: LinkMenu,
  Loading,
  TinyLoading: Loading,
  Locales,
  TinyLocales: Locales,
  LogonUser,
  TinyLogonUser: LogonUser,
  Logout,
  TinyLogout: Logout,
  Message,
  TinyMessage: Message,
  Milestone,
  TinyMilestone: Milestone,
  MindMap,
  TinyMindMap: MindMap,
  Modal,
  TinyModal: Modal,
  MonthRange,
  TinyMonthRange: MonthRange,
  MonthTable,
  TinyMonthTable: MonthTable,
  NavMenu,
  TinyNavMenu: NavMenu,
  Notify,
  TinyNotify: Notify,
  Numeric,
  TinyNumeric: Numeric,
  Option,
  TinyOption: Option,
  OptionGroup,
  TinyOptionGroup: OptionGroup,
  Pager,
  TinyPager: Pager,
  PagerItem,
  TinyPagerItem: PagerItem,
  Panel,
  TinyPanel: Panel,
  Picker,
  TinyPicker: Picker,
  PopUpload,
  TinyPopUpload: PopUpload,
  Popconfirm,
  TinyPopconfirm: Popconfirm,
  Popeditor,
  TinyPopeditor: Popeditor,
  Popover,
  TinyPopover: Popover,
  Popup,
  TinyPopup: Popup,
  Progress,
  TinyProgress: Progress,
  QrCode,
  TinyQrCode: QrCode,
  QuarterPanel,
  TinyQuarterPanel: QuarterPanel,
  QueryBuilder,
  TinyQueryBuilder: QueryBuilder,
  Radio,
  TinyRadio: Radio,
  RadioButton,
  TinyRadioButton: RadioButton,
  RadioGroup,
  TinyRadioGroup: RadioGroup,
  Rate,
  TinyRate: Rate,
  RecycleScroller,
  TinyRecycleScroller: RecycleScroller,
  RichTextEditor,
  TinyRichTextEditor: RichTextEditor,
  River,
  TinyRiver: River,
  Roles,
  TinyRoles: Roles,
  Row,
  TinyRow: Row,
  ScrollText,
  TinyScrollText: ScrollText,
  Scrollbar,
  TinyScrollbar: Scrollbar,
  Search,
  TinySearch: Search,
  Select,
  TinySelect: Select,
  SelectDropdown,
  TinySelectDropdown: SelectDropdown,
  SelectedBox,
  TinySelectedBox: SelectedBox,
  Skeleton,
  TinySkeleton: Skeleton,
  SkeletonItem,
  TinySkeletonItem: SkeletonItem,
  Slider,
  TinySlider: Slider,
  Split,
  TinySplit: Split,
  Statistic,
  TinyStatistic: Statistic,
  Steps,
  TinySteps: Steps,
  Sticky,
  TinySticky: Sticky,
  Switch,
  TinySwitch: Switch,
  TabItem,
  TinyTabItem: TabItem,
  Table,
  TinyTable: Table,
  Tabs,
  TinyTabs: Tabs,
  Tag,
  TinyTag: Tag,
  TagGroup,
  TinyTagGroup: TagGroup,
  TextPopup,
  TinyTextPopup: TextPopup,
  Time,
  TinyTime: Time,
  TimeLine,
  TinyTimeLine: TimeLine,
  TimeLineNew,
  TinyTimeLineNew: TimeLineNew,
  TimePanel,
  TinyTimePanel: TimePanel,
  TimePicker,
  TinyTimePicker: TimePicker,
  TimeRange,
  TinyTimeRange: TimeRange,
  TimeSelect,
  TinyTimeSelect: TimeSelect,
  TimeSpinner,
  TinyTimeSpinner: TimeSpinner,
  TimelineItem,
  TinyTimelineItem: TimelineItem,
  ToggleMenu,
  TinyToggleMenu: ToggleMenu,
  Tooltip,
  TinyTooltip: Tooltip,
  TopBox,
  TinyTopBox: TopBox,
  Transfer,
  TinyTransfer: Transfer,
  TransferPanel,
  TinyTransferPanel: TransferPanel,
  Tree,
  TinyTree: Tree,
  TreeMenu,
  TinyTreeMenu: TreeMenu,
  TreeSelect,
  TinyTreeSelect: TreeSelect,
  Upload,
  TinyUpload: Upload,
  UploadDragger,
  TinyUploadDragger: UploadDragger,
  UploadList,
  TinyUploadList: UploadList,
  User,
  TinyUser: User,
  UserAccount,
  TinyUserAccount: UserAccount,
  UserContact,
  TinyUserContact: UserContact,
  UserHead,
  TinyUserHead: UserHead,
  UserLink,
  TinyUserLink: UserLink,
  VirtualScrollBox,
  TinyVirtualScrollBox: VirtualScrollBox,
  VirtualTree,
  TinyVirtualTree: VirtualTree,
  Watermark,
  TinyWatermark: Watermark,
  Wizard,
  TinyWizard: Wizard,
  YearRange,
  TinyYearRange: YearRange,
  YearTable,
  TinyYearTable: YearTable,
  install
};
export {
  default2 as ActionMenu,
  default4 as Alert,
  default6 as Amount,
  default8 as Anchor,
  default10 as Area,
  default12 as Autocomplete,
  default14 as Badge,
  default16 as BaseSelect,
  default18 as Breadcrumb,
  default20 as BreadcrumbItem,
  default22 as BulletinBoard,
  default24 as Button,
  default26 as ButtonGroup,
  default28 as Calendar,
  default30 as CalendarView,
  default32 as Card,
  default34 as CardGroup,
  default36 as CardTemplate,
  default38 as Carousel,
  default40 as CarouselItem,
  default42 as Cascader,
  default44 as CascaderMenu,
  default46 as CascaderNode,
  default48 as CascaderPanel,
  default152 as Charts,
  default155 as ChartsAmap,
  default158 as ChartsBar,
  default161 as ChartsBmap,
  default164 as ChartsBoxplot,
  default167 as ChartsCandle,
  default170 as ChartsCore,
  default173 as ChartsFunnel,
  default176 as ChartsGauge,
  default179 as ChartsGraph,
  default182 as ChartsHeatmap,
  default185 as ChartsHistogram,
  default188 as ChartsLine,
  default191 as ChartsLiquidfill,
  default194 as ChartsMap,
  default197 as ChartsPie,
  default200 as ChartsProcess,
  default203 as ChartsRadar,
  default206 as ChartsRing,
  default209 as ChartsSankey,
  default212 as ChartsScatter,
  default215 as ChartsSunburst,
  default218 as ChartsTree,
  default221 as ChartsWaterfall,
  default224 as ChartsWordcloud,
  default50 as Checkbox,
  default52 as CheckboxButton,
  default54 as CheckboxGroup,
  default56 as Col,
  default58 as Collapse,
  default60 as CollapseItem,
  default62 as CollapseTransition,
  default64 as ColorPicker,
  default66 as ColorSelectPanel,
  default68 as Company,
  default70 as ConfigProvider,
  default72 as Container,
  default74 as Country,
  default76 as Crop,
  default78 as Currency,
  default80 as DatePanel,
  default82 as DatePicker,
  default84 as DatePickerMobileFirst,
  default86 as DateRange,
  default88 as DateTable,
  default90 as Dept,
  default92 as DialogBox,
  default94 as DialogSelect,
  default96 as Divider,
  default98 as Drawer,
  default100 as DropRoles,
  default102 as DropTimes,
  default104 as Dropdown,
  default106 as DropdownItem,
  default108 as DropdownMenu,
  default110 as DynamicScroller,
  default112 as DynamicScrollerItem,
  default114 as Espace,
  default116 as FallMenu,
  default118 as FileUpload,
  default120 as FilterBox,
  default122 as FilterPanel,
  default124 as FloatButton,
  default126 as Floatbar,
  default128 as Flowchart,
  default130 as FluentEditor,
  default132 as Form,
  default134 as FormItem,
  default136 as Fullscreen,
  default138 as Grid,
  default140 as GridColumn,
  default142 as GridManager,
  default144 as GridSelect,
  default146 as GridToolbar,
  default148 as Guide,
  default150 as Hrapprover,
  default153 as Huicharts,
  default156 as HuichartsAmap,
  default159 as HuichartsBar,
  default162 as HuichartsBmap,
  default165 as HuichartsBoxplot,
  default168 as HuichartsCandle,
  default171 as HuichartsCore,
  default174 as HuichartsFunnel,
  default177 as HuichartsGauge,
  default180 as HuichartsGraph,
  default183 as HuichartsHeatmap,
  default186 as HuichartsHistogram,
  default189 as HuichartsLine,
  default192 as HuichartsLiquidfill,
  default195 as HuichartsMap,
  default198 as HuichartsPie,
  default201 as HuichartsProcess,
  default204 as HuichartsRadar,
  default207 as HuichartsRing,
  default210 as HuichartsSankey,
  default213 as HuichartsScatter,
  default216 as HuichartsSunburst,
  default219 as HuichartsTree,
  default222 as HuichartsWaterfall,
  default225 as HuichartsWordcloud,
  default227 as Image,
  default229 as ImageViewer,
  default231 as IndexBarAnchor,
  default233 as Input,
  default235 as IpAddress,
  default237 as Layout,
  default239 as Link,
  default241 as LinkMenu,
  default243 as Loading,
  default245 as Locales,
  default247 as LogonUser,
  default249 as Logout,
  default251 as Message,
  default253 as Milestone,
  default255 as MindMap,
  default257 as Modal,
  default259 as MonthRange,
  default261 as MonthTable,
  default263 as NavMenu,
  default265 as Notify,
  default267 as Numeric,
  default269 as Option,
  default271 as OptionGroup,
  default273 as Pager,
  default275 as PagerItem,
  default277 as Panel,
  default279 as Picker,
  default281 as PopUpload,
  default283 as Popconfirm,
  default285 as Popeditor,
  default287 as Popover,
  default289 as Popup,
  default291 as Progress,
  default293 as QrCode,
  default295 as QuarterPanel,
  default297 as QueryBuilder,
  default299 as Radio,
  default301 as RadioButton,
  default303 as RadioGroup,
  default305 as Rate,
  default307 as RecycleScroller,
  default309 as RichTextEditor,
  default311 as River,
  default313 as Roles,
  default315 as Row,
  default317 as ScrollText,
  default319 as Scrollbar,
  default321 as Search,
  default323 as Select,
  default325 as SelectDropdown,
  default327 as SelectedBox,
  default329 as Skeleton,
  default331 as SkeletonItem,
  default333 as Slider,
  default335 as Split,
  default337 as Statistic,
  default339 as Steps,
  default341 as Sticky,
  default343 as Switch,
  default345 as TabItem,
  default347 as Table,
  default349 as Tabs,
  default351 as Tag,
  default353 as TagGroup,
  default355 as TextPopup,
  default357 as Time,
  default359 as TimeLine,
  default361 as TimeLineNew,
  default363 as TimePanel,
  default365 as TimePicker,
  default367 as TimeRange,
  default369 as TimeSelect,
  default371 as TimeSpinner,
  default373 as TimelineItem,
  default3 as TinyActionMenu,
  default5 as TinyAlert,
  default7 as TinyAmount,
  default9 as TinyAnchor,
  default11 as TinyArea,
  default13 as TinyAutocomplete,
  default15 as TinyBadge,
  default17 as TinyBaseSelect,
  default19 as TinyBreadcrumb,
  default21 as TinyBreadcrumbItem,
  default23 as TinyBulletinBoard,
  default25 as TinyButton,
  default27 as TinyButtonGroup,
  default29 as TinyCalendar,
  default31 as TinyCalendarView,
  default33 as TinyCard,
  default35 as TinyCardGroup,
  default37 as TinyCardTemplate,
  default39 as TinyCarousel,
  default41 as TinyCarouselItem,
  default43 as TinyCascader,
  default45 as TinyCascaderMenu,
  default47 as TinyCascaderNode,
  default49 as TinyCascaderPanel,
  default51 as TinyCheckbox,
  default53 as TinyCheckboxButton,
  default55 as TinyCheckboxGroup,
  default57 as TinyCol,
  default59 as TinyCollapse,
  default61 as TinyCollapseItem,
  default63 as TinyCollapseTransition,
  default65 as TinyColorPicker,
  default67 as TinyColorSelectPanel,
  default69 as TinyCompany,
  default71 as TinyConfigProvider,
  default73 as TinyContainer,
  default75 as TinyCountry,
  default77 as TinyCrop,
  default79 as TinyCurrency,
  default81 as TinyDatePanel,
  default83 as TinyDatePicker,
  default85 as TinyDatePickerMobileFirst,
  default87 as TinyDateRange,
  default89 as TinyDateTable,
  default91 as TinyDept,
  default93 as TinyDialogBox,
  default95 as TinyDialogSelect,
  default97 as TinyDivider,
  default99 as TinyDrawer,
  default101 as TinyDropRoles,
  default103 as TinyDropTimes,
  default105 as TinyDropdown,
  default107 as TinyDropdownItem,
  default109 as TinyDropdownMenu,
  default111 as TinyDynamicScroller,
  default113 as TinyDynamicScrollerItem,
  default115 as TinyEspace,
  default117 as TinyFallMenu,
  default119 as TinyFileUpload,
  default121 as TinyFilterBox,
  default123 as TinyFilterPanel,
  default125 as TinyFloatButton,
  default127 as TinyFloatbar,
  default129 as TinyFlowchart,
  default131 as TinyFluentEditor,
  default133 as TinyForm,
  default135 as TinyFormItem,
  default137 as TinyFullscreen,
  default139 as TinyGrid,
  default141 as TinyGridColumn,
  default143 as TinyGridManager,
  default145 as TinyGridSelect,
  default147 as TinyGridToolbar,
  default149 as TinyGuide,
  default151 as TinyHrapprover,
  default154 as TinyHuicharts,
  default157 as TinyHuichartsAmap,
  default160 as TinyHuichartsBar,
  default163 as TinyHuichartsBmap,
  default166 as TinyHuichartsBoxplot,
  default169 as TinyHuichartsCandle,
  default172 as TinyHuichartsCore,
  default175 as TinyHuichartsFunnel,
  default178 as TinyHuichartsGauge,
  default181 as TinyHuichartsGraph,
  default184 as TinyHuichartsHeatmap,
  default187 as TinyHuichartsHistogram,
  default190 as TinyHuichartsLine,
  default193 as TinyHuichartsLiquidfill,
  default196 as TinyHuichartsMap,
  default199 as TinyHuichartsPie,
  default202 as TinyHuichartsProcess,
  default205 as TinyHuichartsRadar,
  default208 as TinyHuichartsRing,
  default211 as TinyHuichartsSankey,
  default214 as TinyHuichartsScatter,
  default217 as TinyHuichartsSunburst,
  default220 as TinyHuichartsTree,
  default223 as TinyHuichartsWaterfall,
  default226 as TinyHuichartsWordcloud,
  default228 as TinyImage,
  default230 as TinyImageViewer,
  default232 as TinyIndexBarAnchor,
  default234 as TinyInput,
  default236 as TinyIpAddress,
  default238 as TinyLayout,
  default240 as TinyLink,
  default242 as TinyLinkMenu,
  default244 as TinyLoading,
  default246 as TinyLocales,
  default248 as TinyLogonUser,
  default250 as TinyLogout,
  default252 as TinyMessage,
  default254 as TinyMilestone,
  default256 as TinyMindMap,
  default258 as TinyModal,
  default260 as TinyMonthRange,
  default262 as TinyMonthTable,
  default264 as TinyNavMenu,
  default266 as TinyNotify,
  default268 as TinyNumeric,
  default270 as TinyOption,
  default272 as TinyOptionGroup,
  default274 as TinyPager,
  default276 as TinyPagerItem,
  default278 as TinyPanel,
  default280 as TinyPicker,
  default282 as TinyPopUpload,
  default284 as TinyPopconfirm,
  default286 as TinyPopeditor,
  default288 as TinyPopover,
  default290 as TinyPopup,
  default292 as TinyProgress,
  default294 as TinyQrCode,
  default296 as TinyQuarterPanel,
  default298 as TinyQueryBuilder,
  default300 as TinyRadio,
  default302 as TinyRadioButton,
  default304 as TinyRadioGroup,
  default306 as TinyRate,
  default308 as TinyRecycleScroller,
  default310 as TinyRichTextEditor,
  default312 as TinyRiver,
  default314 as TinyRoles,
  default316 as TinyRow,
  default318 as TinyScrollText,
  default320 as TinyScrollbar,
  default322 as TinySearch,
  default324 as TinySelect,
  default326 as TinySelectDropdown,
  default328 as TinySelectedBox,
  default330 as TinySkeleton,
  default332 as TinySkeletonItem,
  default334 as TinySlider,
  default336 as TinySplit,
  default338 as TinyStatistic,
  default340 as TinySteps,
  default342 as TinySticky,
  default344 as TinySwitch,
  default346 as TinyTabItem,
  default348 as TinyTable,
  default350 as TinyTabs,
  default352 as TinyTag,
  default354 as TinyTagGroup,
  default356 as TinyTextPopup,
  default358 as TinyTime,
  default360 as TinyTimeLine,
  default362 as TinyTimeLineNew,
  default364 as TinyTimePanel,
  default366 as TinyTimePicker,
  default368 as TinyTimeRange,
  default370 as TinyTimeSelect,
  default372 as TinyTimeSpinner,
  default374 as TinyTimelineItem,
  default375 as TinyToggleMenu,
  default377 as TinyTooltip,
  default379 as TinyTopBox,
  default381 as TinyTransfer,
  default383 as TinyTransferPanel,
  default385 as TinyTree,
  default387 as TinyTreeMenu,
  default389 as TinyTreeSelect,
  default391 as TinyUpload,
  default393 as TinyUploadDragger,
  default395 as TinyUploadList,
  default397 as TinyUser,
  default399 as TinyUserAccount,
  default401 as TinyUserContact,
  default403 as TinyUserHead,
  default405 as TinyUserLink,
  default407 as TinyVirtualScrollBox,
  default409 as TinyVirtualTree,
  default411 as TinyWatermark,
  default413 as TinyWizard,
  default415 as TinyYearRange,
  default417 as TinyYearTable,
  default376 as ToggleMenu,
  default378 as Tooltip,
  default380 as TopBox,
  default382 as Transfer,
  default384 as TransferPanel,
  default386 as Tree,
  default388 as TreeMenu,
  default390 as TreeSelect,
  default392 as Upload,
  default394 as UploadDragger,
  default396 as UploadList,
  default398 as User,
  default400 as UserAccount,
  default402 as UserContact,
  default404 as UserHead,
  default406 as UserLink,
  default408 as VirtualScrollBox,
  default410 as VirtualTree,
  default412 as Watermark,
  default414 as Wizard,
  default416 as YearRange,
  default418 as YearTable,
  pc as default,
  install,
  version
};
