// @opentiny/huicharts@1.0.1/util/type.js
/**
 * Copyright (c) 2024 - present OpenTiny HUICharts Authors.
 * Copyright (c) 2024 - present Huawei Cloud Computing Technologies Co., Ltd.
 *
 * Use of this source code is governed by an MIT-style license.
 *
 * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
 * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
 * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
 *
 */
function isArray(value) {
  return Object.prototype.toString.call(value) === '[object Array]' ? true : false;
}
var isObject = function isObject(value) {
  return value !== null && typeof value === 'object';
};
var isFunction = function isFunction(value) {
  return typeof value === 'function';
};
var isString = function isString(value) {
  return typeof value === 'string';
};
var isBoolean = function isBoolean(value) {
  return typeof value === 'boolean';
};
var isNumber = function isNumber(value) {
  return typeof value === 'number';
};
var isDOM = typeof HTMLElement === 'object' ? function (dom) {
  return dom instanceof HTMLElement;
} : function (dom) {
  return dom && typeof dom === 'object' && dom.nodeType === 1 && typeof dom.nodeName === 'string';
};
var toArray = function toArray(value) {
  if (!isArray(value)) {
    return [value];
  } else {
    return value;
  }
};
export { isArray, isBoolean, isDOM, isFunction, isNumber, isObject, isString, toArray };
