// @opentiny/huicharts@1.0.1/components/CircleProcessChart/handleSeries.js
import chartToken from './chartToken.js';

/**
 * Copyright (c) 2024 - present OpenTiny HUICharts Authors.
 * Copyright (c) 2024 - present Huawei Cloud Computing Technologies Co., Ltd.
 *
 * Use of this source code is governed by an MIT-style license.
 *
 * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
 * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
 * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
 *
 */

function getSeriesInit() {
  return {
    type: 'bar',
    coordinateSystem: 'polar',
    emphasis: {
      focus: 'series'
    },
    animation: false,
    stack: 'total',
    showBackground: true,
    backgroundStyle: {
      color: ''
    },
    data: []
  };
}
function setSeries(seriesData, iChartOption) {
  var data = iChartOption.data;
  var series = [];
  data.forEach(function (item, i) {
    var seriesUnit = getSeriesInit();
    seriesUnit.name = item.name;
    seriesUnit.data = seriesData[i];
    seriesUnit.backgroundStyle.color = chartToken.background;
    series.push(seriesUnit);
  });
  return series;
}
export { setSeries };
