// @opentiny/vue-exception@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/exception/vue';
import { defineComponent, setup, $props } from '@opentiny/vue-common';
import Button from '@opentiny/vue-button';
import { iconNoData, iconNoNews, iconNoPerm, iconNoResult, iconWeaknet, iconPageNoperm, iconPageNothing, iconPageServererror, iconPageWeaknet } from '@opentiny/vue-icon';
import { openBlock, createElementBlock, normalizeClass, createElementVNode, createBlock, resolveDynamicComponent, createCommentVNode, renderSlot, toDisplayString } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var $constants = {
  INTERNALCONFIG: {
    NODATA: "ui.exception.nodatamf",
    NOPERM: "ui.exception.nopermmf",
    WEAKNET: "ui.exception.weaknetmf",
    NORESULT: "ui.exception.noresult",
    NONEWS: "ui.exception.nonews",
    PAGENOPERM: "ui.exception.pagenoperm",
    PAGEWEAKNET: "ui.exception.pageweaknet",
    PAGENOTHING: "ui.exception.pagenothing",
    PAGESERVERERROR: "ui.exception.pageservererror"
  },
  ICONCONFIG: {
    NODATA: "icon-no-data",
    NOPERM: "icon-no-perm",
    NONEWS: "icon-no-news",
    WEAKNET: "icon-weaknet",
    NORESULT: "icon-no-result",
    PAGENOPERM: "icon-page-noperm",
    PAGENOTHING: "icon-page-nothing",
    PAGESERVERERROR: "icon-page-servererror",
    PAGEWEAKNET: "icon-page-weaknet"
  }
};
var _sfc_main = defineComponent({
  components: {
    TinyButton: Button,
    IconNoData: iconNoData(),
    IconNoNews: iconNoNews(),
    IconNoPerm: iconNoPerm(),
    IconNoResult: iconNoResult(),
    IconWeaknet: iconWeaknet(),
    IconPageNoperm: iconPageNoperm(),
    IconPageNothing: iconPageNothing(),
    IconPageServererror: iconPageServererror(),
    IconPageWeaknet: iconPageWeaknet()
  },
  emits: ["click"],
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    type: {
      type: String,
      default: "nodata"
    },
    subMessage: String,
    exceptionClass: String,
    pageEmpty: {
      type: Boolean,
      default: false
    },
    componentPage: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup$1(props, context) {
    return setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-exception-body"
};
var _hoisted_2 = {
  class: "bg-cover",
  "data-tag": "tiny-exception-image"
};
var _hoisted_3 = {
  "data-tag": "tiny-exception-description"
};
var _hoisted_4 = {
  key: 0,
  "data-tag": "tiny-exception-submessage",
  class: "leading-5.5 sm:leading-3 sm:text-xs mt-1 sm:mt-2 text-color-icon-placeholder"
};
var _hoisted_5 = {
  class: "sm:mt-4 mt-6 inline-block"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-exception",
      class: normalizeClass(_ctx.m("flex t-0 justify-center items-center w-full h-full bg-transparent sm:bg-color-bg-1 text-center", _ctx.exceptionClass))
    },
    [createElementVNode("div", _hoisted_1, [createElementVNode("div", _hoisted_2, [_ctx.state.urlType ? (openBlock(), createBlock(resolveDynamicComponent(_ctx._constants.ICONCONFIG[_ctx.type.toUpperCase()]), {
      key: 0,
      "custom-class": _ctx.componentPage ? "w-24 h-24" : "w-52 h-40"
    }, null, 8, ["custom-class"])) : createCommentVNode("v-if", true)]), renderSlot(_ctx.$slots, "content", {}, function() {
      return [createElementVNode("div", _hoisted_3, [createElementVNode(
        "div",
        {
          "data-tag": "tiny-exception-componentpage",
          class: normalizeClass(["text-color-text-primary mt-1 text-center text-sm", _ctx.componentPage ? "sm:text-xs" : "sm:text-sm"])
        },
        toDisplayString(_ctx.t(_ctx._constants.INTERNALCONFIG[_ctx.type.toUpperCase()])),
        3
        /* TEXT, CLASS */
      ), _ctx.subMessage ? (openBlock(), createElementBlock(
        "div",
        _hoisted_4,
        toDisplayString(_ctx.subMessage),
        1
        /* TEXT */
      )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_5, [renderSlot(_ctx.$slots, "default")])])];
    })])],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
