// @opentiny/vue-year-range@3.20.0/lib/mobile-first.js
import { renderless, api } from '@opentiny/vue-renderless/year-range/vue';
import { defineComponent, directive, props, setup } from '@opentiny/vue-common';
import Clickoutside from '@opentiny/vue-renderless/common/deps/clickoutside';
import YearTable from '@opentiny/vue-year-table';
import { IconDoubleRight, IconDoubleLeft } from '@opentiny/vue-icon';
import { resolveComponent, openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, renderSlot, createElementBlock, Fragment, renderList, toDisplayString, createCommentVNode, createVNode, vShow } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  directives: directive({
    Clickoutside
  }),
  components: {
    YearTable,
    IconDoubleRight: IconDoubleRight(),
    IconDoubleLeft: IconDoubleLeft()
  },
  props: [].concat(props, ["emitter"]),
  emits: ["dodestroy", "pick"],
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "after:content-[''] after:table after:clear-both"
};
var _hoisted_2 = {
  key: 0,
  class: "absolute top-0 bottom-0 w-28 border-r border-r-color-bg-3 box-border pt-5 bg-color-bg-1 overflow-auto"
};
var _hoisted_3 = ["onClick", "title"];
var _hoisted_4 = {
  class: "m-0 float-left w-1/2 border-r border-r-color-bg-3 relative py-0 px-4"
};
var _hoisted_5 = {
  class: "relative text-center h-12 p-4"
};
var _hoisted_6 = {
  class: "mx-12 text-xs font-medium"
};
var _hoisted_7 = {
  class: "m-0 float-right w-1/2 border-r border-r-color-bg-3 relative py-0 px-4"
};
var _hoisted_8 = {
  class: "relative text-center h-12 p-4"
};
var _hoisted_9 = {
  class: "mx-12 text-xs font-medium"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_double_left = resolveComponent("icon-double-left");
  var _component_year_table = resolveComponent("year-table");
  var _component_icon_double_right = resolveComponent("icon-double-right");
  return openBlock(), createBlock(Transition, {
    onAfterLeave: _cache[2] || (_cache[2] = function($event) {
      return _ctx.$emit("dodestroy");
    }),
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          class: normalizeClass(["w-[552px] text-color-text-primary text-xs shadow-md bg-color-bg-1 rounded leading-4 my-0.5 mx-0", [{
            "w-[664px]": _ctx.slots.sidebar || _ctx.state.shortcuts
          }, _ctx.state.popperClass]])
        },
        [createElementVNode("div", _hoisted_1, [renderSlot(_ctx.$slots, "sidebar", {
          class: "absolute top-0 bottom-0 w-28 border-r border-r-color-bg-3 box-border pt-5 bg-color-bg-1 overflow-auto"
        }), _ctx.state.shortcuts ? (openBlock(), createElementBlock("div", _hoisted_2, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.shortcuts, function(shortcut, key) {
            return openBlock(), createElementBlock("button", {
              type: "button",
              class: "block w-24 my-0 mx-auto rounded border-0 bg-transparent leading-8 text-xs text-color-text-primary pl-4 text-left outline-0 cursor-pointer truncate hover:bg-color-bg-2",
              key,
              onClick: function onClick($event) {
                return _ctx.handleShortcutClick(shortcut);
              },
              title: shortcut.text
            }, toDisplayString(shortcut.text), 9, _hoisted_3);
          }),
          128
          /* KEYED_FRAGMENT */
        ))])) : createCommentVNode("v-if", true), createElementVNode(
          "div",
          {
            class: normalizeClass(["after:content-[''] after:table after:clear-both min-w-[513px]", {
              "ml-28": _ctx.slots.sidebar || _ctx.state.shortcuts
            }])
          },
          [createElementVNode("div", _hoisted_4, [createElementVNode("div", _hoisted_5, [createElementVNode("button", {
            type: "button",
            onClick: _cache[0] || (_cache[0] = function() {
              return _ctx.leftPrevYear && _ctx.leftPrevYear.apply(_ctx, arguments);
            }),
            class: "float-left text-sm leading-4 fill-color-text-primary border-0 bg-left-bottom cursor-pointer outline-0 hover:fill-color-brand"
          }, [createVNode(_component_icon_double_left)]), createElementVNode(
            "div",
            _hoisted_6,
            toDisplayString(_ctx.state.leftLabel),
            1
            /* TEXT */
          )]), createVNode(_component_year_table, {
            ref: "leftYearTable",
            "selection-mode": _ctx.state.selectionMode,
            date: _ctx.state.leftDate,
            "default-value": _ctx.state.defaultValue,
            "min-date": _ctx.state.minDate,
            "max-date": _ctx.state.maxDate,
            "range-state": _ctx.state.rangeState,
            "disabled-date": _ctx.state.disabledDate,
            "start-year": _ctx.state.leftStartYear,
            onChangerange: _ctx.handleChangeRange,
            onPick: _ctx.handleRangePick
          }, null, 8, ["selection-mode", "date", "default-value", "min-date", "max-date", "range-state", "disabled-date", "start-year", "onChangerange", "onPick"])]), createElementVNode("div", _hoisted_7, [createElementVNode("div", _hoisted_8, [createElementVNode("button", {
            type: "button",
            onClick: _cache[1] || (_cache[1] = function() {
              return _ctx.rightNextYear && _ctx.rightNextYear.apply(_ctx, arguments);
            }),
            class: "float-right text-sm leading-4 fill-color-text-primary border-0 bg-left-bottom cursor-pointer outline-0 hover:fill-color-brand"
          }, [createVNode(_component_icon_double_right)]), createElementVNode(
            "div",
            _hoisted_9,
            toDisplayString(_ctx.state.rightLabel),
            1
            /* TEXT */
          )]), createVNode(_component_year_table, {
            ref: "rightRearTable",
            "selection-mode": _ctx.state.selectionMode,
            date: _ctx.state.rightDate,
            "default-value": _ctx.state.defaultValue,
            "min-date": _ctx.state.minDate,
            "max-date": _ctx.state.maxDate,
            "range-state": _ctx.state.rangeState,
            "disabled-date": _ctx.state.disabledDate,
            "start-year": _ctx.state.rightStartYear,
            onChangerange: _ctx.handleChangeRange,
            onPick: _ctx.handleRangePick
          }, null, 8, ["selection-mode", "date", "default-value", "min-date", "max-date", "range-state", "disabled-date", "start-year", "onChangerange", "onPick"])])],
          2
          /* CLASS */
        )])],
        2
        /* CLASS */
      ), [[vShow, _ctx.state.visible]])];
    }),
    _: 3
    /* FORWARDED */
  });
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
