// @opentiny/vue-pull-refresh@3.20.0/lib/mobile-first.js
import { defineComponent, $prefix, props, setup, h } from '@opentiny/vue-common';
import { renderless, api } from '@opentiny/vue-renderless/pull-refresh-mobile-first/vue';
import Loading from '@opentiny/vue-loading';
import { openBlock, createElementBlock, normalizeStyle, createElementVNode, createBlock, resolveDynamicComponent, renderSlot } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

function _extends() {
  return _extends = Object.assign ? Object.assign.bind() : function(n) {
    for (var e = 1; e < arguments.length; e++) {
      var t = arguments[e];
      for (var r in t)
        ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
    }
    return n;
  }, _extends.apply(null, arguments);
}
var _sfc_main = defineComponent({
  name: $prefix + "PullRefresh",
  emits: ["change", "refresh", "update:modelValue"],
  props: [].concat(props, ["disabled", "modelValue", "headHeight", "successText", "pullingText", "loosingText", "loadingText", "pullDistance", "successDuration", "animationDuration", "loadingOptions", "selfSimulate"]),
  setup: function setup$1(props2, context) {
    var apis = setup({
      props: props2,
      context,
      renderless,
      api,
      extendOptions: {
        Loading
      }
    });
    var state = apis.state, getStatusText = apis.getStatusText, constants = apis.constants;
    var _constants$STATUS = constants.STATUS, PULLING = _constants$STATUS.PULLING, LOOSING = _constants$STATUS.LOOSING, SUCCESS = _constants$STATUS.SUCCESS;
    var HeadComponent = {
      render: function render() {
        var slots = context.slots;
        var status = state.status, distance = state.distance;
        var children;
        if (slots[status]) {
          children = [slots[status]({
            distance
          })];
        } else {
          children = [];
          if ([PULLING, LOOSING, SUCCESS].includes(status)) {
            children.push(h("div", {
              attrs: {
                "data-tag": "tiny-pull-refresh__text"
              }
            }, [getStatusText()]));
          }
        }
        return h("div", {
          class: ["absolute left-0 -translate-y-full overflow-hidden text-base text-color-text-info-primary text-center", "w-full h-[var(--tiny-pull-refresh-head-height)] leading-[var(--tiny-pull-refresh-head-height)]"],
          attrs: {
            "data-tag": "tiny-pull-refresh__head"
          }
        }, children);
      }
    };
    return _extends({}, apis, {
      HeadComponent
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      ref: "root",
      "data-tag": "tiny-pull-refresh",
      style: normalizeStyle({
        "--tiny-pull-refresh-head-height": _ctx.state.headerHeight
      })
    },
    [createElementVNode(
      "div",
      {
        ref: "track",
        class: "relative h-fit transition-transform",
        "data-tag": "tiny-pull-refresh__track",
        "data-tiny-touch-simulate-container": "",
        style: normalizeStyle({
          transitionDuration: _ctx.state.duration + "ms",
          transform: _ctx.state.distance ? "translate3d(0," + _ctx.state.distance + "px, 0)" : ""
        }),
        onTouchstartPassive: _cache[0] || (_cache[0] = function() {
          return _ctx.onTouchStart && _ctx.onTouchStart.apply(_ctx, arguments);
        }),
        onTouchend: _cache[1] || (_cache[1] = function() {
          return _ctx.onTouchEnd && _ctx.onTouchEnd.apply(_ctx, arguments);
        }),
        onTouchcancel: _cache[2] || (_cache[2] = function() {
          return _ctx.onTouchEnd && _ctx.onTouchEnd.apply(_ctx, arguments);
        })
      },
      [(openBlock(), createBlock(
        resolveDynamicComponent(_ctx.HeadComponent),
        {
          ref: "head"
        },
        null,
        512
        /* NEED_PATCH */
      )), renderSlot(_ctx.$slots, "default")],
      36
      /* STYLE, NEED_HYDRATION */
    )],
    4
    /* STYLE */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobileFirst as default };
