// @opentiny/vue-checkbox@3.20.0/lib/mobile.js
import { renderless, api } from '@opentiny/vue-renderless/checkbox/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import '@opentiny/vue-theme-mobile/checkbox/index.css';
import { openBlock, createElementBlock, normalizeClass, createElementVNode, withDirectives, withModifiers, vModelCheckbox, renderSlot, createTextVNode, toDisplayString, createCommentVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["modelValue", "text", "events", "label", "indeterminate", "disabled", "checked", "name", "trueLabel", "falseLabel", "id", "iconPosition"]),
  emits: ["update:modelValue", "change", "complete", "click"],
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["id"];
var _hoisted_2 = ["tabindex", "role", "aria-checked"];
var _hoisted_3 = /* @__PURE__ */ createElementVNode(
  "span",
  {
    class: "tiny-mobile-checkbox__inner"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_4 = ["aria-hidden", "name", "disabled", "true-value", "false-value"];
var _hoisted_5 = ["aria-hidden", "disabled", "value", "name"];
var _hoisted_6 = {
  key: 0,
  class: "tiny-mobile-checkbox__label"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("label", {
    class: normalizeClass(["tiny-mobile-checkbox", {
      "is-disabled": _ctx.state.isDisabled,
      "is-checked": _ctx.state.isChecked,
      "is-indeterminate": _ctx.indeterminate,
      "is-focus": _ctx.state.focus,
      "icon-position-top": _ctx.state.iconPosition === "top"
    }]),
    id: _ctx.id
  }, [createElementVNode("span", {
    class: "tiny-mobile-checkbox__input",
    tabindex: _ctx.indeterminate ? 0 : false,
    role: _ctx.indeterminate ? "checkbox" : false,
    "aria-checked": _ctx.indeterminate ? "mixed" : false
  }, [_hoisted_3, _ctx.trueLabel || _ctx.falseLabel ? withDirectives((openBlock(), createElementBlock("input", {
    key: 0,
    class: "tiny-mobile-checkbox__original",
    type: "checkbox",
    "aria-hidden": _ctx.indeterminate ? "true" : "false",
    name: _ctx.name,
    disabled: _ctx.state.isDisabled,
    "true-value": _ctx.trueLabel,
    "false-value": _ctx.falseLabel,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.model = $event;
    }),
    onChange: _cache[1] || (_cache[1] = withModifiers(function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }, ["stop"])),
    onFocus: _cache[2] || (_cache[2] = function($event) {
      return _ctx.state.focus = true;
    }),
    onBlur: _cache[3] || (_cache[3] = function($event) {
      return _ctx.state.focus = false;
    })
  }, null, 40, _hoisted_4)), [[vModelCheckbox, _ctx.state.model]]) : withDirectives((openBlock(), createElementBlock("input", {
    key: 1,
    class: "tiny-mobile-checkbox__original",
    type: "checkbox",
    "aria-hidden": _ctx.indeterminate ? "true" : "false",
    disabled: _ctx.state.isDisabled,
    value: _ctx.label,
    name: _ctx.name,
    "onUpdate:modelValue": _cache[4] || (_cache[4] = function($event) {
      return _ctx.state.model = $event;
    }),
    onChange: _cache[5] || (_cache[5] = withModifiers(function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }, ["stop"])),
    onFocus: _cache[6] || (_cache[6] = function($event) {
      return _ctx.state.focus = true;
    }),
    onBlur: _cache[7] || (_cache[7] = function($event) {
      return _ctx.state.focus = false;
    })
  }, null, 40, _hoisted_5)), [[vModelCheckbox, _ctx.state.model]])], 8, _hoisted_2), _ctx.slots.default && _ctx.slots.default() || _ctx.state.isShowText ? (openBlock(), createElementBlock("span", _hoisted_6, [renderSlot(_ctx.$slots, "default", {}, function() {
    return [createTextVNode(
      toDisplayString(_ctx.state.showText),
      1
      /* TEXT */
    )];
  })])) : createCommentVNode("v-if", true)], 10, _hoisted_1);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { mobile as default };
