// @opentiny/vue-checkbox@3.20.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/checkbox/vue';
import { defineComponent, props, setup } from '@opentiny/vue-common';
import '@opentiny/vue-theme/checkbox/index.css';
import { iconHalfselect, iconCheckedSur, iconCheck } from '@opentiny/vue-icon';
import { AutoTip } from '@opentiny/vue-directive';
import { resolveComponent, resolveDirective, openBlock, createElementBlock, mergeProps, createElementVNode, normalizeClass, withDirectives, withModifiers, vModelCheckbox, createBlock, renderSlot, createTextVNode, toDisplayString, createCommentVNode } from 'vue';

function _createForOfIteratorHelperLoose(r, e) {
  var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
  if (t)
    return (t = t.call(r)).next.bind(t);
  if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) {
    t && (r = t);
    var o = 0;
    return function() {
      return o >= r.length ? { done: true } : { done: false, value: r[o++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(r, a) {
  if (r) {
    if ("string" == typeof r)
      return _arrayLikeToArray(r, a);
    var t = {}.toString.call(r).slice(8, -1);
    return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0;
  }
}
function _arrayLikeToArray(r, a) {
  (null == a || a > r.length) && (a = r.length);
  for (var e = 0, n = Array(a); e < a; e++)
    n[e] = r[e];
  return n;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  // tiny 新增。 renderless中，没有emit('click')的地方。 此处勿声明，否则会造成丢失click事件。
  emits: ["update:modelValue", "change", "complete"],
  directives: {
    AutoTip
  },
  props: [].concat(props, ["modelValue", "text", "events", "label", "indeterminate", "disabled", "checked", "name", "trueLabel", "falseLabel", "id", "controls", "size", "border", "validateEvent", "displayOnly", "shape", "tabindex"]),
  components: {
    IconHalfselect: iconHalfselect(),
    IconCheckedSur: iconCheckedSur(),
    IconCheck: iconCheck()
  },
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["id"];
var _hoisted_2 = ["role", "aria-checked"];
var _hoisted_3 = ["tabindex", "aria-hidden", "name", "disabled", "true-value", "false-value"];
var _hoisted_4 = ["tabindex", "aria-hidden", "disabled", "value", "name"];
var _hoisted_5 = {
  class: "tiny-checkbox__inner"
};
var _hoisted_6 = {
  key: 0,
  class: "tiny-checkbox__label tiny-checkbox-display-only"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_halfselect = resolveComponent("icon-halfselect");
  var _component_icon_checked_sur = resolveComponent("icon-checked-sur");
  var _component_icon_check = resolveComponent("icon-check");
  var _directive_auto_tip = resolveDirective("auto-tip");
  return openBlock(), createElementBlock("label", mergeProps({
    class: ["tiny-checkbox", [_ctx.border && _ctx.state.checkboxSize ? "tiny-checkbox--" + _ctx.state.checkboxSize : "", {
      "is-disabled": _ctx.state.isDisabled || _ctx.state.isDisplayOnly
    }, {
      "is-bordered": _ctx.border
    }, {
      "is-checked": _ctx.state.isChecked
    }, {
      "is-group-display-only": _ctx.state.isGroupDisplayOnly
    }, {
      "is-display-only": _ctx.state.isDisplayOnly
    }, {
      "is-filter": _ctx.state.shape === "filter"
    }]],
    id: _ctx.id,
    tabindex: "-1"
  }, _ctx.a(_ctx.$attrs, ["class", "style", "onClick"], true)), [createElementVNode("span", {
    class: normalizeClass(["tiny-checkbox__input", {
      "is-disabled": _ctx.state.isDisabled || _ctx.state.isDisplayOnly,
      "is-checked": _ctx.state.isChecked,
      "is-indeterminate": _ctx.indeterminate,
      "is-focus": _ctx.state.focus
    }]),
    role: _ctx.indeterminate ? "checkbox" : void 0,
    "aria-checked": _ctx.indeterminate ? "mixed" : false
  }, [_ctx.trueLabel || _ctx.falseLabel ? withDirectives((openBlock(), createElementBlock("input", {
    key: 0,
    class: "tiny-checkbox__original",
    type: "checkbox",
    tabindex: _ctx.tabindex,
    "aria-hidden": _ctx.indeterminate ? "true" : "false",
    name: _ctx.name,
    disabled: _ctx.state.isDisabled || _ctx.state.isDisplayOnly,
    "true-value": _ctx.trueLabel,
    "false-value": _ctx.falseLabel,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.model = $event;
    }),
    onChange: _cache[1] || (_cache[1] = withModifiers(function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }, ["stop"])),
    onFocus: _cache[2] || (_cache[2] = function($event) {
      return _ctx.state.focus = true;
    }),
    onBlur: _cache[3] || (_cache[3] = function($event) {
      return _ctx.state.focus = false;
    }),
    onClick: _cache[4] || (_cache[4] = withModifiers(function() {
    }, ["stop"]))
  }, null, 40, _hoisted_3)), [[vModelCheckbox, _ctx.state.model]]) : withDirectives((openBlock(), createElementBlock("input", {
    key: 1,
    class: "tiny-checkbox__original",
    type: "checkbox",
    tabindex: _ctx.tabindex,
    "aria-hidden": _ctx.indeterminate ? "true" : "false",
    disabled: _ctx.state.isDisabled || _ctx.state.isDisplayOnly,
    value: _ctx.label,
    name: _ctx.name,
    "onUpdate:modelValue": _cache[5] || (_cache[5] = function($event) {
      return _ctx.state.model = $event;
    }),
    onChange: _cache[6] || (_cache[6] = withModifiers(function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }, ["stop"])),
    onFocus: _cache[7] || (_cache[7] = function($event) {
      return _ctx.state.focus = true;
    }),
    onBlur: _cache[8] || (_cache[8] = function($event) {
      return _ctx.state.focus = false;
    }),
    onClick: _cache[9] || (_cache[9] = withModifiers(function() {
    }, ["stop"]))
  }, null, 40, _hoisted_4)), [[vModelCheckbox, _ctx.state.model]]), createElementVNode("span", _hoisted_5, [_ctx.indeterminate && _ctx.state.shape !== "filter" ? (openBlock(), createBlock(_component_icon_halfselect, {
    key: 0,
    class: "tiny-svg-size icon-halfselect"
  })) : _ctx.state.isChecked ? (openBlock(), createBlock(_component_icon_checked_sur, {
    key: 1,
    class: "tiny-svg-size icon-checked-sur"
  })) : (openBlock(), createBlock(_component_icon_check, {
    key: 2,
    class: "tiny-svg-size icon-check"
  }))])], 10, _hoisted_2), _ctx.slots.default && _ctx.slots.default() || _ctx.state.isShowText ? withDirectives((openBlock(), createElementBlock("span", _hoisted_6, [renderSlot(_ctx.$slots, "default", {}, function() {
    return [createTextVNode(
      toDisplayString(_ctx.state.showText),
      1
      /* TEXT */
    )];
  })])), [[_directive_auto_tip]]) : createCommentVNode("v-if", true)], 16, _hoisted_1);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
